{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "82bcb386-c921-44c4-9813-7afcaae07b19",
   "metadata": {},
   "source": [
    "# Protein Metadata\n",
    "\n",
    "**Created**: 19 February 2022"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af98c29f-3850-4b1c-ba79-7d204ee7015e",
   "metadata": {},
   "source": [
    "I will generate metadata for the proteins from Yuxin's MS Proteomics data. This will be a file that maps UniProt IDs to Ensembl IDs and gene information. This table will be useful for performing colocalization down the line."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14ac401e-bd10-4be3-814d-0c62e781962d",
   "metadata": {},
   "source": [
    "## Environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2720ef84-32e1-4941-992e-01f9fd545dd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a249ead0-1b73-4c10-928a-9e1d36fec99e",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/nfs/users/nfs_n/nm18/eQTL_pQTL_Characterization/')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "737f1e17-76b4-44aa-adce-cfd8a2c48e97",
   "metadata": {},
   "source": [
    "## Load Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ee4bcdfb-1540-4cac-a982-761b31af0fe1",
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot = pd.read_csv('02_pQTL_Mapping/data/UniProt/HUMAN_9606_idmapping_selected.tab', sep='\\t', header=None, low_memory=False)\n",
    "uniprot = uniprot.iloc[:,[0, 18]]\n",
    "uniprot.columns = ['UniProt_ID', 'Gene_IDs']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5e497d91-6a54-46d4-9342-f6e4b29ec549",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProt_ID</th>\n",
       "      <th>Gene_IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P31946</td>\n",
       "      <td>ENSG00000166913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P62258</td>\n",
       "      <td>ENSG00000108953; ENSG00000274474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q04917</td>\n",
       "      <td>ENSG00000128245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P61981</td>\n",
       "      <td>ENSG00000170027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P31947</td>\n",
       "      <td>ENSG00000175793</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProt_ID                          Gene_IDs\n",
       "0     P31946                   ENSG00000166913\n",
       "1     P62258  ENSG00000108953; ENSG00000274474\n",
       "2     Q04917                   ENSG00000128245\n",
       "3     P61981                   ENSG00000170027\n",
       "4     P31947                   ENSG00000175793"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d856838a-0ce6-4ac8-9593-b399096b8bc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_info = pd.read_csv('/nfs/users/nfs_n/nm18/gains_team282/proteomics/MS2019_processed_data/protein_info_291_MS2019.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "dd858815-8239-4e6a-82bb-dd2a1fff26c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene.Names</th>\n",
       "      <th>Protein Length</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Protein Existence</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sp|A0A075B6I9|LV746_HUMAN</td>\n",
       "      <td>A0A075B6I9</td>\n",
       "      <td>LV746_HUMAN</td>\n",
       "      <td>IGLV7-46</td>\n",
       "      <td>117</td>\n",
       "      <td>76.9</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin lambda variable 7-46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sp|A0A075B6P5|KV228_HUMAN</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>KV228_HUMAN</td>\n",
       "      <td>IGKV2-28</td>\n",
       "      <td>120</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sp|A0A087WW87|KV240_HUMAN</td>\n",
       "      <td>A0A087WW87</td>\n",
       "      <td>KV240_HUMAN</td>\n",
       "      <td>IGKV2-40</td>\n",
       "      <td>121</td>\n",
       "      <td>81.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sp|A0A0B4J1V0|HV315_HUMAN</td>\n",
       "      <td>A0A0B4J1V0</td>\n",
       "      <td>HV315_HUMAN</td>\n",
       "      <td>IGHV3-15</td>\n",
       "      <td>119</td>\n",
       "      <td>84.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin heavy variable 3-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sp|A0A0B4J1V2|HV226_HUMAN</td>\n",
       "      <td>A0A0B4J1V2</td>\n",
       "      <td>HV226_HUMAN</td>\n",
       "      <td>IGHV2-26</td>\n",
       "      <td>119</td>\n",
       "      <td>70.6</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin heavy variable 2-26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Protein   Accession   Entry Name Gene.Names  \\\n",
       "0  sp|A0A075B6I9|LV746_HUMAN  A0A075B6I9  LV746_HUMAN   IGLV7-46   \n",
       "1  sp|A0A075B6P5|KV228_HUMAN  A0A075B6P5  KV228_HUMAN   IGKV2-28   \n",
       "2  sp|A0A087WW87|KV240_HUMAN  A0A087WW87  KV240_HUMAN   IGKV2-40   \n",
       "3  sp|A0A0B4J1V0|HV315_HUMAN  A0A0B4J1V0  HV315_HUMAN   IGHV3-15   \n",
       "4  sp|A0A0B4J1V2|HV226_HUMAN  A0A0B4J1V2  HV226_HUMAN   IGHV2-26   \n",
       "\n",
       "   Protein Length  Coverage                 Protein Existence  \\\n",
       "0             117      76.9  3:Protein inferred from homology   \n",
       "1             120     100.0  3:Protein inferred from homology   \n",
       "2             121      81.0  3:Protein inferred from homology   \n",
       "3             119      84.0  3:Protein inferred from homology   \n",
       "4             119      70.6  3:Protein inferred from homology   \n",
       "\n",
       "                           Description  \n",
       "0  Immunoglobulin lambda variable 7-46  \n",
       "1   Immunoglobulin kappa variable 2-28  \n",
       "2   Immunoglobulin kappa variable 2-40  \n",
       "3   Immunoglobulin heavy variable 3-15  \n",
       "4   Immunoglobulin heavy variable 2-26  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d5a9e0eb-a79b-49fc-9631-6e678ade32ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_info = pd.read_table('/lustre/scratch118/humgen/resources/rna_seq_genomes/Homo_sapiens.GRCh38.99.gtf', sep='\\t', skiprows=5, header=None, low_memory=False)\n",
    "gene_info.columns = ['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attribute']\n",
    "gene_info = gene_info[gene_info.feature == 'gene']\n",
    "gene_info = gene_info.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5d8777b2-9ab9-47f2-a742-3d5258b76a41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>attribute</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>mirbase</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"5\"; g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>mirbase</td>\n",
       "      <td>gene</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000284332\"; gene_version \"1\"; g...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqname   source feature  start    end score strand frame  \\\n",
       "0       1   havana    gene  11869  14409     .      +     .   \n",
       "1       1   havana    gene  14404  29570     .      -     .   \n",
       "2       1  mirbase    gene  17369  17436     .      -     .   \n",
       "3       1   havana    gene  29554  31109     .      +     .   \n",
       "4       1  mirbase    gene  30366  30503     .      +     .   \n",
       "\n",
       "                                           attribute  \n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"5\"; g...  \n",
       "4  gene_id \"ENSG00000284332\"; gene_version \"1\"; g...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "15960a90-0f13-4c84-ab85-f7914d557f97",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_info_attributes = {\n",
    "    'gene_id': list(),\n",
    "    'gene_name': list(),\n",
    "    'gene_biotype': list()\n",
    "}\n",
    "\n",
    "for attribute_str in gene_info.attribute.values:\n",
    "    attributes = [x.split() for x in attribute_str.split(';') if x]\n",
    "    attributes = {x[0]: re.sub('\"', '', x[1]) for x in attributes}\n",
    "    for key in gene_info_attributes:\n",
    "        gene_info_attributes[key].append(attributes[key])\n",
    "\n",
    "gene_info.pop('attribute')\n",
    "\n",
    "gene_info = pd.concat((gene_info, pd.DataFrame(gene_info_attributes)), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "20676182-e9f4-405e-b8e1-e0afdab5cb90",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_info['tss'] = gene_info.apply(lambda x: x.start if x.strand == '+' else x.end, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0ea89bc6-6b7f-4951-a4da-7a4fdb241e60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>11869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>29570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>mirbase</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>17436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>havana</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>29554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>mirbase</td>\n",
       "      <td>gene</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000284332</td>\n",
       "      <td>MIR1302-2</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>30366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqname   source feature  start    end score strand frame          gene_id  \\\n",
       "0       1   havana    gene  11869  14409     .      +     .  ENSG00000223972   \n",
       "1       1   havana    gene  14404  29570     .      -     .  ENSG00000227232   \n",
       "2       1  mirbase    gene  17369  17436     .      -     .  ENSG00000278267   \n",
       "3       1   havana    gene  29554  31109     .      +     .  ENSG00000243485   \n",
       "4       1  mirbase    gene  30366  30503     .      +     .  ENSG00000284332   \n",
       "\n",
       "     gene_name                        gene_biotype    tss  \n",
       "0      DDX11L1  transcribed_unprocessed_pseudogene  11869  \n",
       "1       WASH7P              unprocessed_pseudogene  29570  \n",
       "2    MIR6859-1                               miRNA  17436  \n",
       "3  MIR1302-2HG                              lncRNA  29554  \n",
       "4    MIR1302-2                               miRNA  30366  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_info.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a7ca897-a1a5-4db8-b295-3f192bad715f",
   "metadata": {},
   "source": [
    "## Generate Combined Table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3a79767-0653-473f-a2bc-4fadf895eba1",
   "metadata": {},
   "source": [
    "We only care about the genes on the autosomes and the X chromosome. Filter gene information to these only."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a0e18d7b-6780-4fb8-aa86-47565fc54d8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['1', '2', '3', '4', '5', '6', '7', 'X', '8', '9', '11', '10', '12',\n",
       "       '13', '14', '15', '16', '17', '18', '20', '19', 'Y', '22', '21',\n",
       "       'MT', 'KI270728.1', 'KI270727.1', 'KI270442.1', 'GL000225.1',\n",
       "       'GL000009.2', 'GL000194.1', 'GL000205.2', 'GL000195.1',\n",
       "       'KI270733.1', 'GL000219.1', 'GL000216.2', 'KI270744.1',\n",
       "       'KI270734.1', 'GL000213.1', 'GL000220.1', 'GL000218.1',\n",
       "       'KI270731.1', 'KI270750.1', 'KI270721.1', 'KI270726.1',\n",
       "       'KI270711.1', 'KI270713.1'], dtype=object)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_info.seqname.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0f9a3245-1e7e-4932-8353-15b8125ab2fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosomes = [str(x) for x in range(1, 23)] + ['X']\n",
    "gene_info_auto_X = gene_info[gene_info.seqname.isin(chromosomes)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8b73cff-e2bf-462b-a6a6-39f42d30d855",
   "metadata": {},
   "source": [
    "I will merge the UniProt data with the protein information from Yuxin's MS Proteomics. This will provide Ensembl Gene IDs for each protein. Some proteins are not associated with an Ensembl Gene ID, which I remove. Some proteins are associated with multiple Ensembl Gene IDs (an example being the hemoglobin peptide HBA1, with associated genes HBA1 and HBA2)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "05f05413-55f6-41d5-a65b-ec2dea9909d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_info_all = prot_info.merge(uniprot, left_on='Accession', right_on='UniProt_ID')\n",
    "prot_info_all.Gene_IDs = prot_info_all.Gene_IDs.fillna('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2f6753e4-8cf0-4354-b57c-9d528c99cc03",
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_info_all_genes_expanded = prot_info_all.Gene_IDs.str.split('; ', expand=True)\n",
    "prot_info_all = prot_info_all \\\n",
    "    .join(prot_info_all_genes_expanded) \\\n",
    "    .melt(id_vars=prot_info_all.columns, value_vars=prot_info_all_genes_expanded.columns) \\\n",
    "    .dropna(subset=['value'])\n",
    "\n",
    "del prot_info_all['Gene_IDs']\n",
    "del prot_info_all['variable']\n",
    "prot_info_all = prot_info_all.rename(columns={'value': 'Gene_ID'})\n",
    "prot_info_all = prot_info_all.sort_values(by='Accession').reset_index(drop=True)\n",
    "prot_info_all.Gene_ID = prot_info_all.Gene_ID.replace({'': np.nan})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "d16600f7-5f66-4cf9-9349-c08bec6caecf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene.Names</th>\n",
       "      <th>Protein Length</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Protein Existence</th>\n",
       "      <th>Description</th>\n",
       "      <th>UniProt_ID</th>\n",
       "      <th>Gene_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sp|A0A075B6I9|LV746_HUMAN</td>\n",
       "      <td>A0A075B6I9</td>\n",
       "      <td>LV746_HUMAN</td>\n",
       "      <td>IGLV7-46</td>\n",
       "      <td>117</td>\n",
       "      <td>76.9</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin lambda variable 7-46</td>\n",
       "      <td>A0A075B6I9</td>\n",
       "      <td>ENSG00000211649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sp|A0A075B6P5|KV228_HUMAN</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>KV228_HUMAN</td>\n",
       "      <td>IGKV2-28</td>\n",
       "      <td>120</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-28</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>ENSG00000244116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sp|A0A075B6P5|KV228_HUMAN</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>KV228_HUMAN</td>\n",
       "      <td>IGKV2-28</td>\n",
       "      <td>120</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-28</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>ENSG00000282025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sp|A0A087WSY6|KVD15_HUMAN</td>\n",
       "      <td>A0A087WSY6</td>\n",
       "      <td>KVD15_HUMAN</td>\n",
       "      <td>IGKV3D-15</td>\n",
       "      <td>115</td>\n",
       "      <td>96.5</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 3D-15</td>\n",
       "      <td>A0A087WSY6</td>\n",
       "      <td>ENSG00000224041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sp|A0A087WW87|KV240_HUMAN</td>\n",
       "      <td>A0A087WW87</td>\n",
       "      <td>KV240_HUMAN</td>\n",
       "      <td>IGKV2-40</td>\n",
       "      <td>121</td>\n",
       "      <td>81.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-40</td>\n",
       "      <td>A0A087WW87</td>\n",
       "      <td>ENSG00000273962</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Protein   Accession   Entry Name Gene.Names  \\\n",
       "0  sp|A0A075B6I9|LV746_HUMAN  A0A075B6I9  LV746_HUMAN   IGLV7-46   \n",
       "1  sp|A0A075B6P5|KV228_HUMAN  A0A075B6P5  KV228_HUMAN   IGKV2-28   \n",
       "2  sp|A0A075B6P5|KV228_HUMAN  A0A075B6P5  KV228_HUMAN   IGKV2-28   \n",
       "3  sp|A0A087WSY6|KVD15_HUMAN  A0A087WSY6  KVD15_HUMAN  IGKV3D-15   \n",
       "4  sp|A0A087WW87|KV240_HUMAN  A0A087WW87  KV240_HUMAN   IGKV2-40   \n",
       "\n",
       "   Protein Length  Coverage                 Protein Existence  \\\n",
       "0             117      76.9  3:Protein inferred from homology   \n",
       "1             120     100.0  3:Protein inferred from homology   \n",
       "2             120     100.0  3:Protein inferred from homology   \n",
       "3             115      96.5  3:Protein inferred from homology   \n",
       "4             121      81.0  3:Protein inferred from homology   \n",
       "\n",
       "                           Description  UniProt_ID          Gene_ID  \n",
       "0  Immunoglobulin lambda variable 7-46  A0A075B6I9  ENSG00000211649  \n",
       "1   Immunoglobulin kappa variable 2-28  A0A075B6P5  ENSG00000244116  \n",
       "2   Immunoglobulin kappa variable 2-28  A0A075B6P5  ENSG00000282025  \n",
       "3  Immunoglobulin kappa variable 3D-15  A0A087WSY6  ENSG00000224041  \n",
       "4   Immunoglobulin kappa variable 2-40  A0A087WW87  ENSG00000273962  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_info_all.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c19ccca3-825c-4734-ad49-54f62f95f497",
   "metadata": {},
   "source": [
    "Merge protein information with gene information from the GTF file used for RNA-Seq mapping. This is from Ensembl Version 99 (GRCh38)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "72d5766d-50b5-43cd-a910-82822ba67be2",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = prot_info_all.merge(gene_info, left_on='Gene_ID', right_on='gene_id', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "45cc7669-e2c1-4d35-b6f1-79f4df677baf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene.Names</th>\n",
       "      <th>Protein Length</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Protein Existence</th>\n",
       "      <th>Description</th>\n",
       "      <th>UniProt_ID</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>...</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sp|A0A075B6I9|LV746_HUMAN</td>\n",
       "      <td>A0A075B6I9</td>\n",
       "      <td>LV746_HUMAN</td>\n",
       "      <td>IGLV7-46</td>\n",
       "      <td>117</td>\n",
       "      <td>76.9</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin lambda variable 7-46</td>\n",
       "      <td>A0A075B6I9</td>\n",
       "      <td>ENSG00000211649</td>\n",
       "      <td>...</td>\n",
       "      <td>gene</td>\n",
       "      <td>22369614.0</td>\n",
       "      <td>22370087.0</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000211649</td>\n",
       "      <td>IGLV7-46</td>\n",
       "      <td>IG_V_gene</td>\n",
       "      <td>22369614.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sp|A0A075B6P5|KV228_HUMAN</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>KV228_HUMAN</td>\n",
       "      <td>IGKV2-28</td>\n",
       "      <td>120</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-28</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>ENSG00000244116</td>\n",
       "      <td>...</td>\n",
       "      <td>gene</td>\n",
       "      <td>89221698.0</td>\n",
       "      <td>89222461.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000244116</td>\n",
       "      <td>IGKV2-28</td>\n",
       "      <td>IG_V_gene</td>\n",
       "      <td>89222461.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sp|A0A075B6P5|KV228_HUMAN</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>KV228_HUMAN</td>\n",
       "      <td>IGKV2-28</td>\n",
       "      <td>120</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-28</td>\n",
       "      <td>A0A075B6P5</td>\n",
       "      <td>ENSG00000282025</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sp|A0A087WSY6|KVD15_HUMAN</td>\n",
       "      <td>A0A087WSY6</td>\n",
       "      <td>KVD15_HUMAN</td>\n",
       "      <td>IGKV3D-15</td>\n",
       "      <td>115</td>\n",
       "      <td>96.5</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 3D-15</td>\n",
       "      <td>A0A087WSY6</td>\n",
       "      <td>ENSG00000224041</td>\n",
       "      <td>...</td>\n",
       "      <td>gene</td>\n",
       "      <td>90114838.0</td>\n",
       "      <td>90115402.0</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000224041</td>\n",
       "      <td>IGKV3D-15</td>\n",
       "      <td>IG_V_gene</td>\n",
       "      <td>90114838.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sp|A0A087WW87|KV240_HUMAN</td>\n",
       "      <td>A0A087WW87</td>\n",
       "      <td>KV240_HUMAN</td>\n",
       "      <td>IGKV2-40</td>\n",
       "      <td>121</td>\n",
       "      <td>81.0</td>\n",
       "      <td>3:Protein inferred from homology</td>\n",
       "      <td>Immunoglobulin kappa variable 2-40</td>\n",
       "      <td>A0A087WW87</td>\n",
       "      <td>ENSG00000273962</td>\n",
       "      <td>...</td>\n",
       "      <td>gene</td>\n",
       "      <td>89330110.0</td>\n",
       "      <td>89330429.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000273962</td>\n",
       "      <td>IGKV2-40</td>\n",
       "      <td>IG_V_gene</td>\n",
       "      <td>89330429.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Protein   Accession   Entry Name Gene.Names  \\\n",
       "0  sp|A0A075B6I9|LV746_HUMAN  A0A075B6I9  LV746_HUMAN   IGLV7-46   \n",
       "1  sp|A0A075B6P5|KV228_HUMAN  A0A075B6P5  KV228_HUMAN   IGKV2-28   \n",
       "2  sp|A0A075B6P5|KV228_HUMAN  A0A075B6P5  KV228_HUMAN   IGKV2-28   \n",
       "3  sp|A0A087WSY6|KVD15_HUMAN  A0A087WSY6  KVD15_HUMAN  IGKV3D-15   \n",
       "4  sp|A0A087WW87|KV240_HUMAN  A0A087WW87  KV240_HUMAN   IGKV2-40   \n",
       "\n",
       "   Protein Length  Coverage                 Protein Existence  \\\n",
       "0             117      76.9  3:Protein inferred from homology   \n",
       "1             120     100.0  3:Protein inferred from homology   \n",
       "2             120     100.0  3:Protein inferred from homology   \n",
       "3             115      96.5  3:Protein inferred from homology   \n",
       "4             121      81.0  3:Protein inferred from homology   \n",
       "\n",
       "                           Description  UniProt_ID          Gene_ID  ...  \\\n",
       "0  Immunoglobulin lambda variable 7-46  A0A075B6I9  ENSG00000211649  ...   \n",
       "1   Immunoglobulin kappa variable 2-28  A0A075B6P5  ENSG00000244116  ...   \n",
       "2   Immunoglobulin kappa variable 2-28  A0A075B6P5  ENSG00000282025  ...   \n",
       "3  Immunoglobulin kappa variable 3D-15  A0A087WSY6  ENSG00000224041  ...   \n",
       "4   Immunoglobulin kappa variable 2-40  A0A087WW87  ENSG00000273962  ...   \n",
       "\n",
       "  feature       start         end  score  strand frame          gene_id  \\\n",
       "0    gene  22369614.0  22370087.0      .       +     .  ENSG00000211649   \n",
       "1    gene  89221698.0  89222461.0      .       -     .  ENSG00000244116   \n",
       "2     NaN         NaN         NaN    NaN     NaN   NaN              NaN   \n",
       "3    gene  90114838.0  90115402.0      .       +     .  ENSG00000224041   \n",
       "4    gene  89330110.0  89330429.0      .       -     .  ENSG00000273962   \n",
       "\n",
       "   gene_name gene_biotype         tss  \n",
       "0   IGLV7-46    IG_V_gene  22369614.0  \n",
       "1   IGKV2-28    IG_V_gene  89222461.0  \n",
       "2        NaN          NaN         NaN  \n",
       "3  IGKV3D-15    IG_V_gene  90114838.0  \n",
       "4   IGKV2-40    IG_V_gene  89330429.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "37ff3aa4-88b6-4c46-b30d-fe96b4093f51",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata.to_csv('/nfs/users/nfs_n/nm18/gains_team282/nikhil/colocalization/eQTL_pQTL_metadata.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ba29a4e-5404-47d4-b604-b503b70f2464",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
