{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "44598895-d393-467a-a6c3-ccb29b1aeb14",
   "metadata": {},
   "source": [
    "## QTL Ensembl VEP\n",
    "\n",
    "**Created**: 10 June 2022"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81b8724d-9b37-4a8f-8005-528a9c9cdf8e",
   "metadata": {},
   "source": [
    "## Environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a4868818-8c2a-4531-9e03-762a5e3bfca8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)\n",
    "\n",
    "setwd(\"~/eQTL_pQTL_Characterization/\")\n",
    "\n",
    "source(\"03_Functional_Interpretation/scripts/utils/ggplot_theme.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87749052-952e-4235-9887-af639da85412",
   "metadata": {},
   "source": [
    "## Load Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "760130f2-4f46-4134-9361-e39c6327595e",
   "metadata": {},
   "outputs": [],
   "source": [
    "vep <- fread(\"~/gains_team282/epigenetics/vep/ensembl_vep/qtl_vep_output.tsv\", sep=\"\\t\", na.strings=\"\") %>%\n",
    "    as.data.frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "41859fc9-760a-49d5-8929-40488ad234ea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 110</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHROM</th><th scope=col>POS</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>Allele</th><th scope=col>Consequence</th><th scope=col>IMPACT</th><th scope=col>SYMBOL</th><th scope=col>Gene</th><th scope=col>⋯</th><th scope=col>REVEL</th><th scope=col>SpliceAI_pred_DP_AG</th><th scope=col>SpliceAI_pred_DP_AL</th><th scope=col>SpliceAI_pred_DP_DG</th><th scope=col>SpliceAI_pred_DP_DL</th><th scope=col>SpliceAI_pred_DS_AG</th><th scope=col>SpliceAI_pred_DS_AL</th><th scope=col>SpliceAI_pred_DS_DG</th><th scope=col>SpliceAI_pred_DS_DL</th><th scope=col>SpliceAI_pred_SYMBOL</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>817341</td><td>rs3131972</td><td>G</td><td>A</td><td>A</td><td>upstream_gene_variant                       </td><td>MODIFIER</td><td>FAM87B</td><td>ENSG00000177757</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>817341</td><td>rs3131972</td><td>G</td><td>A</td><td>A</td><td>intron_variant&amp;non_coding_transcript_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>NA    </span></td><td>ENSG00000230092</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>817341</td><td>rs3131972</td><td>G</td><td>A</td><td>A</td><td>intron_variant&amp;non_coding_transcript_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>NA    </span></td><td>ENSG00000230021</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>817341</td><td>rs3131972</td><td>G</td><td>A</td><td>A</td><td>intron_variant&amp;non_coding_transcript_variant</td><td>MODIFIER</td><td><span style=white-space:pre-wrap>NA    </span></td><td>ENSG00000230021</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>817341</td><td>rs3131972</td><td>G</td><td>A</td><td>A</td><td>regulatory_region_variant                   </td><td>MODIFIER</td><td>NA    </td><td>NA             </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>817341</td><td>rs3131972</td><td>G</td><td>A</td><td>A</td><td>TF_binding_site_variant                     </td><td>MODIFIER</td><td>NA    </td><td>NA             </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 110\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & CHROM & POS & ID & REF & ALT & Allele & Consequence & IMPACT & SYMBOL & Gene & ⋯ & REVEL & SpliceAI\\_pred\\_DP\\_AG & SpliceAI\\_pred\\_DP\\_AL & SpliceAI\\_pred\\_DP\\_DG & SpliceAI\\_pred\\_DP\\_DL & SpliceAI\\_pred\\_DS\\_AG & SpliceAI\\_pred\\_DS\\_AL & SpliceAI\\_pred\\_DS\\_DG & SpliceAI\\_pred\\_DS\\_DL & SpliceAI\\_pred\\_SYMBOL\\\\\n",
       "  & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <dbl> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & 817341 & rs3131972 & G & A & A & upstream\\_gene\\_variant                        & MODIFIER & FAM87B & ENSG00000177757 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t2 & 1 & 817341 & rs3131972 & G & A & A & intron\\_variant\\&non\\_coding\\_transcript\\_variant & MODIFIER & NA     & ENSG00000230092 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t3 & 1 & 817341 & rs3131972 & G & A & A & intron\\_variant\\&non\\_coding\\_transcript\\_variant & MODIFIER & NA     & ENSG00000230021 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t4 & 1 & 817341 & rs3131972 & G & A & A & intron\\_variant\\&non\\_coding\\_transcript\\_variant & MODIFIER & NA     & ENSG00000230021 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t5 & 1 & 817341 & rs3131972 & G & A & A & regulatory\\_region\\_variant                    & MODIFIER & NA     & NA              & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t6 & 1 & 817341 & rs3131972 & G & A & A & TF\\_binding\\_site\\_variant                      & MODIFIER & NA     & NA              & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 110\n",
       "\n",
       "| <!--/--> | CHROM &lt;int&gt; | POS &lt;int&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | Allele &lt;chr&gt; | Consequence &lt;chr&gt; | IMPACT &lt;chr&gt; | SYMBOL &lt;chr&gt; | Gene &lt;chr&gt; | ⋯ ⋯ | REVEL &lt;dbl&gt; | SpliceAI_pred_DP_AG &lt;int&gt; | SpliceAI_pred_DP_AL &lt;int&gt; | SpliceAI_pred_DP_DG &lt;int&gt; | SpliceAI_pred_DP_DL &lt;int&gt; | SpliceAI_pred_DS_AG &lt;dbl&gt; | SpliceAI_pred_DS_AL &lt;dbl&gt; | SpliceAI_pred_DS_DG &lt;dbl&gt; | SpliceAI_pred_DS_DL &lt;dbl&gt; | SpliceAI_pred_SYMBOL &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | 817341 | rs3131972 | G | A | A | upstream_gene_variant                        | MODIFIER | FAM87B | ENSG00000177757 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 2 | 1 | 817341 | rs3131972 | G | A | A | intron_variant&amp;non_coding_transcript_variant | MODIFIER | NA     | ENSG00000230092 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 3 | 1 | 817341 | rs3131972 | G | A | A | intron_variant&amp;non_coding_transcript_variant | MODIFIER | NA     | ENSG00000230021 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 4 | 1 | 817341 | rs3131972 | G | A | A | intron_variant&amp;non_coding_transcript_variant | MODIFIER | NA     | ENSG00000230021 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 5 | 1 | 817341 | rs3131972 | G | A | A | regulatory_region_variant                    | MODIFIER | NA     | NA              | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 6 | 1 | 817341 | rs3131972 | G | A | A | TF_binding_site_variant                      | MODIFIER | NA     | NA              | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  CHROM POS    ID        REF ALT Allele\n",
       "1 1     817341 rs3131972 G   A   A     \n",
       "2 1     817341 rs3131972 G   A   A     \n",
       "3 1     817341 rs3131972 G   A   A     \n",
       "4 1     817341 rs3131972 G   A   A     \n",
       "5 1     817341 rs3131972 G   A   A     \n",
       "6 1     817341 rs3131972 G   A   A     \n",
       "  Consequence                                  IMPACT   SYMBOL Gene           \n",
       "1 upstream_gene_variant                        MODIFIER FAM87B ENSG00000177757\n",
       "2 intron_variant&non_coding_transcript_variant MODIFIER NA     ENSG00000230092\n",
       "3 intron_variant&non_coding_transcript_variant MODIFIER NA     ENSG00000230021\n",
       "4 intron_variant&non_coding_transcript_variant MODIFIER NA     ENSG00000230021\n",
       "5 regulatory_region_variant                    MODIFIER NA     NA             \n",
       "6 TF_binding_site_variant                      MODIFIER NA     NA             \n",
       "  ⋯ REVEL SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL SpliceAI_pred_DP_DG\n",
       "1 ⋯ NA    NA                  NA                  NA                 \n",
       "2 ⋯ NA    NA                  NA                  NA                 \n",
       "3 ⋯ NA    NA                  NA                  NA                 \n",
       "4 ⋯ NA    NA                  NA                  NA                 \n",
       "5 ⋯ NA    NA                  NA                  NA                 \n",
       "6 ⋯ NA    NA                  NA                  NA                 \n",
       "  SpliceAI_pred_DP_DL SpliceAI_pred_DS_AG SpliceAI_pred_DS_AL\n",
       "1 NA                  NA                  NA                 \n",
       "2 NA                  NA                  NA                 \n",
       "3 NA                  NA                  NA                 \n",
       "4 NA                  NA                  NA                 \n",
       "5 NA                  NA                  NA                 \n",
       "6 NA                  NA                  NA                 \n",
       "  SpliceAI_pred_DS_DG SpliceAI_pred_DS_DL SpliceAI_pred_SYMBOL\n",
       "1 NA                  NA                  NA                  \n",
       "2 NA                  NA                  NA                  \n",
       "3 NA                  NA                  NA                  \n",
       "4 NA                  NA                  NA                  \n",
       "5 NA                  NA                  NA                  \n",
       "6 NA                  NA                  NA                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(vep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d12949a7-f5d4-436a-b3af-248e4946a6c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CHROM'</li><li>'POS'</li><li>'ID'</li><li>'REF'</li><li>'ALT'</li><li>'Allele'</li><li>'Consequence'</li><li>'IMPACT'</li><li>'SYMBOL'</li><li>'Gene'</li><li>'Feature_type'</li><li>'Feature'</li><li>'BIOTYPE'</li><li>'EXON'</li><li>'INTRON'</li><li>'HGVSc'</li><li>'HGVSp'</li><li>'cDNA_position'</li><li>'CDS_position'</li><li>'Protein_position'</li><li>'Amino_acids'</li><li>'Codons'</li><li>'Existing_variation'</li><li>'ALLELE_NUM'</li><li>'DISTANCE'</li><li>'STRAND'</li><li>'FLAGS'</li><li>'VARIANT_CLASS'</li><li>'SYMBOL_SOURCE'</li><li>'HGNC_ID'</li><li>'CANONICAL'</li><li>'MANE_SELECT'</li><li>'MANE_PLUS_CLINICAL'</li><li>'TSL'</li><li>'APPRIS'</li><li>'CCDS'</li><li>'ENSP'</li><li>'SWISSPROT'</li><li>'TREMBL'</li><li>'UNIPARC'</li><li>'UNIPROT_ISOFORM'</li><li>'GENE_PHENO'</li><li>'SIFT'</li><li>'PolyPhen'</li><li>'DOMAINS'</li><li>'miRNA'</li><li>'HGVS_OFFSET'</li><li>'AF'</li><li>'AFR_AF'</li><li>'AMR_AF'</li><li>'EAS_AF'</li><li>'EUR_AF'</li><li>'SAS_AF'</li><li>'AA_AF'</li><li>'EA_AF'</li><li>'gnomAD_AF'</li><li>'gnomAD_AFR_AF'</li><li>'gnomAD_AMR_AF'</li><li>'gnomAD_ASJ_AF'</li><li>'gnomAD_EAS_AF'</li><li>'gnomAD_FIN_AF'</li><li>'gnomAD_NFE_AF'</li><li>'gnomAD_OTH_AF'</li><li>'gnomAD_SAS_AF'</li><li>'MAX_AF'</li><li>'MAX_AF_POPS'</li><li>'CLIN_SIG'</li><li>'SOMATIC'</li><li>'PHENO'</li><li>'PUBMED'</li><li>'MOTIF_NAME'</li><li>'MOTIF_POS'</li><li>'HIGH_INF_POS'</li><li>'MOTIF_SCORE_CHANGE'</li><li>'TRANSCRIPTION_FACTORS'</li><li>'SpliceRegion'</li><li>'GeneSplicer'</li><li>'existing_InFrame_oORFs'</li><li>'existing_OutOfFrame_oORFs'</li><li>'existing_uORFs'</li><li>'five_prime_UTR_variant_annotation'</li><li>'five_prime_UTR_variant_consequence'</li><li>'CADD_PHRED'</li><li>'CADD_RAW'</li><li>'Ensembl_transcriptid'</li><li>'LRT_pred'</li><li>'MutationTaster_pred'</li><li>'Polyphen2_HDIV_pred'</li><li>'Polyphen2_HVAR_pred'</li><li>'SIFT_pred'</li><li>'Uniprot_acc'</li><li>'VEP_canonical'</li><li>'DisGeNET_PMID'</li><li>'DisGeNET_SCORE'</li><li>'PHENOTYPES'</li><li>'Conservation'</li><li>'LoF'</li><li>'LoF_filter'</li><li>'LoF_flags'</li><li>'LoF_info'</li><li>'REVEL'</li><li>'SpliceAI_pred_DP_AG'</li><li>'SpliceAI_pred_DP_AL'</li><li>'SpliceAI_pred_DP_DG'</li><li>'SpliceAI_pred_DP_DL'</li><li>'SpliceAI_pred_DS_AG'</li><li>'SpliceAI_pred_DS_AL'</li><li>'SpliceAI_pred_DS_DG'</li><li>'SpliceAI_pred_DS_DL'</li><li>'SpliceAI_pred_SYMBOL'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CHROM'\n",
       "\\item 'POS'\n",
       "\\item 'ID'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'Allele'\n",
       "\\item 'Consequence'\n",
       "\\item 'IMPACT'\n",
       "\\item 'SYMBOL'\n",
       "\\item 'Gene'\n",
       "\\item 'Feature\\_type'\n",
       "\\item 'Feature'\n",
       "\\item 'BIOTYPE'\n",
       "\\item 'EXON'\n",
       "\\item 'INTRON'\n",
       "\\item 'HGVSc'\n",
       "\\item 'HGVSp'\n",
       "\\item 'cDNA\\_position'\n",
       "\\item 'CDS\\_position'\n",
       "\\item 'Protein\\_position'\n",
       "\\item 'Amino\\_acids'\n",
       "\\item 'Codons'\n",
       "\\item 'Existing\\_variation'\n",
       "\\item 'ALLELE\\_NUM'\n",
       "\\item 'DISTANCE'\n",
       "\\item 'STRAND'\n",
       "\\item 'FLAGS'\n",
       "\\item 'VARIANT\\_CLASS'\n",
       "\\item 'SYMBOL\\_SOURCE'\n",
       "\\item 'HGNC\\_ID'\n",
       "\\item 'CANONICAL'\n",
       "\\item 'MANE\\_SELECT'\n",
       "\\item 'MANE\\_PLUS\\_CLINICAL'\n",
       "\\item 'TSL'\n",
       "\\item 'APPRIS'\n",
       "\\item 'CCDS'\n",
       "\\item 'ENSP'\n",
       "\\item 'SWISSPROT'\n",
       "\\item 'TREMBL'\n",
       "\\item 'UNIPARC'\n",
       "\\item 'UNIPROT\\_ISOFORM'\n",
       "\\item 'GENE\\_PHENO'\n",
       "\\item 'SIFT'\n",
       "\\item 'PolyPhen'\n",
       "\\item 'DOMAINS'\n",
       "\\item 'miRNA'\n",
       "\\item 'HGVS\\_OFFSET'\n",
       "\\item 'AF'\n",
       "\\item 'AFR\\_AF'\n",
       "\\item 'AMR\\_AF'\n",
       "\\item 'EAS\\_AF'\n",
       "\\item 'EUR\\_AF'\n",
       "\\item 'SAS\\_AF'\n",
       "\\item 'AA\\_AF'\n",
       "\\item 'EA\\_AF'\n",
       "\\item 'gnomAD\\_AF'\n",
       "\\item 'gnomAD\\_AFR\\_AF'\n",
       "\\item 'gnomAD\\_AMR\\_AF'\n",
       "\\item 'gnomAD\\_ASJ\\_AF'\n",
       "\\item 'gnomAD\\_EAS\\_AF'\n",
       "\\item 'gnomAD\\_FIN\\_AF'\n",
       "\\item 'gnomAD\\_NFE\\_AF'\n",
       "\\item 'gnomAD\\_OTH\\_AF'\n",
       "\\item 'gnomAD\\_SAS\\_AF'\n",
       "\\item 'MAX\\_AF'\n",
       "\\item 'MAX\\_AF\\_POPS'\n",
       "\\item 'CLIN\\_SIG'\n",
       "\\item 'SOMATIC'\n",
       "\\item 'PHENO'\n",
       "\\item 'PUBMED'\n",
       "\\item 'MOTIF\\_NAME'\n",
       "\\item 'MOTIF\\_POS'\n",
       "\\item 'HIGH\\_INF\\_POS'\n",
       "\\item 'MOTIF\\_SCORE\\_CHANGE'\n",
       "\\item 'TRANSCRIPTION\\_FACTORS'\n",
       "\\item 'SpliceRegion'\n",
       "\\item 'GeneSplicer'\n",
       "\\item 'existing\\_InFrame\\_oORFs'\n",
       "\\item 'existing\\_OutOfFrame\\_oORFs'\n",
       "\\item 'existing\\_uORFs'\n",
       "\\item 'five\\_prime\\_UTR\\_variant\\_annotation'\n",
       "\\item 'five\\_prime\\_UTR\\_variant\\_consequence'\n",
       "\\item 'CADD\\_PHRED'\n",
       "\\item 'CADD\\_RAW'\n",
       "\\item 'Ensembl\\_transcriptid'\n",
       "\\item 'LRT\\_pred'\n",
       "\\item 'MutationTaster\\_pred'\n",
       "\\item 'Polyphen2\\_HDIV\\_pred'\n",
       "\\item 'Polyphen2\\_HVAR\\_pred'\n",
       "\\item 'SIFT\\_pred'\n",
       "\\item 'Uniprot\\_acc'\n",
       "\\item 'VEP\\_canonical'\n",
       "\\item 'DisGeNET\\_PMID'\n",
       "\\item 'DisGeNET\\_SCORE'\n",
       "\\item 'PHENOTYPES'\n",
       "\\item 'Conservation'\n",
       "\\item 'LoF'\n",
       "\\item 'LoF\\_filter'\n",
       "\\item 'LoF\\_flags'\n",
       "\\item 'LoF\\_info'\n",
       "\\item 'REVEL'\n",
       "\\item 'SpliceAI\\_pred\\_DP\\_AG'\n",
       "\\item 'SpliceAI\\_pred\\_DP\\_AL'\n",
       "\\item 'SpliceAI\\_pred\\_DP\\_DG'\n",
       "\\item 'SpliceAI\\_pred\\_DP\\_DL'\n",
       "\\item 'SpliceAI\\_pred\\_DS\\_AG'\n",
       "\\item 'SpliceAI\\_pred\\_DS\\_AL'\n",
       "\\item 'SpliceAI\\_pred\\_DS\\_DG'\n",
       "\\item 'SpliceAI\\_pred\\_DS\\_DL'\n",
       "\\item 'SpliceAI\\_pred\\_SYMBOL'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CHROM'\n",
       "2. 'POS'\n",
       "3. 'ID'\n",
       "4. 'REF'\n",
       "5. 'ALT'\n",
       "6. 'Allele'\n",
       "7. 'Consequence'\n",
       "8. 'IMPACT'\n",
       "9. 'SYMBOL'\n",
       "10. 'Gene'\n",
       "11. 'Feature_type'\n",
       "12. 'Feature'\n",
       "13. 'BIOTYPE'\n",
       "14. 'EXON'\n",
       "15. 'INTRON'\n",
       "16. 'HGVSc'\n",
       "17. 'HGVSp'\n",
       "18. 'cDNA_position'\n",
       "19. 'CDS_position'\n",
       "20. 'Protein_position'\n",
       "21. 'Amino_acids'\n",
       "22. 'Codons'\n",
       "23. 'Existing_variation'\n",
       "24. 'ALLELE_NUM'\n",
       "25. 'DISTANCE'\n",
       "26. 'STRAND'\n",
       "27. 'FLAGS'\n",
       "28. 'VARIANT_CLASS'\n",
       "29. 'SYMBOL_SOURCE'\n",
       "30. 'HGNC_ID'\n",
       "31. 'CANONICAL'\n",
       "32. 'MANE_SELECT'\n",
       "33. 'MANE_PLUS_CLINICAL'\n",
       "34. 'TSL'\n",
       "35. 'APPRIS'\n",
       "36. 'CCDS'\n",
       "37. 'ENSP'\n",
       "38. 'SWISSPROT'\n",
       "39. 'TREMBL'\n",
       "40. 'UNIPARC'\n",
       "41. 'UNIPROT_ISOFORM'\n",
       "42. 'GENE_PHENO'\n",
       "43. 'SIFT'\n",
       "44. 'PolyPhen'\n",
       "45. 'DOMAINS'\n",
       "46. 'miRNA'\n",
       "47. 'HGVS_OFFSET'\n",
       "48. 'AF'\n",
       "49. 'AFR_AF'\n",
       "50. 'AMR_AF'\n",
       "51. 'EAS_AF'\n",
       "52. 'EUR_AF'\n",
       "53. 'SAS_AF'\n",
       "54. 'AA_AF'\n",
       "55. 'EA_AF'\n",
       "56. 'gnomAD_AF'\n",
       "57. 'gnomAD_AFR_AF'\n",
       "58. 'gnomAD_AMR_AF'\n",
       "59. 'gnomAD_ASJ_AF'\n",
       "60. 'gnomAD_EAS_AF'\n",
       "61. 'gnomAD_FIN_AF'\n",
       "62. 'gnomAD_NFE_AF'\n",
       "63. 'gnomAD_OTH_AF'\n",
       "64. 'gnomAD_SAS_AF'\n",
       "65. 'MAX_AF'\n",
       "66. 'MAX_AF_POPS'\n",
       "67. 'CLIN_SIG'\n",
       "68. 'SOMATIC'\n",
       "69. 'PHENO'\n",
       "70. 'PUBMED'\n",
       "71. 'MOTIF_NAME'\n",
       "72. 'MOTIF_POS'\n",
       "73. 'HIGH_INF_POS'\n",
       "74. 'MOTIF_SCORE_CHANGE'\n",
       "75. 'TRANSCRIPTION_FACTORS'\n",
       "76. 'SpliceRegion'\n",
       "77. 'GeneSplicer'\n",
       "78. 'existing_InFrame_oORFs'\n",
       "79. 'existing_OutOfFrame_oORFs'\n",
       "80. 'existing_uORFs'\n",
       "81. 'five_prime_UTR_variant_annotation'\n",
       "82. 'five_prime_UTR_variant_consequence'\n",
       "83. 'CADD_PHRED'\n",
       "84. 'CADD_RAW'\n",
       "85. 'Ensembl_transcriptid'\n",
       "86. 'LRT_pred'\n",
       "87. 'MutationTaster_pred'\n",
       "88. 'Polyphen2_HDIV_pred'\n",
       "89. 'Polyphen2_HVAR_pred'\n",
       "90. 'SIFT_pred'\n",
       "91. 'Uniprot_acc'\n",
       "92. 'VEP_canonical'\n",
       "93. 'DisGeNET_PMID'\n",
       "94. 'DisGeNET_SCORE'\n",
       "95. 'PHENOTYPES'\n",
       "96. 'Conservation'\n",
       "97. 'LoF'\n",
       "98. 'LoF_filter'\n",
       "99. 'LoF_flags'\n",
       "100. 'LoF_info'\n",
       "101. 'REVEL'\n",
       "102. 'SpliceAI_pred_DP_AG'\n",
       "103. 'SpliceAI_pred_DP_AL'\n",
       "104. 'SpliceAI_pred_DP_DG'\n",
       "105. 'SpliceAI_pred_DP_DL'\n",
       "106. 'SpliceAI_pred_DS_AG'\n",
       "107. 'SpliceAI_pred_DS_AL'\n",
       "108. 'SpliceAI_pred_DS_DG'\n",
       "109. 'SpliceAI_pred_DS_DL'\n",
       "110. 'SpliceAI_pred_SYMBOL'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"CHROM\"                              \"POS\"                               \n",
       "  [3] \"ID\"                                 \"REF\"                               \n",
       "  [5] \"ALT\"                                \"Allele\"                            \n",
       "  [7] \"Consequence\"                        \"IMPACT\"                            \n",
       "  [9] \"SYMBOL\"                             \"Gene\"                              \n",
       " [11] \"Feature_type\"                       \"Feature\"                           \n",
       " [13] \"BIOTYPE\"                            \"EXON\"                              \n",
       " [15] \"INTRON\"                             \"HGVSc\"                             \n",
       " [17] \"HGVSp\"                              \"cDNA_position\"                     \n",
       " [19] \"CDS_position\"                       \"Protein_position\"                  \n",
       " [21] \"Amino_acids\"                        \"Codons\"                            \n",
       " [23] \"Existing_variation\"                 \"ALLELE_NUM\"                        \n",
       " [25] \"DISTANCE\"                           \"STRAND\"                            \n",
       " [27] \"FLAGS\"                              \"VARIANT_CLASS\"                     \n",
       " [29] \"SYMBOL_SOURCE\"                      \"HGNC_ID\"                           \n",
       " [31] \"CANONICAL\"                          \"MANE_SELECT\"                       \n",
       " [33] \"MANE_PLUS_CLINICAL\"                 \"TSL\"                               \n",
       " [35] \"APPRIS\"                             \"CCDS\"                              \n",
       " [37] \"ENSP\"                               \"SWISSPROT\"                         \n",
       " [39] \"TREMBL\"                             \"UNIPARC\"                           \n",
       " [41] \"UNIPROT_ISOFORM\"                    \"GENE_PHENO\"                        \n",
       " [43] \"SIFT\"                               \"PolyPhen\"                          \n",
       " [45] \"DOMAINS\"                            \"miRNA\"                             \n",
       " [47] \"HGVS_OFFSET\"                        \"AF\"                                \n",
       " [49] \"AFR_AF\"                             \"AMR_AF\"                            \n",
       " [51] \"EAS_AF\"                             \"EUR_AF\"                            \n",
       " [53] \"SAS_AF\"                             \"AA_AF\"                             \n",
       " [55] \"EA_AF\"                              \"gnomAD_AF\"                         \n",
       " [57] \"gnomAD_AFR_AF\"                      \"gnomAD_AMR_AF\"                     \n",
       " [59] \"gnomAD_ASJ_AF\"                      \"gnomAD_EAS_AF\"                     \n",
       " [61] \"gnomAD_FIN_AF\"                      \"gnomAD_NFE_AF\"                     \n",
       " [63] \"gnomAD_OTH_AF\"                      \"gnomAD_SAS_AF\"                     \n",
       " [65] \"MAX_AF\"                             \"MAX_AF_POPS\"                       \n",
       " [67] \"CLIN_SIG\"                           \"SOMATIC\"                           \n",
       " [69] \"PHENO\"                              \"PUBMED\"                            \n",
       " [71] \"MOTIF_NAME\"                         \"MOTIF_POS\"                         \n",
       " [73] \"HIGH_INF_POS\"                       \"MOTIF_SCORE_CHANGE\"                \n",
       " [75] \"TRANSCRIPTION_FACTORS\"              \"SpliceRegion\"                      \n",
       " [77] \"GeneSplicer\"                        \"existing_InFrame_oORFs\"            \n",
       " [79] \"existing_OutOfFrame_oORFs\"          \"existing_uORFs\"                    \n",
       " [81] \"five_prime_UTR_variant_annotation\"  \"five_prime_UTR_variant_consequence\"\n",
       " [83] \"CADD_PHRED\"                         \"CADD_RAW\"                          \n",
       " [85] \"Ensembl_transcriptid\"               \"LRT_pred\"                          \n",
       " [87] \"MutationTaster_pred\"                \"Polyphen2_HDIV_pred\"               \n",
       " [89] \"Polyphen2_HVAR_pred\"                \"SIFT_pred\"                         \n",
       " [91] \"Uniprot_acc\"                        \"VEP_canonical\"                     \n",
       " [93] \"DisGeNET_PMID\"                      \"DisGeNET_SCORE\"                    \n",
       " [95] \"PHENOTYPES\"                         \"Conservation\"                      \n",
       " [97] \"LoF\"                                \"LoF_filter\"                        \n",
       " [99] \"LoF_flags\"                          \"LoF_info\"                          \n",
       "[101] \"REVEL\"                              \"SpliceAI_pred_DP_AG\"               \n",
       "[103] \"SpliceAI_pred_DP_AL\"                \"SpliceAI_pred_DP_DG\"               \n",
       "[105] \"SpliceAI_pred_DP_DL\"                \"SpliceAI_pred_DS_AG\"               \n",
       "[107] \"SpliceAI_pred_DS_AL\"                \"SpliceAI_pred_DS_DG\"               \n",
       "[109] \"SpliceAI_pred_DS_DL\"                \"SpliceAI_pred_SYMBOL\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(vep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "cc2687b0-3489-4a31-9c2a-075142853248",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene.info <- read.table(\"/nfs/team282/data/gains_team282/gene_info_864_20412_hla.txt\") %>%\n",
    "    dplyr::select(gene_id, gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "e225999a-007c-4524-81ad-c113ba3c367a",
   "metadata": {},
   "outputs": [],
   "source": [
    "cis.eqtl <- read.table(\"~/gains_team282/eqtl/cisresults/eigenMT/ciseqtl_eigenMT_corrected.txt\") %>%\n",
    "    dplyr::filter(Sig) %>%\n",
    "    merge(gene.info, ., by.x=\"gene_id\", by.y=\"gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "bac08a45-bf29-4b2b-b6cb-1b444a6bb1c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene_id</th><th scope=col>gene_name</th><th scope=col>snps</th><th scope=col>statistic</th><th scope=col>pvalue</th><th scope=col>beta</th><th scope=col>se</th><th scope=col>chr</th><th scope=col>SNPpos</th><th scope=col>TSS</th><th scope=col>BF</th><th scope=col>TESTS</th><th scope=col>BF.FDR</th><th scope=col>Sig</th><th scope=col>threshold</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000000457</td><td>SCYL3   </td><td>rs10753794</td><td> 7.037066</td><td>5.192595e-12</td><td> 0.04985873</td><td>0.007085159</td><td>1</td><td>169776143</td><td>169894267</td><td>3.608853e-09</td><td>695</td><td>1.308708e-08</td><td>TRUE</td><td>3.879445e-05</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000000460</td><td>C1orf112</td><td>rs12410806</td><td>-5.942351</td><td>5.426681e-09</td><td>-0.07212698</td><td>0.012137784</td><td>1</td><td>169749095</td><td>169662007</td><td>4.032024e-06</td><td>743</td><td>1.160666e-05</td><td>TRUE</td><td>3.628822e-05</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000000971</td><td>CFH     </td><td>rs12406047</td><td> 4.511592</td><td>7.515235e-06</td><td> 0.17135374</td><td>0.037980769</td><td>1</td><td>196708768</td><td>196652043</td><td>2.818213e-03</td><td>375</td><td>6.073985e-03</td><td>TRUE</td><td>7.189906e-05</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000001036</td><td>FUCA2   </td><td>rs2496591 </td><td> 5.771056</td><td>1.272071e-08</td><td> 0.05556388</td><td>0.009628027</td><td>6</td><td>143416006</td><td>143511720</td><td>1.017657e-05</td><td>800</td><td>2.830516e-05</td><td>TRUE</td><td>3.370268e-05</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000001167</td><td>NFYA    </td><td>rs72856259</td><td> 7.552123</td><td>1.978551e-13</td><td> 0.11980933</td><td>0.015864325</td><td>6</td><td> 41100761</td><td> 41072945</td><td>1.889516e-10</td><td>955</td><td>7.447584e-10</td><td>TRUE</td><td>2.823261e-05</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000001460</td><td>STPG1   </td><td>rs6687210 </td><td> 5.266595</td><td>1.949343e-07</td><td> 0.09241162</td><td>0.017546749</td><td>1</td><td> 24405200</td><td> 24416934</td><td>9.668744e-05</td><td>496</td><td>2.450046e-04</td><td>TRUE</td><td>5.435916e-05</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & gene\\_id & gene\\_name & snps & statistic & pvalue & beta & se & chr & SNPpos & TSS & BF & TESTS & BF.FDR & Sig & threshold\\\\\n",
       "  & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <dbl> & <int> & <dbl> & <lgl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000000457 & SCYL3    & rs10753794 &  7.037066 & 5.192595e-12 &  0.04985873 & 0.007085159 & 1 & 169776143 & 169894267 & 3.608853e-09 & 695 & 1.308708e-08 & TRUE & 3.879445e-05\\\\\n",
       "\t2 & ENSG00000000460 & C1orf112 & rs12410806 & -5.942351 & 5.426681e-09 & -0.07212698 & 0.012137784 & 1 & 169749095 & 169662007 & 4.032024e-06 & 743 & 1.160666e-05 & TRUE & 3.628822e-05\\\\\n",
       "\t3 & ENSG00000000971 & CFH      & rs12406047 &  4.511592 & 7.515235e-06 &  0.17135374 & 0.037980769 & 1 & 196708768 & 196652043 & 2.818213e-03 & 375 & 6.073985e-03 & TRUE & 7.189906e-05\\\\\n",
       "\t4 & ENSG00000001036 & FUCA2    & rs2496591  &  5.771056 & 1.272071e-08 &  0.05556388 & 0.009628027 & 6 & 143416006 & 143511720 & 1.017657e-05 & 800 & 2.830516e-05 & TRUE & 3.370268e-05\\\\\n",
       "\t5 & ENSG00000001167 & NFYA     & rs72856259 &  7.552123 & 1.978551e-13 &  0.11980933 & 0.015864325 & 6 &  41100761 &  41072945 & 1.889516e-10 & 955 & 7.447584e-10 & TRUE & 2.823261e-05\\\\\n",
       "\t6 & ENSG00000001460 & STPG1    & rs6687210  &  5.266595 & 1.949343e-07 &  0.09241162 & 0.017546749 & 1 &  24405200 &  24416934 & 9.668744e-05 & 496 & 2.450046e-04 & TRUE & 5.435916e-05\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | gene_id &lt;chr&gt; | gene_name &lt;chr&gt; | snps &lt;chr&gt; | statistic &lt;dbl&gt; | pvalue &lt;dbl&gt; | beta &lt;dbl&gt; | se &lt;dbl&gt; | chr &lt;int&gt; | SNPpos &lt;int&gt; | TSS &lt;int&gt; | BF &lt;dbl&gt; | TESTS &lt;int&gt; | BF.FDR &lt;dbl&gt; | Sig &lt;lgl&gt; | threshold &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000000457 | SCYL3    | rs10753794 |  7.037066 | 5.192595e-12 |  0.04985873 | 0.007085159 | 1 | 169776143 | 169894267 | 3.608853e-09 | 695 | 1.308708e-08 | TRUE | 3.879445e-05 |\n",
       "| 2 | ENSG00000000460 | C1orf112 | rs12410806 | -5.942351 | 5.426681e-09 | -0.07212698 | 0.012137784 | 1 | 169749095 | 169662007 | 4.032024e-06 | 743 | 1.160666e-05 | TRUE | 3.628822e-05 |\n",
       "| 3 | ENSG00000000971 | CFH      | rs12406047 |  4.511592 | 7.515235e-06 |  0.17135374 | 0.037980769 | 1 | 196708768 | 196652043 | 2.818213e-03 | 375 | 6.073985e-03 | TRUE | 7.189906e-05 |\n",
       "| 4 | ENSG00000001036 | FUCA2    | rs2496591  |  5.771056 | 1.272071e-08 |  0.05556388 | 0.009628027 | 6 | 143416006 | 143511720 | 1.017657e-05 | 800 | 2.830516e-05 | TRUE | 3.370268e-05 |\n",
       "| 5 | ENSG00000001167 | NFYA     | rs72856259 |  7.552123 | 1.978551e-13 |  0.11980933 | 0.015864325 | 6 |  41100761 |  41072945 | 1.889516e-10 | 955 | 7.447584e-10 | TRUE | 2.823261e-05 |\n",
       "| 6 | ENSG00000001460 | STPG1    | rs6687210  |  5.266595 | 1.949343e-07 |  0.09241162 | 0.017546749 | 1 |  24405200 |  24416934 | 9.668744e-05 | 496 | 2.450046e-04 | TRUE | 5.435916e-05 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene_id         gene_name snps       statistic pvalue       beta       \n",
       "1 ENSG00000000457 SCYL3     rs10753794  7.037066 5.192595e-12  0.04985873\n",
       "2 ENSG00000000460 C1orf112  rs12410806 -5.942351 5.426681e-09 -0.07212698\n",
       "3 ENSG00000000971 CFH       rs12406047  4.511592 7.515235e-06  0.17135374\n",
       "4 ENSG00000001036 FUCA2     rs2496591   5.771056 1.272071e-08  0.05556388\n",
       "5 ENSG00000001167 NFYA      rs72856259  7.552123 1.978551e-13  0.11980933\n",
       "6 ENSG00000001460 STPG1     rs6687210   5.266595 1.949343e-07  0.09241162\n",
       "  se          chr SNPpos    TSS       BF           TESTS BF.FDR       Sig \n",
       "1 0.007085159 1   169776143 169894267 3.608853e-09 695   1.308708e-08 TRUE\n",
       "2 0.012137784 1   169749095 169662007 4.032024e-06 743   1.160666e-05 TRUE\n",
       "3 0.037980769 1   196708768 196652043 2.818213e-03 375   6.073985e-03 TRUE\n",
       "4 0.009628027 6   143416006 143511720 1.017657e-05 800   2.830516e-05 TRUE\n",
       "5 0.015864325 6    41100761  41072945 1.889516e-10 955   7.447584e-10 TRUE\n",
       "6 0.017546749 1    24405200  24416934 9.668744e-05 496   2.450046e-04 TRUE\n",
       "  threshold   \n",
       "1 3.879445e-05\n",
       "2 3.628822e-05\n",
       "3 7.189906e-05\n",
       "4 3.370268e-05\n",
       "5 2.823261e-05\n",
       "6 5.435916e-05"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cis.eqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a8519925-93b7-4fcd-a79a-361585c30069",
   "metadata": {},
   "outputs": [],
   "source": [
    "mqtl <- readRDS(\"~/gains_team282/nikhil/expression/eigengene_sva/mqtl.RDS\") %>%\n",
    "    dplyr::group_by(module, qtl.locus) %>%\n",
    "    dplyr::slice_min(p, n=1, with_ties=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "24c8672a-3c22-4b9e-a174-defa3fad9ab3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>snp</th><th scope=col>beta</th><th scope=col>se</th><th scope=col>t</th><th scope=col>p</th><th scope=col>module.qtl</th><th scope=col>module</th><th scope=col>pc</th><th scope=col>qtl.locus</th><th scope=col>qtl.locus.chr</th><th scope=col>qtl.locus.start</th><th scope=col>qtl.locus.end</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>rs16866400</td><td>-0.01081391</td><td>0.0016948840</td><td> -6.380325</td><td> 3.197894e-10</td><td>ME_100_4-2:177556567-179584183</td><td>ME_100</td><td>4</td><td>2:177556567-179584183</td><td>2 </td><td>177556567</td><td>179584183</td></tr>\n",
       "\t<tr><td>rs582717  </td><td>-0.02334898</td><td>0.0014724757</td><td>-15.856953</td><td> 7.745374e-48</td><td>ME_101_2-1:24243590-26454176  </td><td>ME_101</td><td>2</td><td>1:24243590-26454176  </td><td>1 </td><td> 24243590</td><td> 26454176</td></tr>\n",
       "\t<tr><td>rs1131017 </td><td> 0.02409859</td><td>0.0008054830</td><td> 29.918193</td><td>7.188768e-121</td><td>ME_101_3-12:54985643-57100037 </td><td>ME_101</td><td>3</td><td>12:54985643-57100037 </td><td>12</td><td> 54985643</td><td> 57100037</td></tr>\n",
       "\t<tr><td>rs12442141</td><td> 0.04014834</td><td>0.0033260563</td><td> 12.070853</td><td> 1.789047e-30</td><td>ME_101_5-15:30215896-32681425 </td><td>ME_101</td><td>5</td><td>15:30215896-32681425 </td><td>15</td><td> 30215896</td><td> 32681425</td></tr>\n",
       "\t<tr><td>rs629329  </td><td> 0.02986714</td><td>0.0014708263</td><td> 20.306367</td><td> 5.848637e-71</td><td>ME_102_1-17:17145294-19676395 </td><td>ME_102</td><td>1</td><td>17:17145294-19676395 </td><td>17</td><td> 17145294</td><td> 19676395</td></tr>\n",
       "\t<tr><td>rs4761234 </td><td> 0.01989013</td><td>0.0005288609</td><td> 37.609373</td><td>8.488460e-164</td><td>ME_103_3-12:68275923-70376531 </td><td>ME_103</td><td>3</td><td>12:68275923-70376531 </td><td>12</td><td> 68275923</td><td> 70376531</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " snp & beta & se & t & p & module.qtl & module & pc & qtl.locus & qtl.locus.chr & qtl.locus.start & qtl.locus.end\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t rs16866400 & -0.01081391 & 0.0016948840 &  -6.380325 &  3.197894e-10 & ME\\_100\\_4-2:177556567-179584183 & ME\\_100 & 4 & 2:177556567-179584183 & 2  & 177556567 & 179584183\\\\\n",
       "\t rs582717   & -0.02334898 & 0.0014724757 & -15.856953 &  7.745374e-48 & ME\\_101\\_2-1:24243590-26454176   & ME\\_101 & 2 & 1:24243590-26454176   & 1  &  24243590 &  26454176\\\\\n",
       "\t rs1131017  &  0.02409859 & 0.0008054830 &  29.918193 & 7.188768e-121 & ME\\_101\\_3-12:54985643-57100037  & ME\\_101 & 3 & 12:54985643-57100037  & 12 &  54985643 &  57100037\\\\\n",
       "\t rs12442141 &  0.04014834 & 0.0033260563 &  12.070853 &  1.789047e-30 & ME\\_101\\_5-15:30215896-32681425  & ME\\_101 & 5 & 15:30215896-32681425  & 15 &  30215896 &  32681425\\\\\n",
       "\t rs629329   &  0.02986714 & 0.0014708263 &  20.306367 &  5.848637e-71 & ME\\_102\\_1-17:17145294-19676395  & ME\\_102 & 1 & 17:17145294-19676395  & 17 &  17145294 &  19676395\\\\\n",
       "\t rs4761234  &  0.01989013 & 0.0005288609 &  37.609373 & 8.488460e-164 & ME\\_103\\_3-12:68275923-70376531  & ME\\_103 & 3 & 12:68275923-70376531  & 12 &  68275923 &  70376531\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 12\n",
       "\n",
       "| snp &lt;chr&gt; | beta &lt;dbl&gt; | se &lt;dbl&gt; | t &lt;dbl&gt; | p &lt;dbl&gt; | module.qtl &lt;chr&gt; | module &lt;chr&gt; | pc &lt;chr&gt; | qtl.locus &lt;chr&gt; | qtl.locus.chr &lt;chr&gt; | qtl.locus.start &lt;dbl&gt; | qtl.locus.end &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| rs16866400 | -0.01081391 | 0.0016948840 |  -6.380325 |  3.197894e-10 | ME_100_4-2:177556567-179584183 | ME_100 | 4 | 2:177556567-179584183 | 2  | 177556567 | 179584183 |\n",
       "| rs582717   | -0.02334898 | 0.0014724757 | -15.856953 |  7.745374e-48 | ME_101_2-1:24243590-26454176   | ME_101 | 2 | 1:24243590-26454176   | 1  |  24243590 |  26454176 |\n",
       "| rs1131017  |  0.02409859 | 0.0008054830 |  29.918193 | 7.188768e-121 | ME_101_3-12:54985643-57100037  | ME_101 | 3 | 12:54985643-57100037  | 12 |  54985643 |  57100037 |\n",
       "| rs12442141 |  0.04014834 | 0.0033260563 |  12.070853 |  1.789047e-30 | ME_101_5-15:30215896-32681425  | ME_101 | 5 | 15:30215896-32681425  | 15 |  30215896 |  32681425 |\n",
       "| rs629329   |  0.02986714 | 0.0014708263 |  20.306367 |  5.848637e-71 | ME_102_1-17:17145294-19676395  | ME_102 | 1 | 17:17145294-19676395  | 17 |  17145294 |  19676395 |\n",
       "| rs4761234  |  0.01989013 | 0.0005288609 |  37.609373 | 8.488460e-164 | ME_103_3-12:68275923-70376531  | ME_103 | 3 | 12:68275923-70376531  | 12 |  68275923 |  70376531 |\n",
       "\n"
      ],
      "text/plain": [
       "  snp        beta        se           t          p            \n",
       "1 rs16866400 -0.01081391 0.0016948840  -6.380325  3.197894e-10\n",
       "2 rs582717   -0.02334898 0.0014724757 -15.856953  7.745374e-48\n",
       "3 rs1131017   0.02409859 0.0008054830  29.918193 7.188768e-121\n",
       "4 rs12442141  0.04014834 0.0033260563  12.070853  1.789047e-30\n",
       "5 rs629329    0.02986714 0.0014708263  20.306367  5.848637e-71\n",
       "6 rs4761234   0.01989013 0.0005288609  37.609373 8.488460e-164\n",
       "  module.qtl                     module pc qtl.locus             qtl.locus.chr\n",
       "1 ME_100_4-2:177556567-179584183 ME_100 4  2:177556567-179584183 2            \n",
       "2 ME_101_2-1:24243590-26454176   ME_101 2  1:24243590-26454176   1            \n",
       "3 ME_101_3-12:54985643-57100037  ME_101 3  12:54985643-57100037  12           \n",
       "4 ME_101_5-15:30215896-32681425  ME_101 5  15:30215896-32681425  15           \n",
       "5 ME_102_1-17:17145294-19676395  ME_102 1  17:17145294-19676395  17           \n",
       "6 ME_103_3-12:68275923-70376531  ME_103 3  12:68275923-70376531  12           \n",
       "  qtl.locus.start qtl.locus.end\n",
       "1 177556567       179584183    \n",
       "2  24243590        26454176    \n",
       "3  54985643        57100037    \n",
       "4  30215896        32681425    \n",
       "5  17145294        19676395    \n",
       "6  68275923        70376531    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(mqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "95f0d85f-56d8-4164-95f4-d524e26ac166",
   "metadata": {},
   "outputs": [],
   "source": [
    "cis.pqtl <- readRDS(\"/nfs/users/nfs_n/nm18/gains_team282/proteomics/pqtl/pqtl_ms2019/cis_pqtl_all.RDS\") %>%\n",
    "    dplyr::filter(pQTL_pval < (5e-8) / 269) %>%\n",
    "    dplyr::group_by(Gene) %>%\n",
    "    dplyr::slice_min(pQTL_pval, n=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "38af7e8a-159e-4f97-a1d8-903b39c13d93",
   "metadata": {},
   "outputs": [],
   "source": [
    "trans.pqtl <- readRDS(\"/nfs/users/nfs_n/nm18/gains_team282/proteomics/pqtl/pqtl_ms2019/trans_pqtl_all.RDS\") %>%\n",
    "    dplyr::filter(pQTL_pval < (5e-8) / 269) %>%\n",
    "    dplyr::group_by(pQTL_Protein, pQTL_Locus) %>%\n",
    "    dplyr::slice_min(pQTL_pval, n=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "94557708-4788-4bce-bf0e-8c5463a3f5cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>Chr</th><th scope=col>Position</th><th scope=col>pQTL_beta</th><th scope=col>pQTL_SE</th><th scope=col>pQTL_t</th><th scope=col>pQTL_pval</th><th scope=col>pQTL_Protein</th><th scope=col>Gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>rs529541 </td><td> 1</td><td>196750586</td><td>-0.24905935</td><td>0.01325800</td><td>-18.785585</td><td>1.642588e-67</td><td>P08603</td><td>ENSG00000000971</td></tr>\n",
       "\t<tr><td>rs854572 </td><td> 7</td><td> 95325384</td><td> 0.18402863</td><td>0.01585951</td><td> 11.603674</td><td>3.259335e-29</td><td>P27169</td><td>ENSG00000005421</td></tr>\n",
       "\t<tr><td>rs6073958</td><td>20</td><td> 45923216</td><td>-0.14494261</td><td>0.01847291</td><td> -7.846223</td><td>8.488455e-15</td><td>P55058</td><td>ENSG00000100979</td></tr>\n",
       "\t<tr><td>rs7815785</td><td> 8</td><td> 17872829</td><td>-0.22225713</td><td>0.02073768</td><td>-10.717549</td><td>2.077001e-25</td><td>Q08830</td><td>ENSG00000104760</td></tr>\n",
       "\t<tr><td>rs1648722</td><td> 3</td><td>186731200</td><td>-0.06974747</td><td>0.01034678</td><td> -6.740986</td><td>2.709794e-11</td><td>P01042</td><td>ENSG00000113889</td></tr>\n",
       "\t<tr><td>rs1656925</td><td> 3</td><td>186731334</td><td>-0.06974747</td><td>0.01034678</td><td> -6.740986</td><td>2.709794e-11</td><td>P01042</td><td>ENSG00000113889</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " SNP & Chr & Position & pQTL\\_beta & pQTL\\_SE & pQTL\\_t & pQTL\\_pval & pQTL\\_Protein & Gene\\\\\n",
       " <chr> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t rs529541  &  1 & 196750586 & -0.24905935 & 0.01325800 & -18.785585 & 1.642588e-67 & P08603 & ENSG00000000971\\\\\n",
       "\t rs854572  &  7 &  95325384 &  0.18402863 & 0.01585951 &  11.603674 & 3.259335e-29 & P27169 & ENSG00000005421\\\\\n",
       "\t rs6073958 & 20 &  45923216 & -0.14494261 & 0.01847291 &  -7.846223 & 8.488455e-15 & P55058 & ENSG00000100979\\\\\n",
       "\t rs7815785 &  8 &  17872829 & -0.22225713 & 0.02073768 & -10.717549 & 2.077001e-25 & Q08830 & ENSG00000104760\\\\\n",
       "\t rs1648722 &  3 & 186731200 & -0.06974747 & 0.01034678 &  -6.740986 & 2.709794e-11 & P01042 & ENSG00000113889\\\\\n",
       "\t rs1656925 &  3 & 186731334 & -0.06974747 & 0.01034678 &  -6.740986 & 2.709794e-11 & P01042 & ENSG00000113889\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 9\n",
       "\n",
       "| SNP &lt;chr&gt; | Chr &lt;int&gt; | Position &lt;int&gt; | pQTL_beta &lt;dbl&gt; | pQTL_SE &lt;dbl&gt; | pQTL_t &lt;dbl&gt; | pQTL_pval &lt;dbl&gt; | pQTL_Protein &lt;chr&gt; | Gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| rs529541  |  1 | 196750586 | -0.24905935 | 0.01325800 | -18.785585 | 1.642588e-67 | P08603 | ENSG00000000971 |\n",
       "| rs854572  |  7 |  95325384 |  0.18402863 | 0.01585951 |  11.603674 | 3.259335e-29 | P27169 | ENSG00000005421 |\n",
       "| rs6073958 | 20 |  45923216 | -0.14494261 | 0.01847291 |  -7.846223 | 8.488455e-15 | P55058 | ENSG00000100979 |\n",
       "| rs7815785 |  8 |  17872829 | -0.22225713 | 0.02073768 | -10.717549 | 2.077001e-25 | Q08830 | ENSG00000104760 |\n",
       "| rs1648722 |  3 | 186731200 | -0.06974747 | 0.01034678 |  -6.740986 | 2.709794e-11 | P01042 | ENSG00000113889 |\n",
       "| rs1656925 |  3 | 186731334 | -0.06974747 | 0.01034678 |  -6.740986 | 2.709794e-11 | P01042 | ENSG00000113889 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP       Chr Position  pQTL_beta   pQTL_SE    pQTL_t     pQTL_pval   \n",
       "1 rs529541   1  196750586 -0.24905935 0.01325800 -18.785585 1.642588e-67\n",
       "2 rs854572   7   95325384  0.18402863 0.01585951  11.603674 3.259335e-29\n",
       "3 rs6073958 20   45923216 -0.14494261 0.01847291  -7.846223 8.488455e-15\n",
       "4 rs7815785  8   17872829 -0.22225713 0.02073768 -10.717549 2.077001e-25\n",
       "5 rs1648722  3  186731200 -0.06974747 0.01034678  -6.740986 2.709794e-11\n",
       "6 rs1656925  3  186731334 -0.06974747 0.01034678  -6.740986 2.709794e-11\n",
       "  pQTL_Protein Gene           \n",
       "1 P08603       ENSG00000000971\n",
       "2 P27169       ENSG00000005421\n",
       "3 P55058       ENSG00000100979\n",
       "4 Q08830       ENSG00000104760\n",
       "5 P01042       ENSG00000113889\n",
       "6 P01042       ENSG00000113889"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cis.pqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "d80c4593-add9-49f9-a0d7-bb0402812dab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>Chr</th><th scope=col>Position</th><th scope=col>pQTL_beta</th><th scope=col>pQTL_SE</th><th scope=col>pQTL_t</th><th scope=col>pQTL_pval</th><th scope=col>pQTL_Protein</th><th scope=col>pQTL_Locus</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>rs11846959</td><td>14</td><td>94379979</td><td>-0.3065649</td><td>0.01614662</td><td>-18.986322</td><td>3.309735e-69</td><td>P00751</td><td>pQTL_1</td></tr>\n",
       "\t<tr><td>rs11846959</td><td>14</td><td>94379979</td><td>-0.2137874</td><td>0.01793464</td><td>-11.920362</td><td>6.923034e-31</td><td>P02787</td><td>pQTL_1</td></tr>\n",
       "\t<tr><td>rs9302635 </td><td>16</td><td>72110275</td><td> 0.1145890</td><td>0.01466176</td><td>  7.815500</td><td>1.200982e-14</td><td>P10909</td><td>pQTL_1</td></tr>\n",
       "\t<tr><td>rs9302635 </td><td>16</td><td>72110275</td><td> 0.1319606</td><td>0.01429908</td><td>  9.228609</td><td>1.454631e-19</td><td>P19827</td><td>pQTL_1</td></tr>\n",
       "\t<tr><td>rs16917722</td><td> 8</td><td>95461382</td><td>-0.4991650</td><td>0.06949047</td><td> -7.183215</td><td>1.673847e-12</td><td>P35858</td><td>pQTL_1</td></tr>\n",
       "\t<tr><td>rs76368529</td><td> 8</td><td>95464005</td><td>-0.4991650</td><td>0.06949047</td><td> -7.183215</td><td>1.673847e-12</td><td>P35858</td><td>pQTL_1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " SNP & Chr & Position & pQTL\\_beta & pQTL\\_SE & pQTL\\_t & pQTL\\_pval & pQTL\\_Protein & pQTL\\_Locus\\\\\n",
       " <chr> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t rs11846959 & 14 & 94379979 & -0.3065649 & 0.01614662 & -18.986322 & 3.309735e-69 & P00751 & pQTL\\_1\\\\\n",
       "\t rs11846959 & 14 & 94379979 & -0.2137874 & 0.01793464 & -11.920362 & 6.923034e-31 & P02787 & pQTL\\_1\\\\\n",
       "\t rs9302635  & 16 & 72110275 &  0.1145890 & 0.01466176 &   7.815500 & 1.200982e-14 & P10909 & pQTL\\_1\\\\\n",
       "\t rs9302635  & 16 & 72110275 &  0.1319606 & 0.01429908 &   9.228609 & 1.454631e-19 & P19827 & pQTL\\_1\\\\\n",
       "\t rs16917722 &  8 & 95461382 & -0.4991650 & 0.06949047 &  -7.183215 & 1.673847e-12 & P35858 & pQTL\\_1\\\\\n",
       "\t rs76368529 &  8 & 95464005 & -0.4991650 & 0.06949047 &  -7.183215 & 1.673847e-12 & P35858 & pQTL\\_1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 9\n",
       "\n",
       "| SNP &lt;chr&gt; | Chr &lt;int&gt; | Position &lt;int&gt; | pQTL_beta &lt;dbl&gt; | pQTL_SE &lt;dbl&gt; | pQTL_t &lt;dbl&gt; | pQTL_pval &lt;dbl&gt; | pQTL_Protein &lt;chr&gt; | pQTL_Locus &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| rs11846959 | 14 | 94379979 | -0.3065649 | 0.01614662 | -18.986322 | 3.309735e-69 | P00751 | pQTL_1 |\n",
       "| rs11846959 | 14 | 94379979 | -0.2137874 | 0.01793464 | -11.920362 | 6.923034e-31 | P02787 | pQTL_1 |\n",
       "| rs9302635  | 16 | 72110275 |  0.1145890 | 0.01466176 |   7.815500 | 1.200982e-14 | P10909 | pQTL_1 |\n",
       "| rs9302635  | 16 | 72110275 |  0.1319606 | 0.01429908 |   9.228609 | 1.454631e-19 | P19827 | pQTL_1 |\n",
       "| rs16917722 |  8 | 95461382 | -0.4991650 | 0.06949047 |  -7.183215 | 1.673847e-12 | P35858 | pQTL_1 |\n",
       "| rs76368529 |  8 | 95464005 | -0.4991650 | 0.06949047 |  -7.183215 | 1.673847e-12 | P35858 | pQTL_1 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP        Chr Position pQTL_beta  pQTL_SE    pQTL_t     pQTL_pval   \n",
       "1 rs11846959 14  94379979 -0.3065649 0.01614662 -18.986322 3.309735e-69\n",
       "2 rs11846959 14  94379979 -0.2137874 0.01793464 -11.920362 6.923034e-31\n",
       "3 rs9302635  16  72110275  0.1145890 0.01466176   7.815500 1.200982e-14\n",
       "4 rs9302635  16  72110275  0.1319606 0.01429908   9.228609 1.454631e-19\n",
       "5 rs16917722  8  95461382 -0.4991650 0.06949047  -7.183215 1.673847e-12\n",
       "6 rs76368529  8  95464005 -0.4991650 0.06949047  -7.183215 1.673847e-12\n",
       "  pQTL_Protein pQTL_Locus\n",
       "1 P00751       pQTL_1    \n",
       "2 P02787       pQTL_1    \n",
       "3 P10909       pQTL_1    \n",
       "4 P19827       pQTL_1    \n",
       "5 P35858       pQTL_1    \n",
       "6 P35858       pQTL_1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(trans.pqtl)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c77ce625-5566-40cf-a16c-3b0df2a1c044",
   "metadata": {},
   "source": [
    "## Consequence and Impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "b9c5d926-9220-4a28-810b-63ca2514ed3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cis.eqtl.vep <- merge(vep, cis.eqtl, by.x=\"ID\", by.y=\"snps\") %>%\n",
    "    dplyr::select(\n",
    "        gene_id, CHROM, POS, ID, REF, ALT, Allele,\n",
    "        Gene, Feature_type, Feature,\n",
    "        Consequence, IMPACT,\n",
    "        BIOTYPE,\n",
    "        MOTIF_NAME, TRANSCRIPTION_FACTORS,\n",
    "        CADD_PHRED, CADD_RAW, PolyPhen\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "73dff4a4-d145-4054-b874-568845a492f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene_id</th><th scope=col>CHROM</th><th scope=col>POS</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>Allele</th><th scope=col>Gene</th><th scope=col>Feature_type</th><th scope=col>Feature</th><th scope=col>Consequence</th><th scope=col>IMPACT</th><th scope=col>BIOTYPE</th><th scope=col>MOTIF_NAME</th><th scope=col>TRANSCRIPTION_FACTORS</th><th scope=col>CADD_PHRED</th><th scope=col>CADD_RAW</th><th scope=col>PolyPhen</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000272416</td><td>5</td><td>54079901</td><td>5.54079901_A_C</td><td>C</td><td>A</td><td>A</td><td>ENSG00000185305</td><td>Transcript</td><td>ENST00000504924</td><td>intron_variant                              </td><td>MODIFIER</td><td>protein_coding      </td><td>NA</td><td>NA</td><td>1.589</td><td>0.037598</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000272416</td><td>5</td><td>54079901</td><td>5.54079901_A_C</td><td>C</td><td>A</td><td>A</td><td>ENSG00000185305</td><td>Transcript</td><td>ENST00000502271</td><td>intron_variant                              </td><td>MODIFIER</td><td>protein_coding      </td><td>NA</td><td>NA</td><td>1.589</td><td>0.037598</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000272416</td><td>5</td><td>54079901</td><td>5.54079901_A_C</td><td>C</td><td>A</td><td>A</td><td>ENSG00000185305</td><td>Transcript</td><td>ENST00000510591</td><td>intron_variant&amp;non_coding_transcript_variant</td><td>MODIFIER</td><td>processed_transcript</td><td>NA</td><td>NA</td><td>1.589</td><td>0.037598</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000272416</td><td>5</td><td>54079901</td><td>5.54079901_A_C</td><td>C</td><td>A</td><td>A</td><td>ENSG00000265421</td><td>Transcript</td><td>ENST00000581837</td><td>upstream_gene_variant                       </td><td>MODIFIER</td><td>miRNA               </td><td>NA</td><td>NA</td><td>1.589</td><td>0.037598</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000272416</td><td>5</td><td>54079901</td><td>5.54079901_A_C</td><td>A</td><td>C</td><td>C</td><td>ENSG00000185305</td><td>Transcript</td><td>ENST00000504924</td><td>intron_variant                              </td><td>MODIFIER</td><td>protein_coding      </td><td>NA</td><td>NA</td><td>   NA</td><td>      NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000272416</td><td>5</td><td>54079901</td><td>5.54079901_A_C</td><td>C</td><td>A</td><td>A</td><td>ENSG00000185305</td><td>Transcript</td><td>ENST00000620747</td><td>intron_variant                              </td><td>MODIFIER</td><td>protein_coding      </td><td>NA</td><td>NA</td><td>1.589</td><td>0.037598</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & gene\\_id & CHROM & POS & ID & REF & ALT & Allele & Gene & Feature\\_type & Feature & Consequence & IMPACT & BIOTYPE & MOTIF\\_NAME & TRANSCRIPTION\\_FACTORS & CADD\\_PHRED & CADD\\_RAW & PolyPhen\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000272416 & 5 & 54079901 & 5.54079901\\_A\\_C & C & A & A & ENSG00000185305 & Transcript & ENST00000504924 & intron\\_variant                               & MODIFIER & protein\\_coding       & NA & NA & 1.589 & 0.037598 & NA\\\\\n",
       "\t2 & ENSG00000272416 & 5 & 54079901 & 5.54079901\\_A\\_C & C & A & A & ENSG00000185305 & Transcript & ENST00000502271 & intron\\_variant                               & MODIFIER & protein\\_coding       & NA & NA & 1.589 & 0.037598 & NA\\\\\n",
       "\t3 & ENSG00000272416 & 5 & 54079901 & 5.54079901\\_A\\_C & C & A & A & ENSG00000185305 & Transcript & ENST00000510591 & intron\\_variant\\&non\\_coding\\_transcript\\_variant & MODIFIER & processed\\_transcript & NA & NA & 1.589 & 0.037598 & NA\\\\\n",
       "\t4 & ENSG00000272416 & 5 & 54079901 & 5.54079901\\_A\\_C & C & A & A & ENSG00000265421 & Transcript & ENST00000581837 & upstream\\_gene\\_variant                        & MODIFIER & miRNA                & NA & NA & 1.589 & 0.037598 & NA\\\\\n",
       "\t5 & ENSG00000272416 & 5 & 54079901 & 5.54079901\\_A\\_C & A & C & C & ENSG00000185305 & Transcript & ENST00000504924 & intron\\_variant                               & MODIFIER & protein\\_coding       & NA & NA &    NA &       NA & NA\\\\\n",
       "\t6 & ENSG00000272416 & 5 & 54079901 & 5.54079901\\_A\\_C & C & A & A & ENSG00000185305 & Transcript & ENST00000620747 & intron\\_variant                               & MODIFIER & protein\\_coding       & NA & NA & 1.589 & 0.037598 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 18\n",
       "\n",
       "| <!--/--> | gene_id &lt;chr&gt; | CHROM &lt;int&gt; | POS &lt;int&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | Allele &lt;chr&gt; | Gene &lt;chr&gt; | Feature_type &lt;chr&gt; | Feature &lt;chr&gt; | Consequence &lt;chr&gt; | IMPACT &lt;chr&gt; | BIOTYPE &lt;chr&gt; | MOTIF_NAME &lt;chr&gt; | TRANSCRIPTION_FACTORS &lt;chr&gt; | CADD_PHRED &lt;dbl&gt; | CADD_RAW &lt;dbl&gt; | PolyPhen &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000272416 | 5 | 54079901 | 5.54079901_A_C | C | A | A | ENSG00000185305 | Transcript | ENST00000504924 | intron_variant                               | MODIFIER | protein_coding       | NA | NA | 1.589 | 0.037598 | NA |\n",
       "| 2 | ENSG00000272416 | 5 | 54079901 | 5.54079901_A_C | C | A | A | ENSG00000185305 | Transcript | ENST00000502271 | intron_variant                               | MODIFIER | protein_coding       | NA | NA | 1.589 | 0.037598 | NA |\n",
       "| 3 | ENSG00000272416 | 5 | 54079901 | 5.54079901_A_C | C | A | A | ENSG00000185305 | Transcript | ENST00000510591 | intron_variant&amp;non_coding_transcript_variant | MODIFIER | processed_transcript | NA | NA | 1.589 | 0.037598 | NA |\n",
       "| 4 | ENSG00000272416 | 5 | 54079901 | 5.54079901_A_C | C | A | A | ENSG00000265421 | Transcript | ENST00000581837 | upstream_gene_variant                        | MODIFIER | miRNA                | NA | NA | 1.589 | 0.037598 | NA |\n",
       "| 5 | ENSG00000272416 | 5 | 54079901 | 5.54079901_A_C | A | C | C | ENSG00000185305 | Transcript | ENST00000504924 | intron_variant                               | MODIFIER | protein_coding       | NA | NA |    NA |       NA | NA |\n",
       "| 6 | ENSG00000272416 | 5 | 54079901 | 5.54079901_A_C | C | A | A | ENSG00000185305 | Transcript | ENST00000620747 | intron_variant                               | MODIFIER | protein_coding       | NA | NA | 1.589 | 0.037598 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  gene_id         CHROM POS      ID             REF ALT Allele Gene           \n",
       "1 ENSG00000272416 5     54079901 5.54079901_A_C C   A   A      ENSG00000185305\n",
       "2 ENSG00000272416 5     54079901 5.54079901_A_C C   A   A      ENSG00000185305\n",
       "3 ENSG00000272416 5     54079901 5.54079901_A_C C   A   A      ENSG00000185305\n",
       "4 ENSG00000272416 5     54079901 5.54079901_A_C C   A   A      ENSG00000265421\n",
       "5 ENSG00000272416 5     54079901 5.54079901_A_C A   C   C      ENSG00000185305\n",
       "6 ENSG00000272416 5     54079901 5.54079901_A_C C   A   A      ENSG00000185305\n",
       "  Feature_type Feature         Consequence                                 \n",
       "1 Transcript   ENST00000504924 intron_variant                              \n",
       "2 Transcript   ENST00000502271 intron_variant                              \n",
       "3 Transcript   ENST00000510591 intron_variant&non_coding_transcript_variant\n",
       "4 Transcript   ENST00000581837 upstream_gene_variant                       \n",
       "5 Transcript   ENST00000504924 intron_variant                              \n",
       "6 Transcript   ENST00000620747 intron_variant                              \n",
       "  IMPACT   BIOTYPE              MOTIF_NAME TRANSCRIPTION_FACTORS CADD_PHRED\n",
       "1 MODIFIER protein_coding       NA         NA                    1.589     \n",
       "2 MODIFIER protein_coding       NA         NA                    1.589     \n",
       "3 MODIFIER processed_transcript NA         NA                    1.589     \n",
       "4 MODIFIER miRNA                NA         NA                    1.589     \n",
       "5 MODIFIER protein_coding       NA         NA                       NA     \n",
       "6 MODIFIER protein_coding       NA         NA                    1.589     \n",
       "  CADD_RAW PolyPhen\n",
       "1 0.037598 NA      \n",
       "2 0.037598 NA      \n",
       "3 0.037598 NA      \n",
       "4 0.037598 NA      \n",
       "5       NA NA      \n",
       "6 0.037598 NA      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cis.eqtl.vep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "01cc32b0-2995-474b-a127-af28c0a90adf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 67 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_id</th><th scope=col>Consequence</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000022976</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000059691</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 6</td></tr>\n",
       "\t<tr><td>ENSG00000072694</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 6</td></tr>\n",
       "\t<tr><td>ENSG00000087263</td><td>missense_variant&amp;NMD_transcript_variant</td><td>44</td></tr>\n",
       "\t<tr><td>ENSG00000099330</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 4</td></tr>\n",
       "\t<tr><td>ENSG00000099337</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 6</td></tr>\n",
       "\t<tr><td>ENSG00000100938</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000101639</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 4</td></tr>\n",
       "\t<tr><td>ENSG00000104884</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000105255</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000106638</td><td>missense_variant&amp;NMD_transcript_variant</td><td>14</td></tr>\n",
       "\t<tr><td>ENSG00000108826</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000111652</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000112053</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000113456</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000118655</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 4</td></tr>\n",
       "\t<tr><td>ENSG00000126247</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000127311</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 6</td></tr>\n",
       "\t<tr><td>ENSG00000128944</td><td>missense_variant&amp;NMD_transcript_variant</td><td>10</td></tr>\n",
       "\t<tr><td>ENSG00000129559</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000133454</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000141562</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000145241</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 4</td></tr>\n",
       "\t<tr><td>ENSG00000149761</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 4</td></tr>\n",
       "\t<tr><td>ENSG00000150753</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000160183</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 8</td></tr>\n",
       "\t<tr><td>ENSG00000160185</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000161533</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000163322</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>ENSG00000163576</td><td>missense_variant&amp;NMD_transcript_variant</td><td> 2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>ENSG00000185156</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000185710</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000186583</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000187951</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000205560</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000223496</td><td>missense_variant&amp;NMD_transcript_variant</td><td>6</td></tr>\n",
       "\t<tr><td>ENSG00000225217</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000228544</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000230715</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000242110</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000242588</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000243302</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000243679</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000244682</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000250644</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000256826</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000260923</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000261556</td><td>missense_variant&amp;NMD_transcript_variant</td><td>6</td></tr>\n",
       "\t<tr><td>ENSG00000267801</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000269970</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000270179</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000271605</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000271964</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000276231</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000276839</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000279073</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000280828</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000285696</td><td>missense_variant&amp;NMD_transcript_variant</td><td>4</td></tr>\n",
       "\t<tr><td>ENSG00000287725</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "\t<tr><td>ENSG00000288393</td><td>missense_variant&amp;NMD_transcript_variant</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 67 × 3\n",
       "\\begin{tabular}{lll}\n",
       " gene\\_id & Consequence & n\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSG00000022976 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000059691 & missense\\_variant\\&NMD\\_transcript\\_variant &  6\\\\\n",
       "\t ENSG00000072694 & missense\\_variant\\&NMD\\_transcript\\_variant &  6\\\\\n",
       "\t ENSG00000087263 & missense\\_variant\\&NMD\\_transcript\\_variant & 44\\\\\n",
       "\t ENSG00000099330 & missense\\_variant\\&NMD\\_transcript\\_variant &  4\\\\\n",
       "\t ENSG00000099337 & missense\\_variant\\&NMD\\_transcript\\_variant &  6\\\\\n",
       "\t ENSG00000100938 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000101639 & missense\\_variant\\&NMD\\_transcript\\_variant &  4\\\\\n",
       "\t ENSG00000104884 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000105255 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000106638 & missense\\_variant\\&NMD\\_transcript\\_variant & 14\\\\\n",
       "\t ENSG00000108826 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000111652 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000112053 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000113456 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000118655 & missense\\_variant\\&NMD\\_transcript\\_variant &  4\\\\\n",
       "\t ENSG00000126247 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000127311 & missense\\_variant\\&NMD\\_transcript\\_variant &  6\\\\\n",
       "\t ENSG00000128944 & missense\\_variant\\&NMD\\_transcript\\_variant & 10\\\\\n",
       "\t ENSG00000129559 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000133454 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000141562 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000145241 & missense\\_variant\\&NMD\\_transcript\\_variant &  4\\\\\n",
       "\t ENSG00000149761 & missense\\_variant\\&NMD\\_transcript\\_variant &  4\\\\\n",
       "\t ENSG00000150753 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000160183 & missense\\_variant\\&NMD\\_transcript\\_variant &  8\\\\\n",
       "\t ENSG00000160185 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000161533 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000163322 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ENSG00000163576 & missense\\_variant\\&NMD\\_transcript\\_variant &  2\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t ENSG00000185156 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000185710 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000186583 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000187951 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000205560 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000223496 & missense\\_variant\\&NMD\\_transcript\\_variant & 6\\\\\n",
       "\t ENSG00000225217 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000228544 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000230715 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000242110 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000242588 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000243302 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000243679 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000244682 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000250644 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000256826 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000260923 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000261556 & missense\\_variant\\&NMD\\_transcript\\_variant & 6\\\\\n",
       "\t ENSG00000267801 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000269970 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000270179 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000271605 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000271964 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000276231 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000276839 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000279073 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000280828 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000285696 & missense\\_variant\\&NMD\\_transcript\\_variant & 4\\\\\n",
       "\t ENSG00000287725 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\t ENSG00000288393 & missense\\_variant\\&NMD\\_transcript\\_variant & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 67 × 3\n",
       "\n",
       "| gene_id &lt;chr&gt; | Consequence &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000022976 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000059691 | missense_variant&amp;NMD_transcript_variant |  6 |\n",
       "| ENSG00000072694 | missense_variant&amp;NMD_transcript_variant |  6 |\n",
       "| ENSG00000087263 | missense_variant&amp;NMD_transcript_variant | 44 |\n",
       "| ENSG00000099330 | missense_variant&amp;NMD_transcript_variant |  4 |\n",
       "| ENSG00000099337 | missense_variant&amp;NMD_transcript_variant |  6 |\n",
       "| ENSG00000100938 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000101639 | missense_variant&amp;NMD_transcript_variant |  4 |\n",
       "| ENSG00000104884 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000105255 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000106638 | missense_variant&amp;NMD_transcript_variant | 14 |\n",
       "| ENSG00000108826 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000111652 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000112053 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000113456 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000118655 | missense_variant&amp;NMD_transcript_variant |  4 |\n",
       "| ENSG00000126247 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000127311 | missense_variant&amp;NMD_transcript_variant |  6 |\n",
       "| ENSG00000128944 | missense_variant&amp;NMD_transcript_variant | 10 |\n",
       "| ENSG00000129559 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000133454 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000141562 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000145241 | missense_variant&amp;NMD_transcript_variant |  4 |\n",
       "| ENSG00000149761 | missense_variant&amp;NMD_transcript_variant |  4 |\n",
       "| ENSG00000150753 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000160183 | missense_variant&amp;NMD_transcript_variant |  8 |\n",
       "| ENSG00000160185 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000161533 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000163322 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ENSG00000163576 | missense_variant&amp;NMD_transcript_variant |  2 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| ENSG00000185156 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000185710 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000186583 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000187951 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000205560 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000223496 | missense_variant&amp;NMD_transcript_variant | 6 |\n",
       "| ENSG00000225217 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000228544 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000230715 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000242110 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000242588 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000243302 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000243679 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000244682 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000250644 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000256826 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000260923 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000261556 | missense_variant&amp;NMD_transcript_variant | 6 |\n",
       "| ENSG00000267801 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000269970 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000270179 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000271605 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000271964 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000276231 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000276839 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000279073 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000280828 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000285696 | missense_variant&amp;NMD_transcript_variant | 4 |\n",
       "| ENSG00000287725 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "| ENSG00000288393 | missense_variant&amp;NMD_transcript_variant | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_id         Consequence                             n \n",
       "1  ENSG00000022976 missense_variant&NMD_transcript_variant  2\n",
       "2  ENSG00000059691 missense_variant&NMD_transcript_variant  6\n",
       "3  ENSG00000072694 missense_variant&NMD_transcript_variant  6\n",
       "4  ENSG00000087263 missense_variant&NMD_transcript_variant 44\n",
       "5  ENSG00000099330 missense_variant&NMD_transcript_variant  4\n",
       "6  ENSG00000099337 missense_variant&NMD_transcript_variant  6\n",
       "7  ENSG00000100938 missense_variant&NMD_transcript_variant  2\n",
       "8  ENSG00000101639 missense_variant&NMD_transcript_variant  4\n",
       "9  ENSG00000104884 missense_variant&NMD_transcript_variant  2\n",
       "10 ENSG00000105255 missense_variant&NMD_transcript_variant  2\n",
       "11 ENSG00000106638 missense_variant&NMD_transcript_variant 14\n",
       "12 ENSG00000108826 missense_variant&NMD_transcript_variant  2\n",
       "13 ENSG00000111652 missense_variant&NMD_transcript_variant  2\n",
       "14 ENSG00000112053 missense_variant&NMD_transcript_variant  2\n",
       "15 ENSG00000113456 missense_variant&NMD_transcript_variant  2\n",
       "16 ENSG00000118655 missense_variant&NMD_transcript_variant  4\n",
       "17 ENSG00000126247 missense_variant&NMD_transcript_variant  2\n",
       "18 ENSG00000127311 missense_variant&NMD_transcript_variant  6\n",
       "19 ENSG00000128944 missense_variant&NMD_transcript_variant 10\n",
       "20 ENSG00000129559 missense_variant&NMD_transcript_variant  2\n",
       "21 ENSG00000133454 missense_variant&NMD_transcript_variant  2\n",
       "22 ENSG00000141562 missense_variant&NMD_transcript_variant  2\n",
       "23 ENSG00000145241 missense_variant&NMD_transcript_variant  4\n",
       "24 ENSG00000149761 missense_variant&NMD_transcript_variant  4\n",
       "25 ENSG00000150753 missense_variant&NMD_transcript_variant  2\n",
       "26 ENSG00000160183 missense_variant&NMD_transcript_variant  8\n",
       "27 ENSG00000160185 missense_variant&NMD_transcript_variant  2\n",
       "28 ENSG00000161533 missense_variant&NMD_transcript_variant  2\n",
       "29 ENSG00000163322 missense_variant&NMD_transcript_variant  2\n",
       "30 ENSG00000163576 missense_variant&NMD_transcript_variant  2\n",
       "⋮  ⋮               ⋮                                       ⋮ \n",
       "38 ENSG00000185156 missense_variant&NMD_transcript_variant 4 \n",
       "39 ENSG00000185710 missense_variant&NMD_transcript_variant 2 \n",
       "40 ENSG00000186583 missense_variant&NMD_transcript_variant 2 \n",
       "41 ENSG00000187951 missense_variant&NMD_transcript_variant 4 \n",
       "42 ENSG00000205560 missense_variant&NMD_transcript_variant 2 \n",
       "43 ENSG00000223496 missense_variant&NMD_transcript_variant 6 \n",
       "44 ENSG00000225217 missense_variant&NMD_transcript_variant 2 \n",
       "45 ENSG00000228544 missense_variant&NMD_transcript_variant 2 \n",
       "46 ENSG00000230715 missense_variant&NMD_transcript_variant 2 \n",
       "47 ENSG00000242110 missense_variant&NMD_transcript_variant 2 \n",
       "48 ENSG00000242588 missense_variant&NMD_transcript_variant 2 \n",
       "49 ENSG00000243302 missense_variant&NMD_transcript_variant 2 \n",
       "50 ENSG00000243679 missense_variant&NMD_transcript_variant 2 \n",
       "51 ENSG00000244682 missense_variant&NMD_transcript_variant 2 \n",
       "52 ENSG00000250644 missense_variant&NMD_transcript_variant 2 \n",
       "53 ENSG00000256826 missense_variant&NMD_transcript_variant 4 \n",
       "54 ENSG00000260923 missense_variant&NMD_transcript_variant 4 \n",
       "55 ENSG00000261556 missense_variant&NMD_transcript_variant 6 \n",
       "56 ENSG00000267801 missense_variant&NMD_transcript_variant 2 \n",
       "57 ENSG00000269970 missense_variant&NMD_transcript_variant 2 \n",
       "58 ENSG00000270179 missense_variant&NMD_transcript_variant 4 \n",
       "59 ENSG00000271605 missense_variant&NMD_transcript_variant 4 \n",
       "60 ENSG00000271964 missense_variant&NMD_transcript_variant 2 \n",
       "61 ENSG00000276231 missense_variant&NMD_transcript_variant 4 \n",
       "62 ENSG00000276839 missense_variant&NMD_transcript_variant 4 \n",
       "63 ENSG00000279073 missense_variant&NMD_transcript_variant 2 \n",
       "64 ENSG00000280828 missense_variant&NMD_transcript_variant 2 \n",
       "65 ENSG00000285696 missense_variant&NMD_transcript_variant 4 \n",
       "66 ENSG00000287725 missense_variant&NMD_transcript_variant 2 \n",
       "67 ENSG00000288393 missense_variant&NMD_transcript_variant 2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cis.eqtl.vep %>%\n",
    "    dplyr::select(-REF, -ALT, -Allele) %>%\n",
    "    unique() %>%\n",
    "    dplyr::count(gene_id, Consequence) %>%\n",
    "    dplyr::filter(Consequence == \"missense_variant&NMD_transcript_variant\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "ab6abdce-06e9-40e8-809e-df352d9b4aee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 38 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Consequence</th><th scope=col>n</th><th scope=col>egenes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>3_prime_UTR_variant                                               </td><td>  996</td><td> 524</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>3_prime_UTR_variant&amp;NMD_transcript_variant                        </span></td><td><span style=white-space:pre-wrap>  363</span></td><td> 205</td></tr>\n",
       "\t<tr><td>5_prime_UTR_variant                                               </td><td>  501</td><td> 227</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>5_prime_UTR_variant&amp;NMD_transcript_variant                        </span></td><td><span style=white-space:pre-wrap>   72</span></td><td><span style=white-space:pre-wrap>  44</span></td></tr>\n",
       "\t<tr><td>downstream_gene_variant                                           </td><td>14899</td><td>4175</td></tr>\n",
       "\t<tr><td>intergenic_variant                                                </td><td> 1035</td><td>1035</td></tr>\n",
       "\t<tr><td>intron_variant                                                    </td><td>22869</td><td>5871</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>intron_variant&amp;NMD_transcript_variant                             </span></td><td> 5618</td><td>2444</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>intron_variant&amp;non_coding_transcript_variant                      </span></td><td>12813</td><td>4488</td></tr>\n",
       "\t<tr><td>missense_variant                                                  </td><td>  820</td><td> 304</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>missense_variant&amp;NMD_transcript_variant                           </span></td><td><span style=white-space:pre-wrap>  130</span></td><td><span style=white-space:pre-wrap>  67</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>missense_variant&amp;splice_region_variant                            </span></td><td><span style=white-space:pre-wrap>    7</span></td><td><span style=white-space:pre-wrap>   5</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>missense_variant&amp;splice_region_variant&amp;NMD_transcript_variant     </span></td><td><span style=white-space:pre-wrap>    1</span></td><td><span style=white-space:pre-wrap>   1</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>missense_variant&amp;stop_retained_variant                            </span></td><td><span style=white-space:pre-wrap>    2</span></td><td><span style=white-space:pre-wrap>   2</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>missense_variant&amp;stop_retained_variant&amp;NMD_transcript_variant     </span></td><td><span style=white-space:pre-wrap>    1</span></td><td><span style=white-space:pre-wrap>   1</span></td></tr>\n",
       "\t<tr><td>non_coding_transcript_exon_variant                                </td><td> 2285</td><td>1171</td></tr>\n",
       "\t<tr><td>regulatory_region_variant                                         </td><td> 3761</td><td>3464</td></tr>\n",
       "\t<tr><td>splice_acceptor_variant                                           </td><td>   10</td><td>   2</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_acceptor_variant&amp;NMD_transcript_variant                    </span></td><td><span style=white-space:pre-wrap>    5</span></td><td><span style=white-space:pre-wrap>   1</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_acceptor_variant&amp;non_coding_transcript_variant             </span></td><td><span style=white-space:pre-wrap>   12</span></td><td><span style=white-space:pre-wrap>   5</span></td></tr>\n",
       "\t<tr><td>splice_donor_variant                                              </td><td>    3</td><td>   2</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;3_prime_UTR_variant&amp;NMD_transcript_variant  </span></td><td><span style=white-space:pre-wrap>    5</span></td><td><span style=white-space:pre-wrap>   2</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;5_prime_UTR_variant                         </span></td><td><span style=white-space:pre-wrap>    6</span></td><td><span style=white-space:pre-wrap>   4</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;5_prime_UTR_variant&amp;NMD_transcript_variant  </span></td><td><span style=white-space:pre-wrap>    2</span></td><td><span style=white-space:pre-wrap>   2</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;intron_variant                              </span></td><td><span style=white-space:pre-wrap>   92</span></td><td><span style=white-space:pre-wrap>  29</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;intron_variant&amp;NMD_transcript_variant       </span></td><td><span style=white-space:pre-wrap>   53</span></td><td><span style=white-space:pre-wrap>  15</span></td></tr>\n",
       "\t<tr><td>splice_region_variant&amp;intron_variant&amp;non_coding_transcript_variant</td><td><span style=white-space:pre-wrap>   58</span></td><td><span style=white-space:pre-wrap>  31</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;non_coding_transcript_exon_variant          </span></td><td><span style=white-space:pre-wrap>   49</span></td><td><span style=white-space:pre-wrap>  13</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;synonymous_variant                          </span></td><td><span style=white-space:pre-wrap>   21</span></td><td><span style=white-space:pre-wrap>   5</span></td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>splice_region_variant&amp;synonymous_variant&amp;NMD_transcript_variant   </span></td><td><span style=white-space:pre-wrap>    7</span></td><td><span style=white-space:pre-wrap>   3</span></td></tr>\n",
       "\t<tr><td>stop_gained                                                       </td><td>    6</td><td>   2</td></tr>\n",
       "\t<tr><td>stop_lost                                                         </td><td>    8</td><td>   4</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>stop_lost&amp;NMD_transcript_variant                                  </span></td><td><span style=white-space:pre-wrap>    1</span></td><td><span style=white-space:pre-wrap>   1</span></td></tr>\n",
       "\t<tr><td>stop_retained_variant                                             </td><td>    1</td><td>   1</td></tr>\n",
       "\t<tr><td>synonymous_variant                                                </td><td>  648</td><td> 205</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>synonymous_variant&amp;NMD_transcript_variant                         </span></td><td><span style=white-space:pre-wrap>   67</span></td><td><span style=white-space:pre-wrap>  29</span></td></tr>\n",
       "\t<tr><td>TF_binding_site_variant                                           </td><td> 1446</td><td> 532</td></tr>\n",
       "\t<tr><td>upstream_gene_variant                                             </td><td>16712</td><td>4292</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 38 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Consequence & n & egenes\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 3\\_prime\\_UTR\\_variant                                                &   996 &  524\\\\\n",
       "\t 3\\_prime\\_UTR\\_variant\\&NMD\\_transcript\\_variant                         &   363 &  205\\\\\n",
       "\t 5\\_prime\\_UTR\\_variant                                                &   501 &  227\\\\\n",
       "\t 5\\_prime\\_UTR\\_variant\\&NMD\\_transcript\\_variant                         &    72 &   44\\\\\n",
       "\t downstream\\_gene\\_variant                                            & 14899 & 4175\\\\\n",
       "\t intergenic\\_variant                                                 &  1035 & 1035\\\\\n",
       "\t intron\\_variant                                                     & 22869 & 5871\\\\\n",
       "\t intron\\_variant\\&NMD\\_transcript\\_variant                              &  5618 & 2444\\\\\n",
       "\t intron\\_variant\\&non\\_coding\\_transcript\\_variant                       & 12813 & 4488\\\\\n",
       "\t missense\\_variant                                                   &   820 &  304\\\\\n",
       "\t missense\\_variant\\&NMD\\_transcript\\_variant                            &   130 &   67\\\\\n",
       "\t missense\\_variant\\&splice\\_region\\_variant                             &     7 &    5\\\\\n",
       "\t missense\\_variant\\&splice\\_region\\_variant\\&NMD\\_transcript\\_variant      &     1 &    1\\\\\n",
       "\t missense\\_variant\\&stop\\_retained\\_variant                             &     2 &    2\\\\\n",
       "\t missense\\_variant\\&stop\\_retained\\_variant\\&NMD\\_transcript\\_variant      &     1 &    1\\\\\n",
       "\t non\\_coding\\_transcript\\_exon\\_variant                                 &  2285 & 1171\\\\\n",
       "\t regulatory\\_region\\_variant                                          &  3761 & 3464\\\\\n",
       "\t splice\\_acceptor\\_variant                                            &    10 &    2\\\\\n",
       "\t splice\\_acceptor\\_variant\\&NMD\\_transcript\\_variant                     &     5 &    1\\\\\n",
       "\t splice\\_acceptor\\_variant\\&non\\_coding\\_transcript\\_variant              &    12 &    5\\\\\n",
       "\t splice\\_donor\\_variant                                               &     3 &    2\\\\\n",
       "\t splice\\_region\\_variant\\&3\\_prime\\_UTR\\_variant\\&NMD\\_transcript\\_variant   &     5 &    2\\\\\n",
       "\t splice\\_region\\_variant\\&5\\_prime\\_UTR\\_variant                          &     6 &    4\\\\\n",
       "\t splice\\_region\\_variant\\&5\\_prime\\_UTR\\_variant\\&NMD\\_transcript\\_variant   &     2 &    2\\\\\n",
       "\t splice\\_region\\_variant\\&intron\\_variant                               &    92 &   29\\\\\n",
       "\t splice\\_region\\_variant\\&intron\\_variant\\&NMD\\_transcript\\_variant        &    53 &   15\\\\\n",
       "\t splice\\_region\\_variant\\&intron\\_variant\\&non\\_coding\\_transcript\\_variant &    58 &   31\\\\\n",
       "\t splice\\_region\\_variant\\&non\\_coding\\_transcript\\_exon\\_variant           &    49 &   13\\\\\n",
       "\t splice\\_region\\_variant\\&synonymous\\_variant                           &    21 &    5\\\\\n",
       "\t splice\\_region\\_variant\\&synonymous\\_variant\\&NMD\\_transcript\\_variant    &     7 &    3\\\\\n",
       "\t stop\\_gained                                                        &     6 &    2\\\\\n",
       "\t stop\\_lost                                                          &     8 &    4\\\\\n",
       "\t stop\\_lost\\&NMD\\_transcript\\_variant                                   &     1 &    1\\\\\n",
       "\t stop\\_retained\\_variant                                              &     1 &    1\\\\\n",
       "\t synonymous\\_variant                                                 &   648 &  205\\\\\n",
       "\t synonymous\\_variant\\&NMD\\_transcript\\_variant                          &    67 &   29\\\\\n",
       "\t TF\\_binding\\_site\\_variant                                            &  1446 &  532\\\\\n",
       "\t upstream\\_gene\\_variant                                              & 16712 & 4292\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 38 × 3\n",
       "\n",
       "| Consequence &lt;chr&gt; | n &lt;int&gt; | egenes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 3_prime_UTR_variant                                                |   996 |  524 |\n",
       "| 3_prime_UTR_variant&amp;NMD_transcript_variant                         |   363 |  205 |\n",
       "| 5_prime_UTR_variant                                                |   501 |  227 |\n",
       "| 5_prime_UTR_variant&amp;NMD_transcript_variant                         |    72 |   44 |\n",
       "| downstream_gene_variant                                            | 14899 | 4175 |\n",
       "| intergenic_variant                                                 |  1035 | 1035 |\n",
       "| intron_variant                                                     | 22869 | 5871 |\n",
       "| intron_variant&amp;NMD_transcript_variant                              |  5618 | 2444 |\n",
       "| intron_variant&amp;non_coding_transcript_variant                       | 12813 | 4488 |\n",
       "| missense_variant                                                   |   820 |  304 |\n",
       "| missense_variant&amp;NMD_transcript_variant                            |   130 |   67 |\n",
       "| missense_variant&amp;splice_region_variant                             |     7 |    5 |\n",
       "| missense_variant&amp;splice_region_variant&amp;NMD_transcript_variant      |     1 |    1 |\n",
       "| missense_variant&amp;stop_retained_variant                             |     2 |    2 |\n",
       "| missense_variant&amp;stop_retained_variant&amp;NMD_transcript_variant      |     1 |    1 |\n",
       "| non_coding_transcript_exon_variant                                 |  2285 | 1171 |\n",
       "| regulatory_region_variant                                          |  3761 | 3464 |\n",
       "| splice_acceptor_variant                                            |    10 |    2 |\n",
       "| splice_acceptor_variant&amp;NMD_transcript_variant                     |     5 |    1 |\n",
       "| splice_acceptor_variant&amp;non_coding_transcript_variant              |    12 |    5 |\n",
       "| splice_donor_variant                                               |     3 |    2 |\n",
       "| splice_region_variant&amp;3_prime_UTR_variant&amp;NMD_transcript_variant   |     5 |    2 |\n",
       "| splice_region_variant&amp;5_prime_UTR_variant                          |     6 |    4 |\n",
       "| splice_region_variant&amp;5_prime_UTR_variant&amp;NMD_transcript_variant   |     2 |    2 |\n",
       "| splice_region_variant&amp;intron_variant                               |    92 |   29 |\n",
       "| splice_region_variant&amp;intron_variant&amp;NMD_transcript_variant        |    53 |   15 |\n",
       "| splice_region_variant&amp;intron_variant&amp;non_coding_transcript_variant |    58 |   31 |\n",
       "| splice_region_variant&amp;non_coding_transcript_exon_variant           |    49 |   13 |\n",
       "| splice_region_variant&amp;synonymous_variant                           |    21 |    5 |\n",
       "| splice_region_variant&amp;synonymous_variant&amp;NMD_transcript_variant    |     7 |    3 |\n",
       "| stop_gained                                                        |     6 |    2 |\n",
       "| stop_lost                                                          |     8 |    4 |\n",
       "| stop_lost&amp;NMD_transcript_variant                                   |     1 |    1 |\n",
       "| stop_retained_variant                                              |     1 |    1 |\n",
       "| synonymous_variant                                                 |   648 |  205 |\n",
       "| synonymous_variant&amp;NMD_transcript_variant                          |    67 |   29 |\n",
       "| TF_binding_site_variant                                            |  1446 |  532 |\n",
       "| upstream_gene_variant                                              | 16712 | 4292 |\n",
       "\n"
      ],
      "text/plain": [
       "   Consequence                                                        n    \n",
       "1  3_prime_UTR_variant                                                  996\n",
       "2  3_prime_UTR_variant&NMD_transcript_variant                           363\n",
       "3  5_prime_UTR_variant                                                  501\n",
       "4  5_prime_UTR_variant&NMD_transcript_variant                            72\n",
       "5  downstream_gene_variant                                            14899\n",
       "6  intergenic_variant                                                  1035\n",
       "7  intron_variant                                                     22869\n",
       "8  intron_variant&NMD_transcript_variant                               5618\n",
       "9  intron_variant&non_coding_transcript_variant                       12813\n",
       "10 missense_variant                                                     820\n",
       "11 missense_variant&NMD_transcript_variant                              130\n",
       "12 missense_variant&splice_region_variant                                 7\n",
       "13 missense_variant&splice_region_variant&NMD_transcript_variant          1\n",
       "14 missense_variant&stop_retained_variant                                 2\n",
       "15 missense_variant&stop_retained_variant&NMD_transcript_variant          1\n",
       "16 non_coding_transcript_exon_variant                                  2285\n",
       "17 regulatory_region_variant                                           3761\n",
       "18 splice_acceptor_variant                                               10\n",
       "19 splice_acceptor_variant&NMD_transcript_variant                         5\n",
       "20 splice_acceptor_variant&non_coding_transcript_variant                 12\n",
       "21 splice_donor_variant                                                   3\n",
       "22 splice_region_variant&3_prime_UTR_variant&NMD_transcript_variant       5\n",
       "23 splice_region_variant&5_prime_UTR_variant                              6\n",
       "24 splice_region_variant&5_prime_UTR_variant&NMD_transcript_variant       2\n",
       "25 splice_region_variant&intron_variant                                  92\n",
       "26 splice_region_variant&intron_variant&NMD_transcript_variant           53\n",
       "27 splice_region_variant&intron_variant&non_coding_transcript_variant    58\n",
       "28 splice_region_variant&non_coding_transcript_exon_variant              49\n",
       "29 splice_region_variant&synonymous_variant                              21\n",
       "30 splice_region_variant&synonymous_variant&NMD_transcript_variant        7\n",
       "31 stop_gained                                                            6\n",
       "32 stop_lost                                                              8\n",
       "33 stop_lost&NMD_transcript_variant                                       1\n",
       "34 stop_retained_variant                                                  1\n",
       "35 synonymous_variant                                                   648\n",
       "36 synonymous_variant&NMD_transcript_variant                             67\n",
       "37 TF_binding_site_variant                                             1446\n",
       "38 upstream_gene_variant                                              16712\n",
       "   egenes\n",
       "1   524  \n",
       "2   205  \n",
       "3   227  \n",
       "4    44  \n",
       "5  4175  \n",
       "6  1035  \n",
       "7  5871  \n",
       "8  2444  \n",
       "9  4488  \n",
       "10  304  \n",
       "11   67  \n",
       "12    5  \n",
       "13    1  \n",
       "14    2  \n",
       "15    1  \n",
       "16 1171  \n",
       "17 3464  \n",
       "18    2  \n",
       "19    1  \n",
       "20    5  \n",
       "21    2  \n",
       "22    2  \n",
       "23    4  \n",
       "24    2  \n",
       "25   29  \n",
       "26   15  \n",
       "27   31  \n",
       "28   13  \n",
       "29    5  \n",
       "30    3  \n",
       "31    2  \n",
       "32    4  \n",
       "33    1  \n",
       "34    1  \n",
       "35  205  \n",
       "36   29  \n",
       "37  532  \n",
       "38 4292  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cis.eqtl.vep %>%\n",
    "    dplyr::select(-REF, -ALT, -Allele, -CADD_PHRED, -CADD_RAW, -PolyPhen) %>%\n",
    "    unique() %>%\n",
    "    dplyr::count(gene_id, Consequence) %>%\n",
    "    dplyr::group_by(Consequence) %>%\n",
    "    dplyr::summarize(n=sum(n), egenes=n())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "3bbd2e99-b726-4e26-b8c6-43f5a87dfc9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 12 × 3 of type chr</caption>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>gene_id</th><td>ENSG00000059691                        </td><td>ENSG00000059691                        </td><td>ENSG00000059691                        </td></tr>\n",
       "\t<tr><th scope=row>CHROM</th><td>4                                      </td><td>4                                      </td><td>4                                      </td></tr>\n",
       "\t<tr><th scope=row>POS</th><td>151760894                              </td><td>151760894                              </td><td>151760894                              </td></tr>\n",
       "\t<tr><th scope=row>ID</th><td>rs11556167                             </td><td>rs11556167                             </td><td>rs11556167                             </td></tr>\n",
       "\t<tr><th scope=row>Gene</th><td>ENSG00000059691                        </td><td>ENSG00000059691                        </td><td>ENSG00000059691                        </td></tr>\n",
       "\t<tr><th scope=row>Feature_type</th><td>Transcript                             </td><td>Transcript                             </td><td>Transcript                             </td></tr>\n",
       "\t<tr><th scope=row>Feature</th><td>ENST00000511538                        </td><td>ENST00000503160                        </td><td>ENST00000515884                        </td></tr>\n",
       "\t<tr><th scope=row>Consequence</th><td>missense_variant&amp;NMD_transcript_variant</td><td>missense_variant&amp;NMD_transcript_variant</td><td>missense_variant&amp;NMD_transcript_variant</td></tr>\n",
       "\t<tr><th scope=row>IMPACT</th><td>MODERATE                               </td><td>MODERATE                               </td><td>MODERATE                               </td></tr>\n",
       "\t<tr><th scope=row>BIOTYPE</th><td>nonsense_mediated_decay                </td><td>nonsense_mediated_decay                </td><td>nonsense_mediated_decay                </td></tr>\n",
       "\t<tr><th scope=row>MOTIF_NAME</th><td>NA                                     </td><td>NA                                     </td><td>NA                                     </td></tr>\n",
       "\t<tr><th scope=row>TRANSCRIPTION_FACTORS</th><td>NA                                     </td><td>NA                                     </td><td>NA                                     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 12 × 3 of type chr\n",
       "\\begin{tabular}{r|lll}\n",
       "\tgene\\_id & ENSG00000059691                         & ENSG00000059691                         & ENSG00000059691                        \\\\\n",
       "\tCHROM & 4                                       & 4                                       & 4                                      \\\\\n",
       "\tPOS & 151760894                               & 151760894                               & 151760894                              \\\\\n",
       "\tID & rs11556167                              & rs11556167                              & rs11556167                             \\\\\n",
       "\tGene & ENSG00000059691                         & ENSG00000059691                         & ENSG00000059691                        \\\\\n",
       "\tFeature\\_type & Transcript                              & Transcript                              & Transcript                             \\\\\n",
       "\tFeature & ENST00000511538                         & ENST00000503160                         & ENST00000515884                        \\\\\n",
       "\tConsequence & missense\\_variant\\&NMD\\_transcript\\_variant & missense\\_variant\\&NMD\\_transcript\\_variant & missense\\_variant\\&NMD\\_transcript\\_variant\\\\\n",
       "\tIMPACT & MODERATE                                & MODERATE                                & MODERATE                               \\\\\n",
       "\tBIOTYPE & nonsense\\_mediated\\_decay                 & nonsense\\_mediated\\_decay                 & nonsense\\_mediated\\_decay                \\\\\n",
       "\tMOTIF\\_NAME & NA                                      & NA                                      & NA                                     \\\\\n",
       "\tTRANSCRIPTION\\_FACTORS & NA                                      & NA                                      & NA                                     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 12 × 3 of type chr\n",
       "\n",
       "| gene_id | ENSG00000059691                         | ENSG00000059691                         | ENSG00000059691                         |\n",
       "| CHROM | 4                                       | 4                                       | 4                                       |\n",
       "| POS | 151760894                               | 151760894                               | 151760894                               |\n",
       "| ID | rs11556167                              | rs11556167                              | rs11556167                              |\n",
       "| Gene | ENSG00000059691                         | ENSG00000059691                         | ENSG00000059691                         |\n",
       "| Feature_type | Transcript                              | Transcript                              | Transcript                              |\n",
       "| Feature | ENST00000511538                         | ENST00000503160                         | ENST00000515884                         |\n",
       "| Consequence | missense_variant&amp;NMD_transcript_variant | missense_variant&amp;NMD_transcript_variant | missense_variant&amp;NMD_transcript_variant |\n",
       "| IMPACT | MODERATE                                | MODERATE                                | MODERATE                                |\n",
       "| BIOTYPE | nonsense_mediated_decay                 | nonsense_mediated_decay                 | nonsense_mediated_decay                 |\n",
       "| MOTIF_NAME | NA                                      | NA                                      | NA                                      |\n",
       "| TRANSCRIPTION_FACTORS | NA                                      | NA                                      | NA                                      |\n",
       "\n"
      ],
      "text/plain": [
       "                      [,1]                                   \n",
       "gene_id               ENSG00000059691                        \n",
       "CHROM                 4                                      \n",
       "POS                   151760894                              \n",
       "ID                    rs11556167                             \n",
       "Gene                  ENSG00000059691                        \n",
       "Feature_type          Transcript                             \n",
       "Feature               ENST00000511538                        \n",
       "Consequence           missense_variant&NMD_transcript_variant\n",
       "IMPACT                MODERATE                               \n",
       "BIOTYPE               nonsense_mediated_decay                \n",
       "MOTIF_NAME            NA                                     \n",
       "TRANSCRIPTION_FACTORS NA                                     \n",
       "                      [,2]                                   \n",
       "gene_id               ENSG00000059691                        \n",
       "CHROM                 4                                      \n",
       "POS                   151760894                              \n",
       "ID                    rs11556167                             \n",
       "Gene                  ENSG00000059691                        \n",
       "Feature_type          Transcript                             \n",
       "Feature               ENST00000503160                        \n",
       "Consequence           missense_variant&NMD_transcript_variant\n",
       "IMPACT                MODERATE                               \n",
       "BIOTYPE               nonsense_mediated_decay                \n",
       "MOTIF_NAME            NA                                     \n",
       "TRANSCRIPTION_FACTORS NA                                     \n",
       "                      [,3]                                   \n",
       "gene_id               ENSG00000059691                        \n",
       "CHROM                 4                                      \n",
       "POS                   151760894                              \n",
       "ID                    rs11556167                             \n",
       "Gene                  ENSG00000059691                        \n",
       "Feature_type          Transcript                             \n",
       "Feature               ENST00000515884                        \n",
       "Consequence           missense_variant&NMD_transcript_variant\n",
       "IMPACT                MODERATE                               \n",
       "BIOTYPE               nonsense_mediated_decay                \n",
       "MOTIF_NAME            NA                                     \n",
       "TRANSCRIPTION_FACTORS NA                                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "t(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "768405f5-7042-4dd9-8256-7150b419338a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
