{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "40ded556-debe-4c09-a412-cb0c8c47f474",
   "metadata": {},
   "source": [
    "# Eigengene Single-Variant Association Preparation\n",
    "\n",
    "**Created**: 25 November 2021"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "133acb3a-8f08-46dc-bab6-7c8d4afe4803",
   "metadata": {},
   "source": [
    "## Environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5aa6f310-53bc-4e4a-9f26-62e0716b2c2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!requireNamespace(\"BiocManager\", quietly = TRUE)) {\n",
    "    install.packages(\"BiocManager\")\n",
    "}\n",
    "\n",
    "if (!requireNamespace(\"ComplexHeatmap\", quietly=TRUE)) {\n",
    "    BiocManager::install(\"ComplexHeatmap\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b57f682b-54fb-470e-b766-b9f21ae2d9b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.6.2\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "  genomic data. Bioinformatics 2016.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(RColorBrewer)\n",
    "library(ComplexHeatmap)\n",
    "library(data.table)\n",
    "\n",
    "setwd(\"~/eQTL_pQTL_Characterization/\")\n",
    "\n",
    "source(\"04_Expression/scripts/utils/ggplot_theme.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53e08f67-624c-4222-b51c-026831dd085b",
   "metadata": {},
   "source": [
    "## Load Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e6264bda-1395-47a9-bf6d-5b5228984a14",
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics <- readxl::read_xls(\"/nfs/team282/data/gains_team282/ClinicalData/DEMO_12jun2019.xls\", sheet=1) %>%\n",
    "    dplyr::mutate(sex=as.numeric(sex)) %>%\n",
    "    dplyr::mutate(diagnosis=as.numeric(diagnosis)) %>%\n",
    "    dplyr::mutate(SubjectBarCode=gsub(\"^GA\", \"\", SubjectBarCode))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "35ab229e-25c4-404d-8390-01f32b34e94f",
   "metadata": {},
   "outputs": [],
   "source": [
    "srs.info <- read.table(\"/nfs/team282/data/gains_team282/full-gains-SRS-predictions_mNN-RF.tsv\", header=T) %>%\n",
    "    dplyr::mutate(Sample_id=gsub(\"^GA\", \"\", Sample_id)) %>%\n",
    "    dplyr::filter(Assay==\"RNA-seq\")\n",
    "\n",
    "rownames(srs.info) <- srs.info$Sample_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0b1b5e57-4df2-42c1-b77d-10f84cff585d",
   "metadata": {},
   "outputs": [],
   "source": [
    "covs <- read.table(\"~/gains_team282/eqtl/data/covs_and_peer_factors.txt\") %>%\n",
    "    dplyr::mutate(Sample.ID=gsub(\"^GA\", \"\", rownames(.))) %>%\n",
    "    dplyr::mutate(GAinS.ID=gsub(\"\\\\_.\", \"\", Sample.ID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "778d7314-6a62-4e86-a35a-68daaabfe7d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 43</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Neutrophils</th><th scope=col>Lymphocytes</th><th scope=col>Monocytes</th><th scope=col>PC1</th><th scope=col>PC2</th><th scope=col>PC3</th><th scope=col>PC4</th><th scope=col>PC5</th><th scope=col>PC6</th><th scope=col>PC7</th><th scope=col>⋯</th><th scope=col>PEER_21</th><th scope=col>PEER_22</th><th scope=col>PEER_23</th><th scope=col>PEER_24</th><th scope=col>PEER_25</th><th scope=col>PEER_26</th><th scope=col>PEER_27</th><th scope=col>PEER_28</th><th scope=col>PEER_29</th><th scope=col>PEER_30</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>UK01050129_5</th><td>-1.7381318</td><td> 1.8281826</td><td> 1.2829455</td><td> 0.177279994</td><td> 0.051694602</td><td>0.02486030</td><td>-0.035102502</td><td>-0.00455149</td><td>-0.0603707</td><td> 0.03964880</td><td>⋯</td><td>-0.001172263</td><td>-0.05533835</td><td> 0.049742281</td><td> 0.03725176</td><td> 0.04495762</td><td> 0.12696229</td><td> 0.08175036</td><td> 0.05295999</td><td>0.164950684</td><td>-0.10379386</td></tr>\n",
       "\t<tr><th scope=row>UK01060123_3</th><td>-0.3845370</td><td> 0.3878375</td><td> 0.4548023</td><td>-0.000388133</td><td>-0.000674099</td><td>0.00810357</td><td> 0.000212791</td><td> 0.03396150</td><td> 0.0266317</td><td>-0.01885280</td><td>⋯</td><td>-0.012723563</td><td> 0.02226736</td><td>-0.027520355</td><td>-0.05782387</td><td> 0.03653939</td><td> 0.05959492</td><td>-0.02611259</td><td>-0.01790827</td><td>0.012047987</td><td> 0.05554044</td></tr>\n",
       "\t<tr><th scope=row>UK01070117_3</th><td>-0.7738039</td><td> 0.5846032</td><td> 0.9325467</td><td> 0.016432300</td><td>-0.048464298</td><td>0.00813966</td><td>-0.047699600</td><td>-0.07926830</td><td> 0.0263629</td><td>-0.08224740</td><td>⋯</td><td> 0.040162142</td><td> 0.06840651</td><td> 0.000062529</td><td> 0.05292596</td><td> 0.01705806</td><td> 0.05931079</td><td> 0.08962730</td><td> 0.03772458</td><td>0.008254956</td><td>-0.01570964</td></tr>\n",
       "\t<tr><th scope=row>UK01080111_1</th><td> 0.1305646</td><td> 0.2888638</td><td>-0.6232047</td><td> 0.161789000</td><td> 0.036224801</td><td>0.00467903</td><td>-0.048611499</td><td> 0.00180036</td><td>-0.0352357</td><td> 0.05012510</td><td>⋯</td><td>-0.002928361</td><td>-0.03274319</td><td> 0.088269033</td><td> 0.07396927</td><td>-0.01906336</td><td> 0.02454052</td><td> 0.04243371</td><td> 0.03055887</td><td>0.123383760</td><td>-0.02276623</td></tr>\n",
       "\t<tr><th scope=row>UK01110093_1</th><td> 1.0909779</td><td>-1.2488160</td><td>-0.4429439</td><td>-0.006296720</td><td> 0.002821510</td><td>0.00699239</td><td> 0.004351650</td><td> 0.00268802</td><td> 0.0166922</td><td>-0.00733983</td><td>⋯</td><td>-0.108193189</td><td> 0.01590979</td><td>-0.093242317</td><td>-0.03037383</td><td>-0.07385907</td><td>-0.05968965</td><td> 0.09695430</td><td>-0.03975484</td><td>0.130198911</td><td>-0.05011296</td></tr>\n",
       "\t<tr><th scope=row>UK01110093_5</th><td> 0.2302416</td><td> 0.0000000</td><td>-0.3306479</td><td>-0.006296720</td><td> 0.002821510</td><td>0.00699239</td><td> 0.004351650</td><td> 0.00268802</td><td> 0.0166922</td><td>-0.00733983</td><td>⋯</td><td>-0.018054215</td><td> 0.02094322</td><td>-0.008623131</td><td> 0.09560122</td><td>-0.07802416</td><td> 0.02365689</td><td> 0.08040638</td><td>-0.02349218</td><td>0.189585552</td><td>-0.07635343</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 43\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Neutrophils & Lymphocytes & Monocytes & PC1 & PC2 & PC3 & PC4 & PC5 & PC6 & PC7 & ⋯ & PEER\\_21 & PEER\\_22 & PEER\\_23 & PEER\\_24 & PEER\\_25 & PEER\\_26 & PEER\\_27 & PEER\\_28 & PEER\\_29 & PEER\\_30\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tUK01050129\\_5 & -1.7381318 &  1.8281826 &  1.2829455 &  0.177279994 &  0.051694602 & 0.02486030 & -0.035102502 & -0.00455149 & -0.0603707 &  0.03964880 & ⋯ & -0.001172263 & -0.05533835 &  0.049742281 &  0.03725176 &  0.04495762 &  0.12696229 &  0.08175036 &  0.05295999 & 0.164950684 & -0.10379386\\\\\n",
       "\tUK01060123\\_3 & -0.3845370 &  0.3878375 &  0.4548023 & -0.000388133 & -0.000674099 & 0.00810357 &  0.000212791 &  0.03396150 &  0.0266317 & -0.01885280 & ⋯ & -0.012723563 &  0.02226736 & -0.027520355 & -0.05782387 &  0.03653939 &  0.05959492 & -0.02611259 & -0.01790827 & 0.012047987 &  0.05554044\\\\\n",
       "\tUK01070117\\_3 & -0.7738039 &  0.5846032 &  0.9325467 &  0.016432300 & -0.048464298 & 0.00813966 & -0.047699600 & -0.07926830 &  0.0263629 & -0.08224740 & ⋯ &  0.040162142 &  0.06840651 &  0.000062529 &  0.05292596 &  0.01705806 &  0.05931079 &  0.08962730 &  0.03772458 & 0.008254956 & -0.01570964\\\\\n",
       "\tUK01080111\\_1 &  0.1305646 &  0.2888638 & -0.6232047 &  0.161789000 &  0.036224801 & 0.00467903 & -0.048611499 &  0.00180036 & -0.0352357 &  0.05012510 & ⋯ & -0.002928361 & -0.03274319 &  0.088269033 &  0.07396927 & -0.01906336 &  0.02454052 &  0.04243371 &  0.03055887 & 0.123383760 & -0.02276623\\\\\n",
       "\tUK01110093\\_1 &  1.0909779 & -1.2488160 & -0.4429439 & -0.006296720 &  0.002821510 & 0.00699239 &  0.004351650 &  0.00268802 &  0.0166922 & -0.00733983 & ⋯ & -0.108193189 &  0.01590979 & -0.093242317 & -0.03037383 & -0.07385907 & -0.05968965 &  0.09695430 & -0.03975484 & 0.130198911 & -0.05011296\\\\\n",
       "\tUK01110093\\_5 &  0.2302416 &  0.0000000 & -0.3306479 & -0.006296720 &  0.002821510 & 0.00699239 &  0.004351650 &  0.00268802 &  0.0166922 & -0.00733983 & ⋯ & -0.018054215 &  0.02094322 & -0.008623131 &  0.09560122 & -0.07802416 &  0.02365689 &  0.08040638 & -0.02349218 & 0.189585552 & -0.07635343\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 43\n",
       "\n",
       "| <!--/--> | Neutrophils &lt;dbl&gt; | Lymphocytes &lt;dbl&gt; | Monocytes &lt;dbl&gt; | PC1 &lt;dbl&gt; | PC2 &lt;dbl&gt; | PC3 &lt;dbl&gt; | PC4 &lt;dbl&gt; | PC5 &lt;dbl&gt; | PC6 &lt;dbl&gt; | PC7 &lt;dbl&gt; | ⋯ ⋯ | PEER_21 &lt;dbl&gt; | PEER_22 &lt;dbl&gt; | PEER_23 &lt;dbl&gt; | PEER_24 &lt;dbl&gt; | PEER_25 &lt;dbl&gt; | PEER_26 &lt;dbl&gt; | PEER_27 &lt;dbl&gt; | PEER_28 &lt;dbl&gt; | PEER_29 &lt;dbl&gt; | PEER_30 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| UK01050129_5 | -1.7381318 |  1.8281826 |  1.2829455 |  0.177279994 |  0.051694602 | 0.02486030 | -0.035102502 | -0.00455149 | -0.0603707 |  0.03964880 | ⋯ | -0.001172263 | -0.05533835 |  0.049742281 |  0.03725176 |  0.04495762 |  0.12696229 |  0.08175036 |  0.05295999 | 0.164950684 | -0.10379386 |\n",
       "| UK01060123_3 | -0.3845370 |  0.3878375 |  0.4548023 | -0.000388133 | -0.000674099 | 0.00810357 |  0.000212791 |  0.03396150 |  0.0266317 | -0.01885280 | ⋯ | -0.012723563 |  0.02226736 | -0.027520355 | -0.05782387 |  0.03653939 |  0.05959492 | -0.02611259 | -0.01790827 | 0.012047987 |  0.05554044 |\n",
       "| UK01070117_3 | -0.7738039 |  0.5846032 |  0.9325467 |  0.016432300 | -0.048464298 | 0.00813966 | -0.047699600 | -0.07926830 |  0.0263629 | -0.08224740 | ⋯ |  0.040162142 |  0.06840651 |  0.000062529 |  0.05292596 |  0.01705806 |  0.05931079 |  0.08962730 |  0.03772458 | 0.008254956 | -0.01570964 |\n",
       "| UK01080111_1 |  0.1305646 |  0.2888638 | -0.6232047 |  0.161789000 |  0.036224801 | 0.00467903 | -0.048611499 |  0.00180036 | -0.0352357 |  0.05012510 | ⋯ | -0.002928361 | -0.03274319 |  0.088269033 |  0.07396927 | -0.01906336 |  0.02454052 |  0.04243371 |  0.03055887 | 0.123383760 | -0.02276623 |\n",
       "| UK01110093_1 |  1.0909779 | -1.2488160 | -0.4429439 | -0.006296720 |  0.002821510 | 0.00699239 |  0.004351650 |  0.00268802 |  0.0166922 | -0.00733983 | ⋯ | -0.108193189 |  0.01590979 | -0.093242317 | -0.03037383 | -0.07385907 | -0.05968965 |  0.09695430 | -0.03975484 | 0.130198911 | -0.05011296 |\n",
       "| UK01110093_5 |  0.2302416 |  0.0000000 | -0.3306479 | -0.006296720 |  0.002821510 | 0.00699239 |  0.004351650 |  0.00268802 |  0.0166922 | -0.00733983 | ⋯ | -0.018054215 |  0.02094322 | -0.008623131 |  0.09560122 | -0.07802416 |  0.02365689 |  0.08040638 | -0.02349218 | 0.189585552 | -0.07635343 |\n",
       "\n"
      ],
      "text/plain": [
       "             Neutrophils Lymphocytes Monocytes  PC1          PC2         \n",
       "UK01050129_5 -1.7381318   1.8281826   1.2829455  0.177279994  0.051694602\n",
       "UK01060123_3 -0.3845370   0.3878375   0.4548023 -0.000388133 -0.000674099\n",
       "UK01070117_3 -0.7738039   0.5846032   0.9325467  0.016432300 -0.048464298\n",
       "UK01080111_1  0.1305646   0.2888638  -0.6232047  0.161789000  0.036224801\n",
       "UK01110093_1  1.0909779  -1.2488160  -0.4429439 -0.006296720  0.002821510\n",
       "UK01110093_5  0.2302416   0.0000000  -0.3306479 -0.006296720  0.002821510\n",
       "             PC3        PC4          PC5         PC6        PC7         ⋯\n",
       "UK01050129_5 0.02486030 -0.035102502 -0.00455149 -0.0603707  0.03964880 ⋯\n",
       "UK01060123_3 0.00810357  0.000212791  0.03396150  0.0266317 -0.01885280 ⋯\n",
       "UK01070117_3 0.00813966 -0.047699600 -0.07926830  0.0263629 -0.08224740 ⋯\n",
       "UK01080111_1 0.00467903 -0.048611499  0.00180036 -0.0352357  0.05012510 ⋯\n",
       "UK01110093_1 0.00699239  0.004351650  0.00268802  0.0166922 -0.00733983 ⋯\n",
       "UK01110093_5 0.00699239  0.004351650  0.00268802  0.0166922 -0.00733983 ⋯\n",
       "             PEER_21      PEER_22     PEER_23      PEER_24     PEER_25    \n",
       "UK01050129_5 -0.001172263 -0.05533835  0.049742281  0.03725176  0.04495762\n",
       "UK01060123_3 -0.012723563  0.02226736 -0.027520355 -0.05782387  0.03653939\n",
       "UK01070117_3  0.040162142  0.06840651  0.000062529  0.05292596  0.01705806\n",
       "UK01080111_1 -0.002928361 -0.03274319  0.088269033  0.07396927 -0.01906336\n",
       "UK01110093_1 -0.108193189  0.01590979 -0.093242317 -0.03037383 -0.07385907\n",
       "UK01110093_5 -0.018054215  0.02094322 -0.008623131  0.09560122 -0.07802416\n",
       "             PEER_26     PEER_27     PEER_28     PEER_29     PEER_30    \n",
       "UK01050129_5  0.12696229  0.08175036  0.05295999 0.164950684 -0.10379386\n",
       "UK01060123_3  0.05959492 -0.02611259 -0.01790827 0.012047987  0.05554044\n",
       "UK01070117_3  0.05931079  0.08962730  0.03772458 0.008254956 -0.01570964\n",
       "UK01080111_1  0.02454052  0.04243371  0.03055887 0.123383760 -0.02276623\n",
       "UK01110093_1 -0.05968965  0.09695430 -0.03975484 0.130198911 -0.05011296\n",
       "UK01110093_5  0.02365689  0.08040638 -0.02349218 0.189585552 -0.07635343"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "held.out <- c(\"Neutrophils\", \"Lymphocytes\", \"Monocytes\", paste0(\"PC\", 1:7), \"Diagnosis\", \"SRSq\", \"sex\")\n",
    "peer <- paste0(\"PEER_\", 1:30)\n",
    "\n",
    "covs <- covs %>%\n",
    "    merge(., demographics, by.x=\"GAinS.ID\", by.y=\"SubjectBarCode\") %>%\n",
    "    merge(., srs.info, by.x=\"Sample.ID\", by.y=0) %>%\n",
    "    dplyr::select(diagnosis=diagnosis, everything()) %>%\n",
    "    dplyr::select(Sample.ID, any_of(held.out), any_of(peer)) %>%\n",
    "    as.data.frame()\n",
    "                                  \n",
    "rownames(covs) <- covs$Sample.ID\n",
    "covs <- covs %>%\n",
    "    dplyr::select(-Sample.ID)\n",
    "                                  \n",
    "head(covs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "31b17104-f1be-4f75-a7c2-791888eb337f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 106</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ME_1</th><th scope=col>ME_2</th><th scope=col>ME_3</th><th scope=col>ME_4</th><th scope=col>ME_5</th><th scope=col>ME_6</th><th scope=col>ME_7</th><th scope=col>ME_8</th><th scope=col>ME_9</th><th scope=col>ME_10</th><th scope=col>⋯</th><th scope=col>ME_97</th><th scope=col>ME_98</th><th scope=col>ME_99</th><th scope=col>ME_100</th><th scope=col>ME_101</th><th scope=col>ME_102</th><th scope=col>ME_103</th><th scope=col>ME_104</th><th scope=col>ME_105</th><th scope=col>ME_106</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>UK02270173_3</th><td> 0.008640296</td><td> 0.007926439</td><td> 0.0147219682</td><td>-0.011774847</td><td> 0.029042358</td><td>-0.006588595</td><td>-0.004034583</td><td>-0.013647582</td><td> 0.020445938</td><td> 0.03657146</td><td>⋯</td><td> 0.105461618</td><td> 0.014671996</td><td>-0.03186240</td><td>-0.019044546</td><td>-0.001067989</td><td> 0.03791366</td><td> 0.008340969</td><td>-0.002505741</td><td> 0.033258706</td><td>-0.003317747</td></tr>\n",
       "\t<tr><th scope=row>UK15130120_3</th><td>-0.037073495</td><td>-0.033179470</td><td>-0.0311931107</td><td>-0.006189163</td><td> 0.039065099</td><td>-0.011809051</td><td> 0.037283445</td><td>-0.060021257</td><td>-0.036280957</td><td> 0.03133065</td><td>⋯</td><td>-0.006444524</td><td>-0.019642856</td><td> 0.01880373</td><td>-0.044684726</td><td>-0.043663066</td><td>-0.02550290</td><td> 0.031624201</td><td>-0.011931533</td><td> 0.051184001</td><td>-0.012327128</td></tr>\n",
       "\t<tr><th scope=row>UK58000006_3</th><td> 0.029670524</td><td> 0.021490110</td><td> 0.0369543066</td><td>-0.015010027</td><td>-0.004745170</td><td> 0.023133390</td><td>-0.027585817</td><td> 0.010804862</td><td> 0.044649369</td><td>-0.01014143</td><td>⋯</td><td> 0.085257258</td><td>-0.006499496</td><td>-0.01582150</td><td> 0.030709065</td><td> 0.040952839</td><td> 0.04332857</td><td>-0.003376643</td><td> 0.012633897</td><td>-0.006622840</td><td>-0.004740631</td></tr>\n",
       "\t<tr><th scope=row>UK47010004_3</th><td> 0.011954974</td><td> 0.007360515</td><td> 0.0073725862</td><td>-0.011608271</td><td>-0.011341294</td><td>-0.025127169</td><td>-0.009500677</td><td> 0.013387153</td><td> 0.007126337</td><td>-0.02607067</td><td>⋯</td><td> 0.075920669</td><td>-0.016223383</td><td> 0.02515734</td><td>-0.008023902</td><td> 0.018023055</td><td>-0.01461801</td><td> 0.010948287</td><td> 0.023381207</td><td>-0.003072117</td><td> 0.012014706</td></tr>\n",
       "\t<tr><th scope=row>UK42020088_5</th><td> 0.009818957</td><td> 0.008322535</td><td> 0.0004121376</td><td>-0.008882007</td><td>-0.028000468</td><td> 0.002082996</td><td>-0.009532910</td><td> 0.011477800</td><td>-0.001521594</td><td>-0.02223628</td><td>⋯</td><td> 0.035896079</td><td> 0.009274050</td><td> 0.03406072</td><td>-0.005459722</td><td> 0.037120618</td><td>-0.01089588</td><td>-0.008355454</td><td>-0.034402696</td><td>-0.032697222</td><td> 0.007915598</td></tr>\n",
       "\t<tr><th scope=row>UK47490007_3</th><td>-0.021930377</td><td>-0.024756322</td><td>-0.0294342468</td><td>-0.007742353</td><td>-0.005486842</td><td>-0.049842615</td><td> 0.020825112</td><td> 0.007441717</td><td>-0.030615907</td><td>-0.01853294</td><td>⋯</td><td> 0.092538453</td><td> 0.067616309</td><td> 0.02790283</td><td>-0.036887001</td><td>-0.017742254</td><td> 0.01562787</td><td> 0.026717734</td><td>-0.009877409</td><td>-0.003607270</td><td> 0.024871961</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 106\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ME\\_1 & ME\\_2 & ME\\_3 & ME\\_4 & ME\\_5 & ME\\_6 & ME\\_7 & ME\\_8 & ME\\_9 & ME\\_10 & ⋯ & ME\\_97 & ME\\_98 & ME\\_99 & ME\\_100 & ME\\_101 & ME\\_102 & ME\\_103 & ME\\_104 & ME\\_105 & ME\\_106\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tUK02270173\\_3 &  0.008640296 &  0.007926439 &  0.0147219682 & -0.011774847 &  0.029042358 & -0.006588595 & -0.004034583 & -0.013647582 &  0.020445938 &  0.03657146 & ⋯ &  0.105461618 &  0.014671996 & -0.03186240 & -0.019044546 & -0.001067989 &  0.03791366 &  0.008340969 & -0.002505741 &  0.033258706 & -0.003317747\\\\\n",
       "\tUK15130120\\_3 & -0.037073495 & -0.033179470 & -0.0311931107 & -0.006189163 &  0.039065099 & -0.011809051 &  0.037283445 & -0.060021257 & -0.036280957 &  0.03133065 & ⋯ & -0.006444524 & -0.019642856 &  0.01880373 & -0.044684726 & -0.043663066 & -0.02550290 &  0.031624201 & -0.011931533 &  0.051184001 & -0.012327128\\\\\n",
       "\tUK58000006\\_3 &  0.029670524 &  0.021490110 &  0.0369543066 & -0.015010027 & -0.004745170 &  0.023133390 & -0.027585817 &  0.010804862 &  0.044649369 & -0.01014143 & ⋯ &  0.085257258 & -0.006499496 & -0.01582150 &  0.030709065 &  0.040952839 &  0.04332857 & -0.003376643 &  0.012633897 & -0.006622840 & -0.004740631\\\\\n",
       "\tUK47010004\\_3 &  0.011954974 &  0.007360515 &  0.0073725862 & -0.011608271 & -0.011341294 & -0.025127169 & -0.009500677 &  0.013387153 &  0.007126337 & -0.02607067 & ⋯ &  0.075920669 & -0.016223383 &  0.02515734 & -0.008023902 &  0.018023055 & -0.01461801 &  0.010948287 &  0.023381207 & -0.003072117 &  0.012014706\\\\\n",
       "\tUK42020088\\_5 &  0.009818957 &  0.008322535 &  0.0004121376 & -0.008882007 & -0.028000468 &  0.002082996 & -0.009532910 &  0.011477800 & -0.001521594 & -0.02223628 & ⋯ &  0.035896079 &  0.009274050 &  0.03406072 & -0.005459722 &  0.037120618 & -0.01089588 & -0.008355454 & -0.034402696 & -0.032697222 &  0.007915598\\\\\n",
       "\tUK47490007\\_3 & -0.021930377 & -0.024756322 & -0.0294342468 & -0.007742353 & -0.005486842 & -0.049842615 &  0.020825112 &  0.007441717 & -0.030615907 & -0.01853294 & ⋯ &  0.092538453 &  0.067616309 &  0.02790283 & -0.036887001 & -0.017742254 &  0.01562787 &  0.026717734 & -0.009877409 & -0.003607270 &  0.024871961\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 106\n",
       "\n",
       "| <!--/--> | ME_1 &lt;dbl&gt; | ME_2 &lt;dbl&gt; | ME_3 &lt;dbl&gt; | ME_4 &lt;dbl&gt; | ME_5 &lt;dbl&gt; | ME_6 &lt;dbl&gt; | ME_7 &lt;dbl&gt; | ME_8 &lt;dbl&gt; | ME_9 &lt;dbl&gt; | ME_10 &lt;dbl&gt; | ⋯ ⋯ | ME_97 &lt;dbl&gt; | ME_98 &lt;dbl&gt; | ME_99 &lt;dbl&gt; | ME_100 &lt;dbl&gt; | ME_101 &lt;dbl&gt; | ME_102 &lt;dbl&gt; | ME_103 &lt;dbl&gt; | ME_104 &lt;dbl&gt; | ME_105 &lt;dbl&gt; | ME_106 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| UK02270173_3 |  0.008640296 |  0.007926439 |  0.0147219682 | -0.011774847 |  0.029042358 | -0.006588595 | -0.004034583 | -0.013647582 |  0.020445938 |  0.03657146 | ⋯ |  0.105461618 |  0.014671996 | -0.03186240 | -0.019044546 | -0.001067989 |  0.03791366 |  0.008340969 | -0.002505741 |  0.033258706 | -0.003317747 |\n",
       "| UK15130120_3 | -0.037073495 | -0.033179470 | -0.0311931107 | -0.006189163 |  0.039065099 | -0.011809051 |  0.037283445 | -0.060021257 | -0.036280957 |  0.03133065 | ⋯ | -0.006444524 | -0.019642856 |  0.01880373 | -0.044684726 | -0.043663066 | -0.02550290 |  0.031624201 | -0.011931533 |  0.051184001 | -0.012327128 |\n",
       "| UK58000006_3 |  0.029670524 |  0.021490110 |  0.0369543066 | -0.015010027 | -0.004745170 |  0.023133390 | -0.027585817 |  0.010804862 |  0.044649369 | -0.01014143 | ⋯ |  0.085257258 | -0.006499496 | -0.01582150 |  0.030709065 |  0.040952839 |  0.04332857 | -0.003376643 |  0.012633897 | -0.006622840 | -0.004740631 |\n",
       "| UK47010004_3 |  0.011954974 |  0.007360515 |  0.0073725862 | -0.011608271 | -0.011341294 | -0.025127169 | -0.009500677 |  0.013387153 |  0.007126337 | -0.02607067 | ⋯ |  0.075920669 | -0.016223383 |  0.02515734 | -0.008023902 |  0.018023055 | -0.01461801 |  0.010948287 |  0.023381207 | -0.003072117 |  0.012014706 |\n",
       "| UK42020088_5 |  0.009818957 |  0.008322535 |  0.0004121376 | -0.008882007 | -0.028000468 |  0.002082996 | -0.009532910 |  0.011477800 | -0.001521594 | -0.02223628 | ⋯ |  0.035896079 |  0.009274050 |  0.03406072 | -0.005459722 |  0.037120618 | -0.01089588 | -0.008355454 | -0.034402696 | -0.032697222 |  0.007915598 |\n",
       "| UK47490007_3 | -0.021930377 | -0.024756322 | -0.0294342468 | -0.007742353 | -0.005486842 | -0.049842615 |  0.020825112 |  0.007441717 | -0.030615907 | -0.01853294 | ⋯ |  0.092538453 |  0.067616309 |  0.02790283 | -0.036887001 | -0.017742254 |  0.01562787 |  0.026717734 | -0.009877409 | -0.003607270 |  0.024871961 |\n",
       "\n"
      ],
      "text/plain": [
       "             ME_1         ME_2         ME_3          ME_4         ME_5        \n",
       "UK02270173_3  0.008640296  0.007926439  0.0147219682 -0.011774847  0.029042358\n",
       "UK15130120_3 -0.037073495 -0.033179470 -0.0311931107 -0.006189163  0.039065099\n",
       "UK58000006_3  0.029670524  0.021490110  0.0369543066 -0.015010027 -0.004745170\n",
       "UK47010004_3  0.011954974  0.007360515  0.0073725862 -0.011608271 -0.011341294\n",
       "UK42020088_5  0.009818957  0.008322535  0.0004121376 -0.008882007 -0.028000468\n",
       "UK47490007_3 -0.021930377 -0.024756322 -0.0294342468 -0.007742353 -0.005486842\n",
       "             ME_6         ME_7         ME_8         ME_9         ME_10       ⋯\n",
       "UK02270173_3 -0.006588595 -0.004034583 -0.013647582  0.020445938  0.03657146 ⋯\n",
       "UK15130120_3 -0.011809051  0.037283445 -0.060021257 -0.036280957  0.03133065 ⋯\n",
       "UK58000006_3  0.023133390 -0.027585817  0.010804862  0.044649369 -0.01014143 ⋯\n",
       "UK47010004_3 -0.025127169 -0.009500677  0.013387153  0.007126337 -0.02607067 ⋯\n",
       "UK42020088_5  0.002082996 -0.009532910  0.011477800 -0.001521594 -0.02223628 ⋯\n",
       "UK47490007_3 -0.049842615  0.020825112  0.007441717 -0.030615907 -0.01853294 ⋯\n",
       "             ME_97        ME_98        ME_99       ME_100       ME_101      \n",
       "UK02270173_3  0.105461618  0.014671996 -0.03186240 -0.019044546 -0.001067989\n",
       "UK15130120_3 -0.006444524 -0.019642856  0.01880373 -0.044684726 -0.043663066\n",
       "UK58000006_3  0.085257258 -0.006499496 -0.01582150  0.030709065  0.040952839\n",
       "UK47010004_3  0.075920669 -0.016223383  0.02515734 -0.008023902  0.018023055\n",
       "UK42020088_5  0.035896079  0.009274050  0.03406072 -0.005459722  0.037120618\n",
       "UK47490007_3  0.092538453  0.067616309  0.02790283 -0.036887001 -0.017742254\n",
       "             ME_102      ME_103       ME_104       ME_105       ME_106      \n",
       "UK02270173_3  0.03791366  0.008340969 -0.002505741  0.033258706 -0.003317747\n",
       "UK15130120_3 -0.02550290  0.031624201 -0.011931533  0.051184001 -0.012327128\n",
       "UK58000006_3  0.04332857 -0.003376643  0.012633897 -0.006622840 -0.004740631\n",
       "UK47010004_3 -0.01461801  0.010948287  0.023381207 -0.003072117  0.012014706\n",
       "UK42020088_5 -0.01089588 -0.008355454 -0.034402696 -0.032697222  0.007915598\n",
       "UK47490007_3  0.01562787  0.026717734 -0.009877409 -0.003607270  0.024871961"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "eigengenes <- read.csv(\"~/gains_team282/nikhil/expression/gene_expression/eigengenes.csv\", row.names=1)\n",
    "\n",
    "head(eigengenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8a7417ee-130b-40aa-a038-c5e5416e5e92",
   "metadata": {},
   "outputs": [],
   "source": [
    "modules <- read.csv(\"~/gains_team282/nikhil/expression/gene_expression/modules.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bb7738d0-5663-422e-a78d-c79d58a1f014",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>Module</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000001167</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000002330</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000002822</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000005175</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000005194</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000005893</td><td>Module_1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Gene & Module\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000001167 & Module\\_1\\\\\n",
       "\t2 & ENSG00000002330 & Module\\_1\\\\\n",
       "\t3 & ENSG00000002822 & Module\\_1\\\\\n",
       "\t4 & ENSG00000005175 & Module\\_1\\\\\n",
       "\t5 & ENSG00000005194 & Module\\_1\\\\\n",
       "\t6 & ENSG00000005893 & Module\\_1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | Module &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | ENSG00000001167 | Module_1 |\n",
       "| 2 | ENSG00000002330 | Module_1 |\n",
       "| 3 | ENSG00000002822 | Module_1 |\n",
       "| 4 | ENSG00000005175 | Module_1 |\n",
       "| 5 | ENSG00000005194 | Module_1 |\n",
       "| 6 | ENSG00000005893 | Module_1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene            Module  \n",
       "1 ENSG00000001167 Module_1\n",
       "2 ENSG00000002330 Module_1\n",
       "3 ENSG00000002822 Module_1\n",
       "4 ENSG00000005175 Module_1\n",
       "5 ENSG00000005194 Module_1\n",
       "6 ENSG00000005893 Module_1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(modules)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1922a085-a8f3-4bd1-b57b-9126d6f19dd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene.exp <- read.table(\"/lustre/scratch119/humgen/projects/gains_team282/eqtl/data/logcpm_864_20412_hla.txt\")\n",
    "\n",
    "colnames(gene.exp) <- gsub(\"^GA\", \"\", colnames(gene.exp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "50739ed4-280a-427d-8b7d-55e9ce0cb2aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 864</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>UK02270173_3</th><th scope=col>UK15130120_3</th><th scope=col>UK58000006_3</th><th scope=col>UK47010004_3</th><th scope=col>UK42020088_5</th><th scope=col>UK47490007_3</th><th scope=col>UK02770164_3</th><th scope=col>UK02770164_5</th><th scope=col>UK02630151_3</th><th scope=col>UK42150107_1</th><th scope=col>⋯</th><th scope=col>UK59070043_3</th><th scope=col>UK59070043_5</th><th scope=col>UK02510223_3</th><th scope=col>UK02XX0336_5</th><th scope=col>UK29090086_3</th><th scope=col>UK02XX0335_1</th><th scope=col>UK02XX0334_3</th><th scope=col>UK01210130_3</th><th scope=col>UK01210130_5</th><th scope=col>UK01380125_1</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSG00000230521</th><td>0.6644339</td><td>0.58271604</td><td>0.4046756</td><td>0.20905157</td><td>0.2211571</td><td>0.4288519</td><td>0.5133516</td><td>0.57768877</td><td>0.6990766</td><td>0.3812037</td><td>⋯</td><td>0.3823184</td><td>0.5423684</td><td>0.86762630</td><td>0.73833461</td><td>0.6461583</td><td>0.1227063</td><td>0.9621386</td><td>0.65838459</td><td>1.03561474</td><td>0.4457301</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000225864</th><td>0.9817872</td><td>0.43879936</td><td>0.6022344</td><td>0.92340489</td><td>0.2211571</td><td>1.1450686</td><td>0.1921362</td><td>0.57768877</td><td>0.1974851</td><td>1.2290495</td><td>⋯</td><td>0.6842527</td><td>0.8539227</td><td>0.91886777</td><td>0.12581620</td><td>0.5511865</td><td>0.4384007</td><td>0.5465139</td><td>0.40803066</td><td>0.46899151</td><td>0.3544115</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000227766</th><td>1.2117156</td><td>0.30682669</td><td>0.6625010</td><td>0.79963286</td><td>0.6445686</td><td>0.3130290</td><td>0.1183231</td><td>0.16750450</td><td>0.1505900</td><td>1.0533199</td><td>⋯</td><td>0.4901384</td><td>0.4340781</td><td>0.37504164</td><td>0.04316943</td><td>0.6151905</td><td>0.3406488</td><td>0.3147798</td><td>0.43511377</td><td>0.17382742</td><td>0.1157514</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000237669</th><td>0.6420763</td><td>0.69256830</td><td>0.9792012</td><td>0.69929584</td><td>0.3765156</td><td>1.0756817</td><td>0.8912050</td><td>0.60129641</td><td>0.8534668</td><td>1.0533199</td><td>⋯</td><td>1.0790156</td><td>0.6102946</td><td>1.03953331</td><td>0.97965846</td><td>0.7061683</td><td>0.7489857</td><td>0.7919934</td><td>0.85168813</td><td>0.69120049</td><td>1.0798771</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000271581</th><td>2.4744895</td><td>1.33153140</td><td>1.8310464</td><td>2.47531701</td><td>1.8812042</td><td>2.0531366</td><td>0.5972544</td><td>1.00686213</td><td>1.2811361</td><td>2.4244938</td><td>⋯</td><td>2.4728831</td><td>1.5338898</td><td>1.59080138</td><td>0.62923925</td><td>1.2804253</td><td>1.4727511</td><td>1.2187650</td><td>1.46471386</td><td>1.44139443</td><td>1.1896250</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000285647</th><td>0.0000000</td><td>0.03379072</td><td>0.0000000</td><td>0.05517067</td><td>0.6748270</td><td>0.0000000</td><td>0.0000000</td><td>0.03509293</td><td>0.6990766</td><td>1.6556266</td><td>⋯</td><td>3.4117627</td><td>2.4012963</td><td>0.04682116</td><td>0.00000000</td><td>0.0000000</td><td>0.4695662</td><td>0.3762952</td><td>0.07078527</td><td>0.07206309</td><td>0.4159282</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 864\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & UK02270173\\_3 & UK15130120\\_3 & UK58000006\\_3 & UK47010004\\_3 & UK42020088\\_5 & UK47490007\\_3 & UK02770164\\_3 & UK02770164\\_5 & UK02630151\\_3 & UK42150107\\_1 & ⋯ & UK59070043\\_3 & UK59070043\\_5 & UK02510223\\_3 & UK02XX0336\\_5 & UK29090086\\_3 & UK02XX0335\\_1 & UK02XX0334\\_3 & UK01210130\\_3 & UK01210130\\_5 & UK01380125\\_1\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tENSG00000230521 & 0.6644339 & 0.58271604 & 0.4046756 & 0.20905157 & 0.2211571 & 0.4288519 & 0.5133516 & 0.57768877 & 0.6990766 & 0.3812037 & ⋯ & 0.3823184 & 0.5423684 & 0.86762630 & 0.73833461 & 0.6461583 & 0.1227063 & 0.9621386 & 0.65838459 & 1.03561474 & 0.4457301\\\\\n",
       "\tENSG00000225864 & 0.9817872 & 0.43879936 & 0.6022344 & 0.92340489 & 0.2211571 & 1.1450686 & 0.1921362 & 0.57768877 & 0.1974851 & 1.2290495 & ⋯ & 0.6842527 & 0.8539227 & 0.91886777 & 0.12581620 & 0.5511865 & 0.4384007 & 0.5465139 & 0.40803066 & 0.46899151 & 0.3544115\\\\\n",
       "\tENSG00000227766 & 1.2117156 & 0.30682669 & 0.6625010 & 0.79963286 & 0.6445686 & 0.3130290 & 0.1183231 & 0.16750450 & 0.1505900 & 1.0533199 & ⋯ & 0.4901384 & 0.4340781 & 0.37504164 & 0.04316943 & 0.6151905 & 0.3406488 & 0.3147798 & 0.43511377 & 0.17382742 & 0.1157514\\\\\n",
       "\tENSG00000237669 & 0.6420763 & 0.69256830 & 0.9792012 & 0.69929584 & 0.3765156 & 1.0756817 & 0.8912050 & 0.60129641 & 0.8534668 & 1.0533199 & ⋯ & 1.0790156 & 0.6102946 & 1.03953331 & 0.97965846 & 0.7061683 & 0.7489857 & 0.7919934 & 0.85168813 & 0.69120049 & 1.0798771\\\\\n",
       "\tENSG00000271581 & 2.4744895 & 1.33153140 & 1.8310464 & 2.47531701 & 1.8812042 & 2.0531366 & 0.5972544 & 1.00686213 & 1.2811361 & 2.4244938 & ⋯ & 2.4728831 & 1.5338898 & 1.59080138 & 0.62923925 & 1.2804253 & 1.4727511 & 1.2187650 & 1.46471386 & 1.44139443 & 1.1896250\\\\\n",
       "\tENSG00000285647 & 0.0000000 & 0.03379072 & 0.0000000 & 0.05517067 & 0.6748270 & 0.0000000 & 0.0000000 & 0.03509293 & 0.6990766 & 1.6556266 & ⋯ & 3.4117627 & 2.4012963 & 0.04682116 & 0.00000000 & 0.0000000 & 0.4695662 & 0.3762952 & 0.07078527 & 0.07206309 & 0.4159282\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 864\n",
       "\n",
       "| <!--/--> | UK02270173_3 &lt;dbl&gt; | UK15130120_3 &lt;dbl&gt; | UK58000006_3 &lt;dbl&gt; | UK47010004_3 &lt;dbl&gt; | UK42020088_5 &lt;dbl&gt; | UK47490007_3 &lt;dbl&gt; | UK02770164_3 &lt;dbl&gt; | UK02770164_5 &lt;dbl&gt; | UK02630151_3 &lt;dbl&gt; | UK42150107_1 &lt;dbl&gt; | ⋯ ⋯ | UK59070043_3 &lt;dbl&gt; | UK59070043_5 &lt;dbl&gt; | UK02510223_3 &lt;dbl&gt; | UK02XX0336_5 &lt;dbl&gt; | UK29090086_3 &lt;dbl&gt; | UK02XX0335_1 &lt;dbl&gt; | UK02XX0334_3 &lt;dbl&gt; | UK01210130_3 &lt;dbl&gt; | UK01210130_5 &lt;dbl&gt; | UK01380125_1 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000230521 | 0.6644339 | 0.58271604 | 0.4046756 | 0.20905157 | 0.2211571 | 0.4288519 | 0.5133516 | 0.57768877 | 0.6990766 | 0.3812037 | ⋯ | 0.3823184 | 0.5423684 | 0.86762630 | 0.73833461 | 0.6461583 | 0.1227063 | 0.9621386 | 0.65838459 | 1.03561474 | 0.4457301 |\n",
       "| ENSG00000225864 | 0.9817872 | 0.43879936 | 0.6022344 | 0.92340489 | 0.2211571 | 1.1450686 | 0.1921362 | 0.57768877 | 0.1974851 | 1.2290495 | ⋯ | 0.6842527 | 0.8539227 | 0.91886777 | 0.12581620 | 0.5511865 | 0.4384007 | 0.5465139 | 0.40803066 | 0.46899151 | 0.3544115 |\n",
       "| ENSG00000227766 | 1.2117156 | 0.30682669 | 0.6625010 | 0.79963286 | 0.6445686 | 0.3130290 | 0.1183231 | 0.16750450 | 0.1505900 | 1.0533199 | ⋯ | 0.4901384 | 0.4340781 | 0.37504164 | 0.04316943 | 0.6151905 | 0.3406488 | 0.3147798 | 0.43511377 | 0.17382742 | 0.1157514 |\n",
       "| ENSG00000237669 | 0.6420763 | 0.69256830 | 0.9792012 | 0.69929584 | 0.3765156 | 1.0756817 | 0.8912050 | 0.60129641 | 0.8534668 | 1.0533199 | ⋯ | 1.0790156 | 0.6102946 | 1.03953331 | 0.97965846 | 0.7061683 | 0.7489857 | 0.7919934 | 0.85168813 | 0.69120049 | 1.0798771 |\n",
       "| ENSG00000271581 | 2.4744895 | 1.33153140 | 1.8310464 | 2.47531701 | 1.8812042 | 2.0531366 | 0.5972544 | 1.00686213 | 1.2811361 | 2.4244938 | ⋯ | 2.4728831 | 1.5338898 | 1.59080138 | 0.62923925 | 1.2804253 | 1.4727511 | 1.2187650 | 1.46471386 | 1.44139443 | 1.1896250 |\n",
       "| ENSG00000285647 | 0.0000000 | 0.03379072 | 0.0000000 | 0.05517067 | 0.6748270 | 0.0000000 | 0.0000000 | 0.03509293 | 0.6990766 | 1.6556266 | ⋯ | 3.4117627 | 2.4012963 | 0.04682116 | 0.00000000 | 0.0000000 | 0.4695662 | 0.3762952 | 0.07078527 | 0.07206309 | 0.4159282 |\n",
       "\n"
      ],
      "text/plain": [
       "                UK02270173_3 UK15130120_3 UK58000006_3 UK47010004_3\n",
       "ENSG00000230521 0.6644339    0.58271604   0.4046756    0.20905157  \n",
       "ENSG00000225864 0.9817872    0.43879936   0.6022344    0.92340489  \n",
       "ENSG00000227766 1.2117156    0.30682669   0.6625010    0.79963286  \n",
       "ENSG00000237669 0.6420763    0.69256830   0.9792012    0.69929584  \n",
       "ENSG00000271581 2.4744895    1.33153140   1.8310464    2.47531701  \n",
       "ENSG00000285647 0.0000000    0.03379072   0.0000000    0.05517067  \n",
       "                UK42020088_5 UK47490007_3 UK02770164_3 UK02770164_5\n",
       "ENSG00000230521 0.2211571    0.4288519    0.5133516    0.57768877  \n",
       "ENSG00000225864 0.2211571    1.1450686    0.1921362    0.57768877  \n",
       "ENSG00000227766 0.6445686    0.3130290    0.1183231    0.16750450  \n",
       "ENSG00000237669 0.3765156    1.0756817    0.8912050    0.60129641  \n",
       "ENSG00000271581 1.8812042    2.0531366    0.5972544    1.00686213  \n",
       "ENSG00000285647 0.6748270    0.0000000    0.0000000    0.03509293  \n",
       "                UK02630151_3 UK42150107_1 ⋯ UK59070043_3 UK59070043_5\n",
       "ENSG00000230521 0.6990766    0.3812037    ⋯ 0.3823184    0.5423684   \n",
       "ENSG00000225864 0.1974851    1.2290495    ⋯ 0.6842527    0.8539227   \n",
       "ENSG00000227766 0.1505900    1.0533199    ⋯ 0.4901384    0.4340781   \n",
       "ENSG00000237669 0.8534668    1.0533199    ⋯ 1.0790156    0.6102946   \n",
       "ENSG00000271581 1.2811361    2.4244938    ⋯ 2.4728831    1.5338898   \n",
       "ENSG00000285647 0.6990766    1.6556266    ⋯ 3.4117627    2.4012963   \n",
       "                UK02510223_3 UK02XX0336_5 UK29090086_3 UK02XX0335_1\n",
       "ENSG00000230521 0.86762630   0.73833461   0.6461583    0.1227063   \n",
       "ENSG00000225864 0.91886777   0.12581620   0.5511865    0.4384007   \n",
       "ENSG00000227766 0.37504164   0.04316943   0.6151905    0.3406488   \n",
       "ENSG00000237669 1.03953331   0.97965846   0.7061683    0.7489857   \n",
       "ENSG00000271581 1.59080138   0.62923925   1.2804253    1.4727511   \n",
       "ENSG00000285647 0.04682116   0.00000000   0.0000000    0.4695662   \n",
       "                UK02XX0334_3 UK01210130_3 UK01210130_5 UK01380125_1\n",
       "ENSG00000230521 0.9621386    0.65838459   1.03561474   0.4457301   \n",
       "ENSG00000225864 0.5465139    0.40803066   0.46899151   0.3544115   \n",
       "ENSG00000227766 0.3147798    0.43511377   0.17382742   0.1157514   \n",
       "ENSG00000237669 0.7919934    0.85168813   0.69120049   1.0798771   \n",
       "ENSG00000271581 1.2187650    1.46471386   1.44139443   1.1896250   \n",
       "ENSG00000285647 0.3762952    0.07078527   0.07206309   0.4159282   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene.exp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "973adfd9-7295-45ec-ad6e-65ec926c7841",
   "metadata": {},
   "outputs": [],
   "source": [
    "lead.cis.eqtl <- read.table(\"/nfs/users/nfs_n/nm18/gains_team282/eqtl/cisresults/eigenMT/ciseqtl_eigenMT_corrected.txt\") %>%\n",
    "    dplyr::filter(Sig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "f558c85f-1ae0-4c2e-86d8-1de063fe980e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>snps</th><th scope=col>gene</th><th scope=col>statistic</th><th scope=col>pvalue</th><th scope=col>beta</th><th scope=col>se</th><th scope=col>chr</th><th scope=col>SNPpos</th><th scope=col>TSS</th><th scope=col>BF</th><th scope=col>TESTS</th><th scope=col>BF.FDR</th><th scope=col>Sig</th><th scope=col>threshold</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>22</th><td>rs3131972 </td><td>ENSG00000237491</td><td> 5.643405</td><td>2.400875e-08</td><td> 0.07174216</td><td>0.012712570</td><td>1</td><td>817341</td><td>778747</td><td>7.058573e-06</td><td>294</td><td>1.993414e-05</td><td>TRUE</td><td>9.170798e-05</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>rs3131972 </td><td>ENSG00000230092</td><td> 4.756587</td><td>2.329109e-06</td><td> 0.06239499</td><td>0.013117600</td><td>1</td><td>817341</td><td>817712</td><td>6.917454e-04</td><td>297</td><td>1.593483e-03</td><td>TRUE</td><td>9.078164e-05</td></tr>\n",
       "\t<tr><th scope=row>25</th><td>rs3131972 </td><td>ENSG00000225880</td><td> 6.292961</td><td>5.949542e-10</td><td> 0.10016770</td><td>0.015917420</td><td>1</td><td>817341</td><td>827522</td><td>1.767014e-07</td><td>297</td><td>5.668498e-07</td><td>TRUE</td><td>9.078164e-05</td></tr>\n",
       "\t<tr><th scope=row>28</th><td>rs2272757 </td><td>ENSG00000188976</td><td>-7.026045</td><td>4.904859e-12</td><td>-0.05107767</td><td>0.007269761</td><td>1</td><td>946247</td><td>959309</td><td>1.545031e-09</td><td>315</td><td>5.762103e-09</td><td>TRUE</td><td>8.559411e-05</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>rs13303327</td><td>ENSG00000187961</td><td>-7.175012</td><td>1.845538e-12</td><td>-0.13519920</td><td>0.018843060</td><td>1</td><td>960326</td><td>960584</td><td>5.813445e-10</td><td>315</td><td>2.228468e-09</td><td>TRUE</td><td>8.559411e-05</td></tr>\n",
       "\t<tr><th scope=row>30</th><td>rs13303056</td><td>ENSG00000187583</td><td>-9.081031</td><td>1.922802e-18</td><td>-0.29633770</td><td>0.032632610</td><td>1</td><td>953778</td><td>966482</td><td>6.056826e-16</td><td>315</td><td>3.262796e-15</td><td>TRUE</td><td>8.559411e-05</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & snps & gene & statistic & pvalue & beta & se & chr & SNPpos & TSS & BF & TESTS & BF.FDR & Sig & threshold\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <dbl> & <int> & <dbl> & <lgl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t22 & rs3131972  & ENSG00000237491 &  5.643405 & 2.400875e-08 &  0.07174216 & 0.012712570 & 1 & 817341 & 778747 & 7.058573e-06 & 294 & 1.993414e-05 & TRUE & 9.170798e-05\\\\\n",
       "\t23 & rs3131972  & ENSG00000230092 &  4.756587 & 2.329109e-06 &  0.06239499 & 0.013117600 & 1 & 817341 & 817712 & 6.917454e-04 & 297 & 1.593483e-03 & TRUE & 9.078164e-05\\\\\n",
       "\t25 & rs3131972  & ENSG00000225880 &  6.292961 & 5.949542e-10 &  0.10016770 & 0.015917420 & 1 & 817341 & 827522 & 1.767014e-07 & 297 & 5.668498e-07 & TRUE & 9.078164e-05\\\\\n",
       "\t28 & rs2272757  & ENSG00000188976 & -7.026045 & 4.904859e-12 & -0.05107767 & 0.007269761 & 1 & 946247 & 959309 & 1.545031e-09 & 315 & 5.762103e-09 & TRUE & 8.559411e-05\\\\\n",
       "\t29 & rs13303327 & ENSG00000187961 & -7.175012 & 1.845538e-12 & -0.13519920 & 0.018843060 & 1 & 960326 & 960584 & 5.813445e-10 & 315 & 2.228468e-09 & TRUE & 8.559411e-05\\\\\n",
       "\t30 & rs13303056 & ENSG00000187583 & -9.081031 & 1.922802e-18 & -0.29633770 & 0.032632610 & 1 & 953778 & 966482 & 6.056826e-16 & 315 & 3.262796e-15 & TRUE & 8.559411e-05\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 14\n",
       "\n",
       "| <!--/--> | snps &lt;chr&gt; | gene &lt;chr&gt; | statistic &lt;dbl&gt; | pvalue &lt;dbl&gt; | beta &lt;dbl&gt; | se &lt;dbl&gt; | chr &lt;int&gt; | SNPpos &lt;int&gt; | TSS &lt;int&gt; | BF &lt;dbl&gt; | TESTS &lt;int&gt; | BF.FDR &lt;dbl&gt; | Sig &lt;lgl&gt; | threshold &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 22 | rs3131972  | ENSG00000237491 |  5.643405 | 2.400875e-08 |  0.07174216 | 0.012712570 | 1 | 817341 | 778747 | 7.058573e-06 | 294 | 1.993414e-05 | TRUE | 9.170798e-05 |\n",
       "| 23 | rs3131972  | ENSG00000230092 |  4.756587 | 2.329109e-06 |  0.06239499 | 0.013117600 | 1 | 817341 | 817712 | 6.917454e-04 | 297 | 1.593483e-03 | TRUE | 9.078164e-05 |\n",
       "| 25 | rs3131972  | ENSG00000225880 |  6.292961 | 5.949542e-10 |  0.10016770 | 0.015917420 | 1 | 817341 | 827522 | 1.767014e-07 | 297 | 5.668498e-07 | TRUE | 9.078164e-05 |\n",
       "| 28 | rs2272757  | ENSG00000188976 | -7.026045 | 4.904859e-12 | -0.05107767 | 0.007269761 | 1 | 946247 | 959309 | 1.545031e-09 | 315 | 5.762103e-09 | TRUE | 8.559411e-05 |\n",
       "| 29 | rs13303327 | ENSG00000187961 | -7.175012 | 1.845538e-12 | -0.13519920 | 0.018843060 | 1 | 960326 | 960584 | 5.813445e-10 | 315 | 2.228468e-09 | TRUE | 8.559411e-05 |\n",
       "| 30 | rs13303056 | ENSG00000187583 | -9.081031 | 1.922802e-18 | -0.29633770 | 0.032632610 | 1 | 953778 | 966482 | 6.056826e-16 | 315 | 3.262796e-15 | TRUE | 8.559411e-05 |\n",
       "\n"
      ],
      "text/plain": [
       "   snps       gene            statistic pvalue       beta        se         \n",
       "22 rs3131972  ENSG00000237491  5.643405 2.400875e-08  0.07174216 0.012712570\n",
       "23 rs3131972  ENSG00000230092  4.756587 2.329109e-06  0.06239499 0.013117600\n",
       "25 rs3131972  ENSG00000225880  6.292961 5.949542e-10  0.10016770 0.015917420\n",
       "28 rs2272757  ENSG00000188976 -7.026045 4.904859e-12 -0.05107767 0.007269761\n",
       "29 rs13303327 ENSG00000187961 -7.175012 1.845538e-12 -0.13519920 0.018843060\n",
       "30 rs13303056 ENSG00000187583 -9.081031 1.922802e-18 -0.29633770 0.032632610\n",
       "   chr SNPpos TSS    BF           TESTS BF.FDR       Sig  threshold   \n",
       "22 1   817341 778747 7.058573e-06 294   1.993414e-05 TRUE 9.170798e-05\n",
       "23 1   817341 817712 6.917454e-04 297   1.593483e-03 TRUE 9.078164e-05\n",
       "25 1   817341 827522 1.767014e-07 297   5.668498e-07 TRUE 9.078164e-05\n",
       "28 1   946247 959309 1.545031e-09 315   5.762103e-09 TRUE 8.559411e-05\n",
       "29 1   960326 960584 5.813445e-10 315   2.228468e-09 TRUE 8.559411e-05\n",
       "30 1   953778 966482 6.056826e-16 315   3.262796e-15 TRUE 8.559411e-05"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(lead.cis.eqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "6db21f2d-6677-4c50-9764-1746d3bf6fed",
   "metadata": {},
   "outputs": [],
   "source": [
    "conditional.cis.eqtl <- readRDS(\"/nfs/users/nfs_n/nm18/gains_team282/eqtl/cisresults/conditionalanalysis/conditional_eQTL_results_final.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c7d7570f-52a8-43a2-94ec-9bd724d5d165",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SNP</th><th scope=col>Gene</th><th scope=col>eQTL_beta</th><th scope=col>eQTL_SE</th><th scope=col>pvalue</th><th scope=col>Number</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>rs10753794</td><td>ENSG00000000457</td><td>0.0498587312408011 </td><td>0.00708515851579373</td><td>5.19259471106013e-12</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>rs10919255</td><td>ENSG00000000460</td><td>-0.0726264894633498</td><td>0.0136853304567293 </td><td>1.72815915933528e-07</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>rs77006036</td><td>ENSG00000000460</td><td>0.166236949102582  </td><td>0.0334472538699365 </td><td>8.47290238610159e-07</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>rs12406047</td><td>ENSG00000000971</td><td>0.17135374034767   </td><td>0.0379807691167313 </td><td>7.51523462853891e-06</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>rs6696136 </td><td>ENSG00000001460</td><td>0.45884434698694   </td><td>0.0871081626842823 </td><td>1.94887370606405e-07</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>rs6676449 </td><td>ENSG00000001460</td><td>-0.371531778822554 </td><td>0.0880672792106829 </td><td>2.8471651589813e-05 </td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & SNP & Gene & eQTL\\_beta & eQTL\\_SE & pvalue & Number\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & rs10753794 & ENSG00000000457 & 0.0498587312408011  & 0.00708515851579373 & 5.19259471106013e-12 & 1\\\\\n",
       "\t2 & rs10919255 & ENSG00000000460 & -0.0726264894633498 & 0.0136853304567293  & 1.72815915933528e-07 & 1\\\\\n",
       "\t3 & rs77006036 & ENSG00000000460 & 0.166236949102582   & 0.0334472538699365  & 8.47290238610159e-07 & 2\\\\\n",
       "\t4 & rs12406047 & ENSG00000000971 & 0.17135374034767    & 0.0379807691167313  & 7.51523462853891e-06 & 1\\\\\n",
       "\t5 & rs6696136  & ENSG00000001460 & 0.45884434698694    & 0.0871081626842823  & 1.94887370606405e-07 & 1\\\\\n",
       "\t6 & rs6676449  & ENSG00000001460 & -0.371531778822554  & 0.0880672792106829  & 2.8471651589813e-05  & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | SNP &lt;chr&gt; | Gene &lt;chr&gt; | eQTL_beta &lt;chr&gt; | eQTL_SE &lt;chr&gt; | pvalue &lt;chr&gt; | Number &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | rs10753794 | ENSG00000000457 | 0.0498587312408011  | 0.00708515851579373 | 5.19259471106013e-12 | 1 |\n",
       "| 2 | rs10919255 | ENSG00000000460 | -0.0726264894633498 | 0.0136853304567293  | 1.72815915933528e-07 | 1 |\n",
       "| 3 | rs77006036 | ENSG00000000460 | 0.166236949102582   | 0.0334472538699365  | 8.47290238610159e-07 | 2 |\n",
       "| 4 | rs12406047 | ENSG00000000971 | 0.17135374034767    | 0.0379807691167313  | 7.51523462853891e-06 | 1 |\n",
       "| 5 | rs6696136  | ENSG00000001460 | 0.45884434698694    | 0.0871081626842823  | 1.94887370606405e-07 | 1 |\n",
       "| 6 | rs6676449  | ENSG00000001460 | -0.371531778822554  | 0.0880672792106829  | 2.8471651589813e-05  | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP        Gene            eQTL_beta           eQTL_SE            \n",
       "1 rs10753794 ENSG00000000457 0.0498587312408011  0.00708515851579373\n",
       "2 rs10919255 ENSG00000000460 -0.0726264894633498 0.0136853304567293 \n",
       "3 rs77006036 ENSG00000000460 0.166236949102582   0.0334472538699365 \n",
       "4 rs12406047 ENSG00000000971 0.17135374034767    0.0379807691167313 \n",
       "5 rs6696136  ENSG00000001460 0.45884434698694    0.0871081626842823 \n",
       "6 rs6676449  ENSG00000001460 -0.371531778822554  0.0880672792106829 \n",
       "  pvalue               Number\n",
       "1 5.19259471106013e-12 1     \n",
       "2 1.72815915933528e-07 1     \n",
       "3 8.47290238610159e-07 2     \n",
       "4 7.51523462853891e-06 1     \n",
       "5 1.94887370606405e-07 1     \n",
       "6 2.8471651589813e-05  2     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(conditional.cis.eqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3bcec61e-1c77-42fe-b8a5-c2f9c87482de",
   "metadata": {},
   "outputs": [],
   "source": [
    "ebi.studies <- fread(\"04_Expression/data/gwas_catalog_v1.0.2-studies_r2022-02-21.tsv\", header=TRUE, quote=\"\") %>%\n",
    "    as.data.frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "53805170-af73-4d97-870b-23d9c68a25ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 16</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>DATE ADDED TO CATALOG</th><th scope=col>PUBMEDID</th><th scope=col>FIRST AUTHOR</th><th scope=col>DATE</th><th scope=col>JOURNAL</th><th scope=col>LINK</th><th scope=col>STUDY</th><th scope=col>DISEASE/TRAIT</th><th scope=col>INITIAL SAMPLE SIZE</th><th scope=col>REPLICATION SAMPLE SIZE</th><th scope=col>PLATFORM [SNPS PASSING QC]</th><th scope=col>ASSOCIATION COUNT</th><th scope=col>MAPPED_TRAIT</th><th scope=col>MAPPED_TRAIT_URI</th><th scope=col>STUDY ACCESSION</th><th scope=col>GENOTYPING TECHNOLOGY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>2021-08-24</td><td>34124712</td><td>Sherva R</td><td>2021-02-28</td><td>Explor Med</td><td>www.ncbi.nlm.nih.gov/pubmed/34124712</td><td>Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes.</td><td>Cocaine dependence (time to event)</td><td>3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls</td><td>572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls</td><td>Illumina [NR] (imputed)</td><td>5</td><td>cocaine dependence</td><td>http://www.ebi.ac.uk/efo/EFO_0002610</td><td>GCST012225</td><td>Genome-wide genotyping array</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 16\n",
       "\\begin{tabular}{r|llllllllllllllll}\n",
       "  & DATE ADDED TO CATALOG & PUBMEDID & FIRST AUTHOR & DATE & JOURNAL & LINK & STUDY & DISEASE/TRAIT & INITIAL SAMPLE SIZE & REPLICATION SAMPLE SIZE & PLATFORM {[}SNPS PASSING QC{]} & ASSOCIATION COUNT & MAPPED\\_TRAIT & MAPPED\\_TRAIT\\_URI & STUDY ACCESSION & GENOTYPING TECHNOLOGY\\\\\n",
       "  & <date> & <int> & <chr> & <date> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 2021-08-24 & 34124712 & Sherva R & 2021-02-28 & Explor Med & www.ncbi.nlm.nih.gov/pubmed/34124712 & Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes. & Cocaine dependence (time to event) & 3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls & 572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls & Illumina {[}NR{]} (imputed) & 5 & cocaine dependence & http://www.ebi.ac.uk/efo/EFO\\_0002610 & GCST012225 & Genome-wide genotyping array\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 16\n",
       "\n",
       "| <!--/--> | DATE ADDED TO CATALOG &lt;date&gt; | PUBMEDID &lt;int&gt; | FIRST AUTHOR &lt;chr&gt; | DATE &lt;date&gt; | JOURNAL &lt;chr&gt; | LINK &lt;chr&gt; | STUDY &lt;chr&gt; | DISEASE/TRAIT &lt;chr&gt; | INITIAL SAMPLE SIZE &lt;chr&gt; | REPLICATION SAMPLE SIZE &lt;chr&gt; | PLATFORM [SNPS PASSING QC] &lt;chr&gt; | ASSOCIATION COUNT &lt;int&gt; | MAPPED_TRAIT &lt;chr&gt; | MAPPED_TRAIT_URI &lt;chr&gt; | STUDY ACCESSION &lt;chr&gt; | GENOTYPING TECHNOLOGY &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 2021-08-24 | 34124712 | Sherva R | 2021-02-28 | Explor Med | www.ncbi.nlm.nih.gov/pubmed/34124712 | Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes. | Cocaine dependence (time to event) | 3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls | 572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls | Illumina [NR] (imputed) | 5 | cocaine dependence | http://www.ebi.ac.uk/efo/EFO_0002610 | GCST012225 | Genome-wide genotyping array |\n",
       "\n"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG PUBMEDID FIRST AUTHOR DATE       JOURNAL   \n",
       "1 2021-08-24            34124712 Sherva R     2021-02-28 Explor Med\n",
       "  LINK                                \n",
       "1 www.ncbi.nlm.nih.gov/pubmed/34124712\n",
       "  STUDY                                                                                                                                     \n",
       "1 Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes.\n",
       "  DISEASE/TRAIT                     \n",
       "1 Cocaine dependence (time to event)\n",
       "  INITIAL SAMPLE SIZE                                                                                                       \n",
       "1 3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls\n",
       "  REPLICATION SAMPLE SIZE                                                                                                 \n",
       "1 572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls\n",
       "  PLATFORM [SNPS PASSING QC] ASSOCIATION COUNT MAPPED_TRAIT      \n",
       "1 Illumina [NR] (imputed)    5                 cocaine dependence\n",
       "  MAPPED_TRAIT_URI                     STUDY ACCESSION\n",
       "1 http://www.ebi.ac.uk/efo/EFO_0002610 GCST012225     \n",
       "  GENOTYPING TECHNOLOGY       \n",
       "1 Genome-wide genotyping array"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ebi.studies, n=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "e0460a82-daa1-4eaf-877b-b119f710f4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "ebi.assoc <- fread(\"04_Expression/data/gwas_catalog_v1.0.2-associations_e105_r2022-02-21.tsv\", header=TRUE, quote=\"\") %>%\n",
    "    as.data.frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d5fe892a-b746-4ce0-a639-8f577e9ee36e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 38</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>DATE ADDED TO CATALOG</th><th scope=col>PUBMEDID</th><th scope=col>FIRST AUTHOR</th><th scope=col>DATE</th><th scope=col>JOURNAL</th><th scope=col>LINK</th><th scope=col>STUDY</th><th scope=col>DISEASE/TRAIT</th><th scope=col>INITIAL SAMPLE SIZE</th><th scope=col>REPLICATION SAMPLE SIZE</th><th scope=col>⋯</th><th scope=col>PVALUE_MLOG</th><th scope=col>P-VALUE (TEXT)</th><th scope=col>OR or BETA</th><th scope=col>95% CI (TEXT)</th><th scope=col>PLATFORM [SNPS PASSING QC]</th><th scope=col>CNV</th><th scope=col>MAPPED_TRAIT</th><th scope=col>MAPPED_TRAIT_URI</th><th scope=col>STUDY ACCESSION</th><th scope=col>GENOTYPING TECHNOLOGY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>2021-07-07</td><td>34127860</td><td>Robertson CC</td><td>2021-06-14</td><td>Nat Genet</td><td>www.ncbi.nlm.nih.gov/pubmed/34127860</td><td>Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes.</td><td>Type 1 diabetes</td><td>20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls</td><td>NA</td><td>⋯</td><td>8.522879</td><td></td><td>0.912</td><td></td><td>Illumina [715631] (imputed)</td><td>N</td><td>type 1 diabetes mellitus</td><td>http://purl.obolibrary.org/obo/MONDO_0005147</td><td>GCST90013445</td><td>Targeted genotyping array [ImmunoChip]</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 38\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & DATE ADDED TO CATALOG & PUBMEDID & FIRST AUTHOR & DATE & JOURNAL & LINK & STUDY & DISEASE/TRAIT & INITIAL SAMPLE SIZE & REPLICATION SAMPLE SIZE & ⋯ & PVALUE\\_MLOG & P-VALUE (TEXT) & OR or BETA & 95\\% CI (TEXT) & PLATFORM {[}SNPS PASSING QC{]} & CNV & MAPPED\\_TRAIT & MAPPED\\_TRAIT\\_URI & STUDY ACCESSION & GENOTYPING TECHNOLOGY\\\\\n",
       "  & <date> & <int> & <chr> & <date> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <dbl> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 2021-07-07 & 34127860 & Robertson CC & 2021-06-14 & Nat Genet & www.ncbi.nlm.nih.gov/pubmed/34127860 & Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes. & Type 1 diabetes & 20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls & NA & ⋯ & 8.522879 &  & 0.912 &  & Illumina {[}715631{]} (imputed) & N & type 1 diabetes mellitus & http://purl.obolibrary.org/obo/MONDO\\_0005147 & GCST90013445 & Targeted genotyping array {[}ImmunoChip{]}\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 38\n",
       "\n",
       "| <!--/--> | DATE ADDED TO CATALOG &lt;date&gt; | PUBMEDID &lt;int&gt; | FIRST AUTHOR &lt;chr&gt; | DATE &lt;date&gt; | JOURNAL &lt;chr&gt; | LINK &lt;chr&gt; | STUDY &lt;chr&gt; | DISEASE/TRAIT &lt;chr&gt; | INITIAL SAMPLE SIZE &lt;chr&gt; | REPLICATION SAMPLE SIZE &lt;chr&gt; | ⋯ ⋯ | PVALUE_MLOG &lt;dbl&gt; | P-VALUE (TEXT) &lt;chr&gt; | OR or BETA &lt;dbl&gt; | 95% CI (TEXT) &lt;chr&gt; | PLATFORM [SNPS PASSING QC] &lt;chr&gt; | CNV &lt;chr&gt; | MAPPED_TRAIT &lt;chr&gt; | MAPPED_TRAIT_URI &lt;chr&gt; | STUDY ACCESSION &lt;chr&gt; | GENOTYPING TECHNOLOGY &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 2021-07-07 | 34127860 | Robertson CC | 2021-06-14 | Nat Genet | www.ncbi.nlm.nih.gov/pubmed/34127860 | Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes. | Type 1 diabetes | 20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls | NA | ⋯ | 8.522879 | <!----> | 0.912 | <!----> | Illumina [715631] (imputed) | N | type 1 diabetes mellitus | http://purl.obolibrary.org/obo/MONDO_0005147 | GCST90013445 | Targeted genotyping array [ImmunoChip] |\n",
       "\n"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG PUBMEDID FIRST AUTHOR DATE       JOURNAL  \n",
       "1 2021-07-07            34127860 Robertson CC 2021-06-14 Nat Genet\n",
       "  LINK                                \n",
       "1 www.ncbi.nlm.nih.gov/pubmed/34127860\n",
       "  STUDY                                                                                                                          \n",
       "1 Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes.\n",
       "  DISEASE/TRAIT  \n",
       "1 Type 1 diabetes\n",
       "  INITIAL SAMPLE SIZE                                                                                                                                                                                        \n",
       "1 20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls\n",
       "  REPLICATION SAMPLE SIZE ⋯ PVALUE_MLOG P-VALUE (TEXT) OR or BETA 95% CI (TEXT)\n",
       "1 NA                      ⋯ 8.522879                   0.912                   \n",
       "  PLATFORM [SNPS PASSING QC]  CNV MAPPED_TRAIT            \n",
       "1 Illumina [715631] (imputed) N   type 1 diabetes mellitus\n",
       "  MAPPED_TRAIT_URI                             STUDY ACCESSION\n",
       "1 http://purl.obolibrary.org/obo/MONDO_0005147 GCST90013445   \n",
       "  GENOTYPING TECHNOLOGY                 \n",
       "1 Targeted genotyping array [ImmunoChip]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ebi.assoc, n=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4d128a00-384a-4941-baf2-e4f0f200079c",
   "metadata": {},
   "outputs": [],
   "source": [
    "geno.bim <- fread(\"~/gains_team282/Genotyping/All_genotyping_merged_filtered_b38_refiltered_rsID.bim\") %>%\n",
    "    as.data.frame()\n",
    "colnames(geno.bim) <- c(\"chr\", \"snps\", \"cM\", \"pos\", \"minor\", \"major\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f1c567db-258f-49f6-9fd1-f019e2dbd440",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>snps</th><th scope=col>cM</th><th scope=col>pos</th><th scope=col>minor</th><th scope=col>major</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs3131972   </td><td>0</td><td>817341</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs546843995 </td><td>0</td><td>818053</td><td>0</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs553916047 </td><td>0</td><td>818359</td><td>0</td><td>A</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1:818740_T_C</td><td>0</td><td>818740</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs145604921 </td><td>0</td><td>819378</td><td>0</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs535256652 </td><td>0</td><td>821053</td><td>0</td><td>T</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & chr & snps & cM & pos & minor & major\\\\\n",
       "  & <chr> & <chr> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs3131972    & 0 & 817341 & A & G\\\\\n",
       "\t2 & 1 & rs546843995  & 0 & 818053 & 0 & G\\\\\n",
       "\t3 & 1 & rs553916047  & 0 & 818359 & 0 & A\\\\\n",
       "\t4 & 1 & 1:818740\\_T\\_C & 0 & 818740 & T & C\\\\\n",
       "\t5 & 1 & rs145604921  & 0 & 819378 & 0 & C\\\\\n",
       "\t6 & 1 & rs535256652  & 0 & 821053 & 0 & T\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | snps &lt;chr&gt; | cM &lt;int&gt; | pos &lt;int&gt; | minor &lt;chr&gt; | major &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs3131972    | 0 | 817341 | A | G |\n",
       "| 2 | 1 | rs546843995  | 0 | 818053 | 0 | G |\n",
       "| 3 | 1 | rs553916047  | 0 | 818359 | 0 | A |\n",
       "| 4 | 1 | 1:818740_T_C | 0 | 818740 | T | C |\n",
       "| 5 | 1 | rs145604921  | 0 | 819378 | 0 | C |\n",
       "| 6 | 1 | rs535256652  | 0 | 821053 | 0 | T |\n",
       "\n"
      ],
      "text/plain": [
       "  chr snps         cM pos    minor major\n",
       "1 1   rs3131972    0  817341 A     G    \n",
       "2 1   rs546843995  0  818053 0     G    \n",
       "3 1   rs553916047  0  818359 0     A    \n",
       "4 1   1:818740_T_C 0  818740 T     C    \n",
       "5 1   rs145604921  0  819378 0     C    \n",
       "6 1   rs535256652  0  821053 0     T    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(geno.bim)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26c34029-3efa-455d-9485-103384dd4786",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Heatmap"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "076c174a-88e5-4cc9-984f-a19230862509",
   "metadata": {},
   "source": [
    "Before performing single-variant association analysis, we want to check if any covariates are strongly associated with the module eigengenes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6df18220-afd6-43ac-b99f-f15563ff1cca",
   "metadata": {},
   "outputs": [],
   "source": [
    "all.vars <- merge(covs, eigengenes, by.x=0, by.y=0) %>%\n",
    "    dplyr::select(-Row.names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "970676d3-5339-4770-89ea-d97d779932b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colors <- colorRampPalette(brewer.pal(11, \"RdBu\"))(101)\n",
    "\n",
    "cor.mtx <- cor(all.vars)\n",
    "cor.mtx.peer <- cor.mtx[peer, colnames(eigengenes)]\n",
    "cor.mtx.held.out <- cor.mtx[held.out, colnames(eigengenes)]\n",
    "\n",
    "h1 <- Heatmap(cor.mtx.peer, col=colors, heatmap_legend_param=list(\n",
    "    title=\"Correlation\",\n",
    "    at=c(-1, 0, 1)\n",
    "))\n",
    "\n",
    "h2 <- Heatmap(cor.mtx.held.out, col=colors, cluster_rows=F, heatmap_legend_param=list(\n",
    "    title=\"Correlation\",\n",
    "    at=c(-1, 0, 1)\n",
    "))\n",
    "\n",
    "svg(\"04_Expression/results/eigengene_peer_factor_correlation.svg\", width=12, height=8)\n",
    "h1 %v% h2\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfa945dc-6083-47e8-982c-d3c403cf6375",
   "metadata": {},
   "source": [
    "The PEER factors are built using gene expression data. It's not surprising that many of the eigengenes are associated with the PEER factors. WGCNA discarded around 8000 genes. The residual gene expression variation from these genes is likely captured by some of the PEER factors. We model this specifically using 30 gene expression PCs from the discarded genes.\n",
    "\n",
    "The eigengenes are not associated with genotyping PCs (which is good)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8eb0417-04be-46dc-a543-1e58f94529fc",
   "metadata": {},
   "source": [
    "![](../results/eigengene_peer_factor_correlation.svg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32443e95-d375-40f2-a8e2-b2aee70dcd64",
   "metadata": {},
   "source": [
    "Ideally, I will include only sex and genotyping PCs as covariates in this mapping, since everything else (cell counts, SRS group) are important signatures that may be correlated with module eigengenes. I will include the 30 PCs from the unassigned gene expression data to control for technical confounders."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7e66a4f-1e9c-490b-84e5-ad4193d9036d",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Save Covariates"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18223439-bc5d-4e9d-9bed-57fbd2228210",
   "metadata": {},
   "source": [
    "Save list of patients (family ID in the first column and individual ID in the second column) to subset the genotyping data using PLINK. There are 638 unique patients with genotyping information."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "13413b35-1092-4b7a-97ef-7a734fdd0dd5",
   "metadata": {},
   "outputs": [],
   "source": [
    "eigengene.patients <- sapply(strsplit(rownames(eigengenes), \"_\"), function(x) { x[1] })\n",
    "\n",
    "geno.fam <- fread(\"~/gains_team282/Genotyping/All_genotyping_merged_filtered_b38_refiltered_rsID.fam\") %>%\n",
    "    dplyr::select(Family.ID=1, Individual.ID=2) %>%\n",
    "    dplyr::mutate(GAinS.ID=gsub(\"^GA\", \"\", Individual.ID)) %>%\n",
    "    dplyr::filter(GAinS.ID %in% eigengene.patients) %>%\n",
    "    unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "93a8597a-1a50-423b-a6e9-fc179ac88d39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>638</li><li>3</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 638\n",
       "\\item 3\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 638\n",
       "2. 3\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 638   3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(geno.fam)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "61dae4de-5a7a-4686-a22c-c981fee7c1b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(\n",
    "    geno.fam %>% dplyr::select(Family.ID, Individual.ID), \n",
    "    \"~/gains_team282/nikhil/expression/eigengene_sva/mapping_patients.txt\", row.names=F, quote=F, col.names=F, sep=\"\\t\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af790d1e-583b-4a87-848d-c44edf00a4c5",
   "metadata": {},
   "source": [
    "Save the list of module eigengenes as a text file. This will be used by NextFlow to parallelize the association mapping."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "0b100493-d60e-4f89-b9cf-c966916b8fae",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(colnames(eigengenes), \"~/gains_team282/nikhil/expression/eigengene_sva/mapping_eigengenes.txt\", row.names=F, quote=F, col.names=F, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a99a350-ac3e-4c19-b846-80171620e7b1",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Initial Pass"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "702d915b-5cd6-4276-a5a1-e038d0248a28",
   "metadata": {},
   "source": [
    "I will be using a linear mixed model (LMM) to test for association between genotypes and eigengene expression. In the past, the lab has used a likelihood test (F-Test) to compare a null model of the covariates against an alternative model where the genotype is included. I will be building the following model:\n",
    "\n",
    "1. Let $\\mathbf{E}_i\\in\\mathbb{R}^n$ be a vector representing the values of the $i$-th eigengene.\n",
    "2. Let $\\mathbf{Y}\\in\\mathbb{R}^{n\\times c}$ be a matrix of covariates. These covariates include Sex, 16 PEER factors, and 7 Genotyping PCs.\n",
    "3. Let $\\mathbf{Z}\\in\\mathbb{R}^n$ be a vector representing the random effects. The only random effect in this model is the Patient ID.\n",
    "4. Let $\\mathbf{X}\\in\\mathbb{R}^n$ be a vector representing the genotypes of the patients.\n",
    "5. Let $\\beta\\in\\mathbb{R}$ be a scalar value representing the genotypic effect on eigengene expression.\n",
    "6. Let $\\mathbf{\\alpha}\\in\\mathbb{R}^c$ be a vector of covariate effects on eigengene expression.\n",
    "7. Let $\\gamma\\in\\mathbb{R}$ be a scalar value representing the random effect on eigengene expression.\n",
    "\n",
    "The null model is:\n",
    "\n",
    "$$\\mathbf{E}_i \\sim \\mathbf{Y}\\alpha + \\mathbf{Z}\\gamma$$\n",
    "\n",
    "The alternative model is:\n",
    "\n",
    "$$\\mathbf{E}_i \\sim \\mathbf{X}\\beta + \\mathbf{Y}\\alpha + \\mathbf{Z}\\gamma$$"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e20133d-11f0-480f-9bec-4869a8279161",
   "metadata": {},
   "source": [
    "There are 823 samples from the RNA-Seq data that have genotypes as well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "2c9a6515-9ca4-4974-abb0-c5fabf338edc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "823"
      ],
      "text/latex": [
       "823"
      ],
      "text/markdown": [
       "823"
      ],
      "text/plain": [
       "[1] 823"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "samples.with.genotypes <- rownames(eigengenes)[\n",
    "    sapply(\n",
    "        strsplit(rownames(eigengenes), \"_\"),\n",
    "        function(x) { x[1] %in% geno.fam$GAinS.ID }\n",
    "    )\n",
    "]\n",
    "\n",
    "length(samples.with.genotypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e8ae2e1e-aee5-4217-bd0a-538807c882b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'PEER_1'</li><li>'PEER_2'</li><li>'PEER_3'</li><li>'PEER_4'</li><li>'PEER_5'</li><li>'PEER_6'</li><li>'PEER_7'</li><li>'PEER_8'</li><li>'PEER_9'</li><li>'PEER_10'</li><li>'PEER_11'</li><li>'PEER_12'</li><li>'PEER_13'</li><li>'PEER_14'</li><li>'PEER_15'</li><li>'PEER_16'</li><li>'PEER_17'</li><li>'PEER_18'</li><li>'PEER_19'</li><li>'PEER_20'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'PEER\\_1'\n",
       "\\item 'PEER\\_2'\n",
       "\\item 'PEER\\_3'\n",
       "\\item 'PEER\\_4'\n",
       "\\item 'PEER\\_5'\n",
       "\\item 'PEER\\_6'\n",
       "\\item 'PEER\\_7'\n",
       "\\item 'PEER\\_8'\n",
       "\\item 'PEER\\_9'\n",
       "\\item 'PEER\\_10'\n",
       "\\item 'PEER\\_11'\n",
       "\\item 'PEER\\_12'\n",
       "\\item 'PEER\\_13'\n",
       "\\item 'PEER\\_14'\n",
       "\\item 'PEER\\_15'\n",
       "\\item 'PEER\\_16'\n",
       "\\item 'PEER\\_17'\n",
       "\\item 'PEER\\_18'\n",
       "\\item 'PEER\\_19'\n",
       "\\item 'PEER\\_20'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'PEER_1'\n",
       "2. 'PEER_2'\n",
       "3. 'PEER_3'\n",
       "4. 'PEER_4'\n",
       "5. 'PEER_5'\n",
       "6. 'PEER_6'\n",
       "7. 'PEER_7'\n",
       "8. 'PEER_8'\n",
       "9. 'PEER_9'\n",
       "10. 'PEER_10'\n",
       "11. 'PEER_11'\n",
       "12. 'PEER_12'\n",
       "13. 'PEER_13'\n",
       "14. 'PEER_14'\n",
       "15. 'PEER_15'\n",
       "16. 'PEER_16'\n",
       "17. 'PEER_17'\n",
       "18. 'PEER_18'\n",
       "19. 'PEER_19'\n",
       "20. 'PEER_20'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"PEER_1\"  \"PEER_2\"  \"PEER_3\"  \"PEER_4\"  \"PEER_5\"  \"PEER_6\"  \"PEER_7\" \n",
       " [8] \"PEER_8\"  \"PEER_9\"  \"PEER_10\" \"PEER_11\" \"PEER_12\" \"PEER_13\" \"PEER_14\"\n",
       "[15] \"PEER_15\" \"PEER_16\" \"PEER_17\" \"PEER_18\" \"PEER_19\" \"PEER_20\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sex'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'Neutrophils'</li><li>'Lymphocytes'</li><li>'Monocytes'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sex'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'Neutrophils'\n",
       "\\item 'Lymphocytes'\n",
       "\\item 'Monocytes'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sex'\n",
       "2. 'PC1'\n",
       "3. 'PC2'\n",
       "4. 'PC3'\n",
       "5. 'PC4'\n",
       "6. 'PC5'\n",
       "7. 'PC6'\n",
       "8. 'PC7'\n",
       "9. 'Neutrophils'\n",
       "10. 'Lymphocytes'\n",
       "11. 'Monocytes'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sex\"         \"PC1\"         \"PC2\"         \"PC3\"         \"PC4\"        \n",
       " [6] \"PC5\"         \"PC6\"         \"PC7\"         \"Neutrophils\" \"Lymphocytes\"\n",
       "[11] \"Monocytes\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'GAinS.ID'"
      ],
      "text/latex": [
       "'GAinS.ID'"
      ],
      "text/markdown": [
       "'GAinS.ID'"
      ],
      "text/plain": [
       "[1] \"GAinS.ID\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "peer.included <- peer[1:20]\n",
    "peer.included\n",
    "\n",
    "cov.names <- c(\"sex\", paste0(\"PC\", 1:7), \"Neutrophils\", \"Lymphocytes\", \"Monocytes\")\n",
    "cov.names\n",
    "\n",
    "rand.effect.names <- c(\"GAinS.ID\")\n",
    "rand.effect.names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "0b68e365-3db2-4446-8fe9-f71ab5fdd4eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping.data <- merge(eigengenes, covs, by=0) %>%\n",
    "    dplyr::filter(Row.names %in% samples.with.genotypes) %>%\n",
    "    dplyr::select(Sample.ID=Row.names, everything()) %>%\n",
    "    dplyr::mutate(GAinS.ID.NonPrefix=sapply(strsplit(Sample.ID, \"_\"), function(x) { x[1] })) %>%\n",
    "    merge(., geno.fam, by.x=\"GAinS.ID.NonPrefix\", by.y=\"GAinS.ID\") %>%\n",
    "    dplyr::select(GAinS.ID=Individual.ID, everything()) %>%\n",
    "    dplyr::select(Sample.ID, any_of(rand.effect.names), any_of(colnames(eigengenes)), any_of(cov.names), any_of(peer.included))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "35b5728d-c41e-42b9-9bad-844187829c68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 139</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample.ID</th><th scope=col>GAinS.ID</th><th scope=col>ME_1</th><th scope=col>ME_2</th><th scope=col>ME_3</th><th scope=col>ME_4</th><th scope=col>ME_5</th><th scope=col>ME_6</th><th scope=col>ME_7</th><th scope=col>ME_8</th><th scope=col>⋯</th><th scope=col>PEER_11</th><th scope=col>PEER_12</th><th scope=col>PEER_13</th><th scope=col>PEER_14</th><th scope=col>PEER_15</th><th scope=col>PEER_16</th><th scope=col>PEER_17</th><th scope=col>PEER_18</th><th scope=col>PEER_19</th><th scope=col>PEER_20</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;I&lt;chr&gt;&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>UK01050129_5</td><td>UK01050129</td><td>-0.011965012</td><td>-0.008135170</td><td> 0.02189999</td><td>-0.01050068</td><td> 0.0739432434</td><td> 0.0443109416</td><td> 0.014796688</td><td>-0.063930040</td><td>⋯</td><td> 0.047811605</td><td>-0.016539395</td><td> 0.051095940</td><td>-0.026148196</td><td>-0.021135228</td><td> 0.002443923</td><td> 0.032696567</td><td> 0.035078023</td><td> 0.03328779</td><td>-0.072080098</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>UK01060123_3</td><td>UK01060123</td><td>-0.002691514</td><td>-0.003642363</td><td> 0.00948328</td><td>-0.01205738</td><td> 0.0369338683</td><td>-0.0251717916</td><td> 0.004446174</td><td>-0.033855082</td><td>⋯</td><td>-0.007224238</td><td> 0.016796881</td><td>-0.010660484</td><td>-0.017610908</td><td> 0.031190908</td><td> 0.005303205</td><td> 0.040227711</td><td> 0.023455137</td><td>-0.06972641</td><td>-0.016429750</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>UK01070117_3</td><td>UK01070117</td><td>-0.027671656</td><td>-0.018676902</td><td>-0.02153764</td><td> 0.02309788</td><td> 0.0213639304</td><td>-0.0033192787</td><td> 0.027935528</td><td>-0.043763863</td><td>⋯</td><td> 0.002017805</td><td> 0.013472254</td><td> 0.044990435</td><td> 0.017342746</td><td> 0.040239949</td><td>-0.037767619</td><td>-0.002149181</td><td> 0.010321584</td><td>-0.03437537</td><td>-0.008777559</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>UK01080111_1</td><td>UK01080111</td><td> 0.021899313</td><td> 0.011468622</td><td> 0.02719409</td><td>-0.01318245</td><td> 0.0004065718</td><td>-0.0360484365</td><td>-0.022309759</td><td>-0.001370201</td><td>⋯</td><td>-0.063317396</td><td>-0.072918773</td><td>-0.010724876</td><td>-0.016138442</td><td>-0.003775158</td><td>-0.032603394</td><td>-0.068149783</td><td>-0.007441163</td><td>-0.04162392</td><td>-0.056151949</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>UK01110093_1</td><td>UK01110093</td><td>-0.053133706</td><td>-0.054170247</td><td>-0.09207201</td><td>-0.00263561</td><td>-0.0545464726</td><td>-0.0527906474</td><td> 0.047146445</td><td> 0.024393979</td><td>⋯</td><td>-0.009035902</td><td>-0.001971741</td><td>-0.003907994</td><td>-0.001863737</td><td>-0.031021271</td><td>-0.001989635</td><td> 0.010791964</td><td>-0.013796184</td><td> 0.09261347</td><td>-0.072555415</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>UK01110093_5</td><td>UK01110093</td><td>-0.097745041</td><td>-0.093620110</td><td>-0.11078002</td><td> 0.02619148</td><td>-0.0161876898</td><td> 0.0004345917</td><td> 0.087194280</td><td>-0.028478758</td><td>⋯</td><td> 0.076265849</td><td>-0.071181759</td><td> 0.045724034</td><td>-0.083036296</td><td> 0.009541817</td><td>-0.043817975</td><td> 0.079534695</td><td> 0.008144007</td><td> 0.09541912</td><td>-0.201759934</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 139\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Sample.ID & GAinS.ID & ME\\_1 & ME\\_2 & ME\\_3 & ME\\_4 & ME\\_5 & ME\\_6 & ME\\_7 & ME\\_8 & ⋯ & PEER\\_11 & PEER\\_12 & PEER\\_13 & PEER\\_14 & PEER\\_15 & PEER\\_16 & PEER\\_17 & PEER\\_18 & PEER\\_19 & PEER\\_20\\\\\n",
       "  & <I<chr>> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & UK01050129\\_5 & UK01050129 & -0.011965012 & -0.008135170 &  0.02189999 & -0.01050068 &  0.0739432434 &  0.0443109416 &  0.014796688 & -0.063930040 & ⋯ &  0.047811605 & -0.016539395 &  0.051095940 & -0.026148196 & -0.021135228 &  0.002443923 &  0.032696567 &  0.035078023 &  0.03328779 & -0.072080098\\\\\n",
       "\t2 & UK01060123\\_3 & UK01060123 & -0.002691514 & -0.003642363 &  0.00948328 & -0.01205738 &  0.0369338683 & -0.0251717916 &  0.004446174 & -0.033855082 & ⋯ & -0.007224238 &  0.016796881 & -0.010660484 & -0.017610908 &  0.031190908 &  0.005303205 &  0.040227711 &  0.023455137 & -0.06972641 & -0.016429750\\\\\n",
       "\t3 & UK01070117\\_3 & UK01070117 & -0.027671656 & -0.018676902 & -0.02153764 &  0.02309788 &  0.0213639304 & -0.0033192787 &  0.027935528 & -0.043763863 & ⋯ &  0.002017805 &  0.013472254 &  0.044990435 &  0.017342746 &  0.040239949 & -0.037767619 & -0.002149181 &  0.010321584 & -0.03437537 & -0.008777559\\\\\n",
       "\t4 & UK01080111\\_1 & UK01080111 &  0.021899313 &  0.011468622 &  0.02719409 & -0.01318245 &  0.0004065718 & -0.0360484365 & -0.022309759 & -0.001370201 & ⋯ & -0.063317396 & -0.072918773 & -0.010724876 & -0.016138442 & -0.003775158 & -0.032603394 & -0.068149783 & -0.007441163 & -0.04162392 & -0.056151949\\\\\n",
       "\t5 & UK01110093\\_1 & UK01110093 & -0.053133706 & -0.054170247 & -0.09207201 & -0.00263561 & -0.0545464726 & -0.0527906474 &  0.047146445 &  0.024393979 & ⋯ & -0.009035902 & -0.001971741 & -0.003907994 & -0.001863737 & -0.031021271 & -0.001989635 &  0.010791964 & -0.013796184 &  0.09261347 & -0.072555415\\\\\n",
       "\t6 & UK01110093\\_5 & UK01110093 & -0.097745041 & -0.093620110 & -0.11078002 &  0.02619148 & -0.0161876898 &  0.0004345917 &  0.087194280 & -0.028478758 & ⋯ &  0.076265849 & -0.071181759 &  0.045724034 & -0.083036296 &  0.009541817 & -0.043817975 &  0.079534695 &  0.008144007 &  0.09541912 & -0.201759934\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 139\n",
       "\n",
       "| <!--/--> | Sample.ID &lt;I&lt;chr&gt;&gt; | GAinS.ID &lt;chr&gt; | ME_1 &lt;dbl&gt; | ME_2 &lt;dbl&gt; | ME_3 &lt;dbl&gt; | ME_4 &lt;dbl&gt; | ME_5 &lt;dbl&gt; | ME_6 &lt;dbl&gt; | ME_7 &lt;dbl&gt; | ME_8 &lt;dbl&gt; | ⋯ ⋯ | PEER_11 &lt;dbl&gt; | PEER_12 &lt;dbl&gt; | PEER_13 &lt;dbl&gt; | PEER_14 &lt;dbl&gt; | PEER_15 &lt;dbl&gt; | PEER_16 &lt;dbl&gt; | PEER_17 &lt;dbl&gt; | PEER_18 &lt;dbl&gt; | PEER_19 &lt;dbl&gt; | PEER_20 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | UK01050129_5 | UK01050129 | -0.011965012 | -0.008135170 |  0.02189999 | -0.01050068 |  0.0739432434 |  0.0443109416 |  0.014796688 | -0.063930040 | ⋯ |  0.047811605 | -0.016539395 |  0.051095940 | -0.026148196 | -0.021135228 |  0.002443923 |  0.032696567 |  0.035078023 |  0.03328779 | -0.072080098 |\n",
       "| 2 | UK01060123_3 | UK01060123 | -0.002691514 | -0.003642363 |  0.00948328 | -0.01205738 |  0.0369338683 | -0.0251717916 |  0.004446174 | -0.033855082 | ⋯ | -0.007224238 |  0.016796881 | -0.010660484 | -0.017610908 |  0.031190908 |  0.005303205 |  0.040227711 |  0.023455137 | -0.06972641 | -0.016429750 |\n",
       "| 3 | UK01070117_3 | UK01070117 | -0.027671656 | -0.018676902 | -0.02153764 |  0.02309788 |  0.0213639304 | -0.0033192787 |  0.027935528 | -0.043763863 | ⋯ |  0.002017805 |  0.013472254 |  0.044990435 |  0.017342746 |  0.040239949 | -0.037767619 | -0.002149181 |  0.010321584 | -0.03437537 | -0.008777559 |\n",
       "| 4 | UK01080111_1 | UK01080111 |  0.021899313 |  0.011468622 |  0.02719409 | -0.01318245 |  0.0004065718 | -0.0360484365 | -0.022309759 | -0.001370201 | ⋯ | -0.063317396 | -0.072918773 | -0.010724876 | -0.016138442 | -0.003775158 | -0.032603394 | -0.068149783 | -0.007441163 | -0.04162392 | -0.056151949 |\n",
       "| 5 | UK01110093_1 | UK01110093 | -0.053133706 | -0.054170247 | -0.09207201 | -0.00263561 | -0.0545464726 | -0.0527906474 |  0.047146445 |  0.024393979 | ⋯ | -0.009035902 | -0.001971741 | -0.003907994 | -0.001863737 | -0.031021271 | -0.001989635 |  0.010791964 | -0.013796184 |  0.09261347 | -0.072555415 |\n",
       "| 6 | UK01110093_5 | UK01110093 | -0.097745041 | -0.093620110 | -0.11078002 |  0.02619148 | -0.0161876898 |  0.0004345917 |  0.087194280 | -0.028478758 | ⋯ |  0.076265849 | -0.071181759 |  0.045724034 | -0.083036296 |  0.009541817 | -0.043817975 |  0.079534695 |  0.008144007 |  0.09541912 | -0.201759934 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample.ID    GAinS.ID   ME_1         ME_2         ME_3        ME_4       \n",
       "1 UK01050129_5 UK01050129 -0.011965012 -0.008135170  0.02189999 -0.01050068\n",
       "2 UK01060123_3 UK01060123 -0.002691514 -0.003642363  0.00948328 -0.01205738\n",
       "3 UK01070117_3 UK01070117 -0.027671656 -0.018676902 -0.02153764  0.02309788\n",
       "4 UK01080111_1 UK01080111  0.021899313  0.011468622  0.02719409 -0.01318245\n",
       "5 UK01110093_1 UK01110093 -0.053133706 -0.054170247 -0.09207201 -0.00263561\n",
       "6 UK01110093_5 UK01110093 -0.097745041 -0.093620110 -0.11078002  0.02619148\n",
       "  ME_5          ME_6          ME_7         ME_8         ⋯ PEER_11     \n",
       "1  0.0739432434  0.0443109416  0.014796688 -0.063930040 ⋯  0.047811605\n",
       "2  0.0369338683 -0.0251717916  0.004446174 -0.033855082 ⋯ -0.007224238\n",
       "3  0.0213639304 -0.0033192787  0.027935528 -0.043763863 ⋯  0.002017805\n",
       "4  0.0004065718 -0.0360484365 -0.022309759 -0.001370201 ⋯ -0.063317396\n",
       "5 -0.0545464726 -0.0527906474  0.047146445  0.024393979 ⋯ -0.009035902\n",
       "6 -0.0161876898  0.0004345917  0.087194280 -0.028478758 ⋯  0.076265849\n",
       "  PEER_12      PEER_13      PEER_14      PEER_15      PEER_16      PEER_17     \n",
       "1 -0.016539395  0.051095940 -0.026148196 -0.021135228  0.002443923  0.032696567\n",
       "2  0.016796881 -0.010660484 -0.017610908  0.031190908  0.005303205  0.040227711\n",
       "3  0.013472254  0.044990435  0.017342746  0.040239949 -0.037767619 -0.002149181\n",
       "4 -0.072918773 -0.010724876 -0.016138442 -0.003775158 -0.032603394 -0.068149783\n",
       "5 -0.001971741 -0.003907994 -0.001863737 -0.031021271 -0.001989635  0.010791964\n",
       "6 -0.071181759  0.045724034 -0.083036296  0.009541817 -0.043817975  0.079534695\n",
       "  PEER_18      PEER_19     PEER_20     \n",
       "1  0.035078023  0.03328779 -0.072080098\n",
       "2  0.023455137 -0.06972641 -0.016429750\n",
       "3  0.010321584 -0.03437537 -0.008777559\n",
       "4 -0.007441163 -0.04162392 -0.056151949\n",
       "5 -0.013796184  0.09261347 -0.072555415\n",
       "6  0.008144007  0.09541912 -0.201759934"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(mapping.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "b6ec6673-664f-4cd2-862b-9805f25551d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Sample.ID'</li><li>'GAinS.ID'</li><li>'ME_1'</li><li>'ME_2'</li><li>'ME_3'</li><li>'ME_4'</li><li>'ME_5'</li><li>'ME_6'</li><li>'ME_7'</li><li>'ME_8'</li><li>'ME_9'</li><li>'ME_10'</li><li>'ME_11'</li><li>'ME_12'</li><li>'ME_13'</li><li>'ME_14'</li><li>'ME_15'</li><li>'ME_16'</li><li>'ME_17'</li><li>'ME_18'</li><li>'ME_19'</li><li>'ME_20'</li><li>'ME_21'</li><li>'ME_22'</li><li>'ME_23'</li><li>'ME_24'</li><li>'ME_25'</li><li>'ME_26'</li><li>'ME_27'</li><li>'ME_28'</li><li>'ME_29'</li><li>'ME_30'</li><li>'ME_31'</li><li>'ME_32'</li><li>'ME_33'</li><li>'ME_34'</li><li>'ME_35'</li><li>'ME_36'</li><li>'ME_37'</li><li>'ME_38'</li><li>'ME_39'</li><li>'ME_40'</li><li>'ME_41'</li><li>'ME_42'</li><li>'ME_43'</li><li>'ME_44'</li><li>'ME_45'</li><li>'ME_46'</li><li>'ME_47'</li><li>'ME_48'</li><li>'ME_49'</li><li>'ME_50'</li><li>'ME_51'</li><li>'ME_52'</li><li>'ME_53'</li><li>'ME_54'</li><li>'ME_55'</li><li>'ME_56'</li><li>'ME_57'</li><li>'ME_58'</li><li>'ME_59'</li><li>'ME_60'</li><li>'ME_61'</li><li>'ME_62'</li><li>'ME_63'</li><li>'ME_64'</li><li>'ME_65'</li><li>'ME_66'</li><li>'ME_67'</li><li>'ME_68'</li><li>'ME_69'</li><li>'ME_70'</li><li>'ME_71'</li><li>'ME_72'</li><li>'ME_73'</li><li>'ME_74'</li><li>'ME_75'</li><li>'ME_76'</li><li>'ME_77'</li><li>'ME_78'</li><li>'ME_79'</li><li>'ME_80'</li><li>'ME_81'</li><li>'ME_82'</li><li>'ME_83'</li><li>'ME_84'</li><li>'ME_85'</li><li>'ME_86'</li><li>'ME_87'</li><li>'ME_88'</li><li>'ME_89'</li><li>'ME_90'</li><li>'ME_91'</li><li>'ME_92'</li><li>'ME_93'</li><li>'ME_94'</li><li>'ME_95'</li><li>'ME_96'</li><li>'ME_97'</li><li>'ME_98'</li><li>'ME_99'</li><li>'ME_100'</li><li>'ME_101'</li><li>'ME_102'</li><li>'ME_103'</li><li>'ME_104'</li><li>'ME_105'</li><li>'ME_106'</li><li>'sex'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'Neutrophils'</li><li>'Lymphocytes'</li><li>'Monocytes'</li><li>'PEER_1'</li><li>'PEER_2'</li><li>'PEER_3'</li><li>'PEER_4'</li><li>'PEER_5'</li><li>'PEER_6'</li><li>'PEER_7'</li><li>'PEER_8'</li><li>'PEER_9'</li><li>'PEER_10'</li><li>'PEER_11'</li><li>'PEER_12'</li><li>'PEER_13'</li><li>'PEER_14'</li><li>'PEER_15'</li><li>'PEER_16'</li><li>'PEER_17'</li><li>'PEER_18'</li><li>'PEER_19'</li><li>'PEER_20'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Sample.ID'\n",
       "\\item 'GAinS.ID'\n",
       "\\item 'ME\\_1'\n",
       "\\item 'ME\\_2'\n",
       "\\item 'ME\\_3'\n",
       "\\item 'ME\\_4'\n",
       "\\item 'ME\\_5'\n",
       "\\item 'ME\\_6'\n",
       "\\item 'ME\\_7'\n",
       "\\item 'ME\\_8'\n",
       "\\item 'ME\\_9'\n",
       "\\item 'ME\\_10'\n",
       "\\item 'ME\\_11'\n",
       "\\item 'ME\\_12'\n",
       "\\item 'ME\\_13'\n",
       "\\item 'ME\\_14'\n",
       "\\item 'ME\\_15'\n",
       "\\item 'ME\\_16'\n",
       "\\item 'ME\\_17'\n",
       "\\item 'ME\\_18'\n",
       "\\item 'ME\\_19'\n",
       "\\item 'ME\\_20'\n",
       "\\item 'ME\\_21'\n",
       "\\item 'ME\\_22'\n",
       "\\item 'ME\\_23'\n",
       "\\item 'ME\\_24'\n",
       "\\item 'ME\\_25'\n",
       "\\item 'ME\\_26'\n",
       "\\item 'ME\\_27'\n",
       "\\item 'ME\\_28'\n",
       "\\item 'ME\\_29'\n",
       "\\item 'ME\\_30'\n",
       "\\item 'ME\\_31'\n",
       "\\item 'ME\\_32'\n",
       "\\item 'ME\\_33'\n",
       "\\item 'ME\\_34'\n",
       "\\item 'ME\\_35'\n",
       "\\item 'ME\\_36'\n",
       "\\item 'ME\\_37'\n",
       "\\item 'ME\\_38'\n",
       "\\item 'ME\\_39'\n",
       "\\item 'ME\\_40'\n",
       "\\item 'ME\\_41'\n",
       "\\item 'ME\\_42'\n",
       "\\item 'ME\\_43'\n",
       "\\item 'ME\\_44'\n",
       "\\item 'ME\\_45'\n",
       "\\item 'ME\\_46'\n",
       "\\item 'ME\\_47'\n",
       "\\item 'ME\\_48'\n",
       "\\item 'ME\\_49'\n",
       "\\item 'ME\\_50'\n",
       "\\item 'ME\\_51'\n",
       "\\item 'ME\\_52'\n",
       "\\item 'ME\\_53'\n",
       "\\item 'ME\\_54'\n",
       "\\item 'ME\\_55'\n",
       "\\item 'ME\\_56'\n",
       "\\item 'ME\\_57'\n",
       "\\item 'ME\\_58'\n",
       "\\item 'ME\\_59'\n",
       "\\item 'ME\\_60'\n",
       "\\item 'ME\\_61'\n",
       "\\item 'ME\\_62'\n",
       "\\item 'ME\\_63'\n",
       "\\item 'ME\\_64'\n",
       "\\item 'ME\\_65'\n",
       "\\item 'ME\\_66'\n",
       "\\item 'ME\\_67'\n",
       "\\item 'ME\\_68'\n",
       "\\item 'ME\\_69'\n",
       "\\item 'ME\\_70'\n",
       "\\item 'ME\\_71'\n",
       "\\item 'ME\\_72'\n",
       "\\item 'ME\\_73'\n",
       "\\item 'ME\\_74'\n",
       "\\item 'ME\\_75'\n",
       "\\item 'ME\\_76'\n",
       "\\item 'ME\\_77'\n",
       "\\item 'ME\\_78'\n",
       "\\item 'ME\\_79'\n",
       "\\item 'ME\\_80'\n",
       "\\item 'ME\\_81'\n",
       "\\item 'ME\\_82'\n",
       "\\item 'ME\\_83'\n",
       "\\item 'ME\\_84'\n",
       "\\item 'ME\\_85'\n",
       "\\item 'ME\\_86'\n",
       "\\item 'ME\\_87'\n",
       "\\item 'ME\\_88'\n",
       "\\item 'ME\\_89'\n",
       "\\item 'ME\\_90'\n",
       "\\item 'ME\\_91'\n",
       "\\item 'ME\\_92'\n",
       "\\item 'ME\\_93'\n",
       "\\item 'ME\\_94'\n",
       "\\item 'ME\\_95'\n",
       "\\item 'ME\\_96'\n",
       "\\item 'ME\\_97'\n",
       "\\item 'ME\\_98'\n",
       "\\item 'ME\\_99'\n",
       "\\item 'ME\\_100'\n",
       "\\item 'ME\\_101'\n",
       "\\item 'ME\\_102'\n",
       "\\item 'ME\\_103'\n",
       "\\item 'ME\\_104'\n",
       "\\item 'ME\\_105'\n",
       "\\item 'ME\\_106'\n",
       "\\item 'sex'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'Neutrophils'\n",
       "\\item 'Lymphocytes'\n",
       "\\item 'Monocytes'\n",
       "\\item 'PEER\\_1'\n",
       "\\item 'PEER\\_2'\n",
       "\\item 'PEER\\_3'\n",
       "\\item 'PEER\\_4'\n",
       "\\item 'PEER\\_5'\n",
       "\\item 'PEER\\_6'\n",
       "\\item 'PEER\\_7'\n",
       "\\item 'PEER\\_8'\n",
       "\\item 'PEER\\_9'\n",
       "\\item 'PEER\\_10'\n",
       "\\item 'PEER\\_11'\n",
       "\\item 'PEER\\_12'\n",
       "\\item 'PEER\\_13'\n",
       "\\item 'PEER\\_14'\n",
       "\\item 'PEER\\_15'\n",
       "\\item 'PEER\\_16'\n",
       "\\item 'PEER\\_17'\n",
       "\\item 'PEER\\_18'\n",
       "\\item 'PEER\\_19'\n",
       "\\item 'PEER\\_20'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Sample.ID'\n",
       "2. 'GAinS.ID'\n",
       "3. 'ME_1'\n",
       "4. 'ME_2'\n",
       "5. 'ME_3'\n",
       "6. 'ME_4'\n",
       "7. 'ME_5'\n",
       "8. 'ME_6'\n",
       "9. 'ME_7'\n",
       "10. 'ME_8'\n",
       "11. 'ME_9'\n",
       "12. 'ME_10'\n",
       "13. 'ME_11'\n",
       "14. 'ME_12'\n",
       "15. 'ME_13'\n",
       "16. 'ME_14'\n",
       "17. 'ME_15'\n",
       "18. 'ME_16'\n",
       "19. 'ME_17'\n",
       "20. 'ME_18'\n",
       "21. 'ME_19'\n",
       "22. 'ME_20'\n",
       "23. 'ME_21'\n",
       "24. 'ME_22'\n",
       "25. 'ME_23'\n",
       "26. 'ME_24'\n",
       "27. 'ME_25'\n",
       "28. 'ME_26'\n",
       "29. 'ME_27'\n",
       "30. 'ME_28'\n",
       "31. 'ME_29'\n",
       "32. 'ME_30'\n",
       "33. 'ME_31'\n",
       "34. 'ME_32'\n",
       "35. 'ME_33'\n",
       "36. 'ME_34'\n",
       "37. 'ME_35'\n",
       "38. 'ME_36'\n",
       "39. 'ME_37'\n",
       "40. 'ME_38'\n",
       "41. 'ME_39'\n",
       "42. 'ME_40'\n",
       "43. 'ME_41'\n",
       "44. 'ME_42'\n",
       "45. 'ME_43'\n",
       "46. 'ME_44'\n",
       "47. 'ME_45'\n",
       "48. 'ME_46'\n",
       "49. 'ME_47'\n",
       "50. 'ME_48'\n",
       "51. 'ME_49'\n",
       "52. 'ME_50'\n",
       "53. 'ME_51'\n",
       "54. 'ME_52'\n",
       "55. 'ME_53'\n",
       "56. 'ME_54'\n",
       "57. 'ME_55'\n",
       "58. 'ME_56'\n",
       "59. 'ME_57'\n",
       "60. 'ME_58'\n",
       "61. 'ME_59'\n",
       "62. 'ME_60'\n",
       "63. 'ME_61'\n",
       "64. 'ME_62'\n",
       "65. 'ME_63'\n",
       "66. 'ME_64'\n",
       "67. 'ME_65'\n",
       "68. 'ME_66'\n",
       "69. 'ME_67'\n",
       "70. 'ME_68'\n",
       "71. 'ME_69'\n",
       "72. 'ME_70'\n",
       "73. 'ME_71'\n",
       "74. 'ME_72'\n",
       "75. 'ME_73'\n",
       "76. 'ME_74'\n",
       "77. 'ME_75'\n",
       "78. 'ME_76'\n",
       "79. 'ME_77'\n",
       "80. 'ME_78'\n",
       "81. 'ME_79'\n",
       "82. 'ME_80'\n",
       "83. 'ME_81'\n",
       "84. 'ME_82'\n",
       "85. 'ME_83'\n",
       "86. 'ME_84'\n",
       "87. 'ME_85'\n",
       "88. 'ME_86'\n",
       "89. 'ME_87'\n",
       "90. 'ME_88'\n",
       "91. 'ME_89'\n",
       "92. 'ME_90'\n",
       "93. 'ME_91'\n",
       "94. 'ME_92'\n",
       "95. 'ME_93'\n",
       "96. 'ME_94'\n",
       "97. 'ME_95'\n",
       "98. 'ME_96'\n",
       "99. 'ME_97'\n",
       "100. 'ME_98'\n",
       "101. 'ME_99'\n",
       "102. 'ME_100'\n",
       "103. 'ME_101'\n",
       "104. 'ME_102'\n",
       "105. 'ME_103'\n",
       "106. 'ME_104'\n",
       "107. 'ME_105'\n",
       "108. 'ME_106'\n",
       "109. 'sex'\n",
       "110. 'PC1'\n",
       "111. 'PC2'\n",
       "112. 'PC3'\n",
       "113. 'PC4'\n",
       "114. 'PC5'\n",
       "115. 'PC6'\n",
       "116. 'PC7'\n",
       "117. 'Neutrophils'\n",
       "118. 'Lymphocytes'\n",
       "119. 'Monocytes'\n",
       "120. 'PEER_1'\n",
       "121. 'PEER_2'\n",
       "122. 'PEER_3'\n",
       "123. 'PEER_4'\n",
       "124. 'PEER_5'\n",
       "125. 'PEER_6'\n",
       "126. 'PEER_7'\n",
       "127. 'PEER_8'\n",
       "128. 'PEER_9'\n",
       "129. 'PEER_10'\n",
       "130. 'PEER_11'\n",
       "131. 'PEER_12'\n",
       "132. 'PEER_13'\n",
       "133. 'PEER_14'\n",
       "134. 'PEER_15'\n",
       "135. 'PEER_16'\n",
       "136. 'PEER_17'\n",
       "137. 'PEER_18'\n",
       "138. 'PEER_19'\n",
       "139. 'PEER_20'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"Sample.ID\"   \"GAinS.ID\"    \"ME_1\"        \"ME_2\"        \"ME_3\"       \n",
       "  [6] \"ME_4\"        \"ME_5\"        \"ME_6\"        \"ME_7\"        \"ME_8\"       \n",
       " [11] \"ME_9\"        \"ME_10\"       \"ME_11\"       \"ME_12\"       \"ME_13\"      \n",
       " [16] \"ME_14\"       \"ME_15\"       \"ME_16\"       \"ME_17\"       \"ME_18\"      \n",
       " [21] \"ME_19\"       \"ME_20\"       \"ME_21\"       \"ME_22\"       \"ME_23\"      \n",
       " [26] \"ME_24\"       \"ME_25\"       \"ME_26\"       \"ME_27\"       \"ME_28\"      \n",
       " [31] \"ME_29\"       \"ME_30\"       \"ME_31\"       \"ME_32\"       \"ME_33\"      \n",
       " [36] \"ME_34\"       \"ME_35\"       \"ME_36\"       \"ME_37\"       \"ME_38\"      \n",
       " [41] \"ME_39\"       \"ME_40\"       \"ME_41\"       \"ME_42\"       \"ME_43\"      \n",
       " [46] \"ME_44\"       \"ME_45\"       \"ME_46\"       \"ME_47\"       \"ME_48\"      \n",
       " [51] \"ME_49\"       \"ME_50\"       \"ME_51\"       \"ME_52\"       \"ME_53\"      \n",
       " [56] \"ME_54\"       \"ME_55\"       \"ME_56\"       \"ME_57\"       \"ME_58\"      \n",
       " [61] \"ME_59\"       \"ME_60\"       \"ME_61\"       \"ME_62\"       \"ME_63\"      \n",
       " [66] \"ME_64\"       \"ME_65\"       \"ME_66\"       \"ME_67\"       \"ME_68\"      \n",
       " [71] \"ME_69\"       \"ME_70\"       \"ME_71\"       \"ME_72\"       \"ME_73\"      \n",
       " [76] \"ME_74\"       \"ME_75\"       \"ME_76\"       \"ME_77\"       \"ME_78\"      \n",
       " [81] \"ME_79\"       \"ME_80\"       \"ME_81\"       \"ME_82\"       \"ME_83\"      \n",
       " [86] \"ME_84\"       \"ME_85\"       \"ME_86\"       \"ME_87\"       \"ME_88\"      \n",
       " [91] \"ME_89\"       \"ME_90\"       \"ME_91\"       \"ME_92\"       \"ME_93\"      \n",
       " [96] \"ME_94\"       \"ME_95\"       \"ME_96\"       \"ME_97\"       \"ME_98\"      \n",
       "[101] \"ME_99\"       \"ME_100\"      \"ME_101\"      \"ME_102\"      \"ME_103\"     \n",
       "[106] \"ME_104\"      \"ME_105\"      \"ME_106\"      \"sex\"         \"PC1\"        \n",
       "[111] \"PC2\"         \"PC3\"         \"PC4\"         \"PC5\"         \"PC6\"        \n",
       "[116] \"PC7\"         \"Neutrophils\" \"Lymphocytes\" \"Monocytes\"   \"PEER_1\"     \n",
       "[121] \"PEER_2\"      \"PEER_3\"      \"PEER_4\"      \"PEER_5\"      \"PEER_6\"     \n",
       "[126] \"PEER_7\"      \"PEER_8\"      \"PEER_9\"      \"PEER_10\"     \"PEER_11\"    \n",
       "[131] \"PEER_12\"     \"PEER_13\"     \"PEER_14\"     \"PEER_15\"     \"PEER_16\"    \n",
       "[136] \"PEER_17\"     \"PEER_18\"     \"PEER_19\"     \"PEER_20\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(mapping.data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3467b47e-fe69-4d05-a429-96a961835b98",
   "metadata": {},
   "source": [
    "Save the mapping data design matrix for linear mixed modeling."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "0ea89bc0-7ba5-41be-8f25-cf2961c20975",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(mapping.data, \"~/gains_team282/nikhil/expression/eigengene_sva/mapping_data.csv\", row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61f4d47f-529b-4310-b319-7878a5dbaa38",
   "metadata": {},
   "source": [
    "### Identify SNPs for mQTL Mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66e1e685-80a3-4292-98ce-ef867b689896",
   "metadata": {},
   "source": [
    "Since *trans* mapping in a genome-wide fashion will require a lot of testing (the multiple-testing burden will be high), we will use SNPs that we have a more biological hypothesis for. Specifically, I will include all lead eSNPs, all lead eSNPs from the conditional cis-eQTL analysis, and related associations from the EBI GWAS Catalog."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "8b359753-e010-491e-9a22-3a3a9546b14a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "30667"
      ],
      "text/latex": [
       "30667"
      ],
      "text/markdown": [
       "30667"
      ],
      "text/plain": [
       "[1] 30667"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(ebi.studies[,\"DISEASE/TRAIT\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c7befed-55fa-4e35-9097-f3456d9fbb73",
   "metadata": {},
   "source": [
    "After filtering based on the genotyping in GAinS, we have 124,524 studies and 56,218 SNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "28a18096-239a-4f40-ab5e-f5aca97659bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "124524"
      ],
      "text/latex": [
       "124524"
      ],
      "text/markdown": [
       "124524"
      ],
      "text/plain": [
       "[1] 124524"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "56218"
      ],
      "text/latex": [
       "56218"
      ],
      "text/markdown": [
       "56218"
      ],
      "text/plain": [
       "[1] 56218"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Studies with terms specified\n",
    "studies <- ebi.studies[, \"STUDY ACCESSION\"]\n",
    "\n",
    "# Associations from the studies that are genome-wide significant\n",
    "studies.assoc <- ebi.assoc[ebi.assoc[, \"STUDY ACCESSION\"] %in% studies,] %>%\n",
    "    dplyr::mutate(`P-VALUE` = as.numeric(`P-VALUE`)) %>%\n",
    "    dplyr::filter(`P-VALUE` < 5e-8)\n",
    "\n",
    "# Association SNPs that overlap with genotyping data in GAinS\n",
    "# Ignore SNPs that are on the X chromosome\n",
    "studies.assoc <- studies.assoc %>%\n",
    "    dplyr::filter(SNPS %in% geno.bim$snps[geno.bim$chr %in% as.character(1:22)])\n",
    "\n",
    "nrow(studies.assoc)\n",
    "\n",
    "length(unique(studies.assoc$SNPS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "08b11812-a3e3-46ea-87f0-382fd238971b",
   "metadata": {},
   "outputs": [],
   "source": [
    "mqtl.snp.table <- dplyr::bind_rows(\n",
    "    lead.cis.eqtl %>%\n",
    "        dplyr::mutate(source=\"Lead cis-eQTL SNP\") %>%\n",
    "        dplyr::mutate(conditional_number=NA, accession=NA) %>%\n",
    "        dplyr::select(snps, source, egene=gene, conditional_number, accession),\n",
    "    conditional.cis.eqtl %>%\n",
    "        dplyr::mutate(source=\"Conditional cis-eQTL SNP\") %>%\n",
    "        dplyr::mutate(accession=NA) %>%\n",
    "        dplyr::select(snps=SNP, source, egene=Gene, conditional_number=Number, accession),\n",
    "    studies.assoc %>%\n",
    "        dplyr::mutate(source=\"EBI GWAS Catalog\") %>%\n",
    "        dplyr::mutate(egene=NA, conditional_number=NA) %>%\n",
    "        dplyr::select(snps=SNPS, source, egene, conditional_number, accession=37)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "c8a7961b-fc8c-43c7-a058-ba0cdf59eb5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(mqtl.snp.table, \"~/gains_team282/nikhil/expression/eigengene_sva/mqtl_snp_table.csv\", quote=F, row.names=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "ade8fd03-3607-4922-83b4-9bbbf6e41543",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "70968"
      ],
      "text/latex": [
       "70968"
      ],
      "text/markdown": [
       "70968"
      ],
      "text/plain": [
       "[1] 70968"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(mqtl.snp.table$snps))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01268e44-cc71-4f16-a538-0ad41373e8d8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
