{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "40ded556-debe-4c09-a412-cb0c8c47f474",
   "metadata": {},
   "source": [
    "# Eigengene Single-Variant Association Preparation\n",
    "\n",
    "**Created**: 25 November 2021"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "133acb3a-8f08-46dc-bab6-7c8d4afe4803",
   "metadata": {},
   "source": [
    "## Environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5aa6f310-53bc-4e4a-9f26-62e0716b2c2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!requireNamespace(\"BiocManager\", quietly = TRUE)) {\n",
    "    install.packages(\"BiocManager\")\n",
    "}\n",
    "\n",
    "if (!requireNamespace(\"ComplexHeatmap\", quietly=TRUE)) {\n",
    "    BiocManager::install(\"ComplexHeatmap\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b57f682b-54fb-470e-b766-b9f21ae2d9b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.6     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.7     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.6.2\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "  genomic data. Bioinformatics 2016.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(RColorBrewer)\n",
    "library(ComplexHeatmap)\n",
    "library(data.table)\n",
    "\n",
    "setwd(\"~/eQTL_pQTL_Characterization/\")\n",
    "\n",
    "source(\"04_Expression/scripts/utils/ggplot_theme.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53e08f67-624c-4222-b51c-026831dd085b",
   "metadata": {},
   "source": [
    "## Load Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e6264bda-1395-47a9-bf6d-5b5228984a14",
   "metadata": {},
   "outputs": [],
   "source": [
    "demographics <- readxl::read_xls(\"/nfs/team282/data/gains_team282/ClinicalData/DEMO_12jun2019.xls\", sheet=1) %>%\n",
    "    dplyr::mutate(sex=as.numeric(sex)) %>%\n",
    "    dplyr::mutate(diagnosis=as.numeric(diagnosis)) %>%\n",
    "    dplyr::mutate(SubjectBarCode=gsub(\"^GA\", \"\", SubjectBarCode))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "35ab229e-25c4-404d-8390-01f32b34e94f",
   "metadata": {},
   "outputs": [],
   "source": [
    "srs.info <- read.table(\"/nfs/team282/data/gains_team282/full-gains-SRS-predictions_mNN-RF.tsv\", header=T) %>%\n",
    "    dplyr::mutate(Sample_id=gsub(\"^GA\", \"\", Sample_id)) %>%\n",
    "    dplyr::filter(Assay==\"RNA-seq\")\n",
    "\n",
    "rownames(srs.info) <- srs.info$Sample_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0b1b5e57-4df2-42c1-b77d-10f84cff585d",
   "metadata": {},
   "outputs": [],
   "source": [
    "covs <- read.table(\"~/gains_team282/eqtl/data/covs_and_peer_factors.txt\") %>%\n",
    "    dplyr::mutate(Sample.ID=gsub(\"^GA\", \"\", rownames(.))) %>%\n",
    "    dplyr::mutate(GAinS.ID=gsub(\"\\\\_.\", \"\", Sample.ID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "778d7314-6a62-4e86-a35a-68daaabfe7d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 43</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Neutrophils</th><th scope=col>Lymphocytes</th><th scope=col>Monocytes</th><th scope=col>PC1</th><th scope=col>PC2</th><th scope=col>PC3</th><th scope=col>PC4</th><th scope=col>PC5</th><th scope=col>PC6</th><th scope=col>PC7</th><th scope=col>⋯</th><th scope=col>PEER_21</th><th scope=col>PEER_22</th><th scope=col>PEER_23</th><th scope=col>PEER_24</th><th scope=col>PEER_25</th><th scope=col>PEER_26</th><th scope=col>PEER_27</th><th scope=col>PEER_28</th><th scope=col>PEER_29</th><th scope=col>PEER_30</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>UK01050129_5</th><td>-1.7381318</td><td> 1.8281826</td><td> 1.2829455</td><td> 0.177279994</td><td> 0.051694602</td><td>0.02486030</td><td>-0.035102502</td><td>-0.00455149</td><td>-0.0603707</td><td> 0.03964880</td><td>⋯</td><td>-0.001172263</td><td>-0.05533835</td><td> 0.049742281</td><td> 0.03725176</td><td> 0.04495762</td><td> 0.12696229</td><td> 0.08175036</td><td> 0.05295999</td><td>0.164950684</td><td>-0.10379386</td></tr>\n",
       "\t<tr><th scope=row>UK01060123_3</th><td>-0.3845370</td><td> 0.3878375</td><td> 0.4548023</td><td>-0.000388133</td><td>-0.000674099</td><td>0.00810357</td><td> 0.000212791</td><td> 0.03396150</td><td> 0.0266317</td><td>-0.01885280</td><td>⋯</td><td>-0.012723563</td><td> 0.02226736</td><td>-0.027520355</td><td>-0.05782387</td><td> 0.03653939</td><td> 0.05959492</td><td>-0.02611259</td><td>-0.01790827</td><td>0.012047987</td><td> 0.05554044</td></tr>\n",
       "\t<tr><th scope=row>UK01070117_3</th><td>-0.7738039</td><td> 0.5846032</td><td> 0.9325467</td><td> 0.016432300</td><td>-0.048464298</td><td>0.00813966</td><td>-0.047699600</td><td>-0.07926830</td><td> 0.0263629</td><td>-0.08224740</td><td>⋯</td><td> 0.040162142</td><td> 0.06840651</td><td> 0.000062529</td><td> 0.05292596</td><td> 0.01705806</td><td> 0.05931079</td><td> 0.08962730</td><td> 0.03772458</td><td>0.008254956</td><td>-0.01570964</td></tr>\n",
       "\t<tr><th scope=row>UK01080111_1</th><td> 0.1305646</td><td> 0.2888638</td><td>-0.6232047</td><td> 0.161789000</td><td> 0.036224801</td><td>0.00467903</td><td>-0.048611499</td><td> 0.00180036</td><td>-0.0352357</td><td> 0.05012510</td><td>⋯</td><td>-0.002928361</td><td>-0.03274319</td><td> 0.088269033</td><td> 0.07396927</td><td>-0.01906336</td><td> 0.02454052</td><td> 0.04243371</td><td> 0.03055887</td><td>0.123383760</td><td>-0.02276623</td></tr>\n",
       "\t<tr><th scope=row>UK01110093_1</th><td> 1.0909779</td><td>-1.2488160</td><td>-0.4429439</td><td>-0.006296720</td><td> 0.002821510</td><td>0.00699239</td><td> 0.004351650</td><td> 0.00268802</td><td> 0.0166922</td><td>-0.00733983</td><td>⋯</td><td>-0.108193189</td><td> 0.01590979</td><td>-0.093242317</td><td>-0.03037383</td><td>-0.07385907</td><td>-0.05968965</td><td> 0.09695430</td><td>-0.03975484</td><td>0.130198911</td><td>-0.05011296</td></tr>\n",
       "\t<tr><th scope=row>UK01110093_5</th><td> 0.2302416</td><td> 0.0000000</td><td>-0.3306479</td><td>-0.006296720</td><td> 0.002821510</td><td>0.00699239</td><td> 0.004351650</td><td> 0.00268802</td><td> 0.0166922</td><td>-0.00733983</td><td>⋯</td><td>-0.018054215</td><td> 0.02094322</td><td>-0.008623131</td><td> 0.09560122</td><td>-0.07802416</td><td> 0.02365689</td><td> 0.08040638</td><td>-0.02349218</td><td>0.189585552</td><td>-0.07635343</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 43\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Neutrophils & Lymphocytes & Monocytes & PC1 & PC2 & PC3 & PC4 & PC5 & PC6 & PC7 & ⋯ & PEER\\_21 & PEER\\_22 & PEER\\_23 & PEER\\_24 & PEER\\_25 & PEER\\_26 & PEER\\_27 & PEER\\_28 & PEER\\_29 & PEER\\_30\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tUK01050129\\_5 & -1.7381318 &  1.8281826 &  1.2829455 &  0.177279994 &  0.051694602 & 0.02486030 & -0.035102502 & -0.00455149 & -0.0603707 &  0.03964880 & ⋯ & -0.001172263 & -0.05533835 &  0.049742281 &  0.03725176 &  0.04495762 &  0.12696229 &  0.08175036 &  0.05295999 & 0.164950684 & -0.10379386\\\\\n",
       "\tUK01060123\\_3 & -0.3845370 &  0.3878375 &  0.4548023 & -0.000388133 & -0.000674099 & 0.00810357 &  0.000212791 &  0.03396150 &  0.0266317 & -0.01885280 & ⋯ & -0.012723563 &  0.02226736 & -0.027520355 & -0.05782387 &  0.03653939 &  0.05959492 & -0.02611259 & -0.01790827 & 0.012047987 &  0.05554044\\\\\n",
       "\tUK01070117\\_3 & -0.7738039 &  0.5846032 &  0.9325467 &  0.016432300 & -0.048464298 & 0.00813966 & -0.047699600 & -0.07926830 &  0.0263629 & -0.08224740 & ⋯ &  0.040162142 &  0.06840651 &  0.000062529 &  0.05292596 &  0.01705806 &  0.05931079 &  0.08962730 &  0.03772458 & 0.008254956 & -0.01570964\\\\\n",
       "\tUK01080111\\_1 &  0.1305646 &  0.2888638 & -0.6232047 &  0.161789000 &  0.036224801 & 0.00467903 & -0.048611499 &  0.00180036 & -0.0352357 &  0.05012510 & ⋯ & -0.002928361 & -0.03274319 &  0.088269033 &  0.07396927 & -0.01906336 &  0.02454052 &  0.04243371 &  0.03055887 & 0.123383760 & -0.02276623\\\\\n",
       "\tUK01110093\\_1 &  1.0909779 & -1.2488160 & -0.4429439 & -0.006296720 &  0.002821510 & 0.00699239 &  0.004351650 &  0.00268802 &  0.0166922 & -0.00733983 & ⋯ & -0.108193189 &  0.01590979 & -0.093242317 & -0.03037383 & -0.07385907 & -0.05968965 &  0.09695430 & -0.03975484 & 0.130198911 & -0.05011296\\\\\n",
       "\tUK01110093\\_5 &  0.2302416 &  0.0000000 & -0.3306479 & -0.006296720 &  0.002821510 & 0.00699239 &  0.004351650 &  0.00268802 &  0.0166922 & -0.00733983 & ⋯ & -0.018054215 &  0.02094322 & -0.008623131 &  0.09560122 & -0.07802416 &  0.02365689 &  0.08040638 & -0.02349218 & 0.189585552 & -0.07635343\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 43\n",
       "\n",
       "| <!--/--> | Neutrophils &lt;dbl&gt; | Lymphocytes &lt;dbl&gt; | Monocytes &lt;dbl&gt; | PC1 &lt;dbl&gt; | PC2 &lt;dbl&gt; | PC3 &lt;dbl&gt; | PC4 &lt;dbl&gt; | PC5 &lt;dbl&gt; | PC6 &lt;dbl&gt; | PC7 &lt;dbl&gt; | ⋯ ⋯ | PEER_21 &lt;dbl&gt; | PEER_22 &lt;dbl&gt; | PEER_23 &lt;dbl&gt; | PEER_24 &lt;dbl&gt; | PEER_25 &lt;dbl&gt; | PEER_26 &lt;dbl&gt; | PEER_27 &lt;dbl&gt; | PEER_28 &lt;dbl&gt; | PEER_29 &lt;dbl&gt; | PEER_30 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| UK01050129_5 | -1.7381318 |  1.8281826 |  1.2829455 |  0.177279994 |  0.051694602 | 0.02486030 | -0.035102502 | -0.00455149 | -0.0603707 |  0.03964880 | ⋯ | -0.001172263 | -0.05533835 |  0.049742281 |  0.03725176 |  0.04495762 |  0.12696229 |  0.08175036 |  0.05295999 | 0.164950684 | -0.10379386 |\n",
       "| UK01060123_3 | -0.3845370 |  0.3878375 |  0.4548023 | -0.000388133 | -0.000674099 | 0.00810357 |  0.000212791 |  0.03396150 |  0.0266317 | -0.01885280 | ⋯ | -0.012723563 |  0.02226736 | -0.027520355 | -0.05782387 |  0.03653939 |  0.05959492 | -0.02611259 | -0.01790827 | 0.012047987 |  0.05554044 |\n",
       "| UK01070117_3 | -0.7738039 |  0.5846032 |  0.9325467 |  0.016432300 | -0.048464298 | 0.00813966 | -0.047699600 | -0.07926830 |  0.0263629 | -0.08224740 | ⋯ |  0.040162142 |  0.06840651 |  0.000062529 |  0.05292596 |  0.01705806 |  0.05931079 |  0.08962730 |  0.03772458 | 0.008254956 | -0.01570964 |\n",
       "| UK01080111_1 |  0.1305646 |  0.2888638 | -0.6232047 |  0.161789000 |  0.036224801 | 0.00467903 | -0.048611499 |  0.00180036 | -0.0352357 |  0.05012510 | ⋯ | -0.002928361 | -0.03274319 |  0.088269033 |  0.07396927 | -0.01906336 |  0.02454052 |  0.04243371 |  0.03055887 | 0.123383760 | -0.02276623 |\n",
       "| UK01110093_1 |  1.0909779 | -1.2488160 | -0.4429439 | -0.006296720 |  0.002821510 | 0.00699239 |  0.004351650 |  0.00268802 |  0.0166922 | -0.00733983 | ⋯ | -0.108193189 |  0.01590979 | -0.093242317 | -0.03037383 | -0.07385907 | -0.05968965 |  0.09695430 | -0.03975484 | 0.130198911 | -0.05011296 |\n",
       "| UK01110093_5 |  0.2302416 |  0.0000000 | -0.3306479 | -0.006296720 |  0.002821510 | 0.00699239 |  0.004351650 |  0.00268802 |  0.0166922 | -0.00733983 | ⋯ | -0.018054215 |  0.02094322 | -0.008623131 |  0.09560122 | -0.07802416 |  0.02365689 |  0.08040638 | -0.02349218 | 0.189585552 | -0.07635343 |\n",
       "\n"
      ],
      "text/plain": [
       "             Neutrophils Lymphocytes Monocytes  PC1          PC2         \n",
       "UK01050129_5 -1.7381318   1.8281826   1.2829455  0.177279994  0.051694602\n",
       "UK01060123_3 -0.3845370   0.3878375   0.4548023 -0.000388133 -0.000674099\n",
       "UK01070117_3 -0.7738039   0.5846032   0.9325467  0.016432300 -0.048464298\n",
       "UK01080111_1  0.1305646   0.2888638  -0.6232047  0.161789000  0.036224801\n",
       "UK01110093_1  1.0909779  -1.2488160  -0.4429439 -0.006296720  0.002821510\n",
       "UK01110093_5  0.2302416   0.0000000  -0.3306479 -0.006296720  0.002821510\n",
       "             PC3        PC4          PC5         PC6        PC7         ⋯\n",
       "UK01050129_5 0.02486030 -0.035102502 -0.00455149 -0.0603707  0.03964880 ⋯\n",
       "UK01060123_3 0.00810357  0.000212791  0.03396150  0.0266317 -0.01885280 ⋯\n",
       "UK01070117_3 0.00813966 -0.047699600 -0.07926830  0.0263629 -0.08224740 ⋯\n",
       "UK01080111_1 0.00467903 -0.048611499  0.00180036 -0.0352357  0.05012510 ⋯\n",
       "UK01110093_1 0.00699239  0.004351650  0.00268802  0.0166922 -0.00733983 ⋯\n",
       "UK01110093_5 0.00699239  0.004351650  0.00268802  0.0166922 -0.00733983 ⋯\n",
       "             PEER_21      PEER_22     PEER_23      PEER_24     PEER_25    \n",
       "UK01050129_5 -0.001172263 -0.05533835  0.049742281  0.03725176  0.04495762\n",
       "UK01060123_3 -0.012723563  0.02226736 -0.027520355 -0.05782387  0.03653939\n",
       "UK01070117_3  0.040162142  0.06840651  0.000062529  0.05292596  0.01705806\n",
       "UK01080111_1 -0.002928361 -0.03274319  0.088269033  0.07396927 -0.01906336\n",
       "UK01110093_1 -0.108193189  0.01590979 -0.093242317 -0.03037383 -0.07385907\n",
       "UK01110093_5 -0.018054215  0.02094322 -0.008623131  0.09560122 -0.07802416\n",
       "             PEER_26     PEER_27     PEER_28     PEER_29     PEER_30    \n",
       "UK01050129_5  0.12696229  0.08175036  0.05295999 0.164950684 -0.10379386\n",
       "UK01060123_3  0.05959492 -0.02611259 -0.01790827 0.012047987  0.05554044\n",
       "UK01070117_3  0.05931079  0.08962730  0.03772458 0.008254956 -0.01570964\n",
       "UK01080111_1  0.02454052  0.04243371  0.03055887 0.123383760 -0.02276623\n",
       "UK01110093_1 -0.05968965  0.09695430 -0.03975484 0.130198911 -0.05011296\n",
       "UK01110093_5  0.02365689  0.08040638 -0.02349218 0.189585552 -0.07635343"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "held.out <- c(\"Neutrophils\", \"Lymphocytes\", \"Monocytes\", paste0(\"PC\", 1:7), \"Diagnosis\", \"SRS1\", \"sex\")\n",
    "peer <- paste0(\"PEER_\", 1:30)\n",
    "\n",
    "covs <- covs %>%\n",
    "    merge(., demographics, by.x=\"GAinS.ID\", by.y=\"SubjectBarCode\") %>%\n",
    "    merge(., srs.info, by.x=\"Sample.ID\", by.y=0) %>%\n",
    "    dplyr::select(diagnosis=diagnosis, everything()) %>%\n",
    "    dplyr::select(Sample.ID, any_of(held.out), any_of(peer)) %>%\n",
    "    as.data.frame()\n",
    "                                  \n",
    "rownames(covs) <- covs$Sample.ID\n",
    "covs <- covs %>%\n",
    "    dplyr::select(-Sample.ID)\n",
    "                                  \n",
    "head(covs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "31b17104-f1be-4f75-a7c2-791888eb337f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 530</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ME_1_1</th><th scope=col>ME_1_2</th><th scope=col>ME_1_3</th><th scope=col>ME_1_4</th><th scope=col>ME_1_5</th><th scope=col>ME_2_1</th><th scope=col>ME_2_2</th><th scope=col>ME_2_3</th><th scope=col>ME_2_4</th><th scope=col>ME_2_5</th><th scope=col>⋯</th><th scope=col>ME_105_1</th><th scope=col>ME_105_2</th><th scope=col>ME_105_3</th><th scope=col>ME_105_4</th><th scope=col>ME_105_5</th><th scope=col>ME_106_1</th><th scope=col>ME_106_2</th><th scope=col>ME_106_3</th><th scope=col>ME_106_4</th><th scope=col>ME_106_5</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>UK02270173_3</th><td> 0.008640296</td><td>-0.04165815</td><td> 0.027430556</td><td>-0.020437463</td><td> 0.018138657</td><td> 0.007926439</td><td>-0.03881513</td><td> 0.03058366</td><td>-0.014329184</td><td> 0.009126178</td><td>⋯</td><td> 0.033258706</td><td> 0.010337824</td><td> 0.031572946</td><td> 0.0308363822</td><td>-0.01556043</td><td>-0.003317747</td><td> 0.01850385</td><td> 0.001776021</td><td> 0.034254655</td><td> 0.002439359</td></tr>\n",
       "\t<tr><th scope=row>UK15130120_3</th><td>-0.037073495</td><td>-0.01795641</td><td>-0.007283072</td><td>-0.026494085</td><td> 0.003339612</td><td>-0.033179470</td><td>-0.01991901</td><td>-0.01688432</td><td>-0.014852805</td><td> 0.049310271</td><td>⋯</td><td> 0.051184001</td><td>-0.001611603</td><td> 0.005624913</td><td> 0.0273538940</td><td>-0.03670228</td><td>-0.012327128</td><td> 0.04405671</td><td> 0.013144923</td><td> 0.017987682</td><td>-0.001779326</td></tr>\n",
       "\t<tr><th scope=row>UK58000006_3</th><td> 0.029670524</td><td>-0.03481715</td><td> 0.028491140</td><td>-0.009483861</td><td>-0.005862293</td><td> 0.021490110</td><td>-0.03480085</td><td> 0.03014656</td><td>-0.003961906</td><td>-0.032946862</td><td>⋯</td><td>-0.006622840</td><td>-0.045791646</td><td> 0.033491934</td><td>-0.0420056572</td><td> 0.02749077</td><td>-0.004740631</td><td> 0.04638229</td><td>-0.043309799</td><td> 0.028730908</td><td>-0.024051766</td></tr>\n",
       "\t<tr><th scope=row>UK47010004_3</th><td> 0.011954974</td><td> 0.01091971</td><td> 0.038665347</td><td> 0.003726632</td><td> 0.052019153</td><td> 0.007360515</td><td> 0.01763746</td><td> 0.03490663</td><td>-0.005573984</td><td>-0.016190548</td><td>⋯</td><td>-0.003072117</td><td>-0.010005728</td><td> 0.019263388</td><td> 0.0009626484</td><td> 0.06116524</td><td> 0.012014706</td><td> 0.01637633</td><td>-0.003045472</td><td>-0.018223178</td><td>-0.020637855</td></tr>\n",
       "\t<tr><th scope=row>UK42020088_5</th><td> 0.009818957</td><td> 0.02100233</td><td>-0.019852180</td><td>-0.013052613</td><td>-0.030223051</td><td> 0.008322535</td><td> 0.01712902</td><td>-0.02394729</td><td> 0.003193689</td><td> 0.019716419</td><td>⋯</td><td>-0.032697222</td><td>-0.001257679</td><td>-0.019760950</td><td>-0.0079000979</td><td> 0.02988006</td><td> 0.007915598</td><td>-0.01649680</td><td> 0.013250580</td><td>-0.031445321</td><td>-0.014168970</td></tr>\n",
       "\t<tr><th scope=row>UK47490007_3</th><td>-0.021930377</td><td> 0.03807246</td><td> 0.042329128</td><td>-0.025160071</td><td> 0.012732846</td><td>-0.024756322</td><td> 0.03899339</td><td> 0.03987212</td><td>-0.030754242</td><td> 0.038550679</td><td>⋯</td><td>-0.003607270</td><td> 0.043079589</td><td> 0.025888343</td><td> 0.0413386960</td><td> 0.04360340</td><td> 0.024871961</td><td>-0.03349057</td><td> 0.029665102</td><td>-0.009251542</td><td>-0.052908888</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 530\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ME\\_1\\_1 & ME\\_1\\_2 & ME\\_1\\_3 & ME\\_1\\_4 & ME\\_1\\_5 & ME\\_2\\_1 & ME\\_2\\_2 & ME\\_2\\_3 & ME\\_2\\_4 & ME\\_2\\_5 & ⋯ & ME\\_105\\_1 & ME\\_105\\_2 & ME\\_105\\_3 & ME\\_105\\_4 & ME\\_105\\_5 & ME\\_106\\_1 & ME\\_106\\_2 & ME\\_106\\_3 & ME\\_106\\_4 & ME\\_106\\_5\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tUK02270173\\_3 &  0.008640296 & -0.04165815 &  0.027430556 & -0.020437463 &  0.018138657 &  0.007926439 & -0.03881513 &  0.03058366 & -0.014329184 &  0.009126178 & ⋯ &  0.033258706 &  0.010337824 &  0.031572946 &  0.0308363822 & -0.01556043 & -0.003317747 &  0.01850385 &  0.001776021 &  0.034254655 &  0.002439359\\\\\n",
       "\tUK15130120\\_3 & -0.037073495 & -0.01795641 & -0.007283072 & -0.026494085 &  0.003339612 & -0.033179470 & -0.01991901 & -0.01688432 & -0.014852805 &  0.049310271 & ⋯ &  0.051184001 & -0.001611603 &  0.005624913 &  0.0273538940 & -0.03670228 & -0.012327128 &  0.04405671 &  0.013144923 &  0.017987682 & -0.001779326\\\\\n",
       "\tUK58000006\\_3 &  0.029670524 & -0.03481715 &  0.028491140 & -0.009483861 & -0.005862293 &  0.021490110 & -0.03480085 &  0.03014656 & -0.003961906 & -0.032946862 & ⋯ & -0.006622840 & -0.045791646 &  0.033491934 & -0.0420056572 &  0.02749077 & -0.004740631 &  0.04638229 & -0.043309799 &  0.028730908 & -0.024051766\\\\\n",
       "\tUK47010004\\_3 &  0.011954974 &  0.01091971 &  0.038665347 &  0.003726632 &  0.052019153 &  0.007360515 &  0.01763746 &  0.03490663 & -0.005573984 & -0.016190548 & ⋯ & -0.003072117 & -0.010005728 &  0.019263388 &  0.0009626484 &  0.06116524 &  0.012014706 &  0.01637633 & -0.003045472 & -0.018223178 & -0.020637855\\\\\n",
       "\tUK42020088\\_5 &  0.009818957 &  0.02100233 & -0.019852180 & -0.013052613 & -0.030223051 &  0.008322535 &  0.01712902 & -0.02394729 &  0.003193689 &  0.019716419 & ⋯ & -0.032697222 & -0.001257679 & -0.019760950 & -0.0079000979 &  0.02988006 &  0.007915598 & -0.01649680 &  0.013250580 & -0.031445321 & -0.014168970\\\\\n",
       "\tUK47490007\\_3 & -0.021930377 &  0.03807246 &  0.042329128 & -0.025160071 &  0.012732846 & -0.024756322 &  0.03899339 &  0.03987212 & -0.030754242 &  0.038550679 & ⋯ & -0.003607270 &  0.043079589 &  0.025888343 &  0.0413386960 &  0.04360340 &  0.024871961 & -0.03349057 &  0.029665102 & -0.009251542 & -0.052908888\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 530\n",
       "\n",
       "| <!--/--> | ME_1_1 &lt;dbl&gt; | ME_1_2 &lt;dbl&gt; | ME_1_3 &lt;dbl&gt; | ME_1_4 &lt;dbl&gt; | ME_1_5 &lt;dbl&gt; | ME_2_1 &lt;dbl&gt; | ME_2_2 &lt;dbl&gt; | ME_2_3 &lt;dbl&gt; | ME_2_4 &lt;dbl&gt; | ME_2_5 &lt;dbl&gt; | ⋯ ⋯ | ME_105_1 &lt;dbl&gt; | ME_105_2 &lt;dbl&gt; | ME_105_3 &lt;dbl&gt; | ME_105_4 &lt;dbl&gt; | ME_105_5 &lt;dbl&gt; | ME_106_1 &lt;dbl&gt; | ME_106_2 &lt;dbl&gt; | ME_106_3 &lt;dbl&gt; | ME_106_4 &lt;dbl&gt; | ME_106_5 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| UK02270173_3 |  0.008640296 | -0.04165815 |  0.027430556 | -0.020437463 |  0.018138657 |  0.007926439 | -0.03881513 |  0.03058366 | -0.014329184 |  0.009126178 | ⋯ |  0.033258706 |  0.010337824 |  0.031572946 |  0.0308363822 | -0.01556043 | -0.003317747 |  0.01850385 |  0.001776021 |  0.034254655 |  0.002439359 |\n",
       "| UK15130120_3 | -0.037073495 | -0.01795641 | -0.007283072 | -0.026494085 |  0.003339612 | -0.033179470 | -0.01991901 | -0.01688432 | -0.014852805 |  0.049310271 | ⋯ |  0.051184001 | -0.001611603 |  0.005624913 |  0.0273538940 | -0.03670228 | -0.012327128 |  0.04405671 |  0.013144923 |  0.017987682 | -0.001779326 |\n",
       "| UK58000006_3 |  0.029670524 | -0.03481715 |  0.028491140 | -0.009483861 | -0.005862293 |  0.021490110 | -0.03480085 |  0.03014656 | -0.003961906 | -0.032946862 | ⋯ | -0.006622840 | -0.045791646 |  0.033491934 | -0.0420056572 |  0.02749077 | -0.004740631 |  0.04638229 | -0.043309799 |  0.028730908 | -0.024051766 |\n",
       "| UK47010004_3 |  0.011954974 |  0.01091971 |  0.038665347 |  0.003726632 |  0.052019153 |  0.007360515 |  0.01763746 |  0.03490663 | -0.005573984 | -0.016190548 | ⋯ | -0.003072117 | -0.010005728 |  0.019263388 |  0.0009626484 |  0.06116524 |  0.012014706 |  0.01637633 | -0.003045472 | -0.018223178 | -0.020637855 |\n",
       "| UK42020088_5 |  0.009818957 |  0.02100233 | -0.019852180 | -0.013052613 | -0.030223051 |  0.008322535 |  0.01712902 | -0.02394729 |  0.003193689 |  0.019716419 | ⋯ | -0.032697222 | -0.001257679 | -0.019760950 | -0.0079000979 |  0.02988006 |  0.007915598 | -0.01649680 |  0.013250580 | -0.031445321 | -0.014168970 |\n",
       "| UK47490007_3 | -0.021930377 |  0.03807246 |  0.042329128 | -0.025160071 |  0.012732846 | -0.024756322 |  0.03899339 |  0.03987212 | -0.030754242 |  0.038550679 | ⋯ | -0.003607270 |  0.043079589 |  0.025888343 |  0.0413386960 |  0.04360340 |  0.024871961 | -0.03349057 |  0.029665102 | -0.009251542 | -0.052908888 |\n",
       "\n"
      ],
      "text/plain": [
       "             ME_1_1       ME_1_2      ME_1_3       ME_1_4       ME_1_5      \n",
       "UK02270173_3  0.008640296 -0.04165815  0.027430556 -0.020437463  0.018138657\n",
       "UK15130120_3 -0.037073495 -0.01795641 -0.007283072 -0.026494085  0.003339612\n",
       "UK58000006_3  0.029670524 -0.03481715  0.028491140 -0.009483861 -0.005862293\n",
       "UK47010004_3  0.011954974  0.01091971  0.038665347  0.003726632  0.052019153\n",
       "UK42020088_5  0.009818957  0.02100233 -0.019852180 -0.013052613 -0.030223051\n",
       "UK47490007_3 -0.021930377  0.03807246  0.042329128 -0.025160071  0.012732846\n",
       "             ME_2_1       ME_2_2      ME_2_3      ME_2_4       ME_2_5       ⋯\n",
       "UK02270173_3  0.007926439 -0.03881513  0.03058366 -0.014329184  0.009126178 ⋯\n",
       "UK15130120_3 -0.033179470 -0.01991901 -0.01688432 -0.014852805  0.049310271 ⋯\n",
       "UK58000006_3  0.021490110 -0.03480085  0.03014656 -0.003961906 -0.032946862 ⋯\n",
       "UK47010004_3  0.007360515  0.01763746  0.03490663 -0.005573984 -0.016190548 ⋯\n",
       "UK42020088_5  0.008322535  0.01712902 -0.02394729  0.003193689  0.019716419 ⋯\n",
       "UK47490007_3 -0.024756322  0.03899339  0.03987212 -0.030754242  0.038550679 ⋯\n",
       "             ME_105_1     ME_105_2     ME_105_3     ME_105_4      ME_105_5   \n",
       "UK02270173_3  0.033258706  0.010337824  0.031572946  0.0308363822 -0.01556043\n",
       "UK15130120_3  0.051184001 -0.001611603  0.005624913  0.0273538940 -0.03670228\n",
       "UK58000006_3 -0.006622840 -0.045791646  0.033491934 -0.0420056572  0.02749077\n",
       "UK47010004_3 -0.003072117 -0.010005728  0.019263388  0.0009626484  0.06116524\n",
       "UK42020088_5 -0.032697222 -0.001257679 -0.019760950 -0.0079000979  0.02988006\n",
       "UK47490007_3 -0.003607270  0.043079589  0.025888343  0.0413386960  0.04360340\n",
       "             ME_106_1     ME_106_2    ME_106_3     ME_106_4     ME_106_5    \n",
       "UK02270173_3 -0.003317747  0.01850385  0.001776021  0.034254655  0.002439359\n",
       "UK15130120_3 -0.012327128  0.04405671  0.013144923  0.017987682 -0.001779326\n",
       "UK58000006_3 -0.004740631  0.04638229 -0.043309799  0.028730908 -0.024051766\n",
       "UK47010004_3  0.012014706  0.01637633 -0.003045472 -0.018223178 -0.020637855\n",
       "UK42020088_5  0.007915598 -0.01649680  0.013250580 -0.031445321 -0.014168970\n",
       "UK47490007_3  0.024871961 -0.03349057  0.029665102 -0.009251542 -0.052908888"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "eigengenes <- read.csv(\"~/gains_team282/nikhil/expression/gene_expression/eigengenes.multiple.csv\", row.names=1)\n",
    "\n",
    "head(eigengenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8a7417ee-130b-40aa-a038-c5e5416e5e92",
   "metadata": {},
   "outputs": [],
   "source": [
    "modules <- read.csv(\"~/gains_team282/nikhil/expression/gene_expression/modules.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bb7738d0-5663-422e-a78d-c79d58a1f014",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>Module</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000001167</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000002330</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000002822</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000005175</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000005194</td><td>Module_1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000005893</td><td>Module_1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Gene & Module\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000001167 & Module\\_1\\\\\n",
       "\t2 & ENSG00000002330 & Module\\_1\\\\\n",
       "\t3 & ENSG00000002822 & Module\\_1\\\\\n",
       "\t4 & ENSG00000005175 & Module\\_1\\\\\n",
       "\t5 & ENSG00000005194 & Module\\_1\\\\\n",
       "\t6 & ENSG00000005893 & Module\\_1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | Module &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | ENSG00000001167 | Module_1 |\n",
       "| 2 | ENSG00000002330 | Module_1 |\n",
       "| 3 | ENSG00000002822 | Module_1 |\n",
       "| 4 | ENSG00000005175 | Module_1 |\n",
       "| 5 | ENSG00000005194 | Module_1 |\n",
       "| 6 | ENSG00000005893 | Module_1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene            Module  \n",
       "1 ENSG00000001167 Module_1\n",
       "2 ENSG00000002330 Module_1\n",
       "3 ENSG00000002822 Module_1\n",
       "4 ENSG00000005175 Module_1\n",
       "5 ENSG00000005194 Module_1\n",
       "6 ENSG00000005893 Module_1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(modules)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1922a085-a8f3-4bd1-b57b-9126d6f19dd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene.exp <- read.table(\"/lustre/scratch119/humgen/projects/gains_team282/eqtl/data/logcpm_864_20412_hla.txt\")\n",
    "\n",
    "colnames(gene.exp) <- gsub(\"^GA\", \"\", colnames(gene.exp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "50739ed4-280a-427d-8b7d-55e9ce0cb2aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 864</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>UK02270173_3</th><th scope=col>UK15130120_3</th><th scope=col>UK58000006_3</th><th scope=col>UK47010004_3</th><th scope=col>UK42020088_5</th><th scope=col>UK47490007_3</th><th scope=col>UK02770164_3</th><th scope=col>UK02770164_5</th><th scope=col>UK02630151_3</th><th scope=col>UK42150107_1</th><th scope=col>⋯</th><th scope=col>UK59070043_3</th><th scope=col>UK59070043_5</th><th scope=col>UK02510223_3</th><th scope=col>UK02XX0336_5</th><th scope=col>UK29090086_3</th><th scope=col>UK02XX0335_1</th><th scope=col>UK02XX0334_3</th><th scope=col>UK01210130_3</th><th scope=col>UK01210130_5</th><th scope=col>UK01380125_1</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSG00000230521</th><td>0.6644339</td><td>0.58271604</td><td>0.4046756</td><td>0.20905157</td><td>0.2211571</td><td>0.4288519</td><td>0.5133516</td><td>0.57768877</td><td>0.6990766</td><td>0.3812037</td><td>⋯</td><td>0.3823184</td><td>0.5423684</td><td>0.86762630</td><td>0.73833461</td><td>0.6461583</td><td>0.1227063</td><td>0.9621386</td><td>0.65838459</td><td>1.03561474</td><td>0.4457301</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000225864</th><td>0.9817872</td><td>0.43879936</td><td>0.6022344</td><td>0.92340489</td><td>0.2211571</td><td>1.1450686</td><td>0.1921362</td><td>0.57768877</td><td>0.1974851</td><td>1.2290495</td><td>⋯</td><td>0.6842527</td><td>0.8539227</td><td>0.91886777</td><td>0.12581620</td><td>0.5511865</td><td>0.4384007</td><td>0.5465139</td><td>0.40803066</td><td>0.46899151</td><td>0.3544115</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000227766</th><td>1.2117156</td><td>0.30682669</td><td>0.6625010</td><td>0.79963286</td><td>0.6445686</td><td>0.3130290</td><td>0.1183231</td><td>0.16750450</td><td>0.1505900</td><td>1.0533199</td><td>⋯</td><td>0.4901384</td><td>0.4340781</td><td>0.37504164</td><td>0.04316943</td><td>0.6151905</td><td>0.3406488</td><td>0.3147798</td><td>0.43511377</td><td>0.17382742</td><td>0.1157514</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000237669</th><td>0.6420763</td><td>0.69256830</td><td>0.9792012</td><td>0.69929584</td><td>0.3765156</td><td>1.0756817</td><td>0.8912050</td><td>0.60129641</td><td>0.8534668</td><td>1.0533199</td><td>⋯</td><td>1.0790156</td><td>0.6102946</td><td>1.03953331</td><td>0.97965846</td><td>0.7061683</td><td>0.7489857</td><td>0.7919934</td><td>0.85168813</td><td>0.69120049</td><td>1.0798771</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000271581</th><td>2.4744895</td><td>1.33153140</td><td>1.8310464</td><td>2.47531701</td><td>1.8812042</td><td>2.0531366</td><td>0.5972544</td><td>1.00686213</td><td>1.2811361</td><td>2.4244938</td><td>⋯</td><td>2.4728831</td><td>1.5338898</td><td>1.59080138</td><td>0.62923925</td><td>1.2804253</td><td>1.4727511</td><td>1.2187650</td><td>1.46471386</td><td>1.44139443</td><td>1.1896250</td></tr>\n",
       "\t<tr><th scope=row>ENSG00000285647</th><td>0.0000000</td><td>0.03379072</td><td>0.0000000</td><td>0.05517067</td><td>0.6748270</td><td>0.0000000</td><td>0.0000000</td><td>0.03509293</td><td>0.6990766</td><td>1.6556266</td><td>⋯</td><td>3.4117627</td><td>2.4012963</td><td>0.04682116</td><td>0.00000000</td><td>0.0000000</td><td>0.4695662</td><td>0.3762952</td><td>0.07078527</td><td>0.07206309</td><td>0.4159282</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 864\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & UK02270173\\_3 & UK15130120\\_3 & UK58000006\\_3 & UK47010004\\_3 & UK42020088\\_5 & UK47490007\\_3 & UK02770164\\_3 & UK02770164\\_5 & UK02630151\\_3 & UK42150107\\_1 & ⋯ & UK59070043\\_3 & UK59070043\\_5 & UK02510223\\_3 & UK02XX0336\\_5 & UK29090086\\_3 & UK02XX0335\\_1 & UK02XX0334\\_3 & UK01210130\\_3 & UK01210130\\_5 & UK01380125\\_1\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tENSG00000230521 & 0.6644339 & 0.58271604 & 0.4046756 & 0.20905157 & 0.2211571 & 0.4288519 & 0.5133516 & 0.57768877 & 0.6990766 & 0.3812037 & ⋯ & 0.3823184 & 0.5423684 & 0.86762630 & 0.73833461 & 0.6461583 & 0.1227063 & 0.9621386 & 0.65838459 & 1.03561474 & 0.4457301\\\\\n",
       "\tENSG00000225864 & 0.9817872 & 0.43879936 & 0.6022344 & 0.92340489 & 0.2211571 & 1.1450686 & 0.1921362 & 0.57768877 & 0.1974851 & 1.2290495 & ⋯ & 0.6842527 & 0.8539227 & 0.91886777 & 0.12581620 & 0.5511865 & 0.4384007 & 0.5465139 & 0.40803066 & 0.46899151 & 0.3544115\\\\\n",
       "\tENSG00000227766 & 1.2117156 & 0.30682669 & 0.6625010 & 0.79963286 & 0.6445686 & 0.3130290 & 0.1183231 & 0.16750450 & 0.1505900 & 1.0533199 & ⋯ & 0.4901384 & 0.4340781 & 0.37504164 & 0.04316943 & 0.6151905 & 0.3406488 & 0.3147798 & 0.43511377 & 0.17382742 & 0.1157514\\\\\n",
       "\tENSG00000237669 & 0.6420763 & 0.69256830 & 0.9792012 & 0.69929584 & 0.3765156 & 1.0756817 & 0.8912050 & 0.60129641 & 0.8534668 & 1.0533199 & ⋯ & 1.0790156 & 0.6102946 & 1.03953331 & 0.97965846 & 0.7061683 & 0.7489857 & 0.7919934 & 0.85168813 & 0.69120049 & 1.0798771\\\\\n",
       "\tENSG00000271581 & 2.4744895 & 1.33153140 & 1.8310464 & 2.47531701 & 1.8812042 & 2.0531366 & 0.5972544 & 1.00686213 & 1.2811361 & 2.4244938 & ⋯ & 2.4728831 & 1.5338898 & 1.59080138 & 0.62923925 & 1.2804253 & 1.4727511 & 1.2187650 & 1.46471386 & 1.44139443 & 1.1896250\\\\\n",
       "\tENSG00000285647 & 0.0000000 & 0.03379072 & 0.0000000 & 0.05517067 & 0.6748270 & 0.0000000 & 0.0000000 & 0.03509293 & 0.6990766 & 1.6556266 & ⋯ & 3.4117627 & 2.4012963 & 0.04682116 & 0.00000000 & 0.0000000 & 0.4695662 & 0.3762952 & 0.07078527 & 0.07206309 & 0.4159282\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 864\n",
       "\n",
       "| <!--/--> | UK02270173_3 &lt;dbl&gt; | UK15130120_3 &lt;dbl&gt; | UK58000006_3 &lt;dbl&gt; | UK47010004_3 &lt;dbl&gt; | UK42020088_5 &lt;dbl&gt; | UK47490007_3 &lt;dbl&gt; | UK02770164_3 &lt;dbl&gt; | UK02770164_5 &lt;dbl&gt; | UK02630151_3 &lt;dbl&gt; | UK42150107_1 &lt;dbl&gt; | ⋯ ⋯ | UK59070043_3 &lt;dbl&gt; | UK59070043_5 &lt;dbl&gt; | UK02510223_3 &lt;dbl&gt; | UK02XX0336_5 &lt;dbl&gt; | UK29090086_3 &lt;dbl&gt; | UK02XX0335_1 &lt;dbl&gt; | UK02XX0334_3 &lt;dbl&gt; | UK01210130_3 &lt;dbl&gt; | UK01210130_5 &lt;dbl&gt; | UK01380125_1 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000230521 | 0.6644339 | 0.58271604 | 0.4046756 | 0.20905157 | 0.2211571 | 0.4288519 | 0.5133516 | 0.57768877 | 0.6990766 | 0.3812037 | ⋯ | 0.3823184 | 0.5423684 | 0.86762630 | 0.73833461 | 0.6461583 | 0.1227063 | 0.9621386 | 0.65838459 | 1.03561474 | 0.4457301 |\n",
       "| ENSG00000225864 | 0.9817872 | 0.43879936 | 0.6022344 | 0.92340489 | 0.2211571 | 1.1450686 | 0.1921362 | 0.57768877 | 0.1974851 | 1.2290495 | ⋯ | 0.6842527 | 0.8539227 | 0.91886777 | 0.12581620 | 0.5511865 | 0.4384007 | 0.5465139 | 0.40803066 | 0.46899151 | 0.3544115 |\n",
       "| ENSG00000227766 | 1.2117156 | 0.30682669 | 0.6625010 | 0.79963286 | 0.6445686 | 0.3130290 | 0.1183231 | 0.16750450 | 0.1505900 | 1.0533199 | ⋯ | 0.4901384 | 0.4340781 | 0.37504164 | 0.04316943 | 0.6151905 | 0.3406488 | 0.3147798 | 0.43511377 | 0.17382742 | 0.1157514 |\n",
       "| ENSG00000237669 | 0.6420763 | 0.69256830 | 0.9792012 | 0.69929584 | 0.3765156 | 1.0756817 | 0.8912050 | 0.60129641 | 0.8534668 | 1.0533199 | ⋯ | 1.0790156 | 0.6102946 | 1.03953331 | 0.97965846 | 0.7061683 | 0.7489857 | 0.7919934 | 0.85168813 | 0.69120049 | 1.0798771 |\n",
       "| ENSG00000271581 | 2.4744895 | 1.33153140 | 1.8310464 | 2.47531701 | 1.8812042 | 2.0531366 | 0.5972544 | 1.00686213 | 1.2811361 | 2.4244938 | ⋯ | 2.4728831 | 1.5338898 | 1.59080138 | 0.62923925 | 1.2804253 | 1.4727511 | 1.2187650 | 1.46471386 | 1.44139443 | 1.1896250 |\n",
       "| ENSG00000285647 | 0.0000000 | 0.03379072 | 0.0000000 | 0.05517067 | 0.6748270 | 0.0000000 | 0.0000000 | 0.03509293 | 0.6990766 | 1.6556266 | ⋯ | 3.4117627 | 2.4012963 | 0.04682116 | 0.00000000 | 0.0000000 | 0.4695662 | 0.3762952 | 0.07078527 | 0.07206309 | 0.4159282 |\n",
       "\n"
      ],
      "text/plain": [
       "                UK02270173_3 UK15130120_3 UK58000006_3 UK47010004_3\n",
       "ENSG00000230521 0.6644339    0.58271604   0.4046756    0.20905157  \n",
       "ENSG00000225864 0.9817872    0.43879936   0.6022344    0.92340489  \n",
       "ENSG00000227766 1.2117156    0.30682669   0.6625010    0.79963286  \n",
       "ENSG00000237669 0.6420763    0.69256830   0.9792012    0.69929584  \n",
       "ENSG00000271581 2.4744895    1.33153140   1.8310464    2.47531701  \n",
       "ENSG00000285647 0.0000000    0.03379072   0.0000000    0.05517067  \n",
       "                UK42020088_5 UK47490007_3 UK02770164_3 UK02770164_5\n",
       "ENSG00000230521 0.2211571    0.4288519    0.5133516    0.57768877  \n",
       "ENSG00000225864 0.2211571    1.1450686    0.1921362    0.57768877  \n",
       "ENSG00000227766 0.6445686    0.3130290    0.1183231    0.16750450  \n",
       "ENSG00000237669 0.3765156    1.0756817    0.8912050    0.60129641  \n",
       "ENSG00000271581 1.8812042    2.0531366    0.5972544    1.00686213  \n",
       "ENSG00000285647 0.6748270    0.0000000    0.0000000    0.03509293  \n",
       "                UK02630151_3 UK42150107_1 ⋯ UK59070043_3 UK59070043_5\n",
       "ENSG00000230521 0.6990766    0.3812037    ⋯ 0.3823184    0.5423684   \n",
       "ENSG00000225864 0.1974851    1.2290495    ⋯ 0.6842527    0.8539227   \n",
       "ENSG00000227766 0.1505900    1.0533199    ⋯ 0.4901384    0.4340781   \n",
       "ENSG00000237669 0.8534668    1.0533199    ⋯ 1.0790156    0.6102946   \n",
       "ENSG00000271581 1.2811361    2.4244938    ⋯ 2.4728831    1.5338898   \n",
       "ENSG00000285647 0.6990766    1.6556266    ⋯ 3.4117627    2.4012963   \n",
       "                UK02510223_3 UK02XX0336_5 UK29090086_3 UK02XX0335_1\n",
       "ENSG00000230521 0.86762630   0.73833461   0.6461583    0.1227063   \n",
       "ENSG00000225864 0.91886777   0.12581620   0.5511865    0.4384007   \n",
       "ENSG00000227766 0.37504164   0.04316943   0.6151905    0.3406488   \n",
       "ENSG00000237669 1.03953331   0.97965846   0.7061683    0.7489857   \n",
       "ENSG00000271581 1.59080138   0.62923925   1.2804253    1.4727511   \n",
       "ENSG00000285647 0.04682116   0.00000000   0.0000000    0.4695662   \n",
       "                UK02XX0334_3 UK01210130_3 UK01210130_5 UK01380125_1\n",
       "ENSG00000230521 0.9621386    0.65838459   1.03561474   0.4457301   \n",
       "ENSG00000225864 0.5465139    0.40803066   0.46899151   0.3544115   \n",
       "ENSG00000227766 0.3147798    0.43511377   0.17382742   0.1157514   \n",
       "ENSG00000237669 0.7919934    0.85168813   0.69120049   1.0798771   \n",
       "ENSG00000271581 1.2187650    1.46471386   1.44139443   1.1896250   \n",
       "ENSG00000285647 0.3762952    0.07078527   0.07206309   0.4159282   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene.exp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "973adfd9-7295-45ec-ad6e-65ec926c7841",
   "metadata": {},
   "outputs": [],
   "source": [
    "lead.cis.eqtl <- read.table(\"/nfs/users/nfs_n/nm18/gains_team282/eqtl/cisresults/eigenMT/ciseqtl_eigenMT_corrected.txt\") %>%\n",
    "    dplyr::filter(Sig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f558c85f-1ae0-4c2e-86d8-1de063fe980e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>snps</th><th scope=col>gene</th><th scope=col>statistic</th><th scope=col>pvalue</th><th scope=col>beta</th><th scope=col>se</th><th scope=col>chr</th><th scope=col>SNPpos</th><th scope=col>TSS</th><th scope=col>BF</th><th scope=col>TESTS</th><th scope=col>BF.FDR</th><th scope=col>Sig</th><th scope=col>threshold</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>22</th><td>rs3131972 </td><td>ENSG00000237491</td><td> 5.643405</td><td>2.400875e-08</td><td> 0.07174216</td><td>0.012712570</td><td>1</td><td>817341</td><td>778747</td><td>7.058573e-06</td><td>294</td><td>1.993414e-05</td><td>TRUE</td><td>9.170798e-05</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>rs3131972 </td><td>ENSG00000230092</td><td> 4.756587</td><td>2.329109e-06</td><td> 0.06239499</td><td>0.013117600</td><td>1</td><td>817341</td><td>817712</td><td>6.917454e-04</td><td>297</td><td>1.593483e-03</td><td>TRUE</td><td>9.078164e-05</td></tr>\n",
       "\t<tr><th scope=row>25</th><td>rs3131972 </td><td>ENSG00000225880</td><td> 6.292961</td><td>5.949542e-10</td><td> 0.10016770</td><td>0.015917420</td><td>1</td><td>817341</td><td>827522</td><td>1.767014e-07</td><td>297</td><td>5.668498e-07</td><td>TRUE</td><td>9.078164e-05</td></tr>\n",
       "\t<tr><th scope=row>28</th><td>rs2272757 </td><td>ENSG00000188976</td><td>-7.026045</td><td>4.904859e-12</td><td>-0.05107767</td><td>0.007269761</td><td>1</td><td>946247</td><td>959309</td><td>1.545031e-09</td><td>315</td><td>5.762103e-09</td><td>TRUE</td><td>8.559411e-05</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>rs13303327</td><td>ENSG00000187961</td><td>-7.175012</td><td>1.845538e-12</td><td>-0.13519920</td><td>0.018843060</td><td>1</td><td>960326</td><td>960584</td><td>5.813445e-10</td><td>315</td><td>2.228468e-09</td><td>TRUE</td><td>8.559411e-05</td></tr>\n",
       "\t<tr><th scope=row>30</th><td>rs13303056</td><td>ENSG00000187583</td><td>-9.081031</td><td>1.922802e-18</td><td>-0.29633770</td><td>0.032632610</td><td>1</td><td>953778</td><td>966482</td><td>6.056826e-16</td><td>315</td><td>3.262796e-15</td><td>TRUE</td><td>8.559411e-05</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & snps & gene & statistic & pvalue & beta & se & chr & SNPpos & TSS & BF & TESTS & BF.FDR & Sig & threshold\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <dbl> & <int> & <dbl> & <lgl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t22 & rs3131972  & ENSG00000237491 &  5.643405 & 2.400875e-08 &  0.07174216 & 0.012712570 & 1 & 817341 & 778747 & 7.058573e-06 & 294 & 1.993414e-05 & TRUE & 9.170798e-05\\\\\n",
       "\t23 & rs3131972  & ENSG00000230092 &  4.756587 & 2.329109e-06 &  0.06239499 & 0.013117600 & 1 & 817341 & 817712 & 6.917454e-04 & 297 & 1.593483e-03 & TRUE & 9.078164e-05\\\\\n",
       "\t25 & rs3131972  & ENSG00000225880 &  6.292961 & 5.949542e-10 &  0.10016770 & 0.015917420 & 1 & 817341 & 827522 & 1.767014e-07 & 297 & 5.668498e-07 & TRUE & 9.078164e-05\\\\\n",
       "\t28 & rs2272757  & ENSG00000188976 & -7.026045 & 4.904859e-12 & -0.05107767 & 0.007269761 & 1 & 946247 & 959309 & 1.545031e-09 & 315 & 5.762103e-09 & TRUE & 8.559411e-05\\\\\n",
       "\t29 & rs13303327 & ENSG00000187961 & -7.175012 & 1.845538e-12 & -0.13519920 & 0.018843060 & 1 & 960326 & 960584 & 5.813445e-10 & 315 & 2.228468e-09 & TRUE & 8.559411e-05\\\\\n",
       "\t30 & rs13303056 & ENSG00000187583 & -9.081031 & 1.922802e-18 & -0.29633770 & 0.032632610 & 1 & 953778 & 966482 & 6.056826e-16 & 315 & 3.262796e-15 & TRUE & 8.559411e-05\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 14\n",
       "\n",
       "| <!--/--> | snps &lt;chr&gt; | gene &lt;chr&gt; | statistic &lt;dbl&gt; | pvalue &lt;dbl&gt; | beta &lt;dbl&gt; | se &lt;dbl&gt; | chr &lt;int&gt; | SNPpos &lt;int&gt; | TSS &lt;int&gt; | BF &lt;dbl&gt; | TESTS &lt;int&gt; | BF.FDR &lt;dbl&gt; | Sig &lt;lgl&gt; | threshold &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 22 | rs3131972  | ENSG00000237491 |  5.643405 | 2.400875e-08 |  0.07174216 | 0.012712570 | 1 | 817341 | 778747 | 7.058573e-06 | 294 | 1.993414e-05 | TRUE | 9.170798e-05 |\n",
       "| 23 | rs3131972  | ENSG00000230092 |  4.756587 | 2.329109e-06 |  0.06239499 | 0.013117600 | 1 | 817341 | 817712 | 6.917454e-04 | 297 | 1.593483e-03 | TRUE | 9.078164e-05 |\n",
       "| 25 | rs3131972  | ENSG00000225880 |  6.292961 | 5.949542e-10 |  0.10016770 | 0.015917420 | 1 | 817341 | 827522 | 1.767014e-07 | 297 | 5.668498e-07 | TRUE | 9.078164e-05 |\n",
       "| 28 | rs2272757  | ENSG00000188976 | -7.026045 | 4.904859e-12 | -0.05107767 | 0.007269761 | 1 | 946247 | 959309 | 1.545031e-09 | 315 | 5.762103e-09 | TRUE | 8.559411e-05 |\n",
       "| 29 | rs13303327 | ENSG00000187961 | -7.175012 | 1.845538e-12 | -0.13519920 | 0.018843060 | 1 | 960326 | 960584 | 5.813445e-10 | 315 | 2.228468e-09 | TRUE | 8.559411e-05 |\n",
       "| 30 | rs13303056 | ENSG00000187583 | -9.081031 | 1.922802e-18 | -0.29633770 | 0.032632610 | 1 | 953778 | 966482 | 6.056826e-16 | 315 | 3.262796e-15 | TRUE | 8.559411e-05 |\n",
       "\n"
      ],
      "text/plain": [
       "   snps       gene            statistic pvalue       beta        se         \n",
       "22 rs3131972  ENSG00000237491  5.643405 2.400875e-08  0.07174216 0.012712570\n",
       "23 rs3131972  ENSG00000230092  4.756587 2.329109e-06  0.06239499 0.013117600\n",
       "25 rs3131972  ENSG00000225880  6.292961 5.949542e-10  0.10016770 0.015917420\n",
       "28 rs2272757  ENSG00000188976 -7.026045 4.904859e-12 -0.05107767 0.007269761\n",
       "29 rs13303327 ENSG00000187961 -7.175012 1.845538e-12 -0.13519920 0.018843060\n",
       "30 rs13303056 ENSG00000187583 -9.081031 1.922802e-18 -0.29633770 0.032632610\n",
       "   chr SNPpos TSS    BF           TESTS BF.FDR       Sig  threshold   \n",
       "22 1   817341 778747 7.058573e-06 294   1.993414e-05 TRUE 9.170798e-05\n",
       "23 1   817341 817712 6.917454e-04 297   1.593483e-03 TRUE 9.078164e-05\n",
       "25 1   817341 827522 1.767014e-07 297   5.668498e-07 TRUE 9.078164e-05\n",
       "28 1   946247 959309 1.545031e-09 315   5.762103e-09 TRUE 8.559411e-05\n",
       "29 1   960326 960584 5.813445e-10 315   2.228468e-09 TRUE 8.559411e-05\n",
       "30 1   953778 966482 6.056826e-16 315   3.262796e-15 TRUE 8.559411e-05"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(lead.cis.eqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6db21f2d-6677-4c50-9764-1746d3bf6fed",
   "metadata": {},
   "outputs": [],
   "source": [
    "conditional.cis.eqtl <- readRDS(\"/nfs/users/nfs_n/nm18/gains_team282/eqtl/cisresults/conditionalanalysis/conditional_eQTL_results_final.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c7d7570f-52a8-43a2-94ec-9bd724d5d165",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SNP</th><th scope=col>Gene</th><th scope=col>eQTL_beta</th><th scope=col>eQTL_SE</th><th scope=col>pvalue</th><th scope=col>Number</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>rs10753794</td><td>ENSG00000000457</td><td>0.0498587312408011 </td><td>0.00708515851579373</td><td>5.19259471106013e-12</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>rs10919255</td><td>ENSG00000000460</td><td>-0.0726264894633498</td><td>0.0136853304567293 </td><td>1.72815915933528e-07</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>rs77006036</td><td>ENSG00000000460</td><td>0.166236949102582  </td><td>0.0334472538699365 </td><td>8.47290238610159e-07</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>rs12406047</td><td>ENSG00000000971</td><td>0.17135374034767   </td><td>0.0379807691167313 </td><td>7.51523462853891e-06</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>rs6696136 </td><td>ENSG00000001460</td><td>0.45884434698694   </td><td>0.0871081626842823 </td><td>1.94887370606405e-07</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>rs6676449 </td><td>ENSG00000001460</td><td>-0.371531778822554 </td><td>0.0880672792106829 </td><td>2.8471651589813e-05 </td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & SNP & Gene & eQTL\\_beta & eQTL\\_SE & pvalue & Number\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & rs10753794 & ENSG00000000457 & 0.0498587312408011  & 0.00708515851579373 & 5.19259471106013e-12 & 1\\\\\n",
       "\t2 & rs10919255 & ENSG00000000460 & -0.0726264894633498 & 0.0136853304567293  & 1.72815915933528e-07 & 1\\\\\n",
       "\t3 & rs77006036 & ENSG00000000460 & 0.166236949102582   & 0.0334472538699365  & 8.47290238610159e-07 & 2\\\\\n",
       "\t4 & rs12406047 & ENSG00000000971 & 0.17135374034767    & 0.0379807691167313  & 7.51523462853891e-06 & 1\\\\\n",
       "\t5 & rs6696136  & ENSG00000001460 & 0.45884434698694    & 0.0871081626842823  & 1.94887370606405e-07 & 1\\\\\n",
       "\t6 & rs6676449  & ENSG00000001460 & -0.371531778822554  & 0.0880672792106829  & 2.8471651589813e-05  & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | SNP &lt;chr&gt; | Gene &lt;chr&gt; | eQTL_beta &lt;chr&gt; | eQTL_SE &lt;chr&gt; | pvalue &lt;chr&gt; | Number &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | rs10753794 | ENSG00000000457 | 0.0498587312408011  | 0.00708515851579373 | 5.19259471106013e-12 | 1 |\n",
       "| 2 | rs10919255 | ENSG00000000460 | -0.0726264894633498 | 0.0136853304567293  | 1.72815915933528e-07 | 1 |\n",
       "| 3 | rs77006036 | ENSG00000000460 | 0.166236949102582   | 0.0334472538699365  | 8.47290238610159e-07 | 2 |\n",
       "| 4 | rs12406047 | ENSG00000000971 | 0.17135374034767    | 0.0379807691167313  | 7.51523462853891e-06 | 1 |\n",
       "| 5 | rs6696136  | ENSG00000001460 | 0.45884434698694    | 0.0871081626842823  | 1.94887370606405e-07 | 1 |\n",
       "| 6 | rs6676449  | ENSG00000001460 | -0.371531778822554  | 0.0880672792106829  | 2.8471651589813e-05  | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP        Gene            eQTL_beta           eQTL_SE            \n",
       "1 rs10753794 ENSG00000000457 0.0498587312408011  0.00708515851579373\n",
       "2 rs10919255 ENSG00000000460 -0.0726264894633498 0.0136853304567293 \n",
       "3 rs77006036 ENSG00000000460 0.166236949102582   0.0334472538699365 \n",
       "4 rs12406047 ENSG00000000971 0.17135374034767    0.0379807691167313 \n",
       "5 rs6696136  ENSG00000001460 0.45884434698694    0.0871081626842823 \n",
       "6 rs6676449  ENSG00000001460 -0.371531778822554  0.0880672792106829 \n",
       "  pvalue               Number\n",
       "1 5.19259471106013e-12 1     \n",
       "2 1.72815915933528e-07 1     \n",
       "3 8.47290238610159e-07 2     \n",
       "4 7.51523462853891e-06 1     \n",
       "5 1.94887370606405e-07 1     \n",
       "6 2.8471651589813e-05  2     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(conditional.cis.eqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3bcec61e-1c77-42fe-b8a5-c2f9c87482de",
   "metadata": {},
   "outputs": [],
   "source": [
    "ebi.studies <- fread(\"04_Expression/data/gwas_catalog_v1.0.2-studies_r2022-02-21.tsv\", header=TRUE, quote=\"\") %>%\n",
    "    as.data.frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "53805170-af73-4d97-870b-23d9c68a25ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 16</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>DATE ADDED TO CATALOG</th><th scope=col>PUBMEDID</th><th scope=col>FIRST AUTHOR</th><th scope=col>DATE</th><th scope=col>JOURNAL</th><th scope=col>LINK</th><th scope=col>STUDY</th><th scope=col>DISEASE/TRAIT</th><th scope=col>INITIAL SAMPLE SIZE</th><th scope=col>REPLICATION SAMPLE SIZE</th><th scope=col>PLATFORM [SNPS PASSING QC]</th><th scope=col>ASSOCIATION COUNT</th><th scope=col>MAPPED_TRAIT</th><th scope=col>MAPPED_TRAIT_URI</th><th scope=col>STUDY ACCESSION</th><th scope=col>GENOTYPING TECHNOLOGY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>2021-08-24</td><td>34124712</td><td>Sherva R</td><td>2021-02-28</td><td>Explor Med</td><td>www.ncbi.nlm.nih.gov/pubmed/34124712</td><td>Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes.</td><td>Cocaine dependence (time to event)</td><td>3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls</td><td>572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls</td><td>Illumina [NR] (imputed)</td><td>5</td><td>cocaine dependence</td><td>http://www.ebi.ac.uk/efo/EFO_0002610</td><td>GCST012225</td><td>Genome-wide genotyping array</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 16\n",
       "\\begin{tabular}{r|llllllllllllllll}\n",
       "  & DATE ADDED TO CATALOG & PUBMEDID & FIRST AUTHOR & DATE & JOURNAL & LINK & STUDY & DISEASE/TRAIT & INITIAL SAMPLE SIZE & REPLICATION SAMPLE SIZE & PLATFORM {[}SNPS PASSING QC{]} & ASSOCIATION COUNT & MAPPED\\_TRAIT & MAPPED\\_TRAIT\\_URI & STUDY ACCESSION & GENOTYPING TECHNOLOGY\\\\\n",
       "  & <date> & <int> & <chr> & <date> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 2021-08-24 & 34124712 & Sherva R & 2021-02-28 & Explor Med & www.ncbi.nlm.nih.gov/pubmed/34124712 & Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes. & Cocaine dependence (time to event) & 3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls & 572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls & Illumina {[}NR{]} (imputed) & 5 & cocaine dependence & http://www.ebi.ac.uk/efo/EFO\\_0002610 & GCST012225 & Genome-wide genotyping array\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 16\n",
       "\n",
       "| <!--/--> | DATE ADDED TO CATALOG &lt;date&gt; | PUBMEDID &lt;int&gt; | FIRST AUTHOR &lt;chr&gt; | DATE &lt;date&gt; | JOURNAL &lt;chr&gt; | LINK &lt;chr&gt; | STUDY &lt;chr&gt; | DISEASE/TRAIT &lt;chr&gt; | INITIAL SAMPLE SIZE &lt;chr&gt; | REPLICATION SAMPLE SIZE &lt;chr&gt; | PLATFORM [SNPS PASSING QC] &lt;chr&gt; | ASSOCIATION COUNT &lt;int&gt; | MAPPED_TRAIT &lt;chr&gt; | MAPPED_TRAIT_URI &lt;chr&gt; | STUDY ACCESSION &lt;chr&gt; | GENOTYPING TECHNOLOGY &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 2021-08-24 | 34124712 | Sherva R | 2021-02-28 | Explor Med | www.ncbi.nlm.nih.gov/pubmed/34124712 | Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes. | Cocaine dependence (time to event) | 3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls | 572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls | Illumina [NR] (imputed) | 5 | cocaine dependence | http://www.ebi.ac.uk/efo/EFO_0002610 | GCST012225 | Genome-wide genotyping array |\n",
       "\n"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG PUBMEDID FIRST AUTHOR DATE       JOURNAL   \n",
       "1 2021-08-24            34124712 Sherva R     2021-02-28 Explor Med\n",
       "  LINK                                \n",
       "1 www.ncbi.nlm.nih.gov/pubmed/34124712\n",
       "  STUDY                                                                                                                                     \n",
       "1 Genome-wide association study of phenotypes measuring progression from first cocaine or opioid use to dependence reveals novel risk genes.\n",
       "  DISEASE/TRAIT                     \n",
       "1 Cocaine dependence (time to event)\n",
       "  INITIAL SAMPLE SIZE                                                                                                       \n",
       "1 3,554 African American cases, 478 African American controls, 2,712 European ancestry cases, 915 European ancestry controls\n",
       "  REPLICATION SAMPLE SIZE                                                                                                 \n",
       "1 572 African American cases, 416 African American controls, 759 European ancestry cases, 1,620 European ancestry controls\n",
       "  PLATFORM [SNPS PASSING QC] ASSOCIATION COUNT MAPPED_TRAIT      \n",
       "1 Illumina [NR] (imputed)    5                 cocaine dependence\n",
       "  MAPPED_TRAIT_URI                     STUDY ACCESSION\n",
       "1 http://www.ebi.ac.uk/efo/EFO_0002610 GCST012225     \n",
       "  GENOTYPING TECHNOLOGY       \n",
       "1 Genome-wide genotyping array"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ebi.studies, n=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e0460a82-daa1-4eaf-877b-b119f710f4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "ebi.assoc <- fread(\"04_Expression/data/gwas_catalog_v1.0.2-associations_e105_r2022-02-21.tsv\", header=TRUE, quote=\"\") %>%\n",
    "    as.data.frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "d5fe892a-b746-4ce0-a639-8f577e9ee36e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 38</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>DATE ADDED TO CATALOG</th><th scope=col>PUBMEDID</th><th scope=col>FIRST AUTHOR</th><th scope=col>DATE</th><th scope=col>JOURNAL</th><th scope=col>LINK</th><th scope=col>STUDY</th><th scope=col>DISEASE/TRAIT</th><th scope=col>INITIAL SAMPLE SIZE</th><th scope=col>REPLICATION SAMPLE SIZE</th><th scope=col>⋯</th><th scope=col>PVALUE_MLOG</th><th scope=col>P-VALUE (TEXT)</th><th scope=col>OR or BETA</th><th scope=col>95% CI (TEXT)</th><th scope=col>PLATFORM [SNPS PASSING QC]</th><th scope=col>CNV</th><th scope=col>MAPPED_TRAIT</th><th scope=col>MAPPED_TRAIT_URI</th><th scope=col>STUDY ACCESSION</th><th scope=col>GENOTYPING TECHNOLOGY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>2021-07-07</td><td>34127860</td><td>Robertson CC</td><td>2021-06-14</td><td>Nat Genet</td><td>www.ncbi.nlm.nih.gov/pubmed/34127860</td><td>Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes.</td><td>Type 1 diabetes</td><td>20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls</td><td>NA</td><td>⋯</td><td>8.522879</td><td></td><td>0.912</td><td></td><td>Illumina [715631] (imputed)</td><td>N</td><td>type 1 diabetes mellitus</td><td>http://purl.obolibrary.org/obo/MONDO_0005147</td><td>GCST90013445</td><td>Targeted genotyping array [ImmunoChip]</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 38\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & DATE ADDED TO CATALOG & PUBMEDID & FIRST AUTHOR & DATE & JOURNAL & LINK & STUDY & DISEASE/TRAIT & INITIAL SAMPLE SIZE & REPLICATION SAMPLE SIZE & ⋯ & PVALUE\\_MLOG & P-VALUE (TEXT) & OR or BETA & 95\\% CI (TEXT) & PLATFORM {[}SNPS PASSING QC{]} & CNV & MAPPED\\_TRAIT & MAPPED\\_TRAIT\\_URI & STUDY ACCESSION & GENOTYPING TECHNOLOGY\\\\\n",
       "  & <date> & <int> & <chr> & <date> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <dbl> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 2021-07-07 & 34127860 & Robertson CC & 2021-06-14 & Nat Genet & www.ncbi.nlm.nih.gov/pubmed/34127860 & Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes. & Type 1 diabetes & 20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls & NA & ⋯ & 8.522879 &  & 0.912 &  & Illumina {[}715631{]} (imputed) & N & type 1 diabetes mellitus & http://purl.obolibrary.org/obo/MONDO\\_0005147 & GCST90013445 & Targeted genotyping array {[}ImmunoChip{]}\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 38\n",
       "\n",
       "| <!--/--> | DATE ADDED TO CATALOG &lt;date&gt; | PUBMEDID &lt;int&gt; | FIRST AUTHOR &lt;chr&gt; | DATE &lt;date&gt; | JOURNAL &lt;chr&gt; | LINK &lt;chr&gt; | STUDY &lt;chr&gt; | DISEASE/TRAIT &lt;chr&gt; | INITIAL SAMPLE SIZE &lt;chr&gt; | REPLICATION SAMPLE SIZE &lt;chr&gt; | ⋯ ⋯ | PVALUE_MLOG &lt;dbl&gt; | P-VALUE (TEXT) &lt;chr&gt; | OR or BETA &lt;dbl&gt; | 95% CI (TEXT) &lt;chr&gt; | PLATFORM [SNPS PASSING QC] &lt;chr&gt; | CNV &lt;chr&gt; | MAPPED_TRAIT &lt;chr&gt; | MAPPED_TRAIT_URI &lt;chr&gt; | STUDY ACCESSION &lt;chr&gt; | GENOTYPING TECHNOLOGY &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 2021-07-07 | 34127860 | Robertson CC | 2021-06-14 | Nat Genet | www.ncbi.nlm.nih.gov/pubmed/34127860 | Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes. | Type 1 diabetes | 20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls | NA | ⋯ | 8.522879 | <!----> | 0.912 | <!----> | Illumina [715631] (imputed) | N | type 1 diabetes mellitus | http://purl.obolibrary.org/obo/MONDO_0005147 | GCST90013445 | Targeted genotyping array [ImmunoChip] |\n",
       "\n"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG PUBMEDID FIRST AUTHOR DATE       JOURNAL  \n",
       "1 2021-07-07            34127860 Robertson CC 2021-06-14 Nat Genet\n",
       "  LINK                                \n",
       "1 www.ncbi.nlm.nih.gov/pubmed/34127860\n",
       "  STUDY                                                                                                                          \n",
       "1 Fine-mapping, trans-ancestral and genomic analyses identify causal variants, cells, genes and drug targets for type 1 diabetes.\n",
       "  DISEASE/TRAIT  \n",
       "1 Type 1 diabetes\n",
       "  INITIAL SAMPLE SIZE                                                                                                                                                                                        \n",
       "1 20,065 European ancestry cases, 33,065 European ancestry controls, 1,045 other-admixed ancestry cases, 1,103 other-admixed ancestry controls, 1,043 African ancestry cases, 3,206 African ancestry controls\n",
       "  REPLICATION SAMPLE SIZE ⋯ PVALUE_MLOG P-VALUE (TEXT) OR or BETA 95% CI (TEXT)\n",
       "1 NA                      ⋯ 8.522879                   0.912                   \n",
       "  PLATFORM [SNPS PASSING QC]  CNV MAPPED_TRAIT            \n",
       "1 Illumina [715631] (imputed) N   type 1 diabetes mellitus\n",
       "  MAPPED_TRAIT_URI                             STUDY ACCESSION\n",
       "1 http://purl.obolibrary.org/obo/MONDO_0005147 GCST90013445   \n",
       "  GENOTYPING TECHNOLOGY                 \n",
       "1 Targeted genotyping array [ImmunoChip]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ebi.assoc, n=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4d128a00-384a-4941-baf2-e4f0f200079c",
   "metadata": {},
   "outputs": [],
   "source": [
    "geno.bim <- fread(\"~/gains_team282/Genotyping/All_genotyping_merged_filtered_b38_refiltered_rsID.bim\") %>%\n",
    "    as.data.frame()\n",
    "colnames(geno.bim) <- c(\"chr\", \"snps\", \"cM\", \"pos\", \"minor\", \"major\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f1c567db-258f-49f6-9fd1-f019e2dbd440",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>snps</th><th scope=col>cM</th><th scope=col>pos</th><th scope=col>minor</th><th scope=col>major</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs3131972   </td><td>0</td><td>817341</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs546843995 </td><td>0</td><td>818053</td><td>0</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs553916047 </td><td>0</td><td>818359</td><td>0</td><td>A</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>1:818740_T_C</td><td>0</td><td>818740</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs145604921 </td><td>0</td><td>819378</td><td>0</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs535256652 </td><td>0</td><td>821053</td><td>0</td><td>T</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & chr & snps & cM & pos & minor & major\\\\\n",
       "  & <chr> & <chr> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs3131972    & 0 & 817341 & A & G\\\\\n",
       "\t2 & 1 & rs546843995  & 0 & 818053 & 0 & G\\\\\n",
       "\t3 & 1 & rs553916047  & 0 & 818359 & 0 & A\\\\\n",
       "\t4 & 1 & 1:818740\\_T\\_C & 0 & 818740 & T & C\\\\\n",
       "\t5 & 1 & rs145604921  & 0 & 819378 & 0 & C\\\\\n",
       "\t6 & 1 & rs535256652  & 0 & 821053 & 0 & T\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | snps &lt;chr&gt; | cM &lt;int&gt; | pos &lt;int&gt; | minor &lt;chr&gt; | major &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs3131972    | 0 | 817341 | A | G |\n",
       "| 2 | 1 | rs546843995  | 0 | 818053 | 0 | G |\n",
       "| 3 | 1 | rs553916047  | 0 | 818359 | 0 | A |\n",
       "| 4 | 1 | 1:818740_T_C | 0 | 818740 | T | C |\n",
       "| 5 | 1 | rs145604921  | 0 | 819378 | 0 | C |\n",
       "| 6 | 1 | rs535256652  | 0 | 821053 | 0 | T |\n",
       "\n"
      ],
      "text/plain": [
       "  chr snps         cM pos    minor major\n",
       "1 1   rs3131972    0  817341 A     G    \n",
       "2 1   rs546843995  0  818053 0     G    \n",
       "3 1   rs553916047  0  818359 0     A    \n",
       "4 1   1:818740_T_C 0  818740 T     C    \n",
       "5 1   rs145604921  0  819378 0     C    \n",
       "6 1   rs535256652  0  821053 0     T    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(geno.bim)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26c34029-3efa-455d-9485-103384dd4786",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Heatmap"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "076c174a-88e5-4cc9-984f-a19230862509",
   "metadata": {},
   "source": [
    "Before performing single-variant association analysis, we want to check if any covariates are strongly associated with the module eigengenes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "6df18220-afd6-43ac-b99f-f15563ff1cca",
   "metadata": {},
   "outputs": [],
   "source": [
    "all.vars <- merge(covs, eigengenes, by.x=0, by.y=0) %>%\n",
    "    dplyr::select(-Row.names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "970676d3-5339-4770-89ea-d97d779932b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colors <- colorRampPalette(brewer.pal(11, \"RdBu\"))(101)\n",
    "\n",
    "cor.mtx <- cor(all.vars)\n",
    "cor.mtx.peer <- cor.mtx[peer, colnames(eigengenes)]\n",
    "cor.mtx.held.out <- cor.mtx[held.out, colnames(eigengenes)]\n",
    "\n",
    "h1 <- Heatmap(cor.mtx.peer, col=colors, heatmap_legend_param=list(\n",
    "    title=\"Correlation\",\n",
    "    at=c(-1, 0, 1)\n",
    "))\n",
    "\n",
    "h2 <- Heatmap(cor.mtx.held.out, col=colors, cluster_rows=F, heatmap_legend_param=list(\n",
    "    title=\"Correlation\",\n",
    "    at=c(-1, 0, 1)\n",
    "))\n",
    "\n",
    "svg(\"04_Expression/results/eigengene_peer_factor_correlation.svg\", width=12, height=8)\n",
    "h1 %v% h2\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfa945dc-6083-47e8-982c-d3c403cf6375",
   "metadata": {},
   "source": [
    "The PEER factors are built using gene expression data. It's not surprising that many of the eigengenes are associated with the PEER factors. WGCNA discarded around 8000 genes. The residual gene expression variation from these genes is likely captured by some of the PEER factors. We model this specifically using 30 gene expression PCs from the discarded genes.\n",
    "\n",
    "The eigengenes are not associated with genotyping PCs (which is good)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8eb0417-04be-46dc-a543-1e58f94529fc",
   "metadata": {},
   "source": [
    "![](../results/eigengene_peer_factor_correlation.svg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32443e95-d375-40f2-a8e2-b2aee70dcd64",
   "metadata": {},
   "source": [
    "Ideally, I will include only sex and genotyping PCs as covariates in this mapping, since everything else (cell counts, SRS group) are important signatures that may be correlated with module eigengenes. I will include the 30 PCs from the unassigned gene expression data to control for technical confounders."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7e66a4f-1e9c-490b-84e5-ad4193d9036d",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Save Covariates"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18223439-bc5d-4e9d-9bed-57fbd2228210",
   "metadata": {},
   "source": [
    "Save list of patients (family ID in the first column and individual ID in the second column) to subset the genotyping data using PLINK. There are 638 unique patients with genotyping information."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "13413b35-1092-4b7a-97ef-7a734fdd0dd5",
   "metadata": {},
   "outputs": [],
   "source": [
    "eigengene.patients <- sapply(strsplit(rownames(eigengenes), \"_\"), function(x) { x[1] })\n",
    "\n",
    "geno.fam <- fread(\"~/gains_team282/Genotyping/All_genotyping_merged_filtered_b38_refiltered_rsID.fam\") %>%\n",
    "    dplyr::select(Family.ID=1, Individual.ID=2) %>%\n",
    "    dplyr::mutate(GAinS.ID=gsub(\"^GA\", \"\", Individual.ID)) %>%\n",
    "    dplyr::filter(GAinS.ID %in% eigengene.patients) %>%\n",
    "    unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "93a8597a-1a50-423b-a6e9-fc179ac88d39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>638</li><li>3</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 638\n",
       "\\item 3\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 638\n",
       "2. 3\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 638   3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(geno.fam)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "61dae4de-5a7a-4686-a22c-c981fee7c1b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(\n",
    "    geno.fam %>% dplyr::select(Family.ID, Individual.ID), \n",
    "    \"~/gains_team282/nikhil/expression/eigengene_sva/mapping_patients.txt\", row.names=F, quote=F, col.names=F, sep=\"\\t\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af790d1e-583b-4a87-848d-c44edf00a4c5",
   "metadata": {},
   "source": [
    "Save the list of module eigengenes as a text file. This will be used by NextFlow to parallelize the association mapping."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "0b100493-d60e-4f89-b9cf-c966916b8fae",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(colnames(eigengenes), \"~/gains_team282/nikhil/expression/eigengene_sva/mapping_eigengenes.txt\", row.names=F, quote=F, col.names=F, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a99a350-ac3e-4c19-b846-80171620e7b1",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Initial Pass"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "702d915b-5cd6-4276-a5a1-e038d0248a28",
   "metadata": {},
   "source": [
    "I will be using a linear mixed model (LMM) to test for association between genotypes and eigengene expression. In the past, the lab has used a likelihood test (F-Test) to compare a null model of the covariates against an alternative model where the genotype is included. I will be building the following model:\n",
    "\n",
    "1. Let $\\mathbf{E}_i\\in\\mathbb{R}^n$ be a vector representing the values of the $i$-th eigengene.\n",
    "2. Let $\\mathbf{Y}\\in\\mathbb{R}^{n\\times c}$ be a matrix of covariates. These covariates include Sex, 16 PEER factors, and 7 Genotyping PCs.\n",
    "3. Let $\\mathbf{Z}\\in\\mathbb{R}^n$ be a vector representing the random effects. The only random effect in this model is the Patient ID.\n",
    "4. Let $\\mathbf{X}\\in\\mathbb{R}^n$ be a vector representing the genotypes of the patients.\n",
    "5. Let $\\beta\\in\\mathbb{R}$ be a scalar value representing the genotypic effect on eigengene expression.\n",
    "6. Let $\\mathbf{\\alpha}\\in\\mathbb{R}^c$ be a vector of covariate effects on eigengene expression.\n",
    "7. Let $\\gamma\\in\\mathbb{R}$ be a scalar value representing the random effect on eigengene expression.\n",
    "\n",
    "The null model is:\n",
    "\n",
    "$$\\mathbf{E}_i \\sim \\mathbf{Y}\\alpha + \\mathbf{Z}\\gamma$$\n",
    "\n",
    "The alternative model is:\n",
    "\n",
    "$$\\mathbf{E}_i \\sim \\mathbf{X}\\beta + \\mathbf{Y}\\alpha + \\mathbf{Z}\\gamma$$"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e20133d-11f0-480f-9bec-4869a8279161",
   "metadata": {},
   "source": [
    "There are 823 samples from the RNA-Seq data that have genotypes as well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "2c9a6515-9ca4-4974-abb0-c5fabf338edc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "823"
      ],
      "text/latex": [
       "823"
      ],
      "text/markdown": [
       "823"
      ],
      "text/plain": [
       "[1] 823"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "samples.with.genotypes <- rownames(eigengenes)[\n",
    "    sapply(\n",
    "        strsplit(rownames(eigengenes), \"_\"),\n",
    "        function(x) { x[1] %in% geno.fam$GAinS.ID }\n",
    "    )\n",
    "]\n",
    "\n",
    "length(samples.with.genotypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "e8ae2e1e-aee5-4217-bd0a-538807c882b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'PEER_1'</li><li>'PEER_2'</li><li>'PEER_3'</li><li>'PEER_4'</li><li>'PEER_5'</li><li>'PEER_6'</li><li>'PEER_7'</li><li>'PEER_8'</li><li>'PEER_9'</li><li>'PEER_10'</li><li>'PEER_11'</li><li>'PEER_12'</li><li>'PEER_13'</li><li>'PEER_14'</li><li>'PEER_15'</li><li>'PEER_16'</li><li>'PEER_17'</li><li>'PEER_18'</li><li>'PEER_19'</li><li>'PEER_20'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'PEER\\_1'\n",
       "\\item 'PEER\\_2'\n",
       "\\item 'PEER\\_3'\n",
       "\\item 'PEER\\_4'\n",
       "\\item 'PEER\\_5'\n",
       "\\item 'PEER\\_6'\n",
       "\\item 'PEER\\_7'\n",
       "\\item 'PEER\\_8'\n",
       "\\item 'PEER\\_9'\n",
       "\\item 'PEER\\_10'\n",
       "\\item 'PEER\\_11'\n",
       "\\item 'PEER\\_12'\n",
       "\\item 'PEER\\_13'\n",
       "\\item 'PEER\\_14'\n",
       "\\item 'PEER\\_15'\n",
       "\\item 'PEER\\_16'\n",
       "\\item 'PEER\\_17'\n",
       "\\item 'PEER\\_18'\n",
       "\\item 'PEER\\_19'\n",
       "\\item 'PEER\\_20'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'PEER_1'\n",
       "2. 'PEER_2'\n",
       "3. 'PEER_3'\n",
       "4. 'PEER_4'\n",
       "5. 'PEER_5'\n",
       "6. 'PEER_6'\n",
       "7. 'PEER_7'\n",
       "8. 'PEER_8'\n",
       "9. 'PEER_9'\n",
       "10. 'PEER_10'\n",
       "11. 'PEER_11'\n",
       "12. 'PEER_12'\n",
       "13. 'PEER_13'\n",
       "14. 'PEER_14'\n",
       "15. 'PEER_15'\n",
       "16. 'PEER_16'\n",
       "17. 'PEER_17'\n",
       "18. 'PEER_18'\n",
       "19. 'PEER_19'\n",
       "20. 'PEER_20'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"PEER_1\"  \"PEER_2\"  \"PEER_3\"  \"PEER_4\"  \"PEER_5\"  \"PEER_6\"  \"PEER_7\" \n",
       " [8] \"PEER_8\"  \"PEER_9\"  \"PEER_10\" \"PEER_11\" \"PEER_12\" \"PEER_13\" \"PEER_14\"\n",
       "[15] \"PEER_15\" \"PEER_16\" \"PEER_17\" \"PEER_18\" \"PEER_19\" \"PEER_20\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Diagnosis'</li><li>'SRS1'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'Neutrophils'</li><li>'Lymphocytes'</li><li>'Monocytes'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Diagnosis'\n",
       "\\item 'SRS1'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'Neutrophils'\n",
       "\\item 'Lymphocytes'\n",
       "\\item 'Monocytes'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Diagnosis'\n",
       "2. 'SRS1'\n",
       "3. 'PC1'\n",
       "4. 'PC2'\n",
       "5. 'PC3'\n",
       "6. 'PC4'\n",
       "7. 'PC5'\n",
       "8. 'PC6'\n",
       "9. 'PC7'\n",
       "10. 'Neutrophils'\n",
       "11. 'Lymphocytes'\n",
       "12. 'Monocytes'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Diagnosis\"   \"SRS1\"        \"PC1\"         \"PC2\"         \"PC3\"        \n",
       " [6] \"PC4\"         \"PC5\"         \"PC6\"         \"PC7\"         \"Neutrophils\"\n",
       "[11] \"Lymphocytes\" \"Monocytes\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'GAinS.ID'"
      ],
      "text/latex": [
       "'GAinS.ID'"
      ],
      "text/markdown": [
       "'GAinS.ID'"
      ],
      "text/plain": [
       "[1] \"GAinS.ID\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "peer.included <- peer[1:20]\n",
    "peer.included\n",
    "\n",
    "cov.names <- c(\"Diagnosis\", \"SRS1\", paste0(\"PC\", 1:7), \"Neutrophils\", \"Lymphocytes\", \"Monocytes\")\n",
    "cov.names\n",
    "\n",
    "rand.effect.names <- c(\"GAinS.ID\")\n",
    "rand.effect.names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "0b68e365-3db2-4446-8fe9-f71ab5fdd4eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping.data <- merge(eigengenes, covs, by=0) %>%\n",
    "    dplyr::filter(Row.names %in% samples.with.genotypes) %>%\n",
    "    dplyr::select(Sample.ID=Row.names, everything()) %>%\n",
    "    dplyr::mutate(GAinS.ID.NonPrefix=sapply(strsplit(Sample.ID, \"_\"), function(x) { x[1] })) %>%\n",
    "    merge(., geno.fam, by.x=\"GAinS.ID.NonPrefix\", by.y=\"GAinS.ID\") %>%\n",
    "    dplyr::select(GAinS.ID=Individual.ID, everything()) %>%\n",
    "    dplyr::select(Sample.ID, any_of(rand.effect.names), any_of(colnames(eigengenes)), any_of(cov.names), any_of(peer.included))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "35b5728d-c41e-42b9-9bad-844187829c68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 564</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample.ID</th><th scope=col>GAinS.ID</th><th scope=col>ME_1_1</th><th scope=col>ME_1_2</th><th scope=col>ME_1_3</th><th scope=col>ME_1_4</th><th scope=col>ME_1_5</th><th scope=col>ME_2_1</th><th scope=col>ME_2_2</th><th scope=col>ME_2_3</th><th scope=col>⋯</th><th scope=col>PEER_11</th><th scope=col>PEER_12</th><th scope=col>PEER_13</th><th scope=col>PEER_14</th><th scope=col>PEER_15</th><th scope=col>PEER_16</th><th scope=col>PEER_17</th><th scope=col>PEER_18</th><th scope=col>PEER_19</th><th scope=col>PEER_20</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;I&lt;chr&gt;&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>UK01050129_5</td><td>UK01050129</td><td>-0.011965012</td><td>-0.09047520</td><td>-0.017610108</td><td> 0.0007358012</td><td>-0.0151430087</td><td>-0.008135170</td><td>-0.09597010</td><td>-0.012628564</td><td>⋯</td><td> 0.047811605</td><td>-0.016539395</td><td> 0.051095940</td><td>-0.026148196</td><td>-0.021135228</td><td> 0.002443923</td><td> 0.032696567</td><td> 0.035078023</td><td> 0.03328779</td><td>-0.072080098</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>UK01060123_3</td><td>UK01060123</td><td>-0.002691514</td><td>-0.02324379</td><td> 0.040388816</td><td>-0.0194287731</td><td> 0.0097147859</td><td>-0.003642363</td><td>-0.02567566</td><td> 0.042063568</td><td>⋯</td><td>-0.007224238</td><td> 0.016796881</td><td>-0.010660484</td><td>-0.017610908</td><td> 0.031190908</td><td> 0.005303205</td><td> 0.040227711</td><td> 0.023455137</td><td>-0.06972641</td><td>-0.016429750</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>UK01070117_3</td><td>UK01070117</td><td>-0.027671656</td><td>-0.01496809</td><td>-0.008750141</td><td>-0.0280644252</td><td>-0.0200438485</td><td>-0.018676902</td><td>-0.01817045</td><td>-0.010687002</td><td>⋯</td><td> 0.002017805</td><td> 0.013472254</td><td> 0.044990435</td><td> 0.017342746</td><td> 0.040239949</td><td>-0.037767619</td><td>-0.002149181</td><td> 0.010321584</td><td>-0.03437537</td><td>-0.008777559</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>UK01080111_1</td><td>UK01080111</td><td> 0.021899313</td><td>-0.01413116</td><td>-0.004961546</td><td> 0.0193836562</td><td>-0.0677604487</td><td> 0.011468622</td><td>-0.01269462</td><td> 0.000280292</td><td>⋯</td><td>-0.063317396</td><td>-0.072918773</td><td>-0.010724876</td><td>-0.016138442</td><td>-0.003775158</td><td>-0.032603394</td><td>-0.068149783</td><td>-0.007441163</td><td>-0.04162392</td><td>-0.056151949</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>UK01110093_1</td><td>UK01110093</td><td>-0.053133706</td><td> 0.11972216</td><td>-0.029216200</td><td> 0.0042203907</td><td> 0.0216303612</td><td>-0.054170247</td><td> 0.11333090</td><td>-0.036116204</td><td>⋯</td><td>-0.009035902</td><td>-0.001971741</td><td>-0.003907994</td><td>-0.001863737</td><td>-0.031021271</td><td>-0.001989635</td><td> 0.010791964</td><td>-0.013796184</td><td> 0.09261347</td><td>-0.072555415</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>UK01110093_5</td><td>UK01110093</td><td>-0.097745041</td><td> 0.08920047</td><td> 0.001058428</td><td> 0.0131789677</td><td>-0.0001536524</td><td>-0.093620110</td><td> 0.07847404</td><td>-0.007012426</td><td>⋯</td><td> 0.076265849</td><td>-0.071181759</td><td> 0.045724034</td><td>-0.083036296</td><td> 0.009541817</td><td>-0.043817975</td><td> 0.079534695</td><td> 0.008144007</td><td> 0.09541912</td><td>-0.201759934</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 564\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Sample.ID & GAinS.ID & ME\\_1\\_1 & ME\\_1\\_2 & ME\\_1\\_3 & ME\\_1\\_4 & ME\\_1\\_5 & ME\\_2\\_1 & ME\\_2\\_2 & ME\\_2\\_3 & ⋯ & PEER\\_11 & PEER\\_12 & PEER\\_13 & PEER\\_14 & PEER\\_15 & PEER\\_16 & PEER\\_17 & PEER\\_18 & PEER\\_19 & PEER\\_20\\\\\n",
       "  & <I<chr>> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & UK01050129\\_5 & UK01050129 & -0.011965012 & -0.09047520 & -0.017610108 &  0.0007358012 & -0.0151430087 & -0.008135170 & -0.09597010 & -0.012628564 & ⋯ &  0.047811605 & -0.016539395 &  0.051095940 & -0.026148196 & -0.021135228 &  0.002443923 &  0.032696567 &  0.035078023 &  0.03328779 & -0.072080098\\\\\n",
       "\t2 & UK01060123\\_3 & UK01060123 & -0.002691514 & -0.02324379 &  0.040388816 & -0.0194287731 &  0.0097147859 & -0.003642363 & -0.02567566 &  0.042063568 & ⋯ & -0.007224238 &  0.016796881 & -0.010660484 & -0.017610908 &  0.031190908 &  0.005303205 &  0.040227711 &  0.023455137 & -0.06972641 & -0.016429750\\\\\n",
       "\t3 & UK01070117\\_3 & UK01070117 & -0.027671656 & -0.01496809 & -0.008750141 & -0.0280644252 & -0.0200438485 & -0.018676902 & -0.01817045 & -0.010687002 & ⋯ &  0.002017805 &  0.013472254 &  0.044990435 &  0.017342746 &  0.040239949 & -0.037767619 & -0.002149181 &  0.010321584 & -0.03437537 & -0.008777559\\\\\n",
       "\t4 & UK01080111\\_1 & UK01080111 &  0.021899313 & -0.01413116 & -0.004961546 &  0.0193836562 & -0.0677604487 &  0.011468622 & -0.01269462 &  0.000280292 & ⋯ & -0.063317396 & -0.072918773 & -0.010724876 & -0.016138442 & -0.003775158 & -0.032603394 & -0.068149783 & -0.007441163 & -0.04162392 & -0.056151949\\\\\n",
       "\t5 & UK01110093\\_1 & UK01110093 & -0.053133706 &  0.11972216 & -0.029216200 &  0.0042203907 &  0.0216303612 & -0.054170247 &  0.11333090 & -0.036116204 & ⋯ & -0.009035902 & -0.001971741 & -0.003907994 & -0.001863737 & -0.031021271 & -0.001989635 &  0.010791964 & -0.013796184 &  0.09261347 & -0.072555415\\\\\n",
       "\t6 & UK01110093\\_5 & UK01110093 & -0.097745041 &  0.08920047 &  0.001058428 &  0.0131789677 & -0.0001536524 & -0.093620110 &  0.07847404 & -0.007012426 & ⋯ &  0.076265849 & -0.071181759 &  0.045724034 & -0.083036296 &  0.009541817 & -0.043817975 &  0.079534695 &  0.008144007 &  0.09541912 & -0.201759934\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 564\n",
       "\n",
       "| <!--/--> | Sample.ID &lt;I&lt;chr&gt;&gt; | GAinS.ID &lt;chr&gt; | ME_1_1 &lt;dbl&gt; | ME_1_2 &lt;dbl&gt; | ME_1_3 &lt;dbl&gt; | ME_1_4 &lt;dbl&gt; | ME_1_5 &lt;dbl&gt; | ME_2_1 &lt;dbl&gt; | ME_2_2 &lt;dbl&gt; | ME_2_3 &lt;dbl&gt; | ⋯ ⋯ | PEER_11 &lt;dbl&gt; | PEER_12 &lt;dbl&gt; | PEER_13 &lt;dbl&gt; | PEER_14 &lt;dbl&gt; | PEER_15 &lt;dbl&gt; | PEER_16 &lt;dbl&gt; | PEER_17 &lt;dbl&gt; | PEER_18 &lt;dbl&gt; | PEER_19 &lt;dbl&gt; | PEER_20 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | UK01050129_5 | UK01050129 | -0.011965012 | -0.09047520 | -0.017610108 |  0.0007358012 | -0.0151430087 | -0.008135170 | -0.09597010 | -0.012628564 | ⋯ |  0.047811605 | -0.016539395 |  0.051095940 | -0.026148196 | -0.021135228 |  0.002443923 |  0.032696567 |  0.035078023 |  0.03328779 | -0.072080098 |\n",
       "| 2 | UK01060123_3 | UK01060123 | -0.002691514 | -0.02324379 |  0.040388816 | -0.0194287731 |  0.0097147859 | -0.003642363 | -0.02567566 |  0.042063568 | ⋯ | -0.007224238 |  0.016796881 | -0.010660484 | -0.017610908 |  0.031190908 |  0.005303205 |  0.040227711 |  0.023455137 | -0.06972641 | -0.016429750 |\n",
       "| 3 | UK01070117_3 | UK01070117 | -0.027671656 | -0.01496809 | -0.008750141 | -0.0280644252 | -0.0200438485 | -0.018676902 | -0.01817045 | -0.010687002 | ⋯ |  0.002017805 |  0.013472254 |  0.044990435 |  0.017342746 |  0.040239949 | -0.037767619 | -0.002149181 |  0.010321584 | -0.03437537 | -0.008777559 |\n",
       "| 4 | UK01080111_1 | UK01080111 |  0.021899313 | -0.01413116 | -0.004961546 |  0.0193836562 | -0.0677604487 |  0.011468622 | -0.01269462 |  0.000280292 | ⋯ | -0.063317396 | -0.072918773 | -0.010724876 | -0.016138442 | -0.003775158 | -0.032603394 | -0.068149783 | -0.007441163 | -0.04162392 | -0.056151949 |\n",
       "| 5 | UK01110093_1 | UK01110093 | -0.053133706 |  0.11972216 | -0.029216200 |  0.0042203907 |  0.0216303612 | -0.054170247 |  0.11333090 | -0.036116204 | ⋯ | -0.009035902 | -0.001971741 | -0.003907994 | -0.001863737 | -0.031021271 | -0.001989635 |  0.010791964 | -0.013796184 |  0.09261347 | -0.072555415 |\n",
       "| 6 | UK01110093_5 | UK01110093 | -0.097745041 |  0.08920047 |  0.001058428 |  0.0131789677 | -0.0001536524 | -0.093620110 |  0.07847404 | -0.007012426 | ⋯ |  0.076265849 | -0.071181759 |  0.045724034 | -0.083036296 |  0.009541817 | -0.043817975 |  0.079534695 |  0.008144007 |  0.09541912 | -0.201759934 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample.ID    GAinS.ID   ME_1_1       ME_1_2      ME_1_3       ME_1_4       \n",
       "1 UK01050129_5 UK01050129 -0.011965012 -0.09047520 -0.017610108  0.0007358012\n",
       "2 UK01060123_3 UK01060123 -0.002691514 -0.02324379  0.040388816 -0.0194287731\n",
       "3 UK01070117_3 UK01070117 -0.027671656 -0.01496809 -0.008750141 -0.0280644252\n",
       "4 UK01080111_1 UK01080111  0.021899313 -0.01413116 -0.004961546  0.0193836562\n",
       "5 UK01110093_1 UK01110093 -0.053133706  0.11972216 -0.029216200  0.0042203907\n",
       "6 UK01110093_5 UK01110093 -0.097745041  0.08920047  0.001058428  0.0131789677\n",
       "  ME_1_5        ME_2_1       ME_2_2      ME_2_3       ⋯ PEER_11     \n",
       "1 -0.0151430087 -0.008135170 -0.09597010 -0.012628564 ⋯  0.047811605\n",
       "2  0.0097147859 -0.003642363 -0.02567566  0.042063568 ⋯ -0.007224238\n",
       "3 -0.0200438485 -0.018676902 -0.01817045 -0.010687002 ⋯  0.002017805\n",
       "4 -0.0677604487  0.011468622 -0.01269462  0.000280292 ⋯ -0.063317396\n",
       "5  0.0216303612 -0.054170247  0.11333090 -0.036116204 ⋯ -0.009035902\n",
       "6 -0.0001536524 -0.093620110  0.07847404 -0.007012426 ⋯  0.076265849\n",
       "  PEER_12      PEER_13      PEER_14      PEER_15      PEER_16      PEER_17     \n",
       "1 -0.016539395  0.051095940 -0.026148196 -0.021135228  0.002443923  0.032696567\n",
       "2  0.016796881 -0.010660484 -0.017610908  0.031190908  0.005303205  0.040227711\n",
       "3  0.013472254  0.044990435  0.017342746  0.040239949 -0.037767619 -0.002149181\n",
       "4 -0.072918773 -0.010724876 -0.016138442 -0.003775158 -0.032603394 -0.068149783\n",
       "5 -0.001971741 -0.003907994 -0.001863737 -0.031021271 -0.001989635  0.010791964\n",
       "6 -0.071181759  0.045724034 -0.083036296  0.009541817 -0.043817975  0.079534695\n",
       "  PEER_18      PEER_19     PEER_20     \n",
       "1  0.035078023  0.03328779 -0.072080098\n",
       "2  0.023455137 -0.06972641 -0.016429750\n",
       "3  0.010321584 -0.03437537 -0.008777559\n",
       "4 -0.007441163 -0.04162392 -0.056151949\n",
       "5 -0.013796184  0.09261347 -0.072555415\n",
       "6  0.008144007  0.09541912 -0.201759934"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(mapping.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b6ec6673-664f-4cd2-862b-9805f25551d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Sample.ID'</li><li>'GAinS.ID'</li><li>'ME_1_1'</li><li>'ME_1_2'</li><li>'ME_1_3'</li><li>'ME_1_4'</li><li>'ME_1_5'</li><li>'ME_2_1'</li><li>'ME_2_2'</li><li>'ME_2_3'</li><li>'ME_2_4'</li><li>'ME_2_5'</li><li>'ME_3_1'</li><li>'ME_3_2'</li><li>'ME_3_3'</li><li>'ME_3_4'</li><li>'ME_3_5'</li><li>'ME_4_1'</li><li>'ME_4_2'</li><li>'ME_4_3'</li><li>'ME_4_4'</li><li>'ME_4_5'</li><li>'ME_5_1'</li><li>'ME_5_2'</li><li>'ME_5_3'</li><li>'ME_5_4'</li><li>'ME_5_5'</li><li>'ME_6_1'</li><li>'ME_6_2'</li><li>'ME_6_3'</li><li>'ME_6_4'</li><li>'ME_6_5'</li><li>'ME_7_1'</li><li>'ME_7_2'</li><li>'ME_7_3'</li><li>'ME_7_4'</li><li>'ME_7_5'</li><li>'ME_8_1'</li><li>'ME_8_2'</li><li>'ME_8_3'</li><li>'ME_8_4'</li><li>'ME_8_5'</li><li>'ME_9_1'</li><li>'ME_9_2'</li><li>'ME_9_3'</li><li>'ME_9_4'</li><li>'ME_9_5'</li><li>'ME_10_1'</li><li>'ME_10_2'</li><li>'ME_10_3'</li><li>'ME_10_4'</li><li>'ME_10_5'</li><li>'ME_11_1'</li><li>'ME_11_2'</li><li>'ME_11_3'</li><li>'ME_11_4'</li><li>'ME_11_5'</li><li>'ME_12_1'</li><li>'ME_12_2'</li><li>'ME_12_3'</li><li>'ME_12_4'</li><li>'ME_12_5'</li><li>'ME_13_1'</li><li>'ME_13_2'</li><li>'ME_13_3'</li><li>'ME_13_4'</li><li>'ME_13_5'</li><li>'ME_14_1'</li><li>'ME_14_2'</li><li>'ME_14_3'</li><li>'ME_14_4'</li><li>'ME_14_5'</li><li>'ME_15_1'</li><li>'ME_15_2'</li><li>'ME_15_3'</li><li>'ME_15_4'</li><li>'ME_15_5'</li><li>'ME_16_1'</li><li>'ME_16_2'</li><li>'ME_16_3'</li><li>'ME_16_4'</li><li>'ME_16_5'</li><li>'ME_17_1'</li><li>'ME_17_2'</li><li>'ME_17_3'</li><li>'ME_17_4'</li><li>'ME_17_5'</li><li>'ME_18_1'</li><li>'ME_18_2'</li><li>'ME_18_3'</li><li>'ME_18_4'</li><li>'ME_18_5'</li><li>'ME_19_1'</li><li>'ME_19_2'</li><li>'ME_19_3'</li><li>'ME_19_4'</li><li>'ME_19_5'</li><li>'ME_20_1'</li><li>'ME_20_2'</li><li>'ME_20_3'</li><li>'ME_20_4'</li><li>'ME_20_5'</li><li>'ME_21_1'</li><li>'ME_21_2'</li><li>'ME_21_3'</li><li>'ME_21_4'</li><li>'ME_21_5'</li><li>'ME_22_1'</li><li>'ME_22_2'</li><li>'ME_22_3'</li><li>'ME_22_4'</li><li>'ME_22_5'</li><li>'ME_23_1'</li><li>'ME_23_2'</li><li>'ME_23_3'</li><li>'ME_23_4'</li><li>'ME_23_5'</li><li>'ME_24_1'</li><li>'ME_24_2'</li><li>'ME_24_3'</li><li>'ME_24_4'</li><li>'ME_24_5'</li><li>'ME_25_1'</li><li>'ME_25_2'</li><li>'ME_25_3'</li><li>'ME_25_4'</li><li>'ME_25_5'</li><li>'ME_26_1'</li><li>'ME_26_2'</li><li>'ME_26_3'</li><li>'ME_26_4'</li><li>'ME_26_5'</li><li>'ME_27_1'</li><li>'ME_27_2'</li><li>'ME_27_3'</li><li>'ME_27_4'</li><li>'ME_27_5'</li><li>'ME_28_1'</li><li>'ME_28_2'</li><li>'ME_28_3'</li><li>'ME_28_4'</li><li>'ME_28_5'</li><li>'ME_29_1'</li><li>'ME_29_2'</li><li>'ME_29_3'</li><li>'ME_29_4'</li><li>'ME_29_5'</li><li>'ME_30_1'</li><li>'ME_30_2'</li><li>'ME_30_3'</li><li>'ME_30_4'</li><li>'ME_30_5'</li><li>'ME_31_1'</li><li>'ME_31_2'</li><li>'ME_31_3'</li><li>'ME_31_4'</li><li>'ME_31_5'</li><li>'ME_32_1'</li><li>'ME_32_2'</li><li>'ME_32_3'</li><li>'ME_32_4'</li><li>'ME_32_5'</li><li>'ME_33_1'</li><li>'ME_33_2'</li><li>'ME_33_3'</li><li>'ME_33_4'</li><li>'ME_33_5'</li><li>'ME_34_1'</li><li>'ME_34_2'</li><li>'ME_34_3'</li><li>'ME_34_4'</li><li>'ME_34_5'</li><li>'ME_35_1'</li><li>'ME_35_2'</li><li>'ME_35_3'</li><li>'ME_35_4'</li><li>'ME_35_5'</li><li>'ME_36_1'</li><li>'ME_36_2'</li><li>'ME_36_3'</li><li>'ME_36_4'</li><li>'ME_36_5'</li><li>'ME_37_1'</li><li>'ME_37_2'</li><li>'ME_37_3'</li><li>'ME_37_4'</li><li>'ME_37_5'</li><li>'ME_38_1'</li><li>'ME_38_2'</li><li>'ME_38_3'</li><li>'ME_38_4'</li><li>'ME_38_5'</li><li>'ME_39_1'</li><li>'ME_39_2'</li><li>'ME_39_3'</li><li>'ME_39_4'</li><li>'ME_39_5'</li><li>'ME_40_1'</li><li>'ME_40_2'</li><li>'ME_40_3'</li><li>⋯</li><li>'ME_73_3'</li><li>'ME_73_4'</li><li>'ME_73_5'</li><li>'ME_74_1'</li><li>'ME_74_2'</li><li>'ME_74_3'</li><li>'ME_74_4'</li><li>'ME_74_5'</li><li>'ME_75_1'</li><li>'ME_75_2'</li><li>'ME_75_3'</li><li>'ME_75_4'</li><li>'ME_75_5'</li><li>'ME_76_1'</li><li>'ME_76_2'</li><li>'ME_76_3'</li><li>'ME_76_4'</li><li>'ME_76_5'</li><li>'ME_77_1'</li><li>'ME_77_2'</li><li>'ME_77_3'</li><li>'ME_77_4'</li><li>'ME_77_5'</li><li>'ME_78_1'</li><li>'ME_78_2'</li><li>'ME_78_3'</li><li>'ME_78_4'</li><li>'ME_78_5'</li><li>'ME_79_1'</li><li>'ME_79_2'</li><li>'ME_79_3'</li><li>'ME_79_4'</li><li>'ME_79_5'</li><li>'ME_80_1'</li><li>'ME_80_2'</li><li>'ME_80_3'</li><li>'ME_80_4'</li><li>'ME_80_5'</li><li>'ME_81_1'</li><li>'ME_81_2'</li><li>'ME_81_3'</li><li>'ME_81_4'</li><li>'ME_81_5'</li><li>'ME_82_1'</li><li>'ME_82_2'</li><li>'ME_82_3'</li><li>'ME_82_4'</li><li>'ME_82_5'</li><li>'ME_83_1'</li><li>'ME_83_2'</li><li>'ME_83_3'</li><li>'ME_83_4'</li><li>'ME_83_5'</li><li>'ME_84_1'</li><li>'ME_84_2'</li><li>'ME_84_3'</li><li>'ME_84_4'</li><li>'ME_84_5'</li><li>'ME_85_1'</li><li>'ME_85_2'</li><li>'ME_85_3'</li><li>'ME_85_4'</li><li>'ME_85_5'</li><li>'ME_86_1'</li><li>'ME_86_2'</li><li>'ME_86_3'</li><li>'ME_86_4'</li><li>'ME_86_5'</li><li>'ME_87_1'</li><li>'ME_87_2'</li><li>'ME_87_3'</li><li>'ME_87_4'</li><li>'ME_87_5'</li><li>'ME_88_1'</li><li>'ME_88_2'</li><li>'ME_88_3'</li><li>'ME_88_4'</li><li>'ME_88_5'</li><li>'ME_89_1'</li><li>'ME_89_2'</li><li>'ME_89_3'</li><li>'ME_89_4'</li><li>'ME_89_5'</li><li>'ME_90_1'</li><li>'ME_90_2'</li><li>'ME_90_3'</li><li>'ME_90_4'</li><li>'ME_90_5'</li><li>'ME_91_1'</li><li>'ME_91_2'</li><li>'ME_91_3'</li><li>'ME_91_4'</li><li>'ME_91_5'</li><li>'ME_92_1'</li><li>'ME_92_2'</li><li>'ME_92_3'</li><li>'ME_92_4'</li><li>'ME_92_5'</li><li>'ME_93_1'</li><li>'ME_93_2'</li><li>'ME_93_3'</li><li>'ME_93_4'</li><li>'ME_93_5'</li><li>'ME_94_1'</li><li>'ME_94_2'</li><li>'ME_94_3'</li><li>'ME_94_4'</li><li>'ME_94_5'</li><li>'ME_95_1'</li><li>'ME_95_2'</li><li>'ME_95_3'</li><li>'ME_95_4'</li><li>'ME_95_5'</li><li>'ME_96_1'</li><li>'ME_96_2'</li><li>'ME_96_3'</li><li>'ME_96_4'</li><li>'ME_96_5'</li><li>'ME_97_1'</li><li>'ME_97_2'</li><li>'ME_97_3'</li><li>'ME_97_4'</li><li>'ME_97_5'</li><li>'ME_98_1'</li><li>'ME_98_2'</li><li>'ME_98_3'</li><li>'ME_98_4'</li><li>'ME_98_5'</li><li>'ME_99_1'</li><li>'ME_99_2'</li><li>'ME_99_3'</li><li>'ME_99_4'</li><li>'ME_99_5'</li><li>'ME_100_1'</li><li>'ME_100_2'</li><li>'ME_100_3'</li><li>'ME_100_4'</li><li>'ME_100_5'</li><li>'ME_101_1'</li><li>'ME_101_2'</li><li>'ME_101_3'</li><li>'ME_101_4'</li><li>'ME_101_5'</li><li>'ME_102_1'</li><li>'ME_102_2'</li><li>'ME_102_3'</li><li>'ME_102_4'</li><li>'ME_102_5'</li><li>'ME_103_1'</li><li>'ME_103_2'</li><li>'ME_103_3'</li><li>'ME_103_4'</li><li>'ME_103_5'</li><li>'ME_104_1'</li><li>'ME_104_2'</li><li>'ME_104_3'</li><li>'ME_104_4'</li><li>'ME_104_5'</li><li>'ME_105_1'</li><li>'ME_105_2'</li><li>'ME_105_3'</li><li>'ME_105_4'</li><li>'ME_105_5'</li><li>'ME_106_1'</li><li>'ME_106_2'</li><li>'ME_106_3'</li><li>'ME_106_4'</li><li>'ME_106_5'</li><li>'Diagnosis'</li><li>'SRS1'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'Neutrophils'</li><li>'Lymphocytes'</li><li>'Monocytes'</li><li>'PEER_1'</li><li>'PEER_2'</li><li>'PEER_3'</li><li>'PEER_4'</li><li>'PEER_5'</li><li>'PEER_6'</li><li>'PEER_7'</li><li>'PEER_8'</li><li>'PEER_9'</li><li>'PEER_10'</li><li>'PEER_11'</li><li>'PEER_12'</li><li>'PEER_13'</li><li>'PEER_14'</li><li>'PEER_15'</li><li>'PEER_16'</li><li>'PEER_17'</li><li>'PEER_18'</li><li>'PEER_19'</li><li>'PEER_20'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Sample.ID'\n",
       "\\item 'GAinS.ID'\n",
       "\\item 'ME\\_1\\_1'\n",
       "\\item 'ME\\_1\\_2'\n",
       "\\item 'ME\\_1\\_3'\n",
       "\\item 'ME\\_1\\_4'\n",
       "\\item 'ME\\_1\\_5'\n",
       "\\item 'ME\\_2\\_1'\n",
       "\\item 'ME\\_2\\_2'\n",
       "\\item 'ME\\_2\\_3'\n",
       "\\item 'ME\\_2\\_4'\n",
       "\\item 'ME\\_2\\_5'\n",
       "\\item 'ME\\_3\\_1'\n",
       "\\item 'ME\\_3\\_2'\n",
       "\\item 'ME\\_3\\_3'\n",
       "\\item 'ME\\_3\\_4'\n",
       "\\item 'ME\\_3\\_5'\n",
       "\\item 'ME\\_4\\_1'\n",
       "\\item 'ME\\_4\\_2'\n",
       "\\item 'ME\\_4\\_3'\n",
       "\\item 'ME\\_4\\_4'\n",
       "\\item 'ME\\_4\\_5'\n",
       "\\item 'ME\\_5\\_1'\n",
       "\\item 'ME\\_5\\_2'\n",
       "\\item 'ME\\_5\\_3'\n",
       "\\item 'ME\\_5\\_4'\n",
       "\\item 'ME\\_5\\_5'\n",
       "\\item 'ME\\_6\\_1'\n",
       "\\item 'ME\\_6\\_2'\n",
       "\\item 'ME\\_6\\_3'\n",
       "\\item 'ME\\_6\\_4'\n",
       "\\item 'ME\\_6\\_5'\n",
       "\\item 'ME\\_7\\_1'\n",
       "\\item 'ME\\_7\\_2'\n",
       "\\item 'ME\\_7\\_3'\n",
       "\\item 'ME\\_7\\_4'\n",
       "\\item 'ME\\_7\\_5'\n",
       "\\item 'ME\\_8\\_1'\n",
       "\\item 'ME\\_8\\_2'\n",
       "\\item 'ME\\_8\\_3'\n",
       "\\item 'ME\\_8\\_4'\n",
       "\\item 'ME\\_8\\_5'\n",
       "\\item 'ME\\_9\\_1'\n",
       "\\item 'ME\\_9\\_2'\n",
       "\\item 'ME\\_9\\_3'\n",
       "\\item 'ME\\_9\\_4'\n",
       "\\item 'ME\\_9\\_5'\n",
       "\\item 'ME\\_10\\_1'\n",
       "\\item 'ME\\_10\\_2'\n",
       "\\item 'ME\\_10\\_3'\n",
       "\\item 'ME\\_10\\_4'\n",
       "\\item 'ME\\_10\\_5'\n",
       "\\item 'ME\\_11\\_1'\n",
       "\\item 'ME\\_11\\_2'\n",
       "\\item 'ME\\_11\\_3'\n",
       "\\item 'ME\\_11\\_4'\n",
       "\\item 'ME\\_11\\_5'\n",
       "\\item 'ME\\_12\\_1'\n",
       "\\item 'ME\\_12\\_2'\n",
       "\\item 'ME\\_12\\_3'\n",
       "\\item 'ME\\_12\\_4'\n",
       "\\item 'ME\\_12\\_5'\n",
       "\\item 'ME\\_13\\_1'\n",
       "\\item 'ME\\_13\\_2'\n",
       "\\item 'ME\\_13\\_3'\n",
       "\\item 'ME\\_13\\_4'\n",
       "\\item 'ME\\_13\\_5'\n",
       "\\item 'ME\\_14\\_1'\n",
       "\\item 'ME\\_14\\_2'\n",
       "\\item 'ME\\_14\\_3'\n",
       "\\item 'ME\\_14\\_4'\n",
       "\\item 'ME\\_14\\_5'\n",
       "\\item 'ME\\_15\\_1'\n",
       "\\item 'ME\\_15\\_2'\n",
       "\\item 'ME\\_15\\_3'\n",
       "\\item 'ME\\_15\\_4'\n",
       "\\item 'ME\\_15\\_5'\n",
       "\\item 'ME\\_16\\_1'\n",
       "\\item 'ME\\_16\\_2'\n",
       "\\item 'ME\\_16\\_3'\n",
       "\\item 'ME\\_16\\_4'\n",
       "\\item 'ME\\_16\\_5'\n",
       "\\item 'ME\\_17\\_1'\n",
       "\\item 'ME\\_17\\_2'\n",
       "\\item 'ME\\_17\\_3'\n",
       "\\item 'ME\\_17\\_4'\n",
       "\\item 'ME\\_17\\_5'\n",
       "\\item 'ME\\_18\\_1'\n",
       "\\item 'ME\\_18\\_2'\n",
       "\\item 'ME\\_18\\_3'\n",
       "\\item 'ME\\_18\\_4'\n",
       "\\item 'ME\\_18\\_5'\n",
       "\\item 'ME\\_19\\_1'\n",
       "\\item 'ME\\_19\\_2'\n",
       "\\item 'ME\\_19\\_3'\n",
       "\\item 'ME\\_19\\_4'\n",
       "\\item 'ME\\_19\\_5'\n",
       "\\item 'ME\\_20\\_1'\n",
       "\\item 'ME\\_20\\_2'\n",
       "\\item 'ME\\_20\\_3'\n",
       "\\item 'ME\\_20\\_4'\n",
       "\\item 'ME\\_20\\_5'\n",
       "\\item 'ME\\_21\\_1'\n",
       "\\item 'ME\\_21\\_2'\n",
       "\\item 'ME\\_21\\_3'\n",
       "\\item 'ME\\_21\\_4'\n",
       "\\item 'ME\\_21\\_5'\n",
       "\\item 'ME\\_22\\_1'\n",
       "\\item 'ME\\_22\\_2'\n",
       "\\item 'ME\\_22\\_3'\n",
       "\\item 'ME\\_22\\_4'\n",
       "\\item 'ME\\_22\\_5'\n",
       "\\item 'ME\\_23\\_1'\n",
       "\\item 'ME\\_23\\_2'\n",
       "\\item 'ME\\_23\\_3'\n",
       "\\item 'ME\\_23\\_4'\n",
       "\\item 'ME\\_23\\_5'\n",
       "\\item 'ME\\_24\\_1'\n",
       "\\item 'ME\\_24\\_2'\n",
       "\\item 'ME\\_24\\_3'\n",
       "\\item 'ME\\_24\\_4'\n",
       "\\item 'ME\\_24\\_5'\n",
       "\\item 'ME\\_25\\_1'\n",
       "\\item 'ME\\_25\\_2'\n",
       "\\item 'ME\\_25\\_3'\n",
       "\\item 'ME\\_25\\_4'\n",
       "\\item 'ME\\_25\\_5'\n",
       "\\item 'ME\\_26\\_1'\n",
       "\\item 'ME\\_26\\_2'\n",
       "\\item 'ME\\_26\\_3'\n",
       "\\item 'ME\\_26\\_4'\n",
       "\\item 'ME\\_26\\_5'\n",
       "\\item 'ME\\_27\\_1'\n",
       "\\item 'ME\\_27\\_2'\n",
       "\\item 'ME\\_27\\_3'\n",
       "\\item 'ME\\_27\\_4'\n",
       "\\item 'ME\\_27\\_5'\n",
       "\\item 'ME\\_28\\_1'\n",
       "\\item 'ME\\_28\\_2'\n",
       "\\item 'ME\\_28\\_3'\n",
       "\\item 'ME\\_28\\_4'\n",
       "\\item 'ME\\_28\\_5'\n",
       "\\item 'ME\\_29\\_1'\n",
       "\\item 'ME\\_29\\_2'\n",
       "\\item 'ME\\_29\\_3'\n",
       "\\item 'ME\\_29\\_4'\n",
       "\\item 'ME\\_29\\_5'\n",
       "\\item 'ME\\_30\\_1'\n",
       "\\item 'ME\\_30\\_2'\n",
       "\\item 'ME\\_30\\_3'\n",
       "\\item 'ME\\_30\\_4'\n",
       "\\item 'ME\\_30\\_5'\n",
       "\\item 'ME\\_31\\_1'\n",
       "\\item 'ME\\_31\\_2'\n",
       "\\item 'ME\\_31\\_3'\n",
       "\\item 'ME\\_31\\_4'\n",
       "\\item 'ME\\_31\\_5'\n",
       "\\item 'ME\\_32\\_1'\n",
       "\\item 'ME\\_32\\_2'\n",
       "\\item 'ME\\_32\\_3'\n",
       "\\item 'ME\\_32\\_4'\n",
       "\\item 'ME\\_32\\_5'\n",
       "\\item 'ME\\_33\\_1'\n",
       "\\item 'ME\\_33\\_2'\n",
       "\\item 'ME\\_33\\_3'\n",
       "\\item 'ME\\_33\\_4'\n",
       "\\item 'ME\\_33\\_5'\n",
       "\\item 'ME\\_34\\_1'\n",
       "\\item 'ME\\_34\\_2'\n",
       "\\item 'ME\\_34\\_3'\n",
       "\\item 'ME\\_34\\_4'\n",
       "\\item 'ME\\_34\\_5'\n",
       "\\item 'ME\\_35\\_1'\n",
       "\\item 'ME\\_35\\_2'\n",
       "\\item 'ME\\_35\\_3'\n",
       "\\item 'ME\\_35\\_4'\n",
       "\\item 'ME\\_35\\_5'\n",
       "\\item 'ME\\_36\\_1'\n",
       "\\item 'ME\\_36\\_2'\n",
       "\\item 'ME\\_36\\_3'\n",
       "\\item 'ME\\_36\\_4'\n",
       "\\item 'ME\\_36\\_5'\n",
       "\\item 'ME\\_37\\_1'\n",
       "\\item 'ME\\_37\\_2'\n",
       "\\item 'ME\\_37\\_3'\n",
       "\\item 'ME\\_37\\_4'\n",
       "\\item 'ME\\_37\\_5'\n",
       "\\item 'ME\\_38\\_1'\n",
       "\\item 'ME\\_38\\_2'\n",
       "\\item 'ME\\_38\\_3'\n",
       "\\item 'ME\\_38\\_4'\n",
       "\\item 'ME\\_38\\_5'\n",
       "\\item 'ME\\_39\\_1'\n",
       "\\item 'ME\\_39\\_2'\n",
       "\\item 'ME\\_39\\_3'\n",
       "\\item 'ME\\_39\\_4'\n",
       "\\item 'ME\\_39\\_5'\n",
       "\\item 'ME\\_40\\_1'\n",
       "\\item 'ME\\_40\\_2'\n",
       "\\item 'ME\\_40\\_3'\n",
       "\\item ⋯\n",
       "\\item 'ME\\_73\\_3'\n",
       "\\item 'ME\\_73\\_4'\n",
       "\\item 'ME\\_73\\_5'\n",
       "\\item 'ME\\_74\\_1'\n",
       "\\item 'ME\\_74\\_2'\n",
       "\\item 'ME\\_74\\_3'\n",
       "\\item 'ME\\_74\\_4'\n",
       "\\item 'ME\\_74\\_5'\n",
       "\\item 'ME\\_75\\_1'\n",
       "\\item 'ME\\_75\\_2'\n",
       "\\item 'ME\\_75\\_3'\n",
       "\\item 'ME\\_75\\_4'\n",
       "\\item 'ME\\_75\\_5'\n",
       "\\item 'ME\\_76\\_1'\n",
       "\\item 'ME\\_76\\_2'\n",
       "\\item 'ME\\_76\\_3'\n",
       "\\item 'ME\\_76\\_4'\n",
       "\\item 'ME\\_76\\_5'\n",
       "\\item 'ME\\_77\\_1'\n",
       "\\item 'ME\\_77\\_2'\n",
       "\\item 'ME\\_77\\_3'\n",
       "\\item 'ME\\_77\\_4'\n",
       "\\item 'ME\\_77\\_5'\n",
       "\\item 'ME\\_78\\_1'\n",
       "\\item 'ME\\_78\\_2'\n",
       "\\item 'ME\\_78\\_3'\n",
       "\\item 'ME\\_78\\_4'\n",
       "\\item 'ME\\_78\\_5'\n",
       "\\item 'ME\\_79\\_1'\n",
       "\\item 'ME\\_79\\_2'\n",
       "\\item 'ME\\_79\\_3'\n",
       "\\item 'ME\\_79\\_4'\n",
       "\\item 'ME\\_79\\_5'\n",
       "\\item 'ME\\_80\\_1'\n",
       "\\item 'ME\\_80\\_2'\n",
       "\\item 'ME\\_80\\_3'\n",
       "\\item 'ME\\_80\\_4'\n",
       "\\item 'ME\\_80\\_5'\n",
       "\\item 'ME\\_81\\_1'\n",
       "\\item 'ME\\_81\\_2'\n",
       "\\item 'ME\\_81\\_3'\n",
       "\\item 'ME\\_81\\_4'\n",
       "\\item 'ME\\_81\\_5'\n",
       "\\item 'ME\\_82\\_1'\n",
       "\\item 'ME\\_82\\_2'\n",
       "\\item 'ME\\_82\\_3'\n",
       "\\item 'ME\\_82\\_4'\n",
       "\\item 'ME\\_82\\_5'\n",
       "\\item 'ME\\_83\\_1'\n",
       "\\item 'ME\\_83\\_2'\n",
       "\\item 'ME\\_83\\_3'\n",
       "\\item 'ME\\_83\\_4'\n",
       "\\item 'ME\\_83\\_5'\n",
       "\\item 'ME\\_84\\_1'\n",
       "\\item 'ME\\_84\\_2'\n",
       "\\item 'ME\\_84\\_3'\n",
       "\\item 'ME\\_84\\_4'\n",
       "\\item 'ME\\_84\\_5'\n",
       "\\item 'ME\\_85\\_1'\n",
       "\\item 'ME\\_85\\_2'\n",
       "\\item 'ME\\_85\\_3'\n",
       "\\item 'ME\\_85\\_4'\n",
       "\\item 'ME\\_85\\_5'\n",
       "\\item 'ME\\_86\\_1'\n",
       "\\item 'ME\\_86\\_2'\n",
       "\\item 'ME\\_86\\_3'\n",
       "\\item 'ME\\_86\\_4'\n",
       "\\item 'ME\\_86\\_5'\n",
       "\\item 'ME\\_87\\_1'\n",
       "\\item 'ME\\_87\\_2'\n",
       "\\item 'ME\\_87\\_3'\n",
       "\\item 'ME\\_87\\_4'\n",
       "\\item 'ME\\_87\\_5'\n",
       "\\item 'ME\\_88\\_1'\n",
       "\\item 'ME\\_88\\_2'\n",
       "\\item 'ME\\_88\\_3'\n",
       "\\item 'ME\\_88\\_4'\n",
       "\\item 'ME\\_88\\_5'\n",
       "\\item 'ME\\_89\\_1'\n",
       "\\item 'ME\\_89\\_2'\n",
       "\\item 'ME\\_89\\_3'\n",
       "\\item 'ME\\_89\\_4'\n",
       "\\item 'ME\\_89\\_5'\n",
       "\\item 'ME\\_90\\_1'\n",
       "\\item 'ME\\_90\\_2'\n",
       "\\item 'ME\\_90\\_3'\n",
       "\\item 'ME\\_90\\_4'\n",
       "\\item 'ME\\_90\\_5'\n",
       "\\item 'ME\\_91\\_1'\n",
       "\\item 'ME\\_91\\_2'\n",
       "\\item 'ME\\_91\\_3'\n",
       "\\item 'ME\\_91\\_4'\n",
       "\\item 'ME\\_91\\_5'\n",
       "\\item 'ME\\_92\\_1'\n",
       "\\item 'ME\\_92\\_2'\n",
       "\\item 'ME\\_92\\_3'\n",
       "\\item 'ME\\_92\\_4'\n",
       "\\item 'ME\\_92\\_5'\n",
       "\\item 'ME\\_93\\_1'\n",
       "\\item 'ME\\_93\\_2'\n",
       "\\item 'ME\\_93\\_3'\n",
       "\\item 'ME\\_93\\_4'\n",
       "\\item 'ME\\_93\\_5'\n",
       "\\item 'ME\\_94\\_1'\n",
       "\\item 'ME\\_94\\_2'\n",
       "\\item 'ME\\_94\\_3'\n",
       "\\item 'ME\\_94\\_4'\n",
       "\\item 'ME\\_94\\_5'\n",
       "\\item 'ME\\_95\\_1'\n",
       "\\item 'ME\\_95\\_2'\n",
       "\\item 'ME\\_95\\_3'\n",
       "\\item 'ME\\_95\\_4'\n",
       "\\item 'ME\\_95\\_5'\n",
       "\\item 'ME\\_96\\_1'\n",
       "\\item 'ME\\_96\\_2'\n",
       "\\item 'ME\\_96\\_3'\n",
       "\\item 'ME\\_96\\_4'\n",
       "\\item 'ME\\_96\\_5'\n",
       "\\item 'ME\\_97\\_1'\n",
       "\\item 'ME\\_97\\_2'\n",
       "\\item 'ME\\_97\\_3'\n",
       "\\item 'ME\\_97\\_4'\n",
       "\\item 'ME\\_97\\_5'\n",
       "\\item 'ME\\_98\\_1'\n",
       "\\item 'ME\\_98\\_2'\n",
       "\\item 'ME\\_98\\_3'\n",
       "\\item 'ME\\_98\\_4'\n",
       "\\item 'ME\\_98\\_5'\n",
       "\\item 'ME\\_99\\_1'\n",
       "\\item 'ME\\_99\\_2'\n",
       "\\item 'ME\\_99\\_3'\n",
       "\\item 'ME\\_99\\_4'\n",
       "\\item 'ME\\_99\\_5'\n",
       "\\item 'ME\\_100\\_1'\n",
       "\\item 'ME\\_100\\_2'\n",
       "\\item 'ME\\_100\\_3'\n",
       "\\item 'ME\\_100\\_4'\n",
       "\\item 'ME\\_100\\_5'\n",
       "\\item 'ME\\_101\\_1'\n",
       "\\item 'ME\\_101\\_2'\n",
       "\\item 'ME\\_101\\_3'\n",
       "\\item 'ME\\_101\\_4'\n",
       "\\item 'ME\\_101\\_5'\n",
       "\\item 'ME\\_102\\_1'\n",
       "\\item 'ME\\_102\\_2'\n",
       "\\item 'ME\\_102\\_3'\n",
       "\\item 'ME\\_102\\_4'\n",
       "\\item 'ME\\_102\\_5'\n",
       "\\item 'ME\\_103\\_1'\n",
       "\\item 'ME\\_103\\_2'\n",
       "\\item 'ME\\_103\\_3'\n",
       "\\item 'ME\\_103\\_4'\n",
       "\\item 'ME\\_103\\_5'\n",
       "\\item 'ME\\_104\\_1'\n",
       "\\item 'ME\\_104\\_2'\n",
       "\\item 'ME\\_104\\_3'\n",
       "\\item 'ME\\_104\\_4'\n",
       "\\item 'ME\\_104\\_5'\n",
       "\\item 'ME\\_105\\_1'\n",
       "\\item 'ME\\_105\\_2'\n",
       "\\item 'ME\\_105\\_3'\n",
       "\\item 'ME\\_105\\_4'\n",
       "\\item 'ME\\_105\\_5'\n",
       "\\item 'ME\\_106\\_1'\n",
       "\\item 'ME\\_106\\_2'\n",
       "\\item 'ME\\_106\\_3'\n",
       "\\item 'ME\\_106\\_4'\n",
       "\\item 'ME\\_106\\_5'\n",
       "\\item 'Diagnosis'\n",
       "\\item 'SRS1'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'Neutrophils'\n",
       "\\item 'Lymphocytes'\n",
       "\\item 'Monocytes'\n",
       "\\item 'PEER\\_1'\n",
       "\\item 'PEER\\_2'\n",
       "\\item 'PEER\\_3'\n",
       "\\item 'PEER\\_4'\n",
       "\\item 'PEER\\_5'\n",
       "\\item 'PEER\\_6'\n",
       "\\item 'PEER\\_7'\n",
       "\\item 'PEER\\_8'\n",
       "\\item 'PEER\\_9'\n",
       "\\item 'PEER\\_10'\n",
       "\\item 'PEER\\_11'\n",
       "\\item 'PEER\\_12'\n",
       "\\item 'PEER\\_13'\n",
       "\\item 'PEER\\_14'\n",
       "\\item 'PEER\\_15'\n",
       "\\item 'PEER\\_16'\n",
       "\\item 'PEER\\_17'\n",
       "\\item 'PEER\\_18'\n",
       "\\item 'PEER\\_19'\n",
       "\\item 'PEER\\_20'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Sample.ID'\n",
       "2. 'GAinS.ID'\n",
       "3. 'ME_1_1'\n",
       "4. 'ME_1_2'\n",
       "5. 'ME_1_3'\n",
       "6. 'ME_1_4'\n",
       "7. 'ME_1_5'\n",
       "8. 'ME_2_1'\n",
       "9. 'ME_2_2'\n",
       "10. 'ME_2_3'\n",
       "11. 'ME_2_4'\n",
       "12. 'ME_2_5'\n",
       "13. 'ME_3_1'\n",
       "14. 'ME_3_2'\n",
       "15. 'ME_3_3'\n",
       "16. 'ME_3_4'\n",
       "17. 'ME_3_5'\n",
       "18. 'ME_4_1'\n",
       "19. 'ME_4_2'\n",
       "20. 'ME_4_3'\n",
       "21. 'ME_4_4'\n",
       "22. 'ME_4_5'\n",
       "23. 'ME_5_1'\n",
       "24. 'ME_5_2'\n",
       "25. 'ME_5_3'\n",
       "26. 'ME_5_4'\n",
       "27. 'ME_5_5'\n",
       "28. 'ME_6_1'\n",
       "29. 'ME_6_2'\n",
       "30. 'ME_6_3'\n",
       "31. 'ME_6_4'\n",
       "32. 'ME_6_5'\n",
       "33. 'ME_7_1'\n",
       "34. 'ME_7_2'\n",
       "35. 'ME_7_3'\n",
       "36. 'ME_7_4'\n",
       "37. 'ME_7_5'\n",
       "38. 'ME_8_1'\n",
       "39. 'ME_8_2'\n",
       "40. 'ME_8_3'\n",
       "41. 'ME_8_4'\n",
       "42. 'ME_8_5'\n",
       "43. 'ME_9_1'\n",
       "44. 'ME_9_2'\n",
       "45. 'ME_9_3'\n",
       "46. 'ME_9_4'\n",
       "47. 'ME_9_5'\n",
       "48. 'ME_10_1'\n",
       "49. 'ME_10_2'\n",
       "50. 'ME_10_3'\n",
       "51. 'ME_10_4'\n",
       "52. 'ME_10_5'\n",
       "53. 'ME_11_1'\n",
       "54. 'ME_11_2'\n",
       "55. 'ME_11_3'\n",
       "56. 'ME_11_4'\n",
       "57. 'ME_11_5'\n",
       "58. 'ME_12_1'\n",
       "59. 'ME_12_2'\n",
       "60. 'ME_12_3'\n",
       "61. 'ME_12_4'\n",
       "62. 'ME_12_5'\n",
       "63. 'ME_13_1'\n",
       "64. 'ME_13_2'\n",
       "65. 'ME_13_3'\n",
       "66. 'ME_13_4'\n",
       "67. 'ME_13_5'\n",
       "68. 'ME_14_1'\n",
       "69. 'ME_14_2'\n",
       "70. 'ME_14_3'\n",
       "71. 'ME_14_4'\n",
       "72. 'ME_14_5'\n",
       "73. 'ME_15_1'\n",
       "74. 'ME_15_2'\n",
       "75. 'ME_15_3'\n",
       "76. 'ME_15_4'\n",
       "77. 'ME_15_5'\n",
       "78. 'ME_16_1'\n",
       "79. 'ME_16_2'\n",
       "80. 'ME_16_3'\n",
       "81. 'ME_16_4'\n",
       "82. 'ME_16_5'\n",
       "83. 'ME_17_1'\n",
       "84. 'ME_17_2'\n",
       "85. 'ME_17_3'\n",
       "86. 'ME_17_4'\n",
       "87. 'ME_17_5'\n",
       "88. 'ME_18_1'\n",
       "89. 'ME_18_2'\n",
       "90. 'ME_18_3'\n",
       "91. 'ME_18_4'\n",
       "92. 'ME_18_5'\n",
       "93. 'ME_19_1'\n",
       "94. 'ME_19_2'\n",
       "95. 'ME_19_3'\n",
       "96. 'ME_19_4'\n",
       "97. 'ME_19_5'\n",
       "98. 'ME_20_1'\n",
       "99. 'ME_20_2'\n",
       "100. 'ME_20_3'\n",
       "101. 'ME_20_4'\n",
       "102. 'ME_20_5'\n",
       "103. 'ME_21_1'\n",
       "104. 'ME_21_2'\n",
       "105. 'ME_21_3'\n",
       "106. 'ME_21_4'\n",
       "107. 'ME_21_5'\n",
       "108. 'ME_22_1'\n",
       "109. 'ME_22_2'\n",
       "110. 'ME_22_3'\n",
       "111. 'ME_22_4'\n",
       "112. 'ME_22_5'\n",
       "113. 'ME_23_1'\n",
       "114. 'ME_23_2'\n",
       "115. 'ME_23_3'\n",
       "116. 'ME_23_4'\n",
       "117. 'ME_23_5'\n",
       "118. 'ME_24_1'\n",
       "119. 'ME_24_2'\n",
       "120. 'ME_24_3'\n",
       "121. 'ME_24_4'\n",
       "122. 'ME_24_5'\n",
       "123. 'ME_25_1'\n",
       "124. 'ME_25_2'\n",
       "125. 'ME_25_3'\n",
       "126. 'ME_25_4'\n",
       "127. 'ME_25_5'\n",
       "128. 'ME_26_1'\n",
       "129. 'ME_26_2'\n",
       "130. 'ME_26_3'\n",
       "131. 'ME_26_4'\n",
       "132. 'ME_26_5'\n",
       "133. 'ME_27_1'\n",
       "134. 'ME_27_2'\n",
       "135. 'ME_27_3'\n",
       "136. 'ME_27_4'\n",
       "137. 'ME_27_5'\n",
       "138. 'ME_28_1'\n",
       "139. 'ME_28_2'\n",
       "140. 'ME_28_3'\n",
       "141. 'ME_28_4'\n",
       "142. 'ME_28_5'\n",
       "143. 'ME_29_1'\n",
       "144. 'ME_29_2'\n",
       "145. 'ME_29_3'\n",
       "146. 'ME_29_4'\n",
       "147. 'ME_29_5'\n",
       "148. 'ME_30_1'\n",
       "149. 'ME_30_2'\n",
       "150. 'ME_30_3'\n",
       "151. 'ME_30_4'\n",
       "152. 'ME_30_5'\n",
       "153. 'ME_31_1'\n",
       "154. 'ME_31_2'\n",
       "155. 'ME_31_3'\n",
       "156. 'ME_31_4'\n",
       "157. 'ME_31_5'\n",
       "158. 'ME_32_1'\n",
       "159. 'ME_32_2'\n",
       "160. 'ME_32_3'\n",
       "161. 'ME_32_4'\n",
       "162. 'ME_32_5'\n",
       "163. 'ME_33_1'\n",
       "164. 'ME_33_2'\n",
       "165. 'ME_33_3'\n",
       "166. 'ME_33_4'\n",
       "167. 'ME_33_5'\n",
       "168. 'ME_34_1'\n",
       "169. 'ME_34_2'\n",
       "170. 'ME_34_3'\n",
       "171. 'ME_34_4'\n",
       "172. 'ME_34_5'\n",
       "173. 'ME_35_1'\n",
       "174. 'ME_35_2'\n",
       "175. 'ME_35_3'\n",
       "176. 'ME_35_4'\n",
       "177. 'ME_35_5'\n",
       "178. 'ME_36_1'\n",
       "179. 'ME_36_2'\n",
       "180. 'ME_36_3'\n",
       "181. 'ME_36_4'\n",
       "182. 'ME_36_5'\n",
       "183. 'ME_37_1'\n",
       "184. 'ME_37_2'\n",
       "185. 'ME_37_3'\n",
       "186. 'ME_37_4'\n",
       "187. 'ME_37_5'\n",
       "188. 'ME_38_1'\n",
       "189. 'ME_38_2'\n",
       "190. 'ME_38_3'\n",
       "191. 'ME_38_4'\n",
       "192. 'ME_38_5'\n",
       "193. 'ME_39_1'\n",
       "194. 'ME_39_2'\n",
       "195. 'ME_39_3'\n",
       "196. 'ME_39_4'\n",
       "197. 'ME_39_5'\n",
       "198. 'ME_40_1'\n",
       "199. 'ME_40_2'\n",
       "200. 'ME_40_3'\n",
       "201. ⋯\n",
       "202. 'ME_73_3'\n",
       "203. 'ME_73_4'\n",
       "204. 'ME_73_5'\n",
       "205. 'ME_74_1'\n",
       "206. 'ME_74_2'\n",
       "207. 'ME_74_3'\n",
       "208. 'ME_74_4'\n",
       "209. 'ME_74_5'\n",
       "210. 'ME_75_1'\n",
       "211. 'ME_75_2'\n",
       "212. 'ME_75_3'\n",
       "213. 'ME_75_4'\n",
       "214. 'ME_75_5'\n",
       "215. 'ME_76_1'\n",
       "216. 'ME_76_2'\n",
       "217. 'ME_76_3'\n",
       "218. 'ME_76_4'\n",
       "219. 'ME_76_5'\n",
       "220. 'ME_77_1'\n",
       "221. 'ME_77_2'\n",
       "222. 'ME_77_3'\n",
       "223. 'ME_77_4'\n",
       "224. 'ME_77_5'\n",
       "225. 'ME_78_1'\n",
       "226. 'ME_78_2'\n",
       "227. 'ME_78_3'\n",
       "228. 'ME_78_4'\n",
       "229. 'ME_78_5'\n",
       "230. 'ME_79_1'\n",
       "231. 'ME_79_2'\n",
       "232. 'ME_79_3'\n",
       "233. 'ME_79_4'\n",
       "234. 'ME_79_5'\n",
       "235. 'ME_80_1'\n",
       "236. 'ME_80_2'\n",
       "237. 'ME_80_3'\n",
       "238. 'ME_80_4'\n",
       "239. 'ME_80_5'\n",
       "240. 'ME_81_1'\n",
       "241. 'ME_81_2'\n",
       "242. 'ME_81_3'\n",
       "243. 'ME_81_4'\n",
       "244. 'ME_81_5'\n",
       "245. 'ME_82_1'\n",
       "246. 'ME_82_2'\n",
       "247. 'ME_82_3'\n",
       "248. 'ME_82_4'\n",
       "249. 'ME_82_5'\n",
       "250. 'ME_83_1'\n",
       "251. 'ME_83_2'\n",
       "252. 'ME_83_3'\n",
       "253. 'ME_83_4'\n",
       "254. 'ME_83_5'\n",
       "255. 'ME_84_1'\n",
       "256. 'ME_84_2'\n",
       "257. 'ME_84_3'\n",
       "258. 'ME_84_4'\n",
       "259. 'ME_84_5'\n",
       "260. 'ME_85_1'\n",
       "261. 'ME_85_2'\n",
       "262. 'ME_85_3'\n",
       "263. 'ME_85_4'\n",
       "264. 'ME_85_5'\n",
       "265. 'ME_86_1'\n",
       "266. 'ME_86_2'\n",
       "267. 'ME_86_3'\n",
       "268. 'ME_86_4'\n",
       "269. 'ME_86_5'\n",
       "270. 'ME_87_1'\n",
       "271. 'ME_87_2'\n",
       "272. 'ME_87_3'\n",
       "273. 'ME_87_4'\n",
       "274. 'ME_87_5'\n",
       "275. 'ME_88_1'\n",
       "276. 'ME_88_2'\n",
       "277. 'ME_88_3'\n",
       "278. 'ME_88_4'\n",
       "279. 'ME_88_5'\n",
       "280. 'ME_89_1'\n",
       "281. 'ME_89_2'\n",
       "282. 'ME_89_3'\n",
       "283. 'ME_89_4'\n",
       "284. 'ME_89_5'\n",
       "285. 'ME_90_1'\n",
       "286. 'ME_90_2'\n",
       "287. 'ME_90_3'\n",
       "288. 'ME_90_4'\n",
       "289. 'ME_90_5'\n",
       "290. 'ME_91_1'\n",
       "291. 'ME_91_2'\n",
       "292. 'ME_91_3'\n",
       "293. 'ME_91_4'\n",
       "294. 'ME_91_5'\n",
       "295. 'ME_92_1'\n",
       "296. 'ME_92_2'\n",
       "297. 'ME_92_3'\n",
       "298. 'ME_92_4'\n",
       "299. 'ME_92_5'\n",
       "300. 'ME_93_1'\n",
       "301. 'ME_93_2'\n",
       "302. 'ME_93_3'\n",
       "303. 'ME_93_4'\n",
       "304. 'ME_93_5'\n",
       "305. 'ME_94_1'\n",
       "306. 'ME_94_2'\n",
       "307. 'ME_94_3'\n",
       "308. 'ME_94_4'\n",
       "309. 'ME_94_5'\n",
       "310. 'ME_95_1'\n",
       "311. 'ME_95_2'\n",
       "312. 'ME_95_3'\n",
       "313. 'ME_95_4'\n",
       "314. 'ME_95_5'\n",
       "315. 'ME_96_1'\n",
       "316. 'ME_96_2'\n",
       "317. 'ME_96_3'\n",
       "318. 'ME_96_4'\n",
       "319. 'ME_96_5'\n",
       "320. 'ME_97_1'\n",
       "321. 'ME_97_2'\n",
       "322. 'ME_97_3'\n",
       "323. 'ME_97_4'\n",
       "324. 'ME_97_5'\n",
       "325. 'ME_98_1'\n",
       "326. 'ME_98_2'\n",
       "327. 'ME_98_3'\n",
       "328. 'ME_98_4'\n",
       "329. 'ME_98_5'\n",
       "330. 'ME_99_1'\n",
       "331. 'ME_99_2'\n",
       "332. 'ME_99_3'\n",
       "333. 'ME_99_4'\n",
       "334. 'ME_99_5'\n",
       "335. 'ME_100_1'\n",
       "336. 'ME_100_2'\n",
       "337. 'ME_100_3'\n",
       "338. 'ME_100_4'\n",
       "339. 'ME_100_5'\n",
       "340. 'ME_101_1'\n",
       "341. 'ME_101_2'\n",
       "342. 'ME_101_3'\n",
       "343. 'ME_101_4'\n",
       "344. 'ME_101_5'\n",
       "345. 'ME_102_1'\n",
       "346. 'ME_102_2'\n",
       "347. 'ME_102_3'\n",
       "348. 'ME_102_4'\n",
       "349. 'ME_102_5'\n",
       "350. 'ME_103_1'\n",
       "351. 'ME_103_2'\n",
       "352. 'ME_103_3'\n",
       "353. 'ME_103_4'\n",
       "354. 'ME_103_5'\n",
       "355. 'ME_104_1'\n",
       "356. 'ME_104_2'\n",
       "357. 'ME_104_3'\n",
       "358. 'ME_104_4'\n",
       "359. 'ME_104_5'\n",
       "360. 'ME_105_1'\n",
       "361. 'ME_105_2'\n",
       "362. 'ME_105_3'\n",
       "363. 'ME_105_4'\n",
       "364. 'ME_105_5'\n",
       "365. 'ME_106_1'\n",
       "366. 'ME_106_2'\n",
       "367. 'ME_106_3'\n",
       "368. 'ME_106_4'\n",
       "369. 'ME_106_5'\n",
       "370. 'Diagnosis'\n",
       "371. 'SRS1'\n",
       "372. 'PC1'\n",
       "373. 'PC2'\n",
       "374. 'PC3'\n",
       "375. 'PC4'\n",
       "376. 'PC5'\n",
       "377. 'PC6'\n",
       "378. 'PC7'\n",
       "379. 'Neutrophils'\n",
       "380. 'Lymphocytes'\n",
       "381. 'Monocytes'\n",
       "382. 'PEER_1'\n",
       "383. 'PEER_2'\n",
       "384. 'PEER_3'\n",
       "385. 'PEER_4'\n",
       "386. 'PEER_5'\n",
       "387. 'PEER_6'\n",
       "388. 'PEER_7'\n",
       "389. 'PEER_8'\n",
       "390. 'PEER_9'\n",
       "391. 'PEER_10'\n",
       "392. 'PEER_11'\n",
       "393. 'PEER_12'\n",
       "394. 'PEER_13'\n",
       "395. 'PEER_14'\n",
       "396. 'PEER_15'\n",
       "397. 'PEER_16'\n",
       "398. 'PEER_17'\n",
       "399. 'PEER_18'\n",
       "400. 'PEER_19'\n",
       "401. 'PEER_20'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"Sample.ID\"   \"GAinS.ID\"    \"ME_1_1\"      \"ME_1_2\"      \"ME_1_3\"     \n",
       "  [6] \"ME_1_4\"      \"ME_1_5\"      \"ME_2_1\"      \"ME_2_2\"      \"ME_2_3\"     \n",
       " [11] \"ME_2_4\"      \"ME_2_5\"      \"ME_3_1\"      \"ME_3_2\"      \"ME_3_3\"     \n",
       " [16] \"ME_3_4\"      \"ME_3_5\"      \"ME_4_1\"      \"ME_4_2\"      \"ME_4_3\"     \n",
       " [21] \"ME_4_4\"      \"ME_4_5\"      \"ME_5_1\"      \"ME_5_2\"      \"ME_5_3\"     \n",
       " [26] \"ME_5_4\"      \"ME_5_5\"      \"ME_6_1\"      \"ME_6_2\"      \"ME_6_3\"     \n",
       " [31] \"ME_6_4\"      \"ME_6_5\"      \"ME_7_1\"      \"ME_7_2\"      \"ME_7_3\"     \n",
       " [36] \"ME_7_4\"      \"ME_7_5\"      \"ME_8_1\"      \"ME_8_2\"      \"ME_8_3\"     \n",
       " [41] \"ME_8_4\"      \"ME_8_5\"      \"ME_9_1\"      \"ME_9_2\"      \"ME_9_3\"     \n",
       " [46] \"ME_9_4\"      \"ME_9_5\"      \"ME_10_1\"     \"ME_10_2\"     \"ME_10_3\"    \n",
       " [51] \"ME_10_4\"     \"ME_10_5\"     \"ME_11_1\"     \"ME_11_2\"     \"ME_11_3\"    \n",
       " [56] \"ME_11_4\"     \"ME_11_5\"     \"ME_12_1\"     \"ME_12_2\"     \"ME_12_3\"    \n",
       " [61] \"ME_12_4\"     \"ME_12_5\"     \"ME_13_1\"     \"ME_13_2\"     \"ME_13_3\"    \n",
       " [66] \"ME_13_4\"     \"ME_13_5\"     \"ME_14_1\"     \"ME_14_2\"     \"ME_14_3\"    \n",
       " [71] \"ME_14_4\"     \"ME_14_5\"     \"ME_15_1\"     \"ME_15_2\"     \"ME_15_3\"    \n",
       " [76] \"ME_15_4\"     \"ME_15_5\"     \"ME_16_1\"     \"ME_16_2\"     \"ME_16_3\"    \n",
       " [81] \"ME_16_4\"     \"ME_16_5\"     \"ME_17_1\"     \"ME_17_2\"     \"ME_17_3\"    \n",
       " [86] \"ME_17_4\"     \"ME_17_5\"     \"ME_18_1\"     \"ME_18_2\"     \"ME_18_3\"    \n",
       " [91] \"ME_18_4\"     \"ME_18_5\"     \"ME_19_1\"     \"ME_19_2\"     \"ME_19_3\"    \n",
       " [96] \"ME_19_4\"     \"ME_19_5\"     \"ME_20_1\"     \"ME_20_2\"     \"ME_20_3\"    \n",
       "[101] \"ME_20_4\"     \"ME_20_5\"     \"ME_21_1\"     \"ME_21_2\"     \"ME_21_3\"    \n",
       "[106] \"ME_21_4\"     \"ME_21_5\"     \"ME_22_1\"     \"ME_22_2\"     \"ME_22_3\"    \n",
       "[111] \"ME_22_4\"     \"ME_22_5\"     \"ME_23_1\"     \"ME_23_2\"     \"ME_23_3\"    \n",
       "[116] \"ME_23_4\"     \"ME_23_5\"     \"ME_24_1\"     \"ME_24_2\"     \"ME_24_3\"    \n",
       "[121] \"ME_24_4\"     \"ME_24_5\"     \"ME_25_1\"     \"ME_25_2\"     \"ME_25_3\"    \n",
       "[126] \"ME_25_4\"     \"ME_25_5\"     \"ME_26_1\"     \"ME_26_2\"     \"ME_26_3\"    \n",
       "[131] \"ME_26_4\"     \"ME_26_5\"     \"ME_27_1\"     \"ME_27_2\"     \"ME_27_3\"    \n",
       "[136] \"ME_27_4\"     \"ME_27_5\"     \"ME_28_1\"     \"ME_28_2\"     \"ME_28_3\"    \n",
       "[141] \"ME_28_4\"     \"ME_28_5\"     \"ME_29_1\"     \"ME_29_2\"     \"ME_29_3\"    \n",
       "[146] \"ME_29_4\"     \"ME_29_5\"     \"ME_30_1\"     \"ME_30_2\"     \"ME_30_3\"    \n",
       "[151] \"ME_30_4\"     \"ME_30_5\"     \"ME_31_1\"     \"ME_31_2\"     \"ME_31_3\"    \n",
       "[156] \"ME_31_4\"     \"ME_31_5\"     \"ME_32_1\"     \"ME_32_2\"     \"ME_32_3\"    \n",
       "[161] \"ME_32_4\"     \"ME_32_5\"     \"ME_33_1\"     \"ME_33_2\"     \"ME_33_3\"    \n",
       "[166] \"ME_33_4\"     \"ME_33_5\"     \"ME_34_1\"     \"ME_34_2\"     \"ME_34_3\"    \n",
       "[171] \"ME_34_4\"     \"ME_34_5\"     \"ME_35_1\"     \"ME_35_2\"     \"ME_35_3\"    \n",
       "[176] \"ME_35_4\"     \"ME_35_5\"     \"ME_36_1\"     \"ME_36_2\"     \"ME_36_3\"    \n",
       "[181] \"ME_36_4\"     \"ME_36_5\"     \"ME_37_1\"     \"ME_37_2\"     \"ME_37_3\"    \n",
       "[186] \"ME_37_4\"     \"ME_37_5\"     \"ME_38_1\"     \"ME_38_2\"     \"ME_38_3\"    \n",
       "[191] \"ME_38_4\"     \"ME_38_5\"     \"ME_39_1\"     \"ME_39_2\"     \"ME_39_3\"    \n",
       "[196] \"ME_39_4\"     \"ME_39_5\"     \"ME_40_1\"     \"ME_40_2\"     \"ME_40_3\"    \n",
       "[201] \"ME_40_4\"     \"ME_40_5\"     \"ME_41_1\"     \"ME_41_2\"     \"ME_41_3\"    \n",
       "[206] \"ME_41_4\"     \"ME_41_5\"     \"ME_42_1\"     \"ME_42_2\"     \"ME_42_3\"    \n",
       "[211] \"ME_42_4\"     \"ME_42_5\"     \"ME_43_1\"     \"ME_43_2\"     \"ME_43_3\"    \n",
       "[216] \"ME_43_4\"     \"ME_43_5\"     \"ME_44_1\"     \"ME_44_2\"     \"ME_44_3\"    \n",
       "[221] \"ME_44_4\"     \"ME_44_5\"     \"ME_45_1\"     \"ME_45_2\"     \"ME_45_3\"    \n",
       "[226] \"ME_45_4\"     \"ME_45_5\"     \"ME_46_1\"     \"ME_46_2\"     \"ME_46_3\"    \n",
       "[231] \"ME_46_4\"     \"ME_46_5\"     \"ME_47_1\"     \"ME_47_2\"     \"ME_47_3\"    \n",
       "[236] \"ME_47_4\"     \"ME_47_5\"     \"ME_48_1\"     \"ME_48_2\"     \"ME_48_3\"    \n",
       "[241] \"ME_48_4\"     \"ME_48_5\"     \"ME_49_1\"     \"ME_49_2\"     \"ME_49_3\"    \n",
       "[246] \"ME_49_4\"     \"ME_49_5\"     \"ME_50_1\"     \"ME_50_2\"     \"ME_50_3\"    \n",
       "[251] \"ME_50_4\"     \"ME_50_5\"     \"ME_51_1\"     \"ME_51_2\"     \"ME_51_3\"    \n",
       "[256] \"ME_51_4\"     \"ME_51_5\"     \"ME_52_1\"     \"ME_52_2\"     \"ME_52_3\"    \n",
       "[261] \"ME_52_4\"     \"ME_52_5\"     \"ME_53_1\"     \"ME_53_2\"     \"ME_53_3\"    \n",
       "[266] \"ME_53_4\"     \"ME_53_5\"     \"ME_54_1\"     \"ME_54_2\"     \"ME_54_3\"    \n",
       "[271] \"ME_54_4\"     \"ME_54_5\"     \"ME_55_1\"     \"ME_55_2\"     \"ME_55_3\"    \n",
       "[276] \"ME_55_4\"     \"ME_55_5\"     \"ME_56_1\"     \"ME_56_2\"     \"ME_56_3\"    \n",
       "[281] \"ME_56_4\"     \"ME_56_5\"     \"ME_57_1\"     \"ME_57_2\"     \"ME_57_3\"    \n",
       "[286] \"ME_57_4\"     \"ME_57_5\"     \"ME_58_1\"     \"ME_58_2\"     \"ME_58_3\"    \n",
       "[291] \"ME_58_4\"     \"ME_58_5\"     \"ME_59_1\"     \"ME_59_2\"     \"ME_59_3\"    \n",
       "[296] \"ME_59_4\"     \"ME_59_5\"     \"ME_60_1\"     \"ME_60_2\"     \"ME_60_3\"    \n",
       "[301] \"ME_60_4\"     \"ME_60_5\"     \"ME_61_1\"     \"ME_61_2\"     \"ME_61_3\"    \n",
       "[306] \"ME_61_4\"     \"ME_61_5\"     \"ME_62_1\"     \"ME_62_2\"     \"ME_62_3\"    \n",
       "[311] \"ME_62_4\"     \"ME_62_5\"     \"ME_63_1\"     \"ME_63_2\"     \"ME_63_3\"    \n",
       "[316] \"ME_63_4\"     \"ME_63_5\"     \"ME_64_1\"     \"ME_64_2\"     \"ME_64_3\"    \n",
       "[321] \"ME_64_4\"     \"ME_64_5\"     \"ME_65_1\"     \"ME_65_2\"     \"ME_65_3\"    \n",
       "[326] \"ME_65_4\"     \"ME_65_5\"     \"ME_66_1\"     \"ME_66_2\"     \"ME_66_3\"    \n",
       "[331] \"ME_66_4\"     \"ME_66_5\"     \"ME_67_1\"     \"ME_67_2\"     \"ME_67_3\"    \n",
       "[336] \"ME_67_4\"     \"ME_67_5\"     \"ME_68_1\"     \"ME_68_2\"     \"ME_68_3\"    \n",
       "[341] \"ME_68_4\"     \"ME_68_5\"     \"ME_69_1\"     \"ME_69_2\"     \"ME_69_3\"    \n",
       "[346] \"ME_69_4\"     \"ME_69_5\"     \"ME_70_1\"     \"ME_70_2\"     \"ME_70_3\"    \n",
       "[351] \"ME_70_4\"     \"ME_70_5\"     \"ME_71_1\"     \"ME_71_2\"     \"ME_71_3\"    \n",
       "[356] \"ME_71_4\"     \"ME_71_5\"     \"ME_72_1\"     \"ME_72_2\"     \"ME_72_3\"    \n",
       "[361] \"ME_72_4\"     \"ME_72_5\"     \"ME_73_1\"     \"ME_73_2\"     \"ME_73_3\"    \n",
       "[366] \"ME_73_4\"     \"ME_73_5\"     \"ME_74_1\"     \"ME_74_2\"     \"ME_74_3\"    \n",
       "[371] \"ME_74_4\"     \"ME_74_5\"     \"ME_75_1\"     \"ME_75_2\"     \"ME_75_3\"    \n",
       "[376] \"ME_75_4\"     \"ME_75_5\"     \"ME_76_1\"     \"ME_76_2\"     \"ME_76_3\"    \n",
       "[381] \"ME_76_4\"     \"ME_76_5\"     \"ME_77_1\"     \"ME_77_2\"     \"ME_77_3\"    \n",
       "[386] \"ME_77_4\"     \"ME_77_5\"     \"ME_78_1\"     \"ME_78_2\"     \"ME_78_3\"    \n",
       "[391] \"ME_78_4\"     \"ME_78_5\"     \"ME_79_1\"     \"ME_79_2\"     \"ME_79_3\"    \n",
       "[396] \"ME_79_4\"     \"ME_79_5\"     \"ME_80_1\"     \"ME_80_2\"     \"ME_80_3\"    \n",
       "[401] \"ME_80_4\"     \"ME_80_5\"     \"ME_81_1\"     \"ME_81_2\"     \"ME_81_3\"    \n",
       "[406] \"ME_81_4\"     \"ME_81_5\"     \"ME_82_1\"     \"ME_82_2\"     \"ME_82_3\"    \n",
       "[411] \"ME_82_4\"     \"ME_82_5\"     \"ME_83_1\"     \"ME_83_2\"     \"ME_83_3\"    \n",
       "[416] \"ME_83_4\"     \"ME_83_5\"     \"ME_84_1\"     \"ME_84_2\"     \"ME_84_3\"    \n",
       "[421] \"ME_84_4\"     \"ME_84_5\"     \"ME_85_1\"     \"ME_85_2\"     \"ME_85_3\"    \n",
       "[426] \"ME_85_4\"     \"ME_85_5\"     \"ME_86_1\"     \"ME_86_2\"     \"ME_86_3\"    \n",
       "[431] \"ME_86_4\"     \"ME_86_5\"     \"ME_87_1\"     \"ME_87_2\"     \"ME_87_3\"    \n",
       "[436] \"ME_87_4\"     \"ME_87_5\"     \"ME_88_1\"     \"ME_88_2\"     \"ME_88_3\"    \n",
       "[441] \"ME_88_4\"     \"ME_88_5\"     \"ME_89_1\"     \"ME_89_2\"     \"ME_89_3\"    \n",
       "[446] \"ME_89_4\"     \"ME_89_5\"     \"ME_90_1\"     \"ME_90_2\"     \"ME_90_3\"    \n",
       "[451] \"ME_90_4\"     \"ME_90_5\"     \"ME_91_1\"     \"ME_91_2\"     \"ME_91_3\"    \n",
       "[456] \"ME_91_4\"     \"ME_91_5\"     \"ME_92_1\"     \"ME_92_2\"     \"ME_92_3\"    \n",
       "[461] \"ME_92_4\"     \"ME_92_5\"     \"ME_93_1\"     \"ME_93_2\"     \"ME_93_3\"    \n",
       "[466] \"ME_93_4\"     \"ME_93_5\"     \"ME_94_1\"     \"ME_94_2\"     \"ME_94_3\"    \n",
       "[471] \"ME_94_4\"     \"ME_94_5\"     \"ME_95_1\"     \"ME_95_2\"     \"ME_95_3\"    \n",
       "[476] \"ME_95_4\"     \"ME_95_5\"     \"ME_96_1\"     \"ME_96_2\"     \"ME_96_3\"    \n",
       "[481] \"ME_96_4\"     \"ME_96_5\"     \"ME_97_1\"     \"ME_97_2\"     \"ME_97_3\"    \n",
       "[486] \"ME_97_4\"     \"ME_97_5\"     \"ME_98_1\"     \"ME_98_2\"     \"ME_98_3\"    \n",
       "[491] \"ME_98_4\"     \"ME_98_5\"     \"ME_99_1\"     \"ME_99_2\"     \"ME_99_3\"    \n",
       "[496] \"ME_99_4\"     \"ME_99_5\"     \"ME_100_1\"    \"ME_100_2\"    \"ME_100_3\"   \n",
       "[501] \"ME_100_4\"    \"ME_100_5\"    \"ME_101_1\"    \"ME_101_2\"    \"ME_101_3\"   \n",
       "[506] \"ME_101_4\"    \"ME_101_5\"    \"ME_102_1\"    \"ME_102_2\"    \"ME_102_3\"   \n",
       "[511] \"ME_102_4\"    \"ME_102_5\"    \"ME_103_1\"    \"ME_103_2\"    \"ME_103_3\"   \n",
       "[516] \"ME_103_4\"    \"ME_103_5\"    \"ME_104_1\"    \"ME_104_2\"    \"ME_104_3\"   \n",
       "[521] \"ME_104_4\"    \"ME_104_5\"    \"ME_105_1\"    \"ME_105_2\"    \"ME_105_3\"   \n",
       "[526] \"ME_105_4\"    \"ME_105_5\"    \"ME_106_1\"    \"ME_106_2\"    \"ME_106_3\"   \n",
       "[531] \"ME_106_4\"    \"ME_106_5\"    \"Diagnosis\"   \"SRS1\"        \"PC1\"        \n",
       "[536] \"PC2\"         \"PC3\"         \"PC4\"         \"PC5\"         \"PC6\"        \n",
       "[541] \"PC7\"         \"Neutrophils\" \"Lymphocytes\" \"Monocytes\"   \"PEER_1\"     \n",
       "[546] \"PEER_2\"      \"PEER_3\"      \"PEER_4\"      \"PEER_5\"      \"PEER_6\"     \n",
       "[551] \"PEER_7\"      \"PEER_8\"      \"PEER_9\"      \"PEER_10\"     \"PEER_11\"    \n",
       "[556] \"PEER_12\"     \"PEER_13\"     \"PEER_14\"     \"PEER_15\"     \"PEER_16\"    \n",
       "[561] \"PEER_17\"     \"PEER_18\"     \"PEER_19\"     \"PEER_20\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(mapping.data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3467b47e-fe69-4d05-a429-96a961835b98",
   "metadata": {},
   "source": [
    "Save the mapping data design matrix for linear mixed modeling."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "0ea89bc0-7ba5-41be-8f25-cf2961c20975",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(mapping.data, \"~/gains_team282/nikhil/expression/eigengene_sva/mapping_data.csv\", row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61f4d47f-529b-4310-b319-7878a5dbaa38",
   "metadata": {},
   "source": [
    "### Identify SNPs for mQTL Mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66e1e685-80a3-4292-98ce-ef867b689896",
   "metadata": {},
   "source": [
    "Since *trans* mapping in a genome-wide fashion will require a lot of testing (the multiple-testing burden will be high), we will use SNPs that we have a more biological hypothesis for. Specifically, I will include all lead eSNPs, all lead eSNPs from the conditional cis-eQTL analysis, and related associations from the EBI GWAS Catalog."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8b359753-e010-491e-9a22-3a3a9546b14a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "30667"
      ],
      "text/latex": [
       "30667"
      ],
      "text/markdown": [
       "30667"
      ],
      "text/plain": [
       "[1] 30667"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(ebi.studies[,\"DISEASE/TRAIT\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c7befed-55fa-4e35-9097-f3456d9fbb73",
   "metadata": {},
   "source": [
    "After filtering based on the genotyping in GAinS, we have 124,524 studies and 56,218 SNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "28a18096-239a-4f40-ab5e-f5aca97659bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "124524"
      ],
      "text/latex": [
       "124524"
      ],
      "text/markdown": [
       "124524"
      ],
      "text/plain": [
       "[1] 124524"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "56218"
      ],
      "text/latex": [
       "56218"
      ],
      "text/markdown": [
       "56218"
      ],
      "text/plain": [
       "[1] 56218"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Studies with terms specified\n",
    "studies <- ebi.studies[, \"STUDY ACCESSION\"]\n",
    "\n",
    "# Associations from the studies that are genome-wide significant\n",
    "studies.assoc <- ebi.assoc[ebi.assoc[, \"STUDY ACCESSION\"] %in% studies,] %>%\n",
    "    dplyr::mutate(`P-VALUE` = as.numeric(`P-VALUE`)) %>%\n",
    "    dplyr::filter(`P-VALUE` < 5e-8)\n",
    "\n",
    "# Association SNPs that overlap with genotyping data in GAinS\n",
    "# Ignore SNPs that are on the X chromosome\n",
    "studies.assoc <- studies.assoc %>%\n",
    "    dplyr::filter(SNPS %in% geno.bim$snps[geno.bim$chr %in% as.character(1:22)])\n",
    "\n",
    "nrow(studies.assoc)\n",
    "\n",
    "length(unique(studies.assoc$SNPS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "08b11812-a3e3-46ea-87f0-382fd238971b",
   "metadata": {},
   "outputs": [],
   "source": [
    "mqtl.snp.table <- dplyr::bind_rows(\n",
    "    lead.cis.eqtl %>%\n",
    "        dplyr::mutate(source=\"Lead cis-eQTL SNP\") %>%\n",
    "        dplyr::mutate(conditional_number=NA, accession=NA) %>%\n",
    "        dplyr::select(snps, source, egene=gene, conditional_number, accession),\n",
    "    conditional.cis.eqtl %>%\n",
    "        dplyr::mutate(source=\"Conditional cis-eQTL SNP\") %>%\n",
    "        dplyr::mutate(accession=NA) %>%\n",
    "        dplyr::select(snps=SNP, source, egene=Gene, conditional_number=Number, accession),\n",
    "    studies.assoc %>%\n",
    "        dplyr::mutate(source=\"EBI GWAS Catalog\") %>%\n",
    "        dplyr::mutate(egene=NA, conditional_number=NA) %>%\n",
    "        dplyr::select(snps=SNPS, source, egene, conditional_number, accession=37)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "c8a7961b-fc8c-43c7-a058-ba0cdf59eb5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(mqtl.snp.table, \"~/gains_team282/nikhil/expression/eigengene_sva/mqtl_snp_table.csv\", quote=F, row.names=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ade8fd03-3607-4922-83b4-9bbbf6e41543",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "70968"
      ],
      "text/latex": [
       "70968"
      ],
      "text/markdown": [
       "70968"
      ],
      "text/plain": [
       "[1] 70968"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(mqtl.snp.table$snps))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b6e37de3-a4f4-4f9b-a0d0-aaadc0431f9f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
